{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from importlib import reload\n",
    "import src.utils as utils\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "morl_dvd_dir = '/home/srryn/hpchome/DVD/versions/9_1_1/final_outputs/'\n",
    "active_dir = os.path.join(morl_dvd_dir, 'delete_later')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "chroms_names = ['MT', 'X'] + [str(x) for x in range(1,23)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Preprocessing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Constructing df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "chromosome_dir = 'split_vcf_chromosomes_csvs'\n",
    "if not os.path.isdir(chromosome_dir):\n",
    "    os.makedirs(chromosome_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "compressed_dir = 'split_vcf_chromosomes_csvs_compressed'\n",
    "if not os.path.isdir(compressed_dir):\n",
    "    os.mkdir(compressed_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import tarfile\n",
    "\n",
    "# def extract_tarball(tarball_file, keep_tar_file = False):\n",
    "#     file_path, file_name = os.path.dirname(tarball_file), os.path.basename(tarball_file)\n",
    "#     base_file_name = file_name.split('.')[0]\n",
    "#     tar_file = os.path.join(file_path, base_file_name + '.tar')\n",
    "#     archive_file = utils.decompress(tarball_file, tar_file)\n",
    "#     try:\n",
    "#         with tarfile.open(archive_file, 'r') as archive:\n",
    "#             # archive.extractall(output_directory)\n",
    "#             archive.extractall()\n",
    "#         print(\"Unarchiving complete\")\n",
    "#     except FileNotFoundError:\n",
    "#         print(f\"Archive file '{archive_file}' not found.\")\n",
    "#     except Exception as e:\n",
    "#         print(f\"An error occurred: {str(e)}\")\n",
    "        \n",
    "#     if not keep_tar_file:\n",
    "#         os.remove(tar_file)\n",
    "#         print(f\"Removed archive file {tar_file}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (208,209,210,211,212,214,215) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (25,171,199,200,226,229) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (219) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (150,208,209,210,211,212,213,214,215,226) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (177,211,217,227) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (216) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (179,226) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (216,219) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (172,216,226) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (175,207,208,209,210,211,212,213,214,219,227) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (177,226) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (229) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (216) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:5: DtypeWarning: Columns (216) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
      "/tmp/ipykernel_4638/718539697.py:8: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  df = pd.concat(dfs).map(utils.safe_decode)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TYPE</th>\n",
       "      <th>GENE</th>\n",
       "      <th>gnomad_MAX_MAF</th>\n",
       "      <th>gnomad_MAX_MAF_SOURCE</th>\n",
       "      <th>gnomad_AC_afr_pass</th>\n",
       "      <th>gnomad_AC_amr_pass</th>\n",
       "      <th>gnomad_AC_eas_pass</th>\n",
       "      <th>gnomad_AC_nfe_pass</th>\n",
       "      <th>gnomad_AN_afr_pass</th>\n",
       "      <th>gnomad_AN_amr_pass</th>\n",
       "      <th>...</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>Surface Area</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "      <th>in...</th>\n",
       "      <th>second part</th>\n",
       "      <th>calc...</th>\n",
       "      <th>ca...</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12624.0</td>\n",
       "      <td>32814.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>111.511314</td>\n",
       "      <td>0.697034</td>\n",
       "      <td>45.18</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12388.0</td>\n",
       "      <td>32614.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>131.269338</td>\n",
       "      <td>0.951867</td>\n",
       "      <td>42.57</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12242.0</td>\n",
       "      <td>32478.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>114.498499</td>\n",
       "      <td>0.830258</td>\n",
       "      <td>42.35</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11972.0</td>\n",
       "      <td>32270.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>161.350503</td>\n",
       "      <td>1.025505</td>\n",
       "      <td>39.05</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11994.0</td>\n",
       "      <td>32226.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>118.519368</td>\n",
       "      <td>0.859414</td>\n",
       "      <td>42.94</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>SNV</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15462.0</td>\n",
       "      <td>34188.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>5.626061</td>\n",
       "      <td>0.035758</td>\n",
       "      <td>98.35</td>\n",
       "      <td>-0.9</td>\n",
       "      <td>0.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>SNV</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>15448.0</td>\n",
       "      <td>33916.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>66.834983</td>\n",
       "      <td>0.424787</td>\n",
       "      <td>98.27</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>SNV</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>0.000130</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8712.0</td>\n",
       "      <td>848.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>46.255767</td>\n",
       "      <td>0.201331</td>\n",
       "      <td>97.67</td>\n",
       "      <td>-2.6</td>\n",
       "      <td>2.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>SNV</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15436.0</td>\n",
       "      <td>33812.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...</td>\n",
       "      <td>74.926502</td>\n",
       "      <td>0.381969</td>\n",
       "      <td>95.56</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>SNV</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15398.0</td>\n",
       "      <td>33296.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>135.808746</td>\n",
       "      <td>0.686764</td>\n",
       "      <td>85.69</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128786 rows × 236 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             TYPE    GENE  gnomad_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                                \n",
       "22:18561153:A>G               SNV   PEX26        0.000035   \n",
       "22:18561156:C>T               SNV   PEX26        0.000035   \n",
       "22:18561159:C>T               SNV   PEX26        0.000010   \n",
       "22:18561162:C>T               SNV   PEX26        0.000035   \n",
       "22:18561164:T>A               SNV   PEX26        0.000011   \n",
       "...                           ...     ...             ...   \n",
       "15:83561573:G>C               SNV  HOMER2        0.000009   \n",
       "15:83561577:T>G               SNV  HOMER2        0.000018   \n",
       "15:83561579:A>G               SNV  HOMER2        0.000130   \n",
       "15:83561583:T>C               SNV  HOMER2        0.000000   \n",
       "15:83561589:G>C               SNV  HOMER2        0.000009   \n",
       "\n",
       "                                                          gnomad_MAX_MAF_SOURCE  \\\n",
       "Genomic Description (GRCh37)                                                      \n",
       "22:18561153:A>G                                              gnomad_AF_sas_pass   \n",
       "22:18561156:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561159:C>T                                              gnomad_AF_nfe_pass   \n",
       "22:18561162:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561164:T>A                                              gnomad_AF_nfe_pass   \n",
       "...                                                                         ...   \n",
       "15:83561573:G>C                                              gnomad_AF_nfe_pass   \n",
       "15:83561577:T>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561579:A>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561583:T>C               gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...   \n",
       "15:83561589:G>C                                              gnomad_AF_nfe_pass   \n",
       "\n",
       "                              gnomad_AC_afr_pass  gnomad_AC_amr_pass  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "22:18561153:A>G                              0.0                 0.0   \n",
       "22:18561156:C>T                              0.0                 0.0   \n",
       "22:18561159:C>T                              0.0                 0.0   \n",
       "22:18561162:C>T                              0.0                 0.0   \n",
       "22:18561164:T>A                              0.0                 0.0   \n",
       "...                                          ...                 ...   \n",
       "15:83561573:G>C                              0.0                 0.0   \n",
       "15:83561577:T>G                              0.0                 0.0   \n",
       "15:83561579:A>G                              0.0                 0.0   \n",
       "15:83561583:T>C                              0.0                 0.0   \n",
       "15:83561589:G>C                              0.0                 0.0   \n",
       "\n",
       "                              gnomad_AC_eas_pass  gnomad_AC_nfe_pass  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "22:18561153:A>G                              0.0                 0.0   \n",
       "22:18561156:C>T                              0.0                 0.0   \n",
       "22:18561159:C>T                              0.0                 1.0   \n",
       "22:18561162:C>T                              0.0                 0.0   \n",
       "22:18561164:T>A                              0.0                 1.0   \n",
       "...                                          ...                 ...   \n",
       "15:83561573:G>C                              0.0                 1.0   \n",
       "15:83561577:T>G                              0.0                 2.0   \n",
       "15:83561579:A>G                              0.0                 2.0   \n",
       "15:83561583:T>C                              0.0                 0.0   \n",
       "15:83561589:G>C                              0.0                 1.0   \n",
       "\n",
       "                              gnomad_AN_afr_pass  gnomad_AN_amr_pass  ...  \\\n",
       "Genomic Description (GRCh37)                                          ...   \n",
       "22:18561153:A>G                          12624.0             32814.0  ...   \n",
       "22:18561156:C>T                          12388.0             32614.0  ...   \n",
       "22:18561159:C>T                          12242.0             32478.0  ...   \n",
       "22:18561162:C>T                          11972.0             32270.0  ...   \n",
       "22:18561164:T>A                          11994.0             32226.0  ...   \n",
       "...                                          ...                 ...  ...   \n",
       "15:83561573:G>C                          15462.0             34188.0  ...   \n",
       "15:83561577:T>G                          15448.0             33916.0  ...   \n",
       "15:83561579:A>G                           8712.0               848.0  ...   \n",
       "15:83561583:T>C                          15436.0             33812.0  ...   \n",
       "15:83561589:G>C                          15398.0             33296.0  ...   \n",
       "\n",
       "                                                                 Max MAF Source  \\\n",
       "Genomic Description (GRCh37)                                                      \n",
       "22:18561153:A>G                                              gnomad_AF_sas_pass   \n",
       "22:18561156:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561159:C>T                                              gnomad_AF_nfe_pass   \n",
       "22:18561162:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561164:T>A                                              gnomad_AF_nfe_pass   \n",
       "...                                                                         ...   \n",
       "15:83561573:G>C                                              gnomad_AF_nfe_pass   \n",
       "15:83561577:T>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561579:A>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561583:T>C               gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...   \n",
       "15:83561589:G>C                                              gnomad_AF_nfe_pass   \n",
       "\n",
       "                              Surface Area  Normalized SA  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "22:18561153:A>G                 111.511314       0.697034             45.18   \n",
       "22:18561156:C>T                 131.269338       0.951867             42.57   \n",
       "22:18561159:C>T                 114.498499       0.830258             42.35   \n",
       "22:18561162:C>T                 161.350503       1.025505             39.05   \n",
       "22:18561164:T>A                 118.519368       0.859414             42.94   \n",
       "...                                    ...            ...               ...   \n",
       "15:83561573:G>C                   5.626061       0.035758             98.35   \n",
       "15:83561577:T>G                  66.834983       0.424787             98.27   \n",
       "15:83561579:A>G                  46.255767       0.201331             97.67   \n",
       "15:83561583:T>C                  74.926502       0.381969             95.56   \n",
       "15:83561589:G>C                 135.808746       0.686764             85.69   \n",
       "\n",
       "                              ddG  |ddG|  in...  second part  calc...  ca...  \n",
       "Genomic Description (GRCh37)                                                  \n",
       "22:18561153:A>G              -0.7    0.7    NaN          NaN      NaN    NaN  \n",
       "22:18561156:C>T              -0.0    0.0    NaN          NaN      NaN    NaN  \n",
       "22:18561159:C>T              -0.3    0.3    NaN          NaN      NaN    NaN  \n",
       "22:18561162:C>T              -0.0    0.0    NaN          NaN      NaN    NaN  \n",
       "22:18561164:T>A              -0.0    0.0    NaN          NaN      NaN    NaN  \n",
       "...                           ...    ...    ...          ...      ...    ...  \n",
       "15:83561573:G>C              -0.9    0.9    NaN          NaN      NaN    NaN  \n",
       "15:83561577:T>G              -0.6    0.6    NaN          NaN      NaN    NaN  \n",
       "15:83561579:A>G              -2.6    2.6    NaN          NaN      NaN    NaN  \n",
       "15:83561583:T>C              -0.5    0.5    NaN          NaN      NaN    NaN  \n",
       "15:83561589:G>C              -0.3    0.3    NaN          NaN      NaN    NaN  \n",
       "\n",
       "[128786 rows x 236 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = []\n",
    "id_cols = ['CHROM', 'POS', 'REF', 'ALT']\n",
    "dup_cols = [f\"{base_col}_{alt}\" for base_col in id_cols for alt in ['x', 'y']]\n",
    "for f in os.scandir('chromosomes_w_ddg'):\n",
    "    df = pd.read_csv(f.path, index_col=0, encoding='utf-8')\n",
    "    df.drop(columns=dup_cols, inplace=True)\n",
    "    dfs.append(df)\n",
    "df = pd.concat(dfs).map(utils.safe_decode)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Drop multi-nucleotide variants (MNVS), remove TYPE column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TYPE</th>\n",
       "      <th>GENE</th>\n",
       "      <th>gnomad_MAX_MAF</th>\n",
       "      <th>gnomad_MAX_MAF_SOURCE</th>\n",
       "      <th>gnomad_AC_afr_pass</th>\n",
       "      <th>gnomad_AC_amr_pass</th>\n",
       "      <th>gnomad_AC_eas_pass</th>\n",
       "      <th>gnomad_AC_nfe_pass</th>\n",
       "      <th>gnomad_AN_afr_pass</th>\n",
       "      <th>gnomad_AN_amr_pass</th>\n",
       "      <th>...</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>Surface Area</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "      <th>in...</th>\n",
       "      <th>second part</th>\n",
       "      <th>calc...</th>\n",
       "      <th>ca...</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12624.0</td>\n",
       "      <td>32814.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>111.511314</td>\n",
       "      <td>0.697034</td>\n",
       "      <td>45.18</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>0.7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>12388.0</td>\n",
       "      <td>32614.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>131.269338</td>\n",
       "      <td>0.951867</td>\n",
       "      <td>42.57</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>12242.0</td>\n",
       "      <td>32478.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>114.498499</td>\n",
       "      <td>0.830258</td>\n",
       "      <td>42.35</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>0.000035</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11972.0</td>\n",
       "      <td>32270.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>161.350503</td>\n",
       "      <td>1.025505</td>\n",
       "      <td>39.05</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>0.000011</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11994.0</td>\n",
       "      <td>32226.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>118.519368</td>\n",
       "      <td>0.859414</td>\n",
       "      <td>42.94</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>SNV</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15462.0</td>\n",
       "      <td>34188.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>5.626061</td>\n",
       "      <td>0.035758</td>\n",
       "      <td>98.35</td>\n",
       "      <td>-0.9</td>\n",
       "      <td>0.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>SNV</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>0.000018</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>15448.0</td>\n",
       "      <td>33916.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>66.834983</td>\n",
       "      <td>0.424787</td>\n",
       "      <td>98.27</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>SNV</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>0.000130</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8712.0</td>\n",
       "      <td>848.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>46.255767</td>\n",
       "      <td>0.201331</td>\n",
       "      <td>97.67</td>\n",
       "      <td>-2.6</td>\n",
       "      <td>2.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>SNV</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15436.0</td>\n",
       "      <td>33812.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...</td>\n",
       "      <td>74.926502</td>\n",
       "      <td>0.381969</td>\n",
       "      <td>95.56</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>SNV</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>15398.0</td>\n",
       "      <td>33296.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>135.808746</td>\n",
       "      <td>0.686764</td>\n",
       "      <td>85.69</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 236 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             TYPE    GENE  gnomad_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                                \n",
       "22:18561153:A>G               SNV   PEX26        0.000035   \n",
       "22:18561156:C>T               SNV   PEX26        0.000035   \n",
       "22:18561159:C>T               SNV   PEX26        0.000010   \n",
       "22:18561162:C>T               SNV   PEX26        0.000035   \n",
       "22:18561164:T>A               SNV   PEX26        0.000011   \n",
       "...                           ...     ...             ...   \n",
       "15:83561573:G>C               SNV  HOMER2        0.000009   \n",
       "15:83561577:T>G               SNV  HOMER2        0.000018   \n",
       "15:83561579:A>G               SNV  HOMER2        0.000130   \n",
       "15:83561583:T>C               SNV  HOMER2        0.000000   \n",
       "15:83561589:G>C               SNV  HOMER2        0.000009   \n",
       "\n",
       "                                                          gnomad_MAX_MAF_SOURCE  \\\n",
       "Genomic Description (GRCh37)                                                      \n",
       "22:18561153:A>G                                              gnomad_AF_sas_pass   \n",
       "22:18561156:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561159:C>T                                              gnomad_AF_nfe_pass   \n",
       "22:18561162:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561164:T>A                                              gnomad_AF_nfe_pass   \n",
       "...                                                                         ...   \n",
       "15:83561573:G>C                                              gnomad_AF_nfe_pass   \n",
       "15:83561577:T>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561579:A>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561583:T>C               gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...   \n",
       "15:83561589:G>C                                              gnomad_AF_nfe_pass   \n",
       "\n",
       "                              gnomad_AC_afr_pass  gnomad_AC_amr_pass  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "22:18561153:A>G                              0.0                 0.0   \n",
       "22:18561156:C>T                              0.0                 0.0   \n",
       "22:18561159:C>T                              0.0                 0.0   \n",
       "22:18561162:C>T                              0.0                 0.0   \n",
       "22:18561164:T>A                              0.0                 0.0   \n",
       "...                                          ...                 ...   \n",
       "15:83561573:G>C                              0.0                 0.0   \n",
       "15:83561577:T>G                              0.0                 0.0   \n",
       "15:83561579:A>G                              0.0                 0.0   \n",
       "15:83561583:T>C                              0.0                 0.0   \n",
       "15:83561589:G>C                              0.0                 0.0   \n",
       "\n",
       "                              gnomad_AC_eas_pass  gnomad_AC_nfe_pass  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "22:18561153:A>G                              0.0                 0.0   \n",
       "22:18561156:C>T                              0.0                 0.0   \n",
       "22:18561159:C>T                              0.0                 1.0   \n",
       "22:18561162:C>T                              0.0                 0.0   \n",
       "22:18561164:T>A                              0.0                 1.0   \n",
       "...                                          ...                 ...   \n",
       "15:83561573:G>C                              0.0                 1.0   \n",
       "15:83561577:T>G                              0.0                 2.0   \n",
       "15:83561579:A>G                              0.0                 2.0   \n",
       "15:83561583:T>C                              0.0                 0.0   \n",
       "15:83561589:G>C                              0.0                 1.0   \n",
       "\n",
       "                              gnomad_AN_afr_pass  gnomad_AN_amr_pass  ...  \\\n",
       "Genomic Description (GRCh37)                                          ...   \n",
       "22:18561153:A>G                          12624.0             32814.0  ...   \n",
       "22:18561156:C>T                          12388.0             32614.0  ...   \n",
       "22:18561159:C>T                          12242.0             32478.0  ...   \n",
       "22:18561162:C>T                          11972.0             32270.0  ...   \n",
       "22:18561164:T>A                          11994.0             32226.0  ...   \n",
       "...                                          ...                 ...  ...   \n",
       "15:83561573:G>C                          15462.0             34188.0  ...   \n",
       "15:83561577:T>G                          15448.0             33916.0  ...   \n",
       "15:83561579:A>G                           8712.0               848.0  ...   \n",
       "15:83561583:T>C                          15436.0             33812.0  ...   \n",
       "15:83561589:G>C                          15398.0             33296.0  ...   \n",
       "\n",
       "                                                                 Max MAF Source  \\\n",
       "Genomic Description (GRCh37)                                                      \n",
       "22:18561153:A>G                                              gnomad_AF_sas_pass   \n",
       "22:18561156:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561159:C>T                                              gnomad_AF_nfe_pass   \n",
       "22:18561162:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561164:T>A                                              gnomad_AF_nfe_pass   \n",
       "...                                                                         ...   \n",
       "15:83561573:G>C                                              gnomad_AF_nfe_pass   \n",
       "15:83561577:T>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561579:A>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561583:T>C               gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...   \n",
       "15:83561589:G>C                                              gnomad_AF_nfe_pass   \n",
       "\n",
       "                              Surface Area  Normalized SA  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "22:18561153:A>G                 111.511314       0.697034             45.18   \n",
       "22:18561156:C>T                 131.269338       0.951867             42.57   \n",
       "22:18561159:C>T                 114.498499       0.830258             42.35   \n",
       "22:18561162:C>T                 161.350503       1.025505             39.05   \n",
       "22:18561164:T>A                 118.519368       0.859414             42.94   \n",
       "...                                    ...            ...               ...   \n",
       "15:83561573:G>C                   5.626061       0.035758             98.35   \n",
       "15:83561577:T>G                  66.834983       0.424787             98.27   \n",
       "15:83561579:A>G                  46.255767       0.201331             97.67   \n",
       "15:83561583:T>C                  74.926502       0.381969             95.56   \n",
       "15:83561589:G>C                 135.808746       0.686764             85.69   \n",
       "\n",
       "                              ddG  |ddG|  in...  second part  calc...  ca...  \n",
       "Genomic Description (GRCh37)                                                  \n",
       "22:18561153:A>G              -0.7    0.7    NaN          NaN      NaN    NaN  \n",
       "22:18561156:C>T              -0.0    0.0    NaN          NaN      NaN    NaN  \n",
       "22:18561159:C>T              -0.3    0.3    NaN          NaN      NaN    NaN  \n",
       "22:18561162:C>T              -0.0    0.0    NaN          NaN      NaN    NaN  \n",
       "22:18561164:T>A              -0.0    0.0    NaN          NaN      NaN    NaN  \n",
       "...                           ...    ...    ...          ...      ...    ...  \n",
       "15:83561573:G>C              -0.9    0.9    NaN          NaN      NaN    NaN  \n",
       "15:83561577:T>G              -0.6    0.6    NaN          NaN      NaN    NaN  \n",
       "15:83561579:A>G              -2.6    2.6    NaN          NaN      NaN    NaN  \n",
       "15:83561583:T>C              -0.5    0.5    NaN          NaN      NaN    NaN  \n",
       "15:83561589:G>C              -0.3    0.3    NaN          NaN      NaN    NaN  \n",
       "\n",
       "[128712 rows x 236 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.loc[df.TYPE != 'MNV', :]\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TYPE\n",
      "GENE\n",
      "gnomad_MAX_MAF\n",
      "gnomad_MAX_MAF_SOURCE\n",
      "gnomad_AC_afr_pass\n",
      "gnomad_AC_amr_pass\n",
      "gnomad_AC_eas_pass\n",
      "gnomad_AC_nfe_pass\n",
      "gnomad_AN_afr_pass\n",
      "gnomad_AN_amr_pass\n",
      "gnomad_AN_eas_pass\n",
      "gnomad_AN_nfe_pass\n",
      "gnomad_AF_afr_pass\n",
      "gnomad_AF_amr_pass\n",
      "gnomad_AF_eas_pass\n",
      "gnomad_AF_nfe_pass\n",
      "OVERALL_MAX_MAF\n",
      "OVERALL_MAX_MAF_SOURCE\n",
      "PREF_TRANSCRIPT\n",
      "VEP_CONSEQUENCE\n",
      "VEP_CANONICAL\n",
      "VEP_PICK\n",
      "VEP_IMPACT\n",
      "VEP_FEATURE\n",
      "VEP_CADD_RAW\n",
      "VEP_CADD_PHRED\n",
      "VEP_OTHER_FEATURES\n",
      "CLASSIFY_TRAIL\n",
      "CLASSIFY_TRAIL_KEYS\n",
      "FINAL_PATHOGENICITY\n",
      "FINAL_PMID\n",
      "FINAL_PMID_PRIMARY\n",
      "FINAL_COMMENTS\n",
      "FINAL_PATHOGENICITY_SOURCE\n",
      "FINAL_PATHOGENICITY_REASON\n",
      "FINAL_DISEASE\n",
      "CLINVAR_HGMD_CONFLICTED\n",
      "gnomad_AN\n",
      "gnomad_filters\n",
      "gnomad_AC_nfe_seu_pass\n",
      "gnomad_AC_fin_female_pass\n",
      "gnomad_AC_afr_male_pass\n",
      "gnomad_AC_eas_female_pass\n",
      "gnomad_AC_afr_female_pass\n",
      "gnomad_AC_nfe_onf_pass\n",
      "gnomad_AC_fin_male_pass\n",
      "gnomad_AC_nfe_female_pass\n",
      "gnomad_AC_asj_male_pass\n",
      "gnomad_AC_oth_female_pass\n",
      "gnomad_AC_nfe_nwe_pass\n",
      "gnomad_AC_nfe_est_pass\n",
      "gnomad_AC_eas_male_pass\n",
      "gnomad_AC_nfe_male_pass\n",
      "gnomad_AC_asj_female_pass\n",
      "gnomad_AC_amr_male_pass\n",
      "gnomad_AC_amr_female_pass\n",
      "gnomad_AC_oth_male_pass\n",
      "gnomad_AC_asj_pass\n",
      "gnomad_AC_fin_pass\n",
      "gnomad_AC_oth_pass\n",
      "gnomad_AN_nfe_seu_pass\n",
      "gnomad_AN_fin_female_pass\n",
      "gnomad_AN_afr_male_pass\n",
      "gnomad_AN_eas_female_pass\n",
      "gnomad_AN_afr_female_pass\n",
      "gnomad_AN_nfe_onf_pass\n",
      "gnomad_AN_fin_male_pass\n",
      "gnomad_AN_nfe_female_pass\n",
      "gnomad_AN_asj_male_pass\n",
      "gnomad_AN_oth_female_pass\n",
      "gnomad_AN_nfe_nwe_pass\n",
      "gnomad_AN_nfe_est_pass\n",
      "gnomad_AN_eas_male_pass\n",
      "gnomad_AN_nfe_male_pass\n",
      "gnomad_AN_asj_female_pass\n",
      "gnomad_AN_amr_male_pass\n",
      "gnomad_AN_amr_female_pass\n",
      "gnomad_AN_oth_male_pass\n",
      "gnomad_AN_asj_pass\n",
      "gnomad_AN_fin_pass\n",
      "gnomad_AN_oth_pass\n",
      "gnomad_nhomalt_nfe_seu_pass\n",
      "gnomad_nhomalt_fin_female_pass\n",
      "gnomad_nhomalt_afr_male_pass\n",
      "gnomad_nhomalt_afr_pass\n",
      "gnomad_nhomalt_eas_female_pass\n",
      "gnomad_nhomalt_afr_female_pass\n",
      "gnomad_nhomalt_nfe_onf_pass\n",
      "gnomad_nhomalt_fin_male_pass\n",
      "gnomad_nhomalt_nfe_female_pass\n",
      "gnomad_nhomalt_amr_pass\n",
      "gnomad_nhomalt_eas_pass\n",
      "gnomad_nhomalt_asj_male_pass\n",
      "gnomad_nhomalt_oth_female_pass\n",
      "gnomad_nhomalt_nfe_nwe_pass\n",
      "gnomad_nhomalt_nfe_est_pass\n",
      "gnomad_nhomalt_eas_male_pass\n",
      "gnomad_nhomalt_nfe_pass\n",
      "gnomad_nhomalt_fin_pass\n",
      "gnomad_nhomalt_nfe_male_pass\n",
      "gnomad_nhomalt_asj_female_pass\n",
      "gnomad_nhomalt_asj_pass\n",
      "gnomad_nhomalt_oth_pass\n",
      "gnomad_nhomalt_amr_male_pass\n",
      "gnomad_nhomalt_amr_female_pass\n",
      "gnomad_nhomalt_oth_male_pass\n",
      "gnomad_AF_nfe_seu_pass\n",
      "gnomad_AF_fin_female_pass\n",
      "gnomad_AF_afr_male_pass\n",
      "gnomad_AF_eas_female_pass\n",
      "gnomad_AF_afr_female_pass\n",
      "gnomad_AF_nfe_onf_pass\n",
      "gnomad_AF_fin_male_pass\n",
      "gnomad_AF_nfe_female_pass\n",
      "gnomad_AF_asj_male_pass\n",
      "gnomad_AF_oth_female_pass\n",
      "gnomad_AF_nfe_nwe_pass\n",
      "gnomad_AF_nfe_est_pass\n",
      "gnomad_AF_eas_male_pass\n",
      "gnomad_AF_nfe_male_pass\n",
      "gnomad_AF_asj_female_pass\n",
      "gnomad_AF_amr_male_pass\n",
      "gnomad_AF_amr_female_pass\n",
      "gnomad_AF_oth_male_pass\n",
      "gnomad_AF_asj_pass\n",
      "gnomad_AF_fin_pass\n",
      "gnomad_AF_oth_pass\n",
      "CLINVAR_VID\n",
      "CLINVAR_CLNSIG\n",
      "CLINVAR_REVSTAT\n",
      "CLINVAR_HGVS_C\n",
      "CLINVAR_HGVS_P\n",
      "CLINVAR_DISEASE\n",
      "CLINVAR_PMID\n",
      "CLINVAR_PATHOGENIC\n",
      "CLINVAR_MAX_MAF\n",
      "CLINVAR_MAX_MAF_SOURCE\n",
      "VEP_EXON\n",
      "VEP_HGVS_C\n",
      "gnomad_AC_sas_pass\n",
      "gnomad_AN_sas_pass\n",
      "gnomad_AF_sas_pass\n",
      "gnomad_AC_nfe_bgr_pass\n",
      "gnomad_AC_sas_male_pass\n",
      "gnomad_AC_nfe_swe_pass\n",
      "gnomad_AC_eas_jpn_pass\n",
      "gnomad_AC_eas_kor_pass\n",
      "gnomad_AC_eas_oea_pass\n",
      "gnomad_AC_sas_female_pass\n",
      "gnomad_AN_nfe_bgr_pass\n",
      "gnomad_AN_sas_male_pass\n",
      "gnomad_AN_nfe_swe_pass\n",
      "gnomad_AN_eas_jpn_pass\n",
      "gnomad_AN_eas_kor_pass\n",
      "gnomad_AN_eas_oea_pass\n",
      "gnomad_AN_sas_female_pass\n",
      "gnomad_nhomalt_nfe_bgr_pass\n",
      "gnomad_nhomalt_sas_male_pass\n",
      "gnomad_nhomalt_sas_pass\n",
      "gnomad_nhomalt_nfe_swe_pass\n",
      "gnomad_nhomalt_eas_jpn_pass\n",
      "gnomad_nhomalt_eas_kor_pass\n",
      "gnomad_nhomalt_eas_oea_pass\n",
      "gnomad_nhomalt_sas_female_pass\n",
      "gnomad_AF_nfe_bgr_pass\n",
      "gnomad_AF_sas_male_pass\n",
      "gnomad_AF_nfe_swe_pass\n",
      "gnomad_AF_eas_jpn_pass\n",
      "gnomad_AF_eas_kor_pass\n",
      "gnomad_AF_eas_oea_pass\n",
      "gnomad_AF_sas_female_pass\n",
      "VEP_HGVS_P\n",
      "VEP_SIFT_SCORE\n",
      "VEP_SIFT_PRED\n",
      "VEP_POLYPHEN_SCORE\n",
      "VEP_POLYPHEN_PRED\n",
      "VEP_PROTEIN_POS\n",
      "DBNSFP_SIFT_PRED\n",
      "DBNSFP_SIFT_SCORE\n",
      "DBNSFP_POLYPHEN2_HDIV_PRED\n",
      "DBNSFP_POLYPHEN2_HDIV_SCORE\n",
      "DBNSFP_LRT_PRED\n",
      "DBNSFP_LRT_SCORE\n",
      "DBNSFP_MUTATIONTASTER_PRED\n",
      "DBNSFP_MUTATIONTASTER_SCORE\n",
      "DBNSFP_GERP_RS\n",
      "DBNSFP_PHYLOP30WAY_MAMMALIAN\n",
      "DBNSFP_INTERPRO_DOMAIN\n",
      "DBNSFP_GERP_RS_PRED\n",
      "DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED\n",
      "NUM_PATH_PREDS\n",
      "TOTAL_NUM_PREDS\n",
      "FINAL_PRED\n",
      "CLINVAR_CONFLICTED\n",
      "HGMD_CONFIDENCE\n",
      "HGMD_DISEASE\n",
      "HGMD_HGVS\n",
      "HGMD_OMIM_REF\n",
      "HGMD_PMID\n",
      "HGMD_RSID\n",
      "HGMD_VARIANTTYPE\n",
      "HGMD_MAX_MAF\n",
      "HGMD_MAX_MAF_SOURCE\n",
      "CURATED_PATHOGENICITY\n",
      "CURATED_DISEASE\n",
      "CURATED_PMID\n",
      "CURATED_COMMENTS\n",
      "CURATED_DECISION_DATE\n",
      "CURATED_UPDATE_VERSION\n",
      "MORL_MAX_MAF\n",
      "MORL_MAX_MAF_SOURCE\n",
      "VEP_INTRON\n",
      "UNIPROT_DOMAIN\n",
      "UNIPROT_DOMAIN_SOURCE\n",
      "HGVS Nucleotide Change\n",
      "HGVS Protein Change\n",
      "Variant Classification\n",
      "Phenotype\n",
      "Consequence\n",
      "Exon\n",
      "Intron\n",
      "Protein Position\n",
      "Impact\n",
      "CADD Phred\n",
      "Interpro Domain\n",
      "Max MAF (%)\n",
      "Max MAF Source\n",
      "Surface Area\n",
      "Normalized SA\n",
      "Confidence Score\n",
      "ddG\n",
      "|ddG|\n",
      " in...\n",
      " second part\n",
      " calc...\n",
      " ca...\n"
     ]
    }
   ],
   "source": [
    "for x in df.columns:\n",
    "    print(x)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select usable columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>135</th>\n",
       "      <th>136</th>\n",
       "      <th>137</th>\n",
       "      <th>138</th>\n",
       "      <th>139</th>\n",
       "      <th>140</th>\n",
       "      <th>141</th>\n",
       "      <th>142</th>\n",
       "      <th>143</th>\n",
       "      <th>144</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Definitely Keep (used previously)</td>\n",
       "      <td>NUM_PATH_PREDS</td>\n",
       "      <td>TOTAL_NUM_PREDS</td>\n",
       "      <td>FINAL_PRED</td>\n",
       "      <td>GENE</td>\n",
       "      <td>OVERALL_MAX_MAF</td>\n",
       "      <td>OVERALL_MAX_MAF_SOURCE</td>\n",
       "      <td>FINAL_PATHOGENICITY</td>\n",
       "      <td>CURATED_PATHOGENICITY</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Definitely Keep (from ddG file)</td>\n",
       "      <td>Interpro Domain</td>\n",
       "      <td>Max MAF (%)</td>\n",
       "      <td>Max MAF Source</td>\n",
       "      <td>Surface Area</td>\n",
       "      <td>Normalized SA</td>\n",
       "      <td>Confidence Score</td>\n",
       "      <td>ddG</td>\n",
       "      <td>|ddG|</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Very Likely Keep</td>\n",
       "      <td>CLINVAR_PATHOGENIC</td>\n",
       "      <td>VEP_CADD_RAW</td>\n",
       "      <td>VEP_CADD_PHRED</td>\n",
       "      <td>VEP_SIFT_SCORE</td>\n",
       "      <td>VEP_POLYPHEN_SCORE</td>\n",
       "      <td>VEP_POLYPHEN_PRED</td>\n",
       "      <td>UNIPROT_DOMAIN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Likely Keep</td>\n",
       "      <td>CLINVAR_CLNSIG</td>\n",
       "      <td>VEP_CONSEQUENCE</td>\n",
       "      <td>VEP_IMPACT</td>\n",
       "      <td>VEP_FEATURE</td>\n",
       "      <td>DBNSFP_POLYPHEN2_HDIV_PRED</td>\n",
       "      <td>DBNSFP_POLYPHEN2_HDIV_SCORE</td>\n",
       "      <td>DBNSFP_LRT_PRED</td>\n",
       "      <td>DBNSFP_LRT_SCORE</td>\n",
       "      <td>DBNSFP_MUTATIONTASTER_PRED</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Somewhat Likely Keep</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Possible</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Unknown</td>\n",
       "      <td>VEP_PROTEIN_POS</td>\n",
       "      <td>DBNSFP_GERP_RS_PRED</td>\n",
       "      <td>DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Somewhat Likely Not Keep</td>\n",
       "      <td>CLINVAR_MAX_MAF_SOURCE</td>\n",
       "      <td>CLINVAR_HGMD_CONFLICTED</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Likely Not Keep</td>\n",
       "      <td>CLINVAR_VID</td>\n",
       "      <td>CLINVAR_REVSTAT</td>\n",
       "      <td>CLINVAR_DISEASE</td>\n",
       "      <td>CLINVAR_HGVS_C</td>\n",
       "      <td>CLINVAR_HGVS_P</td>\n",
       "      <td>CLINVAR_PMID</td>\n",
       "      <td>VEP_HGVS_P</td>\n",
       "      <td>VEP_HGVS_C</td>\n",
       "      <td>VEP_INTRON</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Very Likely Not Keep</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Kafeen or DVD Related (Not Keep)</td>\n",
       "      <td>CLASSIFY_TRAIL</td>\n",
       "      <td>CLASSIFY_TRAIL_KEYS</td>\n",
       "      <td>FINAL_PMID_PRIMARY</td>\n",
       "      <td>FINAL_PMID</td>\n",
       "      <td>FINAL_COMMENTS</td>\n",
       "      <td>FINAL_PATHOGENICITY_REASON</td>\n",
       "      <td>CURATED_DECISION_DATE</td>\n",
       "      <td>CURATED_PMID</td>\n",
       "      <td>CURATED_DISEASE</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Likely Not Keep (from ddG File)</td>\n",
       "      <td>Variant Classification</td>\n",
       "      <td>HGVS Nucleotide Change</td>\n",
       "      <td>HGVS Protein Change</td>\n",
       "      <td>Consequence</td>\n",
       "      <td>Phenotype</td>\n",
       "      <td>Exon</td>\n",
       "      <td>Intron</td>\n",
       "      <td>Protein Position</td>\n",
       "      <td>Impact</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>HGMD Columns (Not Keeping)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>gnomad Columns (Not Keeping)</td>\n",
       "      <td>gnomad_AN</td>\n",
       "      <td>gnomad_filters</td>\n",
       "      <td>gnomad_AC_nfe_seu_pass</td>\n",
       "      <td>gnomad_AC_fin_female_pass</td>\n",
       "      <td>gnomad_AC_afr_male_pass</td>\n",
       "      <td>gnomad_AC_eas_female_pass</td>\n",
       "      <td>gnomad_AC_afr_female_pass</td>\n",
       "      <td>gnomad_AC_nfe_onf_pass</td>\n",
       "      <td>gnomad_AC_fin_male_pass</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AN_eas_pass</td>\n",
       "      <td>gnomad_AN_nfe_pass</td>\n",
       "      <td>gnomad_AN_sas_pass</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>gnomad_AF_amr_pass</td>\n",
       "      <td>gnomad_AF_eas_pass</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>gnomad_MAX_MAF</td>\n",
       "      <td>gnomad_MAX_MAF_SOURCE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>DBNSFP Columns (duplicates - Not Keeping)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Keep to Validate Consistency of Duplicates</td>\n",
       "      <td>DBNSFP_SIFT_PRED</td>\n",
       "      <td>DBNSFP_SIFT_SCORE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Keep for Filtering</td>\n",
       "      <td>TYPE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Not Keeping (IDS)</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18 rows × 145 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                           0                       1    \\\n",
       "0            Definitely Keep (used previously)          NUM_PATH_PREDS   \n",
       "1              Definitely Keep (from ddG file)         Interpro Domain   \n",
       "2                             Very Likely Keep      CLINVAR_PATHOGENIC   \n",
       "3                                  Likely Keep          CLINVAR_CLNSIG   \n",
       "4                         Somewhat Likely Keep                     NaN   \n",
       "5                                     Possible                     NaN   \n",
       "6                                      Unknown         VEP_PROTEIN_POS   \n",
       "7                     Somewhat Likely Not Keep  CLINVAR_MAX_MAF_SOURCE   \n",
       "8                              Likely Not Keep             CLINVAR_VID   \n",
       "9                         Very Likely Not Keep                     NaN   \n",
       "10            Kafeen or DVD Related (Not Keep)          CLASSIFY_TRAIL   \n",
       "11             Likely Not Keep (from ddG File)  Variant Classification   \n",
       "12                  HGMD Columns (Not Keeping)                     NaN   \n",
       "13                gnomad Columns (Not Keeping)               gnomad_AN   \n",
       "14   DBNSFP Columns (duplicates - Not Keeping)                     NaN   \n",
       "15  Keep to Validate Consistency of Duplicates        DBNSFP_SIFT_PRED   \n",
       "16                          Keep for Filtering                    TYPE   \n",
       "17                           Not Keeping (IDS)                     NaN   \n",
       "\n",
       "                        2                                  3    \\\n",
       "0           TOTAL_NUM_PREDS                         FINAL_PRED   \n",
       "1               Max MAF (%)                     Max MAF Source   \n",
       "2              VEP_CADD_RAW                     VEP_CADD_PHRED   \n",
       "3           VEP_CONSEQUENCE                         VEP_IMPACT   \n",
       "4                       NaN                                NaN   \n",
       "5                       NaN                                NaN   \n",
       "6       DBNSFP_GERP_RS_PRED  DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED   \n",
       "7   CLINVAR_HGMD_CONFLICTED                                NaN   \n",
       "8           CLINVAR_REVSTAT                    CLINVAR_DISEASE   \n",
       "9                       NaN                                NaN   \n",
       "10      CLASSIFY_TRAIL_KEYS                 FINAL_PMID_PRIMARY   \n",
       "11   HGVS Nucleotide Change                HGVS Protein Change   \n",
       "12                      NaN                                NaN   \n",
       "13           gnomad_filters             gnomad_AC_nfe_seu_pass   \n",
       "14                      NaN                                NaN   \n",
       "15        DBNSFP_SIFT_SCORE                                NaN   \n",
       "16                      NaN                                NaN   \n",
       "17                      NaN                                NaN   \n",
       "\n",
       "                          4                           5    \\\n",
       "0                        GENE             OVERALL_MAX_MAF   \n",
       "1                Surface Area               Normalized SA   \n",
       "2              VEP_SIFT_SCORE          VEP_POLYPHEN_SCORE   \n",
       "3                 VEP_FEATURE  DBNSFP_POLYPHEN2_HDIV_PRED   \n",
       "4                         NaN                         NaN   \n",
       "5                         NaN                         NaN   \n",
       "6                         NaN                         NaN   \n",
       "7                         NaN                         NaN   \n",
       "8              CLINVAR_HGVS_C              CLINVAR_HGVS_P   \n",
       "9                         NaN                         NaN   \n",
       "10                 FINAL_PMID              FINAL_COMMENTS   \n",
       "11                Consequence                   Phenotype   \n",
       "12                        NaN                         NaN   \n",
       "13  gnomad_AC_fin_female_pass     gnomad_AC_afr_male_pass   \n",
       "14                        NaN                         NaN   \n",
       "15                        NaN                         NaN   \n",
       "16                        NaN                         NaN   \n",
       "17                        NaN                         NaN   \n",
       "\n",
       "                            6                          7    \\\n",
       "0        OVERALL_MAX_MAF_SOURCE        FINAL_PATHOGENICITY   \n",
       "1              Confidence Score                        ddG   \n",
       "2             VEP_POLYPHEN_PRED             UNIPROT_DOMAIN   \n",
       "3   DBNSFP_POLYPHEN2_HDIV_SCORE            DBNSFP_LRT_PRED   \n",
       "4                           NaN                        NaN   \n",
       "5                           NaN                        NaN   \n",
       "6                           NaN                        NaN   \n",
       "7                           NaN                        NaN   \n",
       "8                  CLINVAR_PMID                 VEP_HGVS_P   \n",
       "9                           NaN                        NaN   \n",
       "10   FINAL_PATHOGENICITY_REASON      CURATED_DECISION_DATE   \n",
       "11                         Exon                     Intron   \n",
       "12                          NaN                        NaN   \n",
       "13    gnomad_AC_eas_female_pass  gnomad_AC_afr_female_pass   \n",
       "14                          NaN                        NaN   \n",
       "15                          NaN                        NaN   \n",
       "16                          NaN                        NaN   \n",
       "17                          NaN                        NaN   \n",
       "\n",
       "                       8                           9    ...  \\\n",
       "0    CURATED_PATHOGENICITY                         NaN  ...   \n",
       "1                    |ddG|                         NaN  ...   \n",
       "2                      NaN                         NaN  ...   \n",
       "3         DBNSFP_LRT_SCORE  DBNSFP_MUTATIONTASTER_PRED  ...   \n",
       "4                      NaN                         NaN  ...   \n",
       "5                      NaN                         NaN  ...   \n",
       "6                      NaN                         NaN  ...   \n",
       "7                      NaN                         NaN  ...   \n",
       "8               VEP_HGVS_C                  VEP_INTRON  ...   \n",
       "9                      NaN                         NaN  ...   \n",
       "10            CURATED_PMID             CURATED_DISEASE  ...   \n",
       "11        Protein Position                      Impact  ...   \n",
       "12                     NaN                         NaN  ...   \n",
       "13  gnomad_AC_nfe_onf_pass     gnomad_AC_fin_male_pass  ...   \n",
       "14                     NaN                         NaN  ...   \n",
       "15                     NaN                         NaN  ...   \n",
       "16                     NaN                         NaN  ...   \n",
       "17                     NaN                         NaN  ...   \n",
       "\n",
       "                   135                 136                 137  \\\n",
       "0                  NaN                 NaN                 NaN   \n",
       "1                  NaN                 NaN                 NaN   \n",
       "2                  NaN                 NaN                 NaN   \n",
       "3                  NaN                 NaN                 NaN   \n",
       "4                  NaN                 NaN                 NaN   \n",
       "5                  NaN                 NaN                 NaN   \n",
       "6                  NaN                 NaN                 NaN   \n",
       "7                  NaN                 NaN                 NaN   \n",
       "8                  NaN                 NaN                 NaN   \n",
       "9                  NaN                 NaN                 NaN   \n",
       "10                 NaN                 NaN                 NaN   \n",
       "11                 NaN                 NaN                 NaN   \n",
       "12                 NaN                 NaN                 NaN   \n",
       "13  gnomad_AN_eas_pass  gnomad_AN_nfe_pass  gnomad_AN_sas_pass   \n",
       "14                 NaN                 NaN                 NaN   \n",
       "15                 NaN                 NaN                 NaN   \n",
       "16                 NaN                 NaN                 NaN   \n",
       "17                 NaN                 NaN                 NaN   \n",
       "\n",
       "                   138                 139                 140  \\\n",
       "0                  NaN                 NaN                 NaN   \n",
       "1                  NaN                 NaN                 NaN   \n",
       "2                  NaN                 NaN                 NaN   \n",
       "3                  NaN                 NaN                 NaN   \n",
       "4                  NaN                 NaN                 NaN   \n",
       "5                  NaN                 NaN                 NaN   \n",
       "6                  NaN                 NaN                 NaN   \n",
       "7                  NaN                 NaN                 NaN   \n",
       "8                  NaN                 NaN                 NaN   \n",
       "9                  NaN                 NaN                 NaN   \n",
       "10                 NaN                 NaN                 NaN   \n",
       "11                 NaN                 NaN                 NaN   \n",
       "12                 NaN                 NaN                 NaN   \n",
       "13  gnomad_AF_afr_pass  gnomad_AF_amr_pass  gnomad_AF_eas_pass   \n",
       "14                 NaN                 NaN                 NaN   \n",
       "15                 NaN                 NaN                 NaN   \n",
       "16                 NaN                 NaN                 NaN   \n",
       "17                 NaN                 NaN                 NaN   \n",
       "\n",
       "                   141                 142             143  \\\n",
       "0                  NaN                 NaN             NaN   \n",
       "1                  NaN                 NaN             NaN   \n",
       "2                  NaN                 NaN             NaN   \n",
       "3                  NaN                 NaN             NaN   \n",
       "4                  NaN                 NaN             NaN   \n",
       "5                  NaN                 NaN             NaN   \n",
       "6                  NaN                 NaN             NaN   \n",
       "7                  NaN                 NaN             NaN   \n",
       "8                  NaN                 NaN             NaN   \n",
       "9                  NaN                 NaN             NaN   \n",
       "10                 NaN                 NaN             NaN   \n",
       "11                 NaN                 NaN             NaN   \n",
       "12                 NaN                 NaN             NaN   \n",
       "13  gnomad_AF_nfe_pass  gnomad_AF_sas_pass  gnomad_MAX_MAF   \n",
       "14                 NaN                 NaN             NaN   \n",
       "15                 NaN                 NaN             NaN   \n",
       "16                 NaN                 NaN             NaN   \n",
       "17                 NaN                 NaN             NaN   \n",
       "\n",
       "                      144  \n",
       "0                     NaN  \n",
       "1                     NaN  \n",
       "2                     NaN  \n",
       "3                     NaN  \n",
       "4                     NaN  \n",
       "5                     NaN  \n",
       "6                     NaN  \n",
       "7                     NaN  \n",
       "8                     NaN  \n",
       "9                     NaN  \n",
       "10                    NaN  \n",
       "11                    NaN  \n",
       "12                    NaN  \n",
       "13  gnomad_MAX_MAF_SOURCE  \n",
       "14                    NaN  \n",
       "15                    NaN  \n",
       "16                    NaN  \n",
       "17                    NaN  \n",
       "\n",
       "[18 rows x 145 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "features_df = pd.read_csv('feature_sorting_updated.csv', header=None)\n",
    "features_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th>HGVS Nucleotide Change</th>\n",
       "      <th>HGVS Protein Change</th>\n",
       "      <th>Variant Classification</th>\n",
       "      <th>Phenotype</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Exon</th>\n",
       "      <th>Intron</th>\n",
       "      <th>Protein Position</th>\n",
       "      <th>Impact</th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>Interpro Domain</th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>Surface Area</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>20:25275622:G&gt;A</td>\n",
       "      <td>NM_015600.5:c.1202C&gt;T</td>\n",
       "      <td>NP_056415.1:p.Ser401Leu</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>13/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>401</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>0.986</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005012</td>\n",
       "      <td>gnomad_AF_eas_pass</td>\n",
       "      <td>83.106924</td>\n",
       "      <td>0.602629</td>\n",
       "      <td>37.05</td>\n",
       "      <td>0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>20:25275628:A&gt;C</td>\n",
       "      <td>NM_015600.5:c.1196T&gt;G</td>\n",
       "      <td>NP_056415.1:p.Met399Arg</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>13/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>399</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>0.549</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000879</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>95.022767</td>\n",
       "      <td>0.438837</td>\n",
       "      <td>32.38</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>20:25275631:C&gt;T</td>\n",
       "      <td>NM_015600.5:c.1193G&gt;A</td>\n",
       "      <td>NP_056415.1:p.Ser398Asn</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>13/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>398</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>12.240</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002891</td>\n",
       "      <td>gnomad_AF_amr_pass</td>\n",
       "      <td>66.517272</td>\n",
       "      <td>0.482334</td>\n",
       "      <td>37.35</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>20:25275632:T&gt;A</td>\n",
       "      <td>NM_015600.5:c.1192A&gt;T</td>\n",
       "      <td>NP_056415.1:p.Ser398Cys</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>13/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>398</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>12.590</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000879</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>66.517272</td>\n",
       "      <td>0.482334</td>\n",
       "      <td>37.35</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>20:25275642:G&gt;C</td>\n",
       "      <td>NM_015600.5:c.1182C&gt;G</td>\n",
       "      <td>NP_056415.1:p.Ser394Arg</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>13/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>394</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>2.456</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002637</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>107.954090</td>\n",
       "      <td>0.782802</td>\n",
       "      <td>31.97</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>300</th>\n",
       "      <td>20:25371324:C&gt;G</td>\n",
       "      <td>NM_001042472.3:c.16G&gt;C</td>\n",
       "      <td>NP_001035937.1:p.Glu6Gln</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>17.790</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.013271</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>157.694735</td>\n",
       "      <td>0.807569</td>\n",
       "      <td>32.75</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>301</th>\n",
       "      <td>20:25371327:T&gt;C</td>\n",
       "      <td>NM_001042472.3:c.13A&gt;G</td>\n",
       "      <td>NP_001035937.1:p.Thr5Ala</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>18.340</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006694</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>103.676396</td>\n",
       "      <td>0.658942</td>\n",
       "      <td>35.66</td>\n",
       "      <td>-0.4</td>\n",
       "      <td>0.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>302</th>\n",
       "      <td>20:25371329:C&gt;T</td>\n",
       "      <td>NM_001042472.3:c.11G&gt;A</td>\n",
       "      <td>NP_001035937.1:p.Arg4Gln</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>31.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>236.116856</td>\n",
       "      <td>0.876371</td>\n",
       "      <td>33.58</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>303</th>\n",
       "      <td>20:25371330:G&gt;A</td>\n",
       "      <td>NM_001042472.3:c.10C&gt;T</td>\n",
       "      <td>NP_001035937.1:p.Arg4Trp</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>31.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>236.116856</td>\n",
       "      <td>0.876371</td>\n",
       "      <td>33.58</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>20:25371331:C&gt;G</td>\n",
       "      <td>NM_001042472.3:c.9G&gt;C</td>\n",
       "      <td>NP_001035937.1:p.Lys3Asn</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1/13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>24.600</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006749</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>205.001994</td>\n",
       "      <td>0.866030</td>\n",
       "      <td>33.61</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>305 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Genomic Description (GRCh37)  HGVS Nucleotide Change  \\\n",
       "0                20:25275622:G>A   NM_015600.5:c.1202C>T   \n",
       "1                20:25275628:A>C   NM_015600.5:c.1196T>G   \n",
       "2                20:25275631:C>T   NM_015600.5:c.1193G>A   \n",
       "3                20:25275632:T>A   NM_015600.5:c.1192A>T   \n",
       "4                20:25275642:G>C   NM_015600.5:c.1182C>G   \n",
       "..                           ...                     ...   \n",
       "300              20:25371324:C>G  NM_001042472.3:c.16G>C   \n",
       "301              20:25371327:T>C  NM_001042472.3:c.13A>G   \n",
       "302              20:25371329:C>T  NM_001042472.3:c.11G>A   \n",
       "303              20:25371330:G>A  NM_001042472.3:c.10C>T   \n",
       "304              20:25371331:C>G   NM_001042472.3:c.9G>C   \n",
       "\n",
       "          HGVS Protein Change Variant Classification Phenotype  \\\n",
       "0     NP_056415.1:p.Ser401Leu          Likely benign       NaN   \n",
       "1     NP_056415.1:p.Met399Arg          Likely benign       NaN   \n",
       "2     NP_056415.1:p.Ser398Asn          Likely benign       NaN   \n",
       "3     NP_056415.1:p.Ser398Cys          Likely benign       NaN   \n",
       "4     NP_056415.1:p.Ser394Arg          Likely benign       NaN   \n",
       "..                        ...                    ...       ...   \n",
       "300  NP_001035937.1:p.Glu6Gln          Likely benign       NaN   \n",
       "301  NP_001035937.1:p.Thr5Ala          Likely benign       NaN   \n",
       "302  NP_001035937.1:p.Arg4Gln   Unknown significance       NaN   \n",
       "303  NP_001035937.1:p.Arg4Trp   Unknown significance       NaN   \n",
       "304  NP_001035937.1:p.Lys3Asn   Unknown significance       NaN   \n",
       "\n",
       "          Consequence   Exon  Intron  Protein Position    Impact  CADD Phred  \\\n",
       "0    missense_variant  13/13     NaN               401  MODERATE       0.986   \n",
       "1    missense_variant  13/13     NaN               399  MODERATE       0.549   \n",
       "2    missense_variant  13/13     NaN               398  MODERATE      12.240   \n",
       "3    missense_variant  13/13     NaN               398  MODERATE      12.590   \n",
       "4    missense_variant  13/13     NaN               394  MODERATE       2.456   \n",
       "..                ...    ...     ...               ...       ...         ...   \n",
       "300  missense_variant   1/13     NaN                 6  MODERATE      17.790   \n",
       "301  missense_variant   1/13     NaN                 5  MODERATE      18.340   \n",
       "302  missense_variant   1/13     NaN                 4  MODERATE      31.000   \n",
       "303  missense_variant   1/13     NaN                 4  MODERATE      31.000   \n",
       "304  missense_variant   1/13     NaN                 3  MODERATE      24.600   \n",
       "\n",
       "    Interpro Domain  Max MAF (%)      Max MAF Source  Surface Area  \\\n",
       "0               NaN     0.005012  gnomad_AF_eas_pass     83.106924   \n",
       "1               NaN     0.000879  gnomad_AF_nfe_pass     95.022767   \n",
       "2               NaN     0.002891  gnomad_AF_amr_pass     66.517272   \n",
       "3               NaN     0.000879  gnomad_AF_nfe_pass     66.517272   \n",
       "4               NaN     0.002637  gnomad_AF_nfe_pass    107.954090   \n",
       "..              ...          ...                 ...           ...   \n",
       "300             NaN     0.013271  gnomad_AF_nfe_pass    157.694735   \n",
       "301             NaN     0.006694  gnomad_AF_sas_pass    103.676396   \n",
       "302             NaN     0.000000                 NaN    236.116856   \n",
       "303             NaN     0.000000                 NaN    236.116856   \n",
       "304             NaN     0.006749  gnomad_AF_sas_pass    205.001994   \n",
       "\n",
       "     Normalized SA  Confidence Score  ddG  |ddG|  \n",
       "0         0.602629             37.05  0.6    0.6  \n",
       "1         0.438837             32.38 -0.5    0.5  \n",
       "2         0.482334             37.35 -0.1    0.1  \n",
       "3         0.482334             37.35 -0.1    0.1  \n",
       "4         0.782802             31.97  0.2    0.2  \n",
       "..             ...               ...  ...    ...  \n",
       "300       0.807569             32.75 -0.1    0.1  \n",
       "301       0.658942             35.66 -0.4    0.4  \n",
       "302       0.876371             33.58 -0.2    0.2  \n",
       "303       0.876371             33.58 -0.2    0.2  \n",
       "304       0.866030             33.61  0.1    0.1  \n",
       "\n",
       "[305 rows x 19 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_df = pd.read_csv('unzipped/featureMappedCsvs/update_ABHD12.csv')\n",
    "gene_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       0.986\n",
       "1       0.549\n",
       "2      12.240\n",
       "3      12.590\n",
       "4       2.456\n",
       "        ...  \n",
       "300    17.790\n",
       "301    18.340\n",
       "302    31.000\n",
       "303    31.000\n",
       "304    24.600\n",
       "Name: CADD Phred, Length: 305, dtype: float64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_df['CADD Phred']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:25275622:G&gt;A</th>\n",
       "      <td>0.986</td>\n",
       "      <td>0.986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:25275628:A&gt;C</th>\n",
       "      <td>0.549</td>\n",
       "      <td>0.549</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:25275631:C&gt;T</th>\n",
       "      <td>12.240</td>\n",
       "      <td>12.240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:25275632:T&gt;A</th>\n",
       "      <td>12.590</td>\n",
       "      <td>12.590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:25275642:G&gt;C</th>\n",
       "      <td>2.456</td>\n",
       "      <td>2.456</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:25371324:C&gt;G</th>\n",
       "      <td>17.790</td>\n",
       "      <td>17.790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:25371327:T&gt;C</th>\n",
       "      <td>18.340</td>\n",
       "      <td>18.340</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:25371329:C&gt;T</th>\n",
       "      <td>31.000</td>\n",
       "      <td>31.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:25371330:G&gt;A</th>\n",
       "      <td>31.000</td>\n",
       "      <td>31.000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:25371331:C&gt;G</th>\n",
       "      <td>24.600</td>\n",
       "      <td>24.600</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>305 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD Phred  VEP_CADD_PHRED\n",
       "Genomic Description (GRCh37)                            \n",
       "20:25275622:G>A                    0.986           0.986\n",
       "20:25275628:A>C                    0.549           0.549\n",
       "20:25275631:C>T                   12.240          12.240\n",
       "20:25275632:T>A                   12.590          12.590\n",
       "20:25275642:G>C                    2.456           2.456\n",
       "...                                  ...             ...\n",
       "20:25371324:C>G                   17.790          17.790\n",
       "20:25371327:T>C                   18.340          18.340\n",
       "20:25371329:C>T                   31.000          31.000\n",
       "20:25371330:G>A                   31.000          31.000\n",
       "20:25371331:C>G                   24.600          24.600\n",
       "\n",
       "[305 rows x 2 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_impact_df = df.loc[df.GENE == 'ABHD12', ['CADD Phred', 'VEP_CADD_PHRED']]\n",
    "gene_impact_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [CADD Phred, VEP_CADD_PHRED]\n",
       "Index: []"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_impact_df[gene_impact_df['CADD Phred'] != gene_impact_df.VEP_CADD_PHRED]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Columns to Drop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "cols_to_drop = []"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['FINAL_PMID',\n",
       " 'FINAL_PMID_PRIMARY',\n",
       " 'CLINVAR_VID',\n",
       " 'CLINVAR_PMID',\n",
       " 'HGMD_PMID',\n",
       " 'HGMD_RSID',\n",
       " 'CURATED_PMID']"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "id_cols = df.columns[df.columns.str.contains('ID') & ~df.columns.str.contains('CONFIDENCE')].tolist()\n",
    "cols_to_drop += id_cols\n",
    "id_cols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### gnomad Filters"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "gnomad_filter_cols = df.columns[df.columns.str.contains('gnomad') & df.columns.str.contains('pass')].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['gnomad_MAX_MAF', 'gnomad_MAX_MAF_SOURCE', 'gnomad_AN',\n",
       "       'gnomad_filters'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns[df.columns.str.contains('gnomad') & ~df.columns.str.contains('pass')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['gnomad_exome_PASS', 'gnomad_exome_PASS|gnomad_genome_PASS',\n",
       "       'gnomad_exome_AC0|gnomad_exome_RF', 'gnomad_exome_AC0',\n",
       "       'gnomad_exome_RF', 'gnomad_genome_PASS', nan,\n",
       "       'gnomad_exome_PASS|gnomad_genome_RF', 'gnomad_genome_RF',\n",
       "       'gnomad_exome_AC0|gnomad_exome_RF|gnomad_genome_PASS',\n",
       "       'gnomad_exome_RF|gnomad_genome_PASS',\n",
       "       'gnomad_exome_AC0|gnomad_genome_PASS',\n",
       "       'gnomad_exome_RF|gnomad_genome_RF',\n",
       "       'gnomad_exome_PASS|gnomad_genome_AC0|gnomad_genome_RF',\n",
       "       'gnomad_exome_AC0|gnomad_exome_RF|gnomad_genome_RF',\n",
       "       'gnomad_genome_AC0|gnomad_genome_RF',\n",
       "       'gnomad_exome_AC0|gnomad_exome_RF|gnomad_genome_AC0|gnomad_genome_RF',\n",
       "       'gnomad_exome_AC0|gnomad_genome_RF',\n",
       "       'gnomad_genome_PASS|gnomad_exome_PASS',\n",
       "       'gnomad_genome_RF|gnomad_exome_PASS',\n",
       "       'gnomad_genome_PASS|gnomad_exome_AC0|gnomad_exome_RF',\n",
       "       'gnomad_genome_PASS|gnomad_exome_AC0',\n",
       "       'gnomad_genome_RF|gnomad_exome_RF',\n",
       "       'gnomad_genome_RF|gnomad_exome_AC0|gnomad_exome_RF',\n",
       "       'gnomad_genome_AC0|gnomad_genome_RF|gnomad_exome_AC0|gnomad_exome_RF',\n",
       "       'gnomad_genome_PASS|gnomad_exome_RF',\n",
       "       'gnomad_genome_AC0|gnomad_exome_PASS',\n",
       "       'gnomad_exome_PASS|gnomad_genome_AC0', 'gnomad_genome_AC0',\n",
       "       'gnomad_genome_AC0|gnomad_genome_RF|gnomad_exome_PASS',\n",
       "       'gnomad_genome_RF|gnomad_exome_AC0',\n",
       "       'gnomad_genome_AC0|gnomad_genome_RF|gnomad_exome_RF',\n",
       "       'gnomad_exome_RF|gnomad_genome_AC0|gnomad_genome_RF',\n",
       "       'gnomad_exome_RF|gnomad_genome_AC0',\n",
       "       'gnomad_genome_InbreedingCoeff|gnomad_genome_RF|gnomad_exome_RF',\n",
       "       'gnomad_genome_AC0|gnomad_genome_RF|gnomad_exome_AC0',\n",
       "       'gnomad_genome_AC0|gnomad_exome_RF',\n",
       "       'gnomad_genome_AC0|gnomad_exome_AC0'], dtype=object)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['gnomad_filters'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "gnomad_filter_cols.append('gnomad_filters')\n",
    "cols_to_drop += gnomad_filter_cols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### MAX MAFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "max_maf_cols = df.columns[df.columns.str.contains('MAF')].tolist()\n",
    "cols_to_drop += max_maf_cols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Errant Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "errant_cols = [' second part', ' in...', ' ca...', ' calc...']\n",
    "cols_to_drop += errant_cols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### DVD/Kafeen Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "dvd_kafeen_cols = ['CLASSIFY_TRAIL', 'CLASSIFY_TRAIL_KEYS', 'FINAL_COMMENTS', 'FINAL_PATHOGENICITY_REASON', 'CURATED_DECISION_DATE', 'PREF_TRANSCRIPT']\n",
    "cols_to_drop += dvd_kafeen_cols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### Comment Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "comment_cols = ['FINAL_COMMENTS', 'CURATED_COMMENTS']\n",
    "cols_to_drop += comment_cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['TYPE', 'GENE', 'VEP_CONSEQUENCE', 'VEP_CANONICAL', 'VEP_PICK',\n",
       "       'VEP_IMPACT', 'VEP_FEATURE', 'VEP_CADD_RAW', 'VEP_CADD_PHRED',\n",
       "       'VEP_OTHER_FEATURES', 'FINAL_PATHOGENICITY',\n",
       "       'FINAL_PATHOGENICITY_SOURCE', 'FINAL_DISEASE',\n",
       "       'CLINVAR_HGMD_CONFLICTED', 'gnomad_AN', 'CLINVAR_CLNSIG',\n",
       "       'CLINVAR_REVSTAT', 'CLINVAR_HGVS_C', 'CLINVAR_HGVS_P',\n",
       "       'CLINVAR_DISEASE', 'CLINVAR_PATHOGENIC', 'VEP_EXON', 'VEP_HGVS_C',\n",
       "       'VEP_HGVS_P', 'VEP_SIFT_SCORE', 'VEP_SIFT_PRED', 'VEP_POLYPHEN_SCORE',\n",
       "       'VEP_POLYPHEN_PRED', 'VEP_PROTEIN_POS', 'DBNSFP_SIFT_PRED',\n",
       "       'DBNSFP_SIFT_SCORE', 'DBNSFP_POLYPHEN2_HDIV_PRED',\n",
       "       'DBNSFP_POLYPHEN2_HDIV_SCORE', 'DBNSFP_LRT_PRED', 'DBNSFP_LRT_SCORE',\n",
       "       'DBNSFP_MUTATIONTASTER_PRED', 'DBNSFP_MUTATIONTASTER_SCORE',\n",
       "       'DBNSFP_GERP_RS', 'DBNSFP_PHYLOP30WAY_MAMMALIAN',\n",
       "       'DBNSFP_INTERPRO_DOMAIN', 'DBNSFP_GERP_RS_PRED',\n",
       "       'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED', 'NUM_PATH_PREDS',\n",
       "       'TOTAL_NUM_PREDS', 'FINAL_PRED', 'CLINVAR_CONFLICTED',\n",
       "       'HGMD_CONFIDENCE', 'HGMD_DISEASE', 'HGMD_HGVS', 'HGMD_OMIM_REF',\n",
       "       'HGMD_VARIANTTYPE', 'CURATED_PATHOGENICITY', 'CURATED_DISEASE',\n",
       "       'CURATED_UPDATE_VERSION', 'VEP_INTRON', 'UNIPROT_DOMAIN',\n",
       "       'UNIPROT_DOMAIN_SOURCE', 'HGVS Nucleotide Change',\n",
       "       'HGVS Protein Change', 'Variant Classification', 'Phenotype',\n",
       "       'Consequence', 'Exon', 'Intron', 'Protein Position', 'Impact',\n",
       "       'CADD Phred', 'Interpro Domain', 'Surface Area', 'Normalized SA',\n",
       "       'Confidence Score', 'ddG', '|ddG|'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.drop(columns=cols_to_drop).columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "genomic_protein_loc_cols = ['VEP_HGVS_P', 'VEP_HGVS_C', 'CLINVAR_HGVS_P', 'CLINVAR_HGVS_C', ]\n",
    "gene_description_cols = ['HGMD_OMIM_REF']\n",
    "\n",
    "cols_to_drop += genomic_protein_loc_cols + gene_description_cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['TYPE', 'GENE', 'VEP_CONSEQUENCE', 'VEP_CANONICAL', 'VEP_PICK',\n",
       "       'VEP_IMPACT', 'VEP_FEATURE', 'VEP_CADD_RAW', 'VEP_CADD_PHRED',\n",
       "       'VEP_OTHER_FEATURES', 'FINAL_PATHOGENICITY',\n",
       "       'FINAL_PATHOGENICITY_SOURCE', 'FINAL_DISEASE',\n",
       "       'CLINVAR_HGMD_CONFLICTED', 'gnomad_AN', 'CLINVAR_CLNSIG',\n",
       "       'CLINVAR_REVSTAT', 'CLINVAR_DISEASE', 'CLINVAR_PATHOGENIC', 'VEP_EXON',\n",
       "       'VEP_SIFT_SCORE', 'VEP_SIFT_PRED', 'VEP_POLYPHEN_SCORE',\n",
       "       'VEP_POLYPHEN_PRED', 'VEP_PROTEIN_POS', 'DBNSFP_SIFT_PRED',\n",
       "       'DBNSFP_SIFT_SCORE', 'DBNSFP_POLYPHEN2_HDIV_PRED',\n",
       "       'DBNSFP_POLYPHEN2_HDIV_SCORE', 'DBNSFP_LRT_PRED', 'DBNSFP_LRT_SCORE',\n",
       "       'DBNSFP_MUTATIONTASTER_PRED', 'DBNSFP_MUTATIONTASTER_SCORE',\n",
       "       'DBNSFP_GERP_RS', 'DBNSFP_PHYLOP30WAY_MAMMALIAN',\n",
       "       'DBNSFP_INTERPRO_DOMAIN', 'DBNSFP_GERP_RS_PRED',\n",
       "       'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED', 'NUM_PATH_PREDS',\n",
       "       'TOTAL_NUM_PREDS', 'FINAL_PRED', 'CLINVAR_CONFLICTED',\n",
       "       'HGMD_CONFIDENCE', 'HGMD_DISEASE', 'HGMD_HGVS', 'HGMD_VARIANTTYPE',\n",
       "       'CURATED_PATHOGENICITY', 'CURATED_DISEASE', 'CURATED_UPDATE_VERSION',\n",
       "       'VEP_INTRON', 'UNIPROT_DOMAIN', 'UNIPROT_DOMAIN_SOURCE',\n",
       "       'HGVS Nucleotide Change', 'HGVS Protein Change',\n",
       "       'Variant Classification', 'Phenotype', 'Consequence', 'Exon', 'Intron',\n",
       "       'Protein Position', 'Impact', 'CADD Phred', 'Interpro Domain',\n",
       "       'Surface Area', 'Normalized SA', 'Confidence Score', 'ddG', '|ddG|'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.drop(columns=cols_to_drop).columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['TYPE', 'GENE', 'VEP_CONSEQUENCE', 'VEP_CANONICAL', 'VEP_PICK',\n",
       "       'VEP_IMPACT', 'VEP_FEATURE', 'VEP_CADD_RAW', 'VEP_CADD_PHRED',\n",
       "       'FINAL_PATHOGENICITY', 'FINAL_PATHOGENICITY_SOURCE', 'FINAL_DISEASE',\n",
       "       'CLINVAR_HGMD_CONFLICTED', 'gnomad_AN', 'CLINVAR_CLNSIG',\n",
       "       'CLINVAR_REVSTAT', 'CLINVAR_DISEASE', 'CLINVAR_PATHOGENIC', 'VEP_EXON',\n",
       "       'VEP_SIFT_SCORE', 'VEP_SIFT_PRED', 'VEP_POLYPHEN_SCORE',\n",
       "       'VEP_POLYPHEN_PRED', 'VEP_PROTEIN_POS', 'DBNSFP_SIFT_PRED',\n",
       "       'DBNSFP_SIFT_SCORE', 'DBNSFP_POLYPHEN2_HDIV_PRED',\n",
       "       'DBNSFP_POLYPHEN2_HDIV_SCORE', 'DBNSFP_LRT_PRED', 'DBNSFP_LRT_SCORE',\n",
       "       'DBNSFP_MUTATIONTASTER_PRED', 'DBNSFP_MUTATIONTASTER_SCORE',\n",
       "       'DBNSFP_GERP_RS', 'DBNSFP_PHYLOP30WAY_MAMMALIAN',\n",
       "       'DBNSFP_INTERPRO_DOMAIN', 'DBNSFP_GERP_RS_PRED',\n",
       "       'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED', 'NUM_PATH_PREDS',\n",
       "       'TOTAL_NUM_PREDS', 'FINAL_PRED', 'CLINVAR_CONFLICTED',\n",
       "       'HGMD_CONFIDENCE', 'HGMD_DISEASE', 'HGMD_HGVS', 'HGMD_VARIANTTYPE',\n",
       "       'CURATED_PATHOGENICITY', 'CURATED_DISEASE', 'CURATED_UPDATE_VERSION',\n",
       "       'VEP_INTRON', 'UNIPROT_DOMAIN', 'UNIPROT_DOMAIN_SOURCE',\n",
       "       'HGVS Nucleotide Change', 'HGVS Protein Change',\n",
       "       'Variant Classification', 'Phenotype', 'Consequence', 'Exon', 'Intron',\n",
       "       'Protein Position', 'Impact', 'CADD Phred', 'Interpro Domain',\n",
       "       'Surface Area', 'Normalized SA', 'Confidence Score', 'ddG', '|ddG|'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.drop(columns=cols_to_drop + ['VEP_OTHER_FEATURES']).columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "keep_cols = ['NUM_PATH_PREDS',\n",
    " 'TOTAL_NUM_PREDS',\n",
    " 'FINAL_PRED',\n",
    " 'GENE',\n",
    " 'OVERALL_MAX_MAF',\n",
    " 'OVERALL_MAX_MAF_SOURCE',\n",
    " 'FINAL_PATHOGENICITY',\n",
    " 'CURATED_PATHOGENICITY']\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "feature_sorting_df = pd.read_csv('feature_sorting_updated.csv', header=None, index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['NUM_PATH_PREDS',\n",
       " 'TOTAL_NUM_PREDS',\n",
       " 'FINAL_PRED',\n",
       " 'GENE',\n",
       " 'OVERALL_MAX_MAF',\n",
       " 'OVERALL_MAX_MAF_SOURCE',\n",
       " 'FINAL_PATHOGENICITY',\n",
       " 'CURATED_PATHOGENICITY']"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_sorting_df.loc['Definitely Keep (used previously)'].dropna().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "kept_features_df = feature_sorting_df.loc[~feature_sorting_df.index.str.contains('Not Keep'), :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "42"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kept_features_list = sorted(kept_features_df.unstack().dropna().unique().tolist())\n",
    "len(kept_features_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "pathogenicity_label_cols = ['CLINVAR_CLNSIG', 'CLINVAR_PATHOGENIC', 'CURATED_PATHOGENICITY']\n",
    "misc = ['VEP_FEATURE']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Selecting Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_CONSEQUENCE</th>\n",
       "      <th>VEP_IMPACT</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>45.18</td>\n",
       "      <td>5.070</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.175758</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>0.869418&amp;0.869418&amp;0.869418</td>\n",
       "      <td>1.312</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>24.700</td>\n",
       "      <td>3.402999</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>4</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>42.57</td>\n",
       "      <td>-2.670</td>\n",
       "      <td>N</td>\n",
       "      <td>.&amp;.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.361412</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>1&amp;1&amp;1</td>\n",
       "      <td>0.133</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>7.598</td>\n",
       "      <td>0.330332</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.013</td>\n",
       "      <td>5</td>\n",
       "      <td>0.24</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>42.35</td>\n",
       "      <td>4.050</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.238861</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>1&amp;1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>23.600</td>\n",
       "      <td>2.995736</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.998</td>\n",
       "      <td>6</td>\n",
       "      <td>0.03</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>39.05</td>\n",
       "      <td>0.354</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.333960</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>1&amp;1&amp;1</td>\n",
       "      <td>-0.267</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>12.740</td>\n",
       "      <td>0.903099</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.003</td>\n",
       "      <td>7</td>\n",
       "      <td>0.16</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>42.94</td>\n",
       "      <td>1.630</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.144994</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>1&amp;1&amp;1</td>\n",
       "      <td>0.197</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>16.050</td>\n",
       "      <td>1.460662</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.896</td>\n",
       "      <td>8</td>\n",
       "      <td>0.02</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>98.35</td>\n",
       "      <td>5.740</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>D&amp;D&amp;D&amp;D</td>\n",
       "      <td>0.999983&amp;0.999983&amp;0.999983&amp;0.999983</td>\n",
       "      <td>1.176</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>23.300</td>\n",
       "      <td>2.865518</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.432</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>-0.9</td>\n",
       "      <td>0.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>98.27</td>\n",
       "      <td>5.740</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>D&amp;D&amp;D&amp;D</td>\n",
       "      <td>1&amp;1&amp;1&amp;1</td>\n",
       "      <td>1.138</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>27.900</td>\n",
       "      <td>4.019092</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>97.67</td>\n",
       "      <td>5.740</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>D&amp;D&amp;D&amp;D</td>\n",
       "      <td>0.999997&amp;0.999997&amp;0.999997&amp;0.999997</td>\n",
       "      <td>1.312</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>28.200</td>\n",
       "      <td>4.062516</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-2.6</td>\n",
       "      <td>2.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>95.56</td>\n",
       "      <td>4.620</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>D&amp;D&amp;D&amp;D</td>\n",
       "      <td>0.999985&amp;0.999985&amp;0.999985&amp;0.999985</td>\n",
       "      <td>1.138</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>23.800</td>\n",
       "      <td>3.099789</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.867</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>85.69</td>\n",
       "      <td>5.740</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>D&amp;D&amp;D&amp;D</td>\n",
       "      <td>1&amp;1&amp;1&amp;1</td>\n",
       "      <td>1.176</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>25.000</td>\n",
       "      <td>3.508386</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.811</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Confidence Score  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "22:18561153:A>G                          45.18           5.070   \n",
       "22:18561156:C>T                          42.57          -2.670   \n",
       "22:18561159:C>T                          42.35           4.050   \n",
       "22:18561162:C>T                          39.05           0.354   \n",
       "22:18561164:T>A                          42.94           1.630   \n",
       "...                                        ...             ...   \n",
       "15:83561573:G>C                          98.35           5.740   \n",
       "15:83561577:T>G                          98.27           5.740   \n",
       "15:83561579:A>G                          97.67           5.740   \n",
       "15:83561583:T>C                          95.56           4.620   \n",
       "15:83561589:G>C                          85.69           5.740   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED  \\\n",
       "Genomic Description (GRCh37)                       \n",
       "22:18561153:A>G                                C   \n",
       "22:18561156:C>T                                N   \n",
       "22:18561159:C>T                                C   \n",
       "22:18561162:C>T                                C   \n",
       "22:18561164:T>A                                C   \n",
       "...                                          ...   \n",
       "15:83561573:G>C                                C   \n",
       "15:83561577:T>G                                C   \n",
       "15:83561579:A>G                                C   \n",
       "15:83561583:T>C                                C   \n",
       "15:83561589:G>C                                C   \n",
       "\n",
       "                                                         DBNSFP_INTERPRO_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                                                      \n",
       "22:18561153:A>G                                                         .&.&.&.   \n",
       "22:18561156:C>T                                                         .&.&.&.   \n",
       "22:18561159:C>T                                                         .&.&.&.   \n",
       "22:18561162:C>T                                                         .&.&.&.   \n",
       "22:18561164:T>A                                                         .&.&.&.   \n",
       "...                                                                         ...   \n",
       "15:83561573:G>C               .&WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "15:83561577:T>G               .&WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "15:83561579:A>G               .&WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "15:83561583:T>C               .&WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "15:83561589:G>C               .&WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "\n",
       "                             DBNSFP_LRT_PRED  DBNSFP_LRT_SCORE  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "22:18561153:A>G                            U          0.175758   \n",
       "22:18561156:C>T                            U          0.361412   \n",
       "22:18561159:C>T                            U          0.238861   \n",
       "22:18561162:C>T                            U          0.333960   \n",
       "22:18561164:T>A                            U          0.144994   \n",
       "...                                      ...               ...   \n",
       "15:83561573:G>C                            D          0.000000   \n",
       "15:83561577:T>G                            D          0.000000   \n",
       "15:83561579:A>G                            D          0.000000   \n",
       "15:83561583:T>C                            D          0.000000   \n",
       "15:83561589:G>C                            D          0.000000   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                              \n",
       "22:18561153:A>G                                   N&N&N   \n",
       "22:18561156:C>T                                   N&N&N   \n",
       "22:18561159:C>T                                   N&N&N   \n",
       "22:18561162:C>T                                   N&N&N   \n",
       "22:18561164:T>A                                   N&N&N   \n",
       "...                                                 ...   \n",
       "15:83561573:G>C                                 D&D&D&D   \n",
       "15:83561577:T>G                                 D&D&D&D   \n",
       "15:83561579:A>G                                 D&D&D&D   \n",
       "15:83561583:T>C                                 D&D&D&D   \n",
       "15:83561589:G>C                                 D&D&D&D   \n",
       "\n",
       "                                      DBNSFP_MUTATIONTASTER_SCORE  \\\n",
       "Genomic Description (GRCh37)                                        \n",
       "22:18561153:A>G                        0.869418&0.869418&0.869418   \n",
       "22:18561156:C>T                                             1&1&1   \n",
       "22:18561159:C>T                                             1&1&1   \n",
       "22:18561162:C>T                                             1&1&1   \n",
       "22:18561164:T>A                                             1&1&1   \n",
       "...                                                           ...   \n",
       "15:83561573:G>C               0.999983&0.999983&0.999983&0.999983   \n",
       "15:83561577:T>G                                           1&1&1&1   \n",
       "15:83561579:A>G               0.999997&0.999997&0.999997&0.999997   \n",
       "15:83561583:T>C               0.999985&0.999985&0.999985&0.999985   \n",
       "15:83561589:G>C                                           1&1&1&1   \n",
       "\n",
       "                              DBNSFP_PHYLOP30WAY_MAMMALIAN  \\\n",
       "Genomic Description (GRCh37)                                 \n",
       "22:18561153:A>G                                      1.312   \n",
       "22:18561156:C>T                                      0.133   \n",
       "22:18561159:C>T                                      1.026   \n",
       "22:18561162:C>T                                     -0.267   \n",
       "22:18561164:T>A                                      0.197   \n",
       "...                                                    ...   \n",
       "15:83561573:G>C                                      1.176   \n",
       "15:83561577:T>G                                      1.138   \n",
       "15:83561579:A>G                                      1.312   \n",
       "15:83561583:T>C                                      1.138   \n",
       "15:83561589:G>C                                      1.176   \n",
       "\n",
       "                             DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED  ...  \\\n",
       "Genomic Description (GRCh37)                                    ...   \n",
       "22:18561153:A>G                                              C  ...   \n",
       "22:18561156:C>T                                              N  ...   \n",
       "22:18561159:C>T                                              C  ...   \n",
       "22:18561162:C>T                                              N  ...   \n",
       "22:18561164:T>A                                              N  ...   \n",
       "...                                                        ...  ...   \n",
       "15:83561573:G>C                                              C  ...   \n",
       "15:83561577:T>G                                              C  ...   \n",
       "15:83561579:A>G                                              C  ...   \n",
       "15:83561583:T>C                                              C  ...   \n",
       "15:83561589:G>C                                              C  ...   \n",
       "\n",
       "                             VEP_CADD_PHRED VEP_CADD_RAW   VEP_CONSEQUENCE  \\\n",
       "Genomic Description (GRCh37)                                                 \n",
       "22:18561153:A>G                      24.700     3.402999  missense_variant   \n",
       "22:18561156:C>T                       7.598     0.330332  missense_variant   \n",
       "22:18561159:C>T                      23.600     2.995736  missense_variant   \n",
       "22:18561162:C>T                      12.740     0.903099  missense_variant   \n",
       "22:18561164:T>A                      16.050     1.460662  missense_variant   \n",
       "...                                     ...          ...               ...   \n",
       "15:83561573:G>C                      23.300     2.865518  missense_variant   \n",
       "15:83561577:T>G                      27.900     4.019092  missense_variant   \n",
       "15:83561579:A>G                      28.200     4.062516  missense_variant   \n",
       "15:83561583:T>C                      23.800     3.099789  missense_variant   \n",
       "15:83561589:G>C                      25.000     3.508386  missense_variant   \n",
       "\n",
       "                             VEP_IMPACT VEP_POLYPHEN_PRED VEP_POLYPHEN_SCORE  \\\n",
       "Genomic Description (GRCh37)                                                   \n",
       "22:18561153:A>G                MODERATE                 D              0.997   \n",
       "22:18561156:C>T                MODERATE                 B              0.013   \n",
       "22:18561159:C>T                MODERATE                 D              0.998   \n",
       "22:18561162:C>T                MODERATE                 B              0.003   \n",
       "22:18561164:T>A                MODERATE                 P              0.896   \n",
       "...                                 ...               ...                ...   \n",
       "15:83561573:G>C                MODERATE                 B              0.432   \n",
       "15:83561577:T>G                MODERATE                 D              1.000   \n",
       "15:83561579:A>G                MODERATE                 D              0.997   \n",
       "15:83561583:T>C                MODERATE                 P              0.867   \n",
       "15:83561589:G>C                MODERATE                 P              0.811   \n",
       "\n",
       "                             VEP_PROTEIN_POS VEP_SIFT_SCORE  ddG |ddG|  \n",
       "Genomic Description (GRCh37)                                            \n",
       "22:18561153:A>G                            4           0.01 -0.7   0.7  \n",
       "22:18561156:C>T                            5           0.24 -0.0   0.0  \n",
       "22:18561159:C>T                            6           0.03 -0.3   0.3  \n",
       "22:18561162:C>T                            7           0.16 -0.0   0.0  \n",
       "22:18561164:T>A                            8           0.02 -0.0   0.0  \n",
       "...                                      ...            ...  ...   ...  \n",
       "15:83561573:G>C                          9.0           0.05 -0.9   0.9  \n",
       "15:83561577:T>G                          8.0           0.00 -0.6   0.6  \n",
       "15:83561579:A>G                          7.0           0.00 -2.6   2.6  \n",
       "15:83561583:T>C                          6.0           0.01 -0.5   0.5  \n",
       "15:83561589:G>C                          4.0           0.00 -0.3   0.3  \n",
       "\n",
       "[128712 rows x 38 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df = df[kept_features_list].drop(columns=pathogenicity_label_cols+misc)\n",
    "working_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### What are the DBNSFP_{} and DBNSFP_{}\\_PRED cols?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>C</td>\n",
       "      <td>5.070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>N</td>\n",
       "      <td>-2.670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>4.050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>0.354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>C</td>\n",
       "      <td>1.630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>C</td>\n",
       "      <td>5.740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>C</td>\n",
       "      <td>5.740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>C</td>\n",
       "      <td>5.740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>C</td>\n",
       "      <td>4.620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>C</td>\n",
       "      <td>5.740</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_GERP_RS_PRED  DBNSFP_GERP_RS\n",
       "Genomic Description (GRCh37)                                    \n",
       "22:18561153:A>G                                C           5.070\n",
       "22:18561156:C>T                                N          -2.670\n",
       "22:18561159:C>T                                C           4.050\n",
       "22:18561162:C>T                                C           0.354\n",
       "22:18561164:T>A                                C           1.630\n",
       "...                                          ...             ...\n",
       "15:83561573:G>C                                C           5.740\n",
       "15:83561577:T>G                                C           5.740\n",
       "15:83561579:A>G                                C           5.740\n",
       "15:83561583:T>C                                C           4.620\n",
       "15:83561589:G>C                                C           5.740\n",
       "\n",
       "[128712 rows x 2 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[['DBNSFP_GERP_RS_PRED', 'DBNSFP_GERP_RS']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>C</td>\n",
       "      <td>5.070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>4.050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>0.354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>C</td>\n",
       "      <td>1.630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561165:C&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>5.070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>C</td>\n",
       "      <td>5.740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>C</td>\n",
       "      <td>5.740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>C</td>\n",
       "      <td>5.740</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>C</td>\n",
       "      <td>4.620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>C</td>\n",
       "      <td>5.740</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>115276 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_GERP_RS_PRED  DBNSFP_GERP_RS\n",
       "Genomic Description (GRCh37)                                    \n",
       "22:18561153:A>G                                C           5.070\n",
       "22:18561159:C>T                                C           4.050\n",
       "22:18561162:C>T                                C           0.354\n",
       "22:18561164:T>A                                C           1.630\n",
       "22:18561165:C>T                                C           5.070\n",
       "...                                          ...             ...\n",
       "15:83561573:G>C                                C           5.740\n",
       "15:83561577:T>G                                C           5.740\n",
       "15:83561579:A>G                                C           5.740\n",
       "15:83561583:T>C                                C           4.620\n",
       "15:83561589:G>C                                C           5.740\n",
       "\n",
       "[115276 rows x 2 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[df.DBNSFP_GERP_RS > 0, ['DBNSFP_GERP_RS_PRED', 'DBNSFP_GERP_RS']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "dbnsfp_cols = working_df.loc[:, working_df.columns.str.contains('DBNSFP')].columns.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['DBNSFP_GERP_RS',\n",
       " 'DBNSFP_GERP_RS_PRED',\n",
       " 'DBNSFP_INTERPRO_DOMAIN',\n",
       " 'DBNSFP_LRT_PRED',\n",
       " 'DBNSFP_LRT_SCORE',\n",
       " 'DBNSFP_MUTATIONTASTER_PRED',\n",
       " 'DBNSFP_MUTATIONTASTER_SCORE',\n",
       " 'DBNSFP_PHYLOP30WAY_MAMMALIAN',\n",
       " 'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED',\n",
       " 'DBNSFP_POLYPHEN2_HDIV_PRED',\n",
       " 'DBNSFP_POLYPHEN2_HDIV_SCORE',\n",
       " 'DBNSFP_SIFT_PRED',\n",
       " 'DBNSFP_SIFT_SCORE']"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dbnsfp_cols"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reviewing other columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "coltons_cols = ['Genomic Description (GRCh37)', 'HGVS Nucleotide Change', 'HGVS Protein Change', 'Phenotype', 'Consequence', 'Exon', 'Intron', 'Protein Position', 'Impact', 'Interpro Domain', 'Max MAF Source', '|ddG|']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>FINAL_PATHOGENICITY</th>\n",
       "      <th>FINAL_PRED</th>\n",
       "      <th>GENE</th>\n",
       "      <th>Interpro Domain</th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>NUM_PATH_PREDS</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>OVERALL_MAX_MAF</th>\n",
       "      <th>...</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_CONSEQUENCE</th>\n",
       "      <th>VEP_IMPACT</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>45.18</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003487</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.697034</td>\n",
       "      <td>3.48651e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>24.700</td>\n",
       "      <td>3.402999</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>4</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>42.57</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003503</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.951867</td>\n",
       "      <td>3.50336e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>7.598</td>\n",
       "      <td>0.330332</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.013</td>\n",
       "      <td>5</td>\n",
       "      <td>0.24</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>42.35</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001050</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.830258</td>\n",
       "      <td>1.04985e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>23.600</td>\n",
       "      <td>2.995736</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.998</td>\n",
       "      <td>6</td>\n",
       "      <td>0.03</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>39.05</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003519</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.025505</td>\n",
       "      <td>3.51865e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>12.740</td>\n",
       "      <td>0.903099</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.003</td>\n",
       "      <td>7</td>\n",
       "      <td>0.16</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>42.94</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001059</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.859414</td>\n",
       "      <td>1.05899e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>16.050</td>\n",
       "      <td>1.460662</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.896</td>\n",
       "      <td>8</td>\n",
       "      <td>0.02</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>98.35</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>0.000887</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.035758</td>\n",
       "      <td>8.87028e-06</td>\n",
       "      <td>...</td>\n",
       "      <td>23.300</td>\n",
       "      <td>2.865518</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.432</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>-0.9</td>\n",
       "      <td>0.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>98.27</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>0.001778</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.424787</td>\n",
       "      <td>1.77803e-05</td>\n",
       "      <td>...</td>\n",
       "      <td>27.900</td>\n",
       "      <td>4.019092</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>97.67</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>0.012962</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.201331</td>\n",
       "      <td>0.000129618</td>\n",
       "      <td>...</td>\n",
       "      <td>28.200</td>\n",
       "      <td>4.062516</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-2.6</td>\n",
       "      <td>2.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>95.56</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.381969</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23.800</td>\n",
       "      <td>3.099789</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.867</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>85.69</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>0.000894</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.686764</td>\n",
       "      <td>8.94374e-06</td>\n",
       "      <td>...</td>\n",
       "      <td>25.000</td>\n",
       "      <td>3.508386</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.811</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Confidence Score   FINAL_PATHOGENICITY  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "22:18561153:A>G                          45.18  Unknown significance   \n",
       "22:18561156:C>T                          42.57         Likely benign   \n",
       "22:18561159:C>T                          42.35  Unknown significance   \n",
       "22:18561162:C>T                          39.05         Likely benign   \n",
       "22:18561164:T>A                          42.94  Unknown significance   \n",
       "...                                        ...                   ...   \n",
       "15:83561573:G>C                          98.35  Unknown significance   \n",
       "15:83561577:T>G                          98.27  Unknown significance   \n",
       "15:83561579:A>G                          97.67  Unknown significance   \n",
       "15:83561583:T>C                          95.56  Unknown significance   \n",
       "15:83561589:G>C                          85.69  Unknown significance   \n",
       "\n",
       "                                        FINAL_PRED    GENE  \\\n",
       "Genomic Description (GRCh37)                                 \n",
       "22:18561153:A>G               Unknown significance   PEX26   \n",
       "22:18561156:C>T                      Likely benign   PEX26   \n",
       "22:18561159:C>T               Unknown significance   PEX26   \n",
       "22:18561162:C>T                      Likely benign   PEX26   \n",
       "22:18561164:T>A               Unknown significance   PEX26   \n",
       "...                                            ...     ...   \n",
       "15:83561573:G>C               Unknown significance  HOMER2   \n",
       "15:83561577:T>G               Unknown significance  HOMER2   \n",
       "15:83561579:A>G               Unknown significance  HOMER2   \n",
       "15:83561583:T>C               Unknown significance  HOMER2   \n",
       "15:83561589:G>C               Unknown significance  HOMER2   \n",
       "\n",
       "                                                              Interpro Domain  \\\n",
       "Genomic Description (GRCh37)                                                    \n",
       "22:18561153:A>G                                                           NaN   \n",
       "22:18561156:C>T                                                           NaN   \n",
       "22:18561159:C>T                                                           NaN   \n",
       "22:18561162:C>T                                                           NaN   \n",
       "22:18561164:T>A                                                           NaN   \n",
       "...                                                                       ...   \n",
       "15:83561573:G>C               WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "15:83561577:T>G               WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "15:83561579:A>G               WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "15:83561583:T>C               WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "15:83561589:G>C               WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "\n",
       "                              Max MAF (%)  \\\n",
       "Genomic Description (GRCh37)                \n",
       "22:18561153:A>G                  0.003487   \n",
       "22:18561156:C>T                  0.003503   \n",
       "22:18561159:C>T                  0.001050   \n",
       "22:18561162:C>T                  0.003519   \n",
       "22:18561164:T>A                  0.001059   \n",
       "...                                   ...   \n",
       "15:83561573:G>C                  0.000887   \n",
       "15:83561577:T>G                  0.001778   \n",
       "15:83561579:A>G                  0.012962   \n",
       "15:83561583:T>C                  0.000000   \n",
       "15:83561589:G>C                  0.000894   \n",
       "\n",
       "                                                                 Max MAF Source  \\\n",
       "Genomic Description (GRCh37)                                                      \n",
       "22:18561153:A>G                                              gnomad_AF_sas_pass   \n",
       "22:18561156:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561159:C>T                                              gnomad_AF_nfe_pass   \n",
       "22:18561162:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561164:T>A                                              gnomad_AF_nfe_pass   \n",
       "...                                                                         ...   \n",
       "15:83561573:G>C                                              gnomad_AF_nfe_pass   \n",
       "15:83561577:T>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561579:A>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561583:T>C               gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...   \n",
       "15:83561589:G>C                                              gnomad_AF_nfe_pass   \n",
       "\n",
       "                              NUM_PATH_PREDS  Normalized SA OVERALL_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "22:18561153:A>G                          4.0       0.697034     3.48651e-05   \n",
       "22:18561156:C>T                          0.0       0.951867     3.50336e-05   \n",
       "22:18561159:C>T                          4.0       0.830258     1.04985e-05   \n",
       "22:18561162:C>T                          1.0       1.025505     3.51865e-05   \n",
       "22:18561164:T>A                          3.0       0.859414     1.05899e-05   \n",
       "...                                      ...            ...             ...   \n",
       "15:83561573:G>C                          5.0       0.035758     8.87028e-06   \n",
       "15:83561577:T>G                          6.0       0.424787     1.77803e-05   \n",
       "15:83561579:A>G                          6.0       0.201331     0.000129618   \n",
       "15:83561583:T>C                          6.0       0.381969             0.0   \n",
       "15:83561589:G>C                          6.0       0.686764     8.94374e-06   \n",
       "\n",
       "                              ... VEP_CADD_PHRED  VEP_CADD_RAW  \\\n",
       "Genomic Description (GRCh37)  ...                                \n",
       "22:18561153:A>G               ...         24.700      3.402999   \n",
       "22:18561156:C>T               ...          7.598      0.330332   \n",
       "22:18561159:C>T               ...         23.600      2.995736   \n",
       "22:18561162:C>T               ...         12.740      0.903099   \n",
       "22:18561164:T>A               ...         16.050      1.460662   \n",
       "...                           ...            ...           ...   \n",
       "15:83561573:G>C               ...         23.300      2.865518   \n",
       "15:83561577:T>G               ...         27.900      4.019092   \n",
       "15:83561579:A>G               ...         28.200      4.062516   \n",
       "15:83561583:T>C               ...         23.800      3.099789   \n",
       "15:83561589:G>C               ...         25.000      3.508386   \n",
       "\n",
       "                               VEP_CONSEQUENCE VEP_IMPACT VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "22:18561153:A>G               missense_variant   MODERATE                 D   \n",
       "22:18561156:C>T               missense_variant   MODERATE                 B   \n",
       "22:18561159:C>T               missense_variant   MODERATE                 D   \n",
       "22:18561162:C>T               missense_variant   MODERATE                 B   \n",
       "22:18561164:T>A               missense_variant   MODERATE                 P   \n",
       "...                                        ...        ...               ...   \n",
       "15:83561573:G>C               missense_variant   MODERATE                 B   \n",
       "15:83561577:T>G               missense_variant   MODERATE                 D   \n",
       "15:83561579:A>G               missense_variant   MODERATE                 D   \n",
       "15:83561583:T>C               missense_variant   MODERATE                 P   \n",
       "15:83561589:G>C               missense_variant   MODERATE                 P   \n",
       "\n",
       "                              VEP_POLYPHEN_SCORE  VEP_PROTEIN_POS  \\\n",
       "Genomic Description (GRCh37)                                        \n",
       "22:18561153:A>G                            0.997                4   \n",
       "22:18561156:C>T                            0.013                5   \n",
       "22:18561159:C>T                            0.998                6   \n",
       "22:18561162:C>T                            0.003                7   \n",
       "22:18561164:T>A                            0.896                8   \n",
       "...                                          ...              ...   \n",
       "15:83561573:G>C                            0.432              9.0   \n",
       "15:83561577:T>G                            1.000              8.0   \n",
       "15:83561579:A>G                            0.997              7.0   \n",
       "15:83561583:T>C                            0.867              6.0   \n",
       "15:83561589:G>C                            0.811              4.0   \n",
       "\n",
       "                             VEP_SIFT_SCORE  ddG |ddG|  \n",
       "Genomic Description (GRCh37)                            \n",
       "22:18561153:A>G                        0.01 -0.7   0.7  \n",
       "22:18561156:C>T                        0.24 -0.0   0.0  \n",
       "22:18561159:C>T                        0.03 -0.3   0.3  \n",
       "22:18561162:C>T                        0.16 -0.0   0.0  \n",
       "22:18561164:T>A                        0.02 -0.0   0.0  \n",
       "...                                     ...  ...   ...  \n",
       "15:83561573:G>C                        0.05 -0.9   0.9  \n",
       "15:83561577:T>G                        0.00 -0.6   0.6  \n",
       "15:83561579:A>G                        0.00 -2.6   2.6  \n",
       "15:83561583:T>C                        0.01 -0.5   0.5  \n",
       "15:83561589:G>C                        0.00 -0.3   0.3  \n",
       "\n",
       "[128712 rows x 25 columns]"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.drop(columns=dbnsfp_cols)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>24.700</td>\n",
       "      <td>3.402999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>7.598</td>\n",
       "      <td>0.330332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>23.600</td>\n",
       "      <td>2.995736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>12.740</td>\n",
       "      <td>0.903099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>16.050</td>\n",
       "      <td>1.460662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>23.300</td>\n",
       "      <td>2.865518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>27.900</td>\n",
       "      <td>4.019092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>28.200</td>\n",
       "      <td>4.062516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>23.800</td>\n",
       "      <td>3.099789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>25.000</td>\n",
       "      <td>3.508386</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_CADD_PHRED  VEP_CADD_RAW\n",
       "Genomic Description (GRCh37)                              \n",
       "22:18561153:A>G                       24.700      3.402999\n",
       "22:18561156:C>T                        7.598      0.330332\n",
       "22:18561159:C>T                       23.600      2.995736\n",
       "22:18561162:C>T                       12.740      0.903099\n",
       "22:18561164:T>A                       16.050      1.460662\n",
       "...                                      ...           ...\n",
       "15:83561573:G>C                       23.300      2.865518\n",
       "15:83561577:T>G                       27.900      4.019092\n",
       "15:83561579:A>G                       28.200      4.062516\n",
       "15:83561583:T>C                       23.800      3.099789\n",
       "15:83561589:G>C                       25.000      3.508386\n",
       "\n",
       "[128712 rows x 2 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[:, working_df.columns.str.contains('CADD') ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>24.700</td>\n",
       "      <td>3.402999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>7.598</td>\n",
       "      <td>0.330332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>23.600</td>\n",
       "      <td>2.995736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>12.740</td>\n",
       "      <td>0.903099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>16.050</td>\n",
       "      <td>1.460662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>23.300</td>\n",
       "      <td>2.865518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>27.900</td>\n",
       "      <td>4.019092</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>28.200</td>\n",
       "      <td>4.062516</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>23.800</td>\n",
       "      <td>3.099789</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>25.000</td>\n",
       "      <td>3.508386</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_CADD_PHRED  VEP_CADD_RAW\n",
       "Genomic Description (GRCh37)                              \n",
       "22:18561153:A>G                       24.700      3.402999\n",
       "22:18561156:C>T                        7.598      0.330332\n",
       "22:18561159:C>T                       23.600      2.995736\n",
       "22:18561162:C>T                       12.740      0.903099\n",
       "22:18561164:T>A                       16.050      1.460662\n",
       "...                                      ...           ...\n",
       "15:83561573:G>C                       23.300      2.865518\n",
       "15:83561577:T>G                       27.900      4.019092\n",
       "15:83561579:A>G                       28.200      4.062516\n",
       "15:83561583:T>C                       23.800      3.099789\n",
       "15:83561589:G>C                       25.000      3.508386\n",
       "\n",
       "[128712 rows x 2 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[:, working_df.columns.str.contains('CADD') | working_df.columns.str.contains('Cadd')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Confidence Score', 'DBNSFP_GERP_RS', 'DBNSFP_GERP_RS_PRED',\n",
       "       'DBNSFP_INTERPRO_DOMAIN', 'DBNSFP_LRT_PRED', 'DBNSFP_LRT_SCORE',\n",
       "       'DBNSFP_MUTATIONTASTER_PRED', 'DBNSFP_MUTATIONTASTER_SCORE',\n",
       "       'DBNSFP_PHYLOP30WAY_MAMMALIAN', 'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED',\n",
       "       'DBNSFP_POLYPHEN2_HDIV_PRED', 'DBNSFP_POLYPHEN2_HDIV_SCORE',\n",
       "       'DBNSFP_SIFT_PRED', 'DBNSFP_SIFT_SCORE', 'FINAL_PATHOGENICITY',\n",
       "       'FINAL_PRED', 'GENE', 'Interpro Domain', 'Max MAF (%)',\n",
       "       'Max MAF Source', 'NUM_PATH_PREDS', 'Normalized SA', 'OVERALL_MAX_MAF',\n",
       "       'OVERALL_MAX_MAF_SOURCE', 'Surface Area', 'TOTAL_NUM_PREDS', 'TYPE',\n",
       "       'UNIPROT_DOMAIN', 'VEP_CADD_PHRED', 'VEP_CADD_RAW', 'VEP_CONSEQUENCE',\n",
       "       'VEP_IMPACT', 'VEP_POLYPHEN_PRED', 'VEP_POLYPHEN_SCORE',\n",
       "       'VEP_PROTEIN_POS', 'VEP_SIFT_SCORE', 'ddG', '|ddG|'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_CONSEQUENCE</th>\n",
       "      <th>VEP_IMPACT</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>24.700</td>\n",
       "      <td>3.402999</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>4</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>7.598</td>\n",
       "      <td>0.330332</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.013</td>\n",
       "      <td>5</td>\n",
       "      <td>0.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>23.600</td>\n",
       "      <td>2.995736</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.998</td>\n",
       "      <td>6</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>12.740</td>\n",
       "      <td>0.903099</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.003</td>\n",
       "      <td>7</td>\n",
       "      <td>0.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>16.050</td>\n",
       "      <td>1.460662</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.896</td>\n",
       "      <td>8</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>23.300</td>\n",
       "      <td>2.865518</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.432</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>27.900</td>\n",
       "      <td>4.019092</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>28.200</td>\n",
       "      <td>4.062516</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>23.800</td>\n",
       "      <td>3.099789</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.867</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>25.000</td>\n",
       "      <td>3.508386</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.811</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_CADD_PHRED  VEP_CADD_RAW   VEP_CONSEQUENCE  \\\n",
       "Genomic Description (GRCh37)                                                   \n",
       "22:18561153:A>G                       24.700      3.402999  missense_variant   \n",
       "22:18561156:C>T                        7.598      0.330332  missense_variant   \n",
       "22:18561159:C>T                       23.600      2.995736  missense_variant   \n",
       "22:18561162:C>T                       12.740      0.903099  missense_variant   \n",
       "22:18561164:T>A                       16.050      1.460662  missense_variant   \n",
       "...                                      ...           ...               ...   \n",
       "15:83561573:G>C                       23.300      2.865518  missense_variant   \n",
       "15:83561577:T>G                       27.900      4.019092  missense_variant   \n",
       "15:83561579:A>G                       28.200      4.062516  missense_variant   \n",
       "15:83561583:T>C                       23.800      3.099789  missense_variant   \n",
       "15:83561589:G>C                       25.000      3.508386  missense_variant   \n",
       "\n",
       "                             VEP_IMPACT VEP_POLYPHEN_PRED  VEP_POLYPHEN_SCORE  \\\n",
       "Genomic Description (GRCh37)                                                    \n",
       "22:18561153:A>G                MODERATE                 D               0.997   \n",
       "22:18561156:C>T                MODERATE                 B               0.013   \n",
       "22:18561159:C>T                MODERATE                 D               0.998   \n",
       "22:18561162:C>T                MODERATE                 B               0.003   \n",
       "22:18561164:T>A                MODERATE                 P               0.896   \n",
       "...                                 ...               ...                 ...   \n",
       "15:83561573:G>C                MODERATE                 B               0.432   \n",
       "15:83561577:T>G                MODERATE                 D               1.000   \n",
       "15:83561579:A>G                MODERATE                 D               0.997   \n",
       "15:83561583:T>C                MODERATE                 P               0.867   \n",
       "15:83561589:G>C                MODERATE                 P               0.811   \n",
       "\n",
       "                             VEP_PROTEIN_POS  VEP_SIFT_SCORE  \n",
       "Genomic Description (GRCh37)                                  \n",
       "22:18561153:A>G                            4            0.01  \n",
       "22:18561156:C>T                            5            0.24  \n",
       "22:18561159:C>T                            6            0.03  \n",
       "22:18561162:C>T                            7            0.16  \n",
       "22:18561164:T>A                            8            0.02  \n",
       "...                                      ...             ...  \n",
       "15:83561573:G>C                          9.0            0.05  \n",
       "15:83561577:T>G                          8.0            0.00  \n",
       "15:83561579:A>G                          7.0            0.00  \n",
       "15:83561583:T>C                          6.0            0.01  \n",
       "15:83561589:G>C                          4.0            0.00  \n",
       "\n",
       "[128712 rows x 8 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>FINAL_PATHOGENICITY</th>\n",
       "      <th>FINAL_PRED</th>\n",
       "      <th>GENE</th>\n",
       "      <th>Interpro Domain</th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>NUM_PATH_PREDS</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>OVERALL_MAX_MAF</th>\n",
       "      <th>OVERALL_MAX_MAF_SOURCE</th>\n",
       "      <th>Surface Area</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>UNIPROT_DOMAIN</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>45.18</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003487</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.697034</td>\n",
       "      <td>3.48651e-05</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>111.511314</td>\n",
       "      <td>5.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>42.57</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003503</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.951867</td>\n",
       "      <td>3.50336e-05</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>131.269338</td>\n",
       "      <td>5.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>42.35</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001050</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.830258</td>\n",
       "      <td>1.04985e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>114.498499</td>\n",
       "      <td>5.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>39.05</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003519</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.025505</td>\n",
       "      <td>3.51865e-05</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>161.350503</td>\n",
       "      <td>5.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>42.94</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>PEX26</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001059</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.859414</td>\n",
       "      <td>1.05899e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>118.519368</td>\n",
       "      <td>5.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>98.35</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>0.000887</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.035758</td>\n",
       "      <td>8.87028e-06</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>5.626061</td>\n",
       "      <td>6.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WH1</td>\n",
       "      <td>-0.9</td>\n",
       "      <td>0.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>98.27</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>0.001778</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.424787</td>\n",
       "      <td>1.77803e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>66.834983</td>\n",
       "      <td>6.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WH1</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>97.67</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>0.012962</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.201331</td>\n",
       "      <td>0.000129618</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>46.255767</td>\n",
       "      <td>6.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WH1</td>\n",
       "      <td>-2.6</td>\n",
       "      <td>2.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>95.56</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.381969</td>\n",
       "      <td>0.0</td>\n",
       "      <td>gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...</td>\n",
       "      <td>74.926502</td>\n",
       "      <td>6.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WH1</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>85.69</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>HOMER2</td>\n",
       "      <td>WH1/EVH1_domain&amp;WH1/EVH1_domain&amp;WH1/EVH1_domain</td>\n",
       "      <td>0.000894</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.686764</td>\n",
       "      <td>8.94374e-06</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>135.808746</td>\n",
       "      <td>6.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WH1</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Confidence Score   FINAL_PATHOGENICITY  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "22:18561153:A>G                          45.18  Unknown significance   \n",
       "22:18561156:C>T                          42.57         Likely benign   \n",
       "22:18561159:C>T                          42.35  Unknown significance   \n",
       "22:18561162:C>T                          39.05         Likely benign   \n",
       "22:18561164:T>A                          42.94  Unknown significance   \n",
       "...                                        ...                   ...   \n",
       "15:83561573:G>C                          98.35  Unknown significance   \n",
       "15:83561577:T>G                          98.27  Unknown significance   \n",
       "15:83561579:A>G                          97.67  Unknown significance   \n",
       "15:83561583:T>C                          95.56  Unknown significance   \n",
       "15:83561589:G>C                          85.69  Unknown significance   \n",
       "\n",
       "                                        FINAL_PRED    GENE  \\\n",
       "Genomic Description (GRCh37)                                 \n",
       "22:18561153:A>G               Unknown significance   PEX26   \n",
       "22:18561156:C>T                      Likely benign   PEX26   \n",
       "22:18561159:C>T               Unknown significance   PEX26   \n",
       "22:18561162:C>T                      Likely benign   PEX26   \n",
       "22:18561164:T>A               Unknown significance   PEX26   \n",
       "...                                            ...     ...   \n",
       "15:83561573:G>C               Unknown significance  HOMER2   \n",
       "15:83561577:T>G               Unknown significance  HOMER2   \n",
       "15:83561579:A>G               Unknown significance  HOMER2   \n",
       "15:83561583:T>C               Unknown significance  HOMER2   \n",
       "15:83561589:G>C               Unknown significance  HOMER2   \n",
       "\n",
       "                                                              Interpro Domain  \\\n",
       "Genomic Description (GRCh37)                                                    \n",
       "22:18561153:A>G                                                           NaN   \n",
       "22:18561156:C>T                                                           NaN   \n",
       "22:18561159:C>T                                                           NaN   \n",
       "22:18561162:C>T                                                           NaN   \n",
       "22:18561164:T>A                                                           NaN   \n",
       "...                                                                       ...   \n",
       "15:83561573:G>C               WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "15:83561577:T>G               WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "15:83561579:A>G               WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "15:83561583:T>C               WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "15:83561589:G>C               WH1/EVH1_domain&WH1/EVH1_domain&WH1/EVH1_domain   \n",
       "\n",
       "                              Max MAF (%)  \\\n",
       "Genomic Description (GRCh37)                \n",
       "22:18561153:A>G                  0.003487   \n",
       "22:18561156:C>T                  0.003503   \n",
       "22:18561159:C>T                  0.001050   \n",
       "22:18561162:C>T                  0.003519   \n",
       "22:18561164:T>A                  0.001059   \n",
       "...                                   ...   \n",
       "15:83561573:G>C                  0.000887   \n",
       "15:83561577:T>G                  0.001778   \n",
       "15:83561579:A>G                  0.012962   \n",
       "15:83561583:T>C                  0.000000   \n",
       "15:83561589:G>C                  0.000894   \n",
       "\n",
       "                                                                 Max MAF Source  \\\n",
       "Genomic Description (GRCh37)                                                      \n",
       "22:18561153:A>G                                              gnomad_AF_sas_pass   \n",
       "22:18561156:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561159:C>T                                              gnomad_AF_nfe_pass   \n",
       "22:18561162:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561164:T>A                                              gnomad_AF_nfe_pass   \n",
       "...                                                                         ...   \n",
       "15:83561573:G>C                                              gnomad_AF_nfe_pass   \n",
       "15:83561577:T>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561579:A>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561583:T>C               gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...   \n",
       "15:83561589:G>C                                              gnomad_AF_nfe_pass   \n",
       "\n",
       "                              NUM_PATH_PREDS  Normalized SA OVERALL_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "22:18561153:A>G                          4.0       0.697034     3.48651e-05   \n",
       "22:18561156:C>T                          0.0       0.951867     3.50336e-05   \n",
       "22:18561159:C>T                          4.0       0.830258     1.04985e-05   \n",
       "22:18561162:C>T                          1.0       1.025505     3.51865e-05   \n",
       "22:18561164:T>A                          3.0       0.859414     1.05899e-05   \n",
       "...                                      ...            ...             ...   \n",
       "15:83561573:G>C                          5.0       0.035758     8.87028e-06   \n",
       "15:83561577:T>G                          6.0       0.424787     1.77803e-05   \n",
       "15:83561579:A>G                          6.0       0.201331     0.000129618   \n",
       "15:83561583:T>C                          6.0       0.381969             0.0   \n",
       "15:83561589:G>C                          6.0       0.686764     8.94374e-06   \n",
       "\n",
       "                                                         OVERALL_MAX_MAF_SOURCE  \\\n",
       "Genomic Description (GRCh37)                                                      \n",
       "22:18561153:A>G                                              gnomad_AF_sas_pass   \n",
       "22:18561156:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561159:C>T                                              gnomad_AF_nfe_pass   \n",
       "22:18561162:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561164:T>A                                              gnomad_AF_nfe_pass   \n",
       "...                                                                         ...   \n",
       "15:83561573:G>C                                              gnomad_AF_nfe_pass   \n",
       "15:83561577:T>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561579:A>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561583:T>C               gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...   \n",
       "15:83561589:G>C                                              gnomad_AF_nfe_pass   \n",
       "\n",
       "                              Surface Area  TOTAL_NUM_PREDS TYPE  \\\n",
       "Genomic Description (GRCh37)                                       \n",
       "22:18561153:A>G                 111.511314              5.0  SNV   \n",
       "22:18561156:C>T                 131.269338              5.0  SNV   \n",
       "22:18561159:C>T                 114.498499              5.0  SNV   \n",
       "22:18561162:C>T                 161.350503              5.0  SNV   \n",
       "22:18561164:T>A                 118.519368              5.0  SNV   \n",
       "...                                    ...              ...  ...   \n",
       "15:83561573:G>C                   5.626061              6.0  SNV   \n",
       "15:83561577:T>G                  66.834983              6.0  SNV   \n",
       "15:83561579:A>G                  46.255767              6.0  SNV   \n",
       "15:83561583:T>C                  74.926502              6.0  SNV   \n",
       "15:83561589:G>C                 135.808746              6.0  SNV   \n",
       "\n",
       "                             UNIPROT_DOMAIN  ddG  |ddG|  \n",
       "Genomic Description (GRCh37)                             \n",
       "22:18561153:A>G                         NaN -0.7    0.7  \n",
       "22:18561156:C>T                         NaN -0.0    0.0  \n",
       "22:18561159:C>T                         NaN -0.3    0.3  \n",
       "22:18561162:C>T                         NaN -0.0    0.0  \n",
       "22:18561164:T>A                         NaN -0.0    0.0  \n",
       "...                                     ...  ...    ...  \n",
       "15:83561573:G>C                         WH1 -0.9    0.9  \n",
       "15:83561577:T>G                         WH1 -0.6    0.6  \n",
       "15:83561579:A>G                         WH1 -2.6    2.6  \n",
       "15:83561583:T>C                         WH1 -0.5    0.5  \n",
       "15:83561589:G>C                         WH1 -0.3    0.3  \n",
       "\n",
       "[128712 rows x 17 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "vep_df = working_df.loc[:, working_df.columns.str.contains('VEP')]\n",
    "display(vep_df)\n",
    "dbnsfp_df = working_df.loc[:, ~(working_df.columns.str.contains('VEP').tolist()+working_df.columns.str.contains('DBNSFP'))]\n",
    "display(dbnsfp_df) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_CONSEQUENCE</th>\n",
       "      <th>VEP_IMPACT</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>FINAL_PATHOGENICITY</th>\n",
       "      <th>...</th>\n",
       "      <th>NUM_PATH_PREDS</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>OVERALL_MAX_MAF</th>\n",
       "      <th>OVERALL_MAX_MAF_SOURCE</th>\n",
       "      <th>Surface Area</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>UNIPROT_DOMAIN</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>24.700</td>\n",
       "      <td>3.402999</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>4</td>\n",
       "      <td>0.01</td>\n",
       "      <td>45.18</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.697034</td>\n",
       "      <td>3.48651e-05</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>111.511314</td>\n",
       "      <td>5.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>7.598</td>\n",
       "      <td>0.330332</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.013</td>\n",
       "      <td>5</td>\n",
       "      <td>0.24</td>\n",
       "      <td>42.57</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.951867</td>\n",
       "      <td>3.50336e-05</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>131.269338</td>\n",
       "      <td>5.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>23.600</td>\n",
       "      <td>2.995736</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.998</td>\n",
       "      <td>6</td>\n",
       "      <td>0.03</td>\n",
       "      <td>42.35</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>...</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.830258</td>\n",
       "      <td>1.04985e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>114.498499</td>\n",
       "      <td>5.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>12.740</td>\n",
       "      <td>0.903099</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.003</td>\n",
       "      <td>7</td>\n",
       "      <td>0.16</td>\n",
       "      <td>39.05</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>...</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.025505</td>\n",
       "      <td>3.51865e-05</td>\n",
       "      <td>gnomad_AF_sas_pass</td>\n",
       "      <td>161.350503</td>\n",
       "      <td>5.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>16.050</td>\n",
       "      <td>1.460662</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.896</td>\n",
       "      <td>8</td>\n",
       "      <td>0.02</td>\n",
       "      <td>42.94</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>...</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.859414</td>\n",
       "      <td>1.05899e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>118.519368</td>\n",
       "      <td>5.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>23.300</td>\n",
       "      <td>2.865518</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.432</td>\n",
       "      <td>9.0</td>\n",
       "      <td>0.05</td>\n",
       "      <td>98.35</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.035758</td>\n",
       "      <td>8.87028e-06</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>5.626061</td>\n",
       "      <td>6.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WH1</td>\n",
       "      <td>-0.9</td>\n",
       "      <td>0.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>27.900</td>\n",
       "      <td>4.019092</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>8.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>98.27</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.424787</td>\n",
       "      <td>1.77803e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>66.834983</td>\n",
       "      <td>6.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WH1</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>28.200</td>\n",
       "      <td>4.062516</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>97.67</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.201331</td>\n",
       "      <td>0.000129618</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>46.255767</td>\n",
       "      <td>6.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WH1</td>\n",
       "      <td>-2.6</td>\n",
       "      <td>2.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>23.800</td>\n",
       "      <td>3.099789</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.867</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.01</td>\n",
       "      <td>95.56</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.381969</td>\n",
       "      <td>0.0</td>\n",
       "      <td>gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...</td>\n",
       "      <td>74.926502</td>\n",
       "      <td>6.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WH1</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>25.000</td>\n",
       "      <td>3.508386</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.811</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.00</td>\n",
       "      <td>85.69</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.686764</td>\n",
       "      <td>8.94374e-06</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>135.808746</td>\n",
       "      <td>6.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WH1</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_CADD_PHRED  VEP_CADD_RAW   VEP_CONSEQUENCE  \\\n",
       "Genomic Description (GRCh37)                                                   \n",
       "22:18561153:A>G                       24.700      3.402999  missense_variant   \n",
       "22:18561156:C>T                        7.598      0.330332  missense_variant   \n",
       "22:18561159:C>T                       23.600      2.995736  missense_variant   \n",
       "22:18561162:C>T                       12.740      0.903099  missense_variant   \n",
       "22:18561164:T>A                       16.050      1.460662  missense_variant   \n",
       "...                                      ...           ...               ...   \n",
       "15:83561573:G>C                       23.300      2.865518  missense_variant   \n",
       "15:83561577:T>G                       27.900      4.019092  missense_variant   \n",
       "15:83561579:A>G                       28.200      4.062516  missense_variant   \n",
       "15:83561583:T>C                       23.800      3.099789  missense_variant   \n",
       "15:83561589:G>C                       25.000      3.508386  missense_variant   \n",
       "\n",
       "                             VEP_IMPACT VEP_POLYPHEN_PRED  VEP_POLYPHEN_SCORE  \\\n",
       "Genomic Description (GRCh37)                                                    \n",
       "22:18561153:A>G                MODERATE                 D               0.997   \n",
       "22:18561156:C>T                MODERATE                 B               0.013   \n",
       "22:18561159:C>T                MODERATE                 D               0.998   \n",
       "22:18561162:C>T                MODERATE                 B               0.003   \n",
       "22:18561164:T>A                MODERATE                 P               0.896   \n",
       "...                                 ...               ...                 ...   \n",
       "15:83561573:G>C                MODERATE                 B               0.432   \n",
       "15:83561577:T>G                MODERATE                 D               1.000   \n",
       "15:83561579:A>G                MODERATE                 D               0.997   \n",
       "15:83561583:T>C                MODERATE                 P               0.867   \n",
       "15:83561589:G>C                MODERATE                 P               0.811   \n",
       "\n",
       "                             VEP_PROTEIN_POS  VEP_SIFT_SCORE  \\\n",
       "Genomic Description (GRCh37)                                   \n",
       "22:18561153:A>G                            4            0.01   \n",
       "22:18561156:C>T                            5            0.24   \n",
       "22:18561159:C>T                            6            0.03   \n",
       "22:18561162:C>T                            7            0.16   \n",
       "22:18561164:T>A                            8            0.02   \n",
       "...                                      ...             ...   \n",
       "15:83561573:G>C                          9.0            0.05   \n",
       "15:83561577:T>G                          8.0            0.00   \n",
       "15:83561579:A>G                          7.0            0.00   \n",
       "15:83561583:T>C                          6.0            0.01   \n",
       "15:83561589:G>C                          4.0            0.00   \n",
       "\n",
       "                              Confidence Score   FINAL_PATHOGENICITY  ...  \\\n",
       "Genomic Description (GRCh37)                                          ...   \n",
       "22:18561153:A>G                          45.18  Unknown significance  ...   \n",
       "22:18561156:C>T                          42.57         Likely benign  ...   \n",
       "22:18561159:C>T                          42.35  Unknown significance  ...   \n",
       "22:18561162:C>T                          39.05         Likely benign  ...   \n",
       "22:18561164:T>A                          42.94  Unknown significance  ...   \n",
       "...                                        ...                   ...  ...   \n",
       "15:83561573:G>C                          98.35  Unknown significance  ...   \n",
       "15:83561577:T>G                          98.27  Unknown significance  ...   \n",
       "15:83561579:A>G                          97.67  Unknown significance  ...   \n",
       "15:83561583:T>C                          95.56  Unknown significance  ...   \n",
       "15:83561589:G>C                          85.69  Unknown significance  ...   \n",
       "\n",
       "                             NUM_PATH_PREDS Normalized SA OVERALL_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                                                \n",
       "22:18561153:A>G                         4.0      0.697034     3.48651e-05   \n",
       "22:18561156:C>T                         0.0      0.951867     3.50336e-05   \n",
       "22:18561159:C>T                         4.0      0.830258     1.04985e-05   \n",
       "22:18561162:C>T                         1.0      1.025505     3.51865e-05   \n",
       "22:18561164:T>A                         3.0      0.859414     1.05899e-05   \n",
       "...                                     ...           ...             ...   \n",
       "15:83561573:G>C                         5.0      0.035758     8.87028e-06   \n",
       "15:83561577:T>G                         6.0      0.424787     1.77803e-05   \n",
       "15:83561579:A>G                         6.0      0.201331     0.000129618   \n",
       "15:83561583:T>C                         6.0      0.381969             0.0   \n",
       "15:83561589:G>C                         6.0      0.686764     8.94374e-06   \n",
       "\n",
       "                                                         OVERALL_MAX_MAF_SOURCE  \\\n",
       "Genomic Description (GRCh37)                                                      \n",
       "22:18561153:A>G                                              gnomad_AF_sas_pass   \n",
       "22:18561156:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561159:C>T                                              gnomad_AF_nfe_pass   \n",
       "22:18561162:C>T                                              gnomad_AF_sas_pass   \n",
       "22:18561164:T>A                                              gnomad_AF_nfe_pass   \n",
       "...                                                                         ...   \n",
       "15:83561573:G>C                                              gnomad_AF_nfe_pass   \n",
       "15:83561577:T>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561579:A>G                                              gnomad_AF_nfe_pass   \n",
       "15:83561583:T>C               gnomad_AF_afr_pass,gnomad_AF_amr_pass,gnomad_A...   \n",
       "15:83561589:G>C                                              gnomad_AF_nfe_pass   \n",
       "\n",
       "                             Surface Area  TOTAL_NUM_PREDS  TYPE  \\\n",
       "Genomic Description (GRCh37)                                       \n",
       "22:18561153:A>G                111.511314              5.0   SNV   \n",
       "22:18561156:C>T                131.269338              5.0   SNV   \n",
       "22:18561159:C>T                114.498499              5.0   SNV   \n",
       "22:18561162:C>T                161.350503              5.0   SNV   \n",
       "22:18561164:T>A                118.519368              5.0   SNV   \n",
       "...                                   ...              ...   ...   \n",
       "15:83561573:G>C                  5.626061              6.0   SNV   \n",
       "15:83561577:T>G                 66.834983              6.0   SNV   \n",
       "15:83561579:A>G                 46.255767              6.0   SNV   \n",
       "15:83561583:T>C                 74.926502              6.0   SNV   \n",
       "15:83561589:G>C                135.808746              6.0   SNV   \n",
       "\n",
       "                             UNIPROT_DOMAIN  ddG  |ddG|  \n",
       "Genomic Description (GRCh37)                             \n",
       "22:18561153:A>G                         NaN -0.7    0.7  \n",
       "22:18561156:C>T                         NaN -0.0    0.0  \n",
       "22:18561159:C>T                         NaN -0.3    0.3  \n",
       "22:18561162:C>T                         NaN -0.0    0.0  \n",
       "22:18561164:T>A                         NaN -0.0    0.0  \n",
       "...                                     ...  ...    ...  \n",
       "15:83561573:G>C                         WH1 -0.9    0.9  \n",
       "15:83561577:T>G                         WH1 -0.6    0.6  \n",
       "15:83561579:A>G                         WH1 -2.6    2.6  \n",
       "15:83561583:T>C                         WH1 -0.5    0.5  \n",
       "15:83561589:G>C                         WH1 -0.3    0.3  \n",
       "\n",
       "[128712 rows x 25 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[:, vep_df.columns.tolist() + dbnsfp_df.columns.tolist()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Confidence Score                     float64\n",
       "DBNSFP_GERP_RS                       float64\n",
       "DBNSFP_GERP_RS_PRED                   object\n",
       "DBNSFP_INTERPRO_DOMAIN                object\n",
       "DBNSFP_LRT_PRED                       object\n",
       "DBNSFP_LRT_SCORE                     float64\n",
       "DBNSFP_MUTATIONTASTER_PRED            object\n",
       "DBNSFP_MUTATIONTASTER_SCORE           object\n",
       "DBNSFP_PHYLOP30WAY_MAMMALIAN         float64\n",
       "DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED     object\n",
       "DBNSFP_POLYPHEN2_HDIV_PRED            object\n",
       "DBNSFP_POLYPHEN2_HDIV_SCORE           object\n",
       "DBNSFP_SIFT_PRED                      object\n",
       "DBNSFP_SIFT_SCORE                     object\n",
       "FINAL_PATHOGENICITY                   object\n",
       "FINAL_PRED                            object\n",
       "GENE                                  object\n",
       "Interpro Domain                       object\n",
       "Max MAF (%)                          float64\n",
       "Max MAF Source                        object\n",
       "NUM_PATH_PREDS                       float64\n",
       "Normalized SA                        float64\n",
       "OVERALL_MAX_MAF                       object\n",
       "OVERALL_MAX_MAF_SOURCE                object\n",
       "Surface Area                         float64\n",
       "TOTAL_NUM_PREDS                      float64\n",
       "TYPE                                  object\n",
       "UNIPROT_DOMAIN                        object\n",
       "VEP_CADD_PHRED                       float64\n",
       "VEP_CADD_RAW                         float64\n",
       "VEP_CONSEQUENCE                       object\n",
       "VEP_IMPACT                            object\n",
       "VEP_POLYPHEN_PRED                     object\n",
       "VEP_POLYPHEN_SCORE                   float64\n",
       "VEP_PROTEIN_POS                       object\n",
       "VEP_SIFT_SCORE                       float64\n",
       "ddG                                  float64\n",
       "|ddG|                                float64\n",
       "dtype: object"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_CONSEQUENCE</th>\n",
       "      <th>VEP_IMPACT</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>45.18</td>\n",
       "      <td>5.070</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.175758</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>0.869418&amp;0.869418&amp;0.869418</td>\n",
       "      <td>1.312</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>24.700</td>\n",
       "      <td>3.402999</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>4</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>42.57</td>\n",
       "      <td>-2.670</td>\n",
       "      <td>N</td>\n",
       "      <td>.&amp;.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.361412</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>1&amp;1&amp;1</td>\n",
       "      <td>0.133</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>7.598</td>\n",
       "      <td>0.330332</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.013</td>\n",
       "      <td>5</td>\n",
       "      <td>0.24</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>42.35</td>\n",
       "      <td>4.050</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.238861</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>1&amp;1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>23.600</td>\n",
       "      <td>2.995736</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.998</td>\n",
       "      <td>6</td>\n",
       "      <td>0.03</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>39.05</td>\n",
       "      <td>0.354</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.333960</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>1&amp;1&amp;1</td>\n",
       "      <td>-0.267</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>12.740</td>\n",
       "      <td>0.903099</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.003</td>\n",
       "      <td>7</td>\n",
       "      <td>0.16</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>42.94</td>\n",
       "      <td>1.630</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.144994</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>1&amp;1&amp;1</td>\n",
       "      <td>0.197</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>16.050</td>\n",
       "      <td>1.460662</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.896</td>\n",
       "      <td>8</td>\n",
       "      <td>0.02</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Confidence Score  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "22:18561153:A>G                          45.18           5.070   \n",
       "22:18561156:C>T                          42.57          -2.670   \n",
       "22:18561159:C>T                          42.35           4.050   \n",
       "22:18561162:C>T                          39.05           0.354   \n",
       "22:18561164:T>A                          42.94           1.630   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_INTERPRO_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "22:18561153:A>G                                C                .&.&.&.   \n",
       "22:18561156:C>T                                N                .&.&.&.   \n",
       "22:18561159:C>T                                C                .&.&.&.   \n",
       "22:18561162:C>T                                C                .&.&.&.   \n",
       "22:18561164:T>A                                C                .&.&.&.   \n",
       "\n",
       "                             DBNSFP_LRT_PRED  DBNSFP_LRT_SCORE  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "22:18561153:A>G                            U          0.175758   \n",
       "22:18561156:C>T                            U          0.361412   \n",
       "22:18561159:C>T                            U          0.238861   \n",
       "22:18561162:C>T                            U          0.333960   \n",
       "22:18561164:T>A                            U          0.144994   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                              \n",
       "22:18561153:A>G                                   N&N&N   \n",
       "22:18561156:C>T                                   N&N&N   \n",
       "22:18561159:C>T                                   N&N&N   \n",
       "22:18561162:C>T                                   N&N&N   \n",
       "22:18561164:T>A                                   N&N&N   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_SCORE  \\\n",
       "Genomic Description (GRCh37)                               \n",
       "22:18561153:A>G               0.869418&0.869418&0.869418   \n",
       "22:18561156:C>T                                    1&1&1   \n",
       "22:18561159:C>T                                    1&1&1   \n",
       "22:18561162:C>T                                    1&1&1   \n",
       "22:18561164:T>A                                    1&1&1   \n",
       "\n",
       "                              DBNSFP_PHYLOP30WAY_MAMMALIAN  \\\n",
       "Genomic Description (GRCh37)                                 \n",
       "22:18561153:A>G                                      1.312   \n",
       "22:18561156:C>T                                      0.133   \n",
       "22:18561159:C>T                                      1.026   \n",
       "22:18561162:C>T                                     -0.267   \n",
       "22:18561164:T>A                                      0.197   \n",
       "\n",
       "                             DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED  ...  \\\n",
       "Genomic Description (GRCh37)                                    ...   \n",
       "22:18561153:A>G                                              C  ...   \n",
       "22:18561156:C>T                                              N  ...   \n",
       "22:18561159:C>T                                              C  ...   \n",
       "22:18561162:C>T                                              N  ...   \n",
       "22:18561164:T>A                                              N  ...   \n",
       "\n",
       "                             VEP_CADD_PHRED VEP_CADD_RAW   VEP_CONSEQUENCE  \\\n",
       "Genomic Description (GRCh37)                                                 \n",
       "22:18561153:A>G                      24.700     3.402999  missense_variant   \n",
       "22:18561156:C>T                       7.598     0.330332  missense_variant   \n",
       "22:18561159:C>T                      23.600     2.995736  missense_variant   \n",
       "22:18561162:C>T                      12.740     0.903099  missense_variant   \n",
       "22:18561164:T>A                      16.050     1.460662  missense_variant   \n",
       "\n",
       "                             VEP_IMPACT VEP_POLYPHEN_PRED VEP_POLYPHEN_SCORE  \\\n",
       "Genomic Description (GRCh37)                                                   \n",
       "22:18561153:A>G                MODERATE                 D              0.997   \n",
       "22:18561156:C>T                MODERATE                 B              0.013   \n",
       "22:18561159:C>T                MODERATE                 D              0.998   \n",
       "22:18561162:C>T                MODERATE                 B              0.003   \n",
       "22:18561164:T>A                MODERATE                 P              0.896   \n",
       "\n",
       "                             VEP_PROTEIN_POS VEP_SIFT_SCORE  ddG |ddG|  \n",
       "Genomic Description (GRCh37)                                            \n",
       "22:18561153:A>G                            4           0.01 -0.7   0.7  \n",
       "22:18561156:C>T                            5           0.24 -0.0   0.0  \n",
       "22:18561159:C>T                            6           0.03 -0.3   0.3  \n",
       "22:18561162:C>T                            7           0.16 -0.0   0.0  \n",
       "22:18561164:T>A                            8           0.02 -0.0   0.0  \n",
       "\n",
       "[5 rows x 38 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Confidence Score', 'DBNSFP_GERP_RS', 'DBNSFP_GERP_RS_PRED',\n",
       "       'DBNSFP_INTERPRO_DOMAIN', 'DBNSFP_LRT_PRED', 'DBNSFP_LRT_SCORE',\n",
       "       'DBNSFP_MUTATIONTASTER_PRED', 'DBNSFP_MUTATIONTASTER_SCORE',\n",
       "       'DBNSFP_PHYLOP30WAY_MAMMALIAN', 'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED',\n",
       "       'DBNSFP_POLYPHEN2_HDIV_PRED', 'DBNSFP_POLYPHEN2_HDIV_SCORE',\n",
       "       'DBNSFP_SIFT_PRED', 'DBNSFP_SIFT_SCORE', 'FINAL_PATHOGENICITY',\n",
       "       'FINAL_PRED', 'GENE', 'Interpro Domain', 'Max MAF (%)',\n",
       "       'Max MAF Source', 'NUM_PATH_PREDS', 'Normalized SA', 'OVERALL_MAX_MAF',\n",
       "       'OVERALL_MAX_MAF_SOURCE', 'Surface Area', 'TOTAL_NUM_PREDS', 'TYPE',\n",
       "       'UNIPROT_DOMAIN', 'VEP_CADD_PHRED', 'VEP_CADD_RAW', 'VEP_CONSEQUENCE',\n",
       "       'VEP_IMPACT', 'VEP_POLYPHEN_PRED', 'VEP_POLYPHEN_SCORE',\n",
       "       'VEP_PROTEIN_POS', 'VEP_SIFT_SCORE', 'ddG', '|ddG|'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn import feature_selection\n",
    "from sklearn import feature_extraction\n",
    "from sklearn.svm import LinearSVC\n",
    "from sklearn.pipeline import Pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "# FINAL_PATHOGENICITY: \"Final curated pathogenicity\" -- includes the DVD manual override\n",
    "# FINAL_PRED: \"Final prediction consensus based on majority vote of prediction scores\" -- should be if NUM_PATH_PREDS / TOTAL_NUM_PREDS >= 0.5 -> Likely pathogenic\n",
    "\n",
    "# Our hypothesis - FINAL_PRED is actually the full Azaiez method result, not just the result of voting from the pathogenicty predictions\n",
    "#                - FINAL_PATHOGENICTY is actually the post curation (overridden) label and is what is displayed on the dvd\n",
    "dvd_labeling_cols = ['NUM_PATH_PREDS', 'TOTAL_NUM_PREDS', 'FINAL_PRED', 'FINAL_PATHOGENICITY']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Up-stream Predictions\n",
    "prediction_cols = ['DBNSFP_GERP_RS_PRED', 'DBNSFP_LRT_PRED', 'DBNSFP_MUTATIONTASTER_PRED', 'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED', 'DBNSFP_POLYPHEN2_HDIV_PRED', 'DBNSFP_SIFT_PRED', 'VEP_POLYPHEN_PRED', 'NUM_PATH_PREDS', 'TOTAL_NUM_PREDS']\n",
    "\n",
    "# Up-stream Scores\n",
    "score_cols = ['DBNSFP_GERP_RS', 'DBNSFP_LRT_SCORE', 'DBNSFP_MUTATIONTASTER_SCORE', 'DBNSFP_PHYLOP30WAY_MAMMALIAN', 'DBNSFP_POLYPHEN2_HDIV_SCORE', 'DBNSFP_SIFT_SCORE', 'VEP_CADD_PHRED', 'VEP_CADD_RAW', 'VEP_POLYPHEN_SCORE', 'VEP_SIFT_SCORE']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED</th>\n",
       "      <th>DBNSFP_POLYPHEN2_HDIV_PRED</th>\n",
       "      <th>DBNSFP_SIFT_PRED</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>NUM_PATH_PREDS</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>C</td>\n",
       "      <td>U</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;D&amp;D&amp;.</td>\n",
       "      <td>.&amp;D&amp;D&amp;D</td>\n",
       "      <td>D</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>N</td>\n",
       "      <td>U</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>N</td>\n",
       "      <td>.&amp;B&amp;B&amp;.</td>\n",
       "      <td>.&amp;T&amp;T&amp;T</td>\n",
       "      <td>B</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>U</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;D&amp;D&amp;.</td>\n",
       "      <td>.&amp;D&amp;D&amp;D</td>\n",
       "      <td>D</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>U</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>N</td>\n",
       "      <td>.&amp;B&amp;B&amp;.</td>\n",
       "      <td>.&amp;T&amp;T&amp;T</td>\n",
       "      <td>B</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>C</td>\n",
       "      <td>U</td>\n",
       "      <td>N&amp;N&amp;N</td>\n",
       "      <td>N</td>\n",
       "      <td>.&amp;P&amp;P&amp;.</td>\n",
       "      <td>.&amp;D&amp;D&amp;D</td>\n",
       "      <td>P</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;D&amp;D&amp;D</td>\n",
       "      <td>C</td>\n",
       "      <td>P&amp;P</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>B</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;D&amp;D&amp;D</td>\n",
       "      <td>C</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>D</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;D&amp;D&amp;D</td>\n",
       "      <td>C</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>D</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;D&amp;D&amp;D</td>\n",
       "      <td>C</td>\n",
       "      <td>P&amp;D</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>P</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;D&amp;D&amp;D</td>\n",
       "      <td>C</td>\n",
       "      <td>P&amp;P</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>P</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_GERP_RS_PRED DBNSFP_LRT_PRED  \\\n",
       "Genomic Description (GRCh37)                                       \n",
       "22:18561153:A>G                                C               U   \n",
       "22:18561156:C>T                                N               U   \n",
       "22:18561159:C>T                                C               U   \n",
       "22:18561162:C>T                                C               U   \n",
       "22:18561164:T>A                                C               U   \n",
       "...                                          ...             ...   \n",
       "15:83561573:G>C                                C               D   \n",
       "15:83561577:T>G                                C               D   \n",
       "15:83561579:A>G                                C               D   \n",
       "15:83561583:T>C                                C               D   \n",
       "15:83561589:G>C                                C               D   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                              \n",
       "22:18561153:A>G                                   N&N&N   \n",
       "22:18561156:C>T                                   N&N&N   \n",
       "22:18561159:C>T                                   N&N&N   \n",
       "22:18561162:C>T                                   N&N&N   \n",
       "22:18561164:T>A                                   N&N&N   \n",
       "...                                                 ...   \n",
       "15:83561573:G>C                                 D&D&D&D   \n",
       "15:83561577:T>G                                 D&D&D&D   \n",
       "15:83561579:A>G                                 D&D&D&D   \n",
       "15:83561583:T>C                                 D&D&D&D   \n",
       "15:83561589:G>C                                 D&D&D&D   \n",
       "\n",
       "                             DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "22:18561153:A>G                                              C   \n",
       "22:18561156:C>T                                              N   \n",
       "22:18561159:C>T                                              C   \n",
       "22:18561162:C>T                                              N   \n",
       "22:18561164:T>A                                              N   \n",
       "...                                                        ...   \n",
       "15:83561573:G>C                                              C   \n",
       "15:83561577:T>G                                              C   \n",
       "15:83561579:A>G                                              C   \n",
       "15:83561583:T>C                                              C   \n",
       "15:83561589:G>C                                              C   \n",
       "\n",
       "                             DBNSFP_POLYPHEN2_HDIV_PRED DBNSFP_SIFT_PRED  \\\n",
       "Genomic Description (GRCh37)                                               \n",
       "22:18561153:A>G                                 .&D&D&.          .&D&D&D   \n",
       "22:18561156:C>T                                 .&B&B&.          .&T&T&T   \n",
       "22:18561159:C>T                                 .&D&D&.          .&D&D&D   \n",
       "22:18561162:C>T                                 .&B&B&.          .&T&T&T   \n",
       "22:18561164:T>A                                 .&P&P&.          .&D&D&D   \n",
       "...                                                 ...              ...   \n",
       "15:83561573:G>C                                     P&P              D&D   \n",
       "15:83561577:T>G                                     D&D              D&D   \n",
       "15:83561579:A>G                                     D&D              D&D   \n",
       "15:83561583:T>C                                     P&D              D&D   \n",
       "15:83561589:G>C                                     P&P              D&D   \n",
       "\n",
       "                             VEP_POLYPHEN_PRED  NUM_PATH_PREDS  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "22:18561153:A>G                              D             4.0   \n",
       "22:18561156:C>T                              B             0.0   \n",
       "22:18561159:C>T                              D             4.0   \n",
       "22:18561162:C>T                              B             1.0   \n",
       "22:18561164:T>A                              P             3.0   \n",
       "...                                        ...             ...   \n",
       "15:83561573:G>C                              B             5.0   \n",
       "15:83561577:T>G                              D             6.0   \n",
       "15:83561579:A>G                              D             6.0   \n",
       "15:83561583:T>C                              P             6.0   \n",
       "15:83561589:G>C                              P             6.0   \n",
       "\n",
       "                              TOTAL_NUM_PREDS  \n",
       "Genomic Description (GRCh37)                   \n",
       "22:18561153:A>G                           5.0  \n",
       "22:18561156:C>T                           5.0  \n",
       "22:18561159:C>T                           5.0  \n",
       "22:18561162:C>T                           5.0  \n",
       "22:18561164:T>A                           5.0  \n",
       "...                                       ...  \n",
       "15:83561573:G>C                           6.0  \n",
       "15:83561577:T>G                           6.0  \n",
       "15:83561579:A>G                           6.0  \n",
       "15:83561583:T>C                           6.0  \n",
       "15:83561589:G>C                           6.0  \n",
       "\n",
       "[128712 rows x 9 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[:, prediction_cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[:, ['NUM_PATH_PREDS', 'TOTAL_NUM_PREDS', 'FINAL_PRED', 'FINAL_PATHOGENICITY', 'FINAL_COMMENTS', 'CURATED_COMMENTS']].to_csv('dvd_pathogenicity_column_subset.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[df.index.str.startswith('5:92929'),  ['NUM_PATH_PREDS', 'TOTAL_NUM_PREDS', 'FINAL_PRED', 'FINAL_PATHOGENICITY', 'FINAL_COMMENTS']].to_csv('out.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NUM_PATH_PREDS</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>FINAL_PRED</th>\n",
       "      <th>FINAL_PATHOGENICITY</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>Likely benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>Likely benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>5.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>Unknown significance</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              NUM_PATH_PREDS  TOTAL_NUM_PREDS  \\\n",
       "Genomic Description (GRCh37)                                    \n",
       "22:18561153:A>G                          4.0              5.0   \n",
       "22:18561156:C>T                          0.0              5.0   \n",
       "22:18561159:C>T                          4.0              5.0   \n",
       "22:18561162:C>T                          1.0              5.0   \n",
       "22:18561164:T>A                          3.0              5.0   \n",
       "...                                      ...              ...   \n",
       "15:83561573:G>C                          5.0              6.0   \n",
       "15:83561577:T>G                          6.0              6.0   \n",
       "15:83561579:A>G                          6.0              6.0   \n",
       "15:83561583:T>C                          6.0              6.0   \n",
       "15:83561589:G>C                          6.0              6.0   \n",
       "\n",
       "                                        FINAL_PRED   FINAL_PATHOGENICITY  \n",
       "Genomic Description (GRCh37)                                              \n",
       "22:18561153:A>G               Unknown significance  Unknown significance  \n",
       "22:18561156:C>T                      Likely benign         Likely benign  \n",
       "22:18561159:C>T               Unknown significance  Unknown significance  \n",
       "22:18561162:C>T                      Likely benign         Likely benign  \n",
       "22:18561164:T>A               Unknown significance  Unknown significance  \n",
       "...                                            ...                   ...  \n",
       "15:83561573:G>C               Unknown significance  Unknown significance  \n",
       "15:83561577:T>G               Unknown significance  Unknown significance  \n",
       "15:83561579:A>G               Unknown significance  Unknown significance  \n",
       "15:83561583:T>C               Unknown significance  Unknown significance  \n",
       "15:83561589:G>C               Unknown significance  Unknown significance  \n",
       "\n",
       "[128712 rows x 4 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df[dvd_labeling_cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>Surface Area</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22:18561153:A&gt;G</th>\n",
       "      <td>3.402999</td>\n",
       "      <td>24.700</td>\n",
       "      <td>0.003487</td>\n",
       "      <td>111.511314</td>\n",
       "      <td>0.697034</td>\n",
       "      <td>45.18</td>\n",
       "      <td>-0.7</td>\n",
       "      <td>0.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561156:C&gt;T</th>\n",
       "      <td>0.330332</td>\n",
       "      <td>7.598</td>\n",
       "      <td>0.003503</td>\n",
       "      <td>131.269338</td>\n",
       "      <td>0.951867</td>\n",
       "      <td>42.57</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561159:C&gt;T</th>\n",
       "      <td>2.995736</td>\n",
       "      <td>23.600</td>\n",
       "      <td>0.001050</td>\n",
       "      <td>114.498499</td>\n",
       "      <td>0.830258</td>\n",
       "      <td>42.35</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561162:C&gt;T</th>\n",
       "      <td>0.903099</td>\n",
       "      <td>12.740</td>\n",
       "      <td>0.003519</td>\n",
       "      <td>161.350503</td>\n",
       "      <td>1.025505</td>\n",
       "      <td>39.05</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22:18561164:T&gt;A</th>\n",
       "      <td>1.460662</td>\n",
       "      <td>16.050</td>\n",
       "      <td>0.001059</td>\n",
       "      <td>118.519368</td>\n",
       "      <td>0.859414</td>\n",
       "      <td>42.94</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561573:G&gt;C</th>\n",
       "      <td>2.865518</td>\n",
       "      <td>23.300</td>\n",
       "      <td>0.000887</td>\n",
       "      <td>5.626061</td>\n",
       "      <td>0.035758</td>\n",
       "      <td>98.35</td>\n",
       "      <td>-0.9</td>\n",
       "      <td>0.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561577:T&gt;G</th>\n",
       "      <td>4.019092</td>\n",
       "      <td>27.900</td>\n",
       "      <td>0.001778</td>\n",
       "      <td>66.834983</td>\n",
       "      <td>0.424787</td>\n",
       "      <td>98.27</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561579:A&gt;G</th>\n",
       "      <td>4.062516</td>\n",
       "      <td>28.200</td>\n",
       "      <td>0.012962</td>\n",
       "      <td>46.255767</td>\n",
       "      <td>0.201331</td>\n",
       "      <td>97.67</td>\n",
       "      <td>-2.6</td>\n",
       "      <td>2.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561583:T&gt;C</th>\n",
       "      <td>3.099789</td>\n",
       "      <td>23.800</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>74.926502</td>\n",
       "      <td>0.381969</td>\n",
       "      <td>95.56</td>\n",
       "      <td>-0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15:83561589:G&gt;C</th>\n",
       "      <td>3.508386</td>\n",
       "      <td>25.000</td>\n",
       "      <td>0.000894</td>\n",
       "      <td>135.808746</td>\n",
       "      <td>0.686764</td>\n",
       "      <td>85.69</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_CADD_RAW  VEP_CADD_PHRED  Max MAF (%)  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "22:18561153:A>G                   3.402999          24.700     0.003487   \n",
       "22:18561156:C>T                   0.330332           7.598     0.003503   \n",
       "22:18561159:C>T                   2.995736          23.600     0.001050   \n",
       "22:18561162:C>T                   0.903099          12.740     0.003519   \n",
       "22:18561164:T>A                   1.460662          16.050     0.001059   \n",
       "...                                    ...             ...          ...   \n",
       "15:83561573:G>C                   2.865518          23.300     0.000887   \n",
       "15:83561577:T>G                   4.019092          27.900     0.001778   \n",
       "15:83561579:A>G                   4.062516          28.200     0.012962   \n",
       "15:83561583:T>C                   3.099789          23.800     0.000000   \n",
       "15:83561589:G>C                   3.508386          25.000     0.000894   \n",
       "\n",
       "                              Surface Area  Normalized SA  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "22:18561153:A>G                 111.511314       0.697034             45.18   \n",
       "22:18561156:C>T                 131.269338       0.951867             42.57   \n",
       "22:18561159:C>T                 114.498499       0.830258             42.35   \n",
       "22:18561162:C>T                 161.350503       1.025505             39.05   \n",
       "22:18561164:T>A                 118.519368       0.859414             42.94   \n",
       "...                                    ...            ...               ...   \n",
       "15:83561573:G>C                   5.626061       0.035758             98.35   \n",
       "15:83561577:T>G                  66.834983       0.424787             98.27   \n",
       "15:83561579:A>G                  46.255767       0.201331             97.67   \n",
       "15:83561583:T>C                  74.926502       0.381969             95.56   \n",
       "15:83561589:G>C                 135.808746       0.686764             85.69   \n",
       "\n",
       "                              ddG  |ddG|  \n",
       "Genomic Description (GRCh37)              \n",
       "22:18561153:A>G              -0.7    0.7  \n",
       "22:18561156:C>T              -0.0    0.0  \n",
       "22:18561159:C>T              -0.3    0.3  \n",
       "22:18561162:C>T              -0.0    0.0  \n",
       "22:18561164:T>A              -0.0    0.0  \n",
       "...                           ...    ...  \n",
       "15:83561573:G>C              -0.9    0.9  \n",
       "15:83561577:T>G              -0.6    0.6  \n",
       "15:83561579:A>G              -2.6    2.6  \n",
       "15:83561583:T>C              -0.5    0.5  \n",
       "15:83561589:G>C              -0.3    0.3  \n",
       "\n",
       "[128712 rows x 8 columns]"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[:, ['VEP_CADD_RAW', 'VEP_CADD_PHRED', 'Max MAF (%)','Surface Area','Normalized SA','Confidence Score', 'ddG', '|ddG|']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'raw_data' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[56], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m y \u001b[39m=\u001b[39m raw_data[\u001b[39m'\u001b[39m\u001b[39mClass\u001b[39m\u001b[39m'\u001b[39m]\n\u001b[1;32m      2\u001b[0m X \u001b[39m=\u001b[39m raw_data\u001b[39m.\u001b[39mdrop(\u001b[39m'\u001b[39m\u001b[39mClass\u001b[39m\u001b[39m'\u001b[39m,axis\u001b[39m=\u001b[39m\u001b[39m1\u001b[39m)\n\u001b[1;32m      4\u001b[0m imputer \u001b[39m=\u001b[39m SimpleImputer(strategy\u001b[39m=\u001b[39m\u001b[39m'\u001b[39m\u001b[39mconstant\u001b[39m\u001b[39m'\u001b[39m,fill_value\u001b[39m=\u001b[39m\u001b[39m0\u001b[39m, missing_values\u001b[39m=\u001b[39mnp\u001b[39m.\u001b[39mnan)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'raw_data' is not defined"
     ]
    }
   ],
   "source": [
    "y = raw_data['Class']\n",
    "X = raw_data.drop('Class',axis=1)\n",
    "\n",
    "imputer = SimpleImputer(strategy='constant',fill_value=0, missing_values=np.nan)\n",
    "imputer = imputer.fit(X[['CADD Phred','MAF','Surface Area','Normalized SA','Confidence Score','ddG']])\n",
    "X[['CADD Phred','MAF','Surface Area','Normalized SA','Confidence Score','ddG']]=imputer.transform(X[['CADD Phred','MAF','Surface Area','Normalized SA','Confidence Score','ddG']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.model_selection import train_test_split\n",
    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2)\n",
    "from sklearn.svm import SVC\n",
    "svclassifier = SVC(kernel='linear',probability=True)\n",
    "svclassifier.fit(X_train, y_train)\n",
    "\n",
    "y_pred_prob = svclassifier.predict_proba(X_test)\n",
    "y_pred_prob\n",
    "y_pred=svclassifier.predict(X_test)\n",
    "display(pd.DataFrame({'svclassifier.predict': y_pred,\n",
    "                     'svclassifier.predict_prob_a[:,0] (Pathogenic)': y_pred_prob[:,0],\n",
    "                     'svclassifier.predict_prob_a[:,1] (Non-Pathogenic)': y_pred_prob[:,1]}).head(100))\n",
    "y_predictions_prob_a = y_pred_prob[:,0]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
