{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from importlib import reload\n",
    "import src.utils as utils\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "chroms_names = ['MT', 'X'] + [str(x) for x in range(1,23)]\n",
    "\n",
    "GENOMIC_DESCRIPTION_COL = 'Genomic Description (GRCh37)'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Detour to Examine the Internal Kafeen State (Azaeiz Method)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'dfs1' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[3], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m df1 \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mconcat(\u001b[43mdfs1\u001b[49m)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'dfs1' is not defined"
     ]
    }
   ],
   "source": [
    "df1 = pd.concat(dfs1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>FINAL_PATHOGENICITY</th>\n",
       "      <th>FINAL_PRED</th>\n",
       "      <th>CURATED_PATHOGENICITY</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>NUM_PATH_PREDS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description GRCh37</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>X:9431334:T&gt;C</th>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X:9431341:T&gt;C</th>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X:9431342:G&gt;GT</th>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X:9431375:G&gt;A</th>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X:9431417:A&gt;G</th>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             FINAL_PATHOGENICITY FINAL_PRED  \\\n",
       "Genomic Description GRCh37                                    \n",
       "X:9431334:T>C               Unknown significance        NaN   \n",
       "X:9431341:T>C               Unknown significance        NaN   \n",
       "X:9431342:G>GT              Unknown significance        NaN   \n",
       "X:9431375:G>A               Unknown significance        NaN   \n",
       "X:9431417:A>G               Unknown significance        NaN   \n",
       "\n",
       "                           CURATED_PATHOGENICITY  TOTAL_NUM_PREDS  \\\n",
       "Genomic Description GRCh37                                          \n",
       "X:9431334:T>C                                NaN              NaN   \n",
       "X:9431341:T>C                                NaN              NaN   \n",
       "X:9431342:G>GT                               NaN              NaN   \n",
       "X:9431375:G>A                                NaN              NaN   \n",
       "X:9431417:A>G                                NaN              NaN   \n",
       "\n",
       "                            NUM_PATH_PREDS  \n",
       "Genomic Description GRCh37                  \n",
       "X:9431334:T>C                          NaN  \n",
       "X:9431341:T>C                          NaN  \n",
       "X:9431342:G>GT                         NaN  \n",
       "X:9431375:G>A                          NaN  \n",
       "X:9431417:A>G                          NaN  "
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df['PREDICTION_RATIO'] = df1['NUM_PATH_PREDS'] / df1['TOTAL_NUM_PREDS']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'df1' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[4], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mdf1\u001b[49m[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mINTERNAL_STATE\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mVUS\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m      2\u001b[0m df1\u001b[38;5;241m.\u001b[39mloc[df1[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mTOTAL_NUM_PREDS\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m>\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m5\u001b[39m, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mINTERNAL_STATE\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mD\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m      3\u001b[0m df1\u001b[38;5;241m.\u001b[39mloc[(df1[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mTOTAL_NUM_PREDS\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m>\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;241m5\u001b[39m) \u001b[38;5;241m&\u001b[39m (df1[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mPREDICTION_RATIO\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m<\u001b[39m \u001b[38;5;241m0.6\u001b[39m), \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mINTERNAL_STATE\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mB\u001b[39m\u001b[38;5;124m'\u001b[39m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'df1' is not defined"
     ]
    }
   ],
   "source": [
    "df1['INTERNAL_STATE'] = 'VUS'\n",
    "df1.loc[df1['TOTAL_NUM_PREDS'] >= 5, 'INTERNAL_STATE'] = 'D'\n",
    "df1.loc[(df1['TOTAL_NUM_PREDS'] >= 5) & (df1['PREDICTION_RATIO'] < 0.6), 'INTERNAL_STATE'] = 'B'\n",
    "df1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "INTERNAL_STATE\n",
       "VUS    2552823\n",
       "D        91470\n",
       "B        48068\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1.INTERNAL_STATE.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FINAL_PATHOGENICITY   INTERNAL_STATE\n",
       "Unknown significance  VUS               2307346\n",
       "Benign                VUS                225772\n",
       "Unknown significance  D                   81886\n",
       "Likely benign         B                   30956\n",
       "Unknown significance  B                   15243\n",
       "Pathogenic            VUS                 10053\n",
       "Likely benign         VUS                  7412\n",
       "Pathogenic            D                    6186\n",
       "Likely pathogenic     VUS                  2217\n",
       "                      D                    2166\n",
       "Benign                B                    1053\n",
       "                      D                     736\n",
       "Pathogenic            B                     450\n",
       "Likely benign         D                     449\n",
       "Likely pathogenic     B                     344\n",
       "Benign*               D                      47\n",
       "                      VUS                    23\n",
       "                      B                      22\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 184,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df1[['FINAL_PATHOGENICITY', 'INTERNAL_STATE']].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "# Data Selection"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Constructing df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "chromosome_dir = 'split_vcf_chromosomes_csvs'\n",
    "\n",
    "if not os.path.isdir(chromosome_dir) or len(os.listdir(chromosome_dir)) == 0:\n",
    "    if not os.path.isdir(chromosome_dir):\n",
    "        os.makedirs(chromosome_dir)\n",
    "\n",
    "    compressed_dir = 'split_vcf_chromosomes_csvs_compressed'\n",
    "    for x in os.scandir(compressed_dir):\n",
    "        utils.extract_tarball(x)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_18_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_18_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_19_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_19_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_X_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_X_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_22_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_22_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_13_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_13_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_14_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_14_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_7_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_7_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_6_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_6_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_1_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_1_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_15_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_15_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_12_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_12_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_21_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_21_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_3_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_3_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_4_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_4_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_10_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_10_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_17_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_17_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_20_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_20_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_MT_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_MT_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_16_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_16_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_11_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_11_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_5_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_5_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_2_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_2_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_8_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_8_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_9_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_9_w_ddg.tar\n"
     ]
    }
   ],
   "source": [
    "compressed_dir = 'compressed_chromosome_w_ddg'\n",
    "if not os.path.isdir(compressed_dir):\n",
    "    os.mkdir(compressed_dir)\n",
    "\n",
    "for x in os.scandir(compressed_dir):\n",
    "    utils.extract_tarball(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/3f/85zthwnn17dfrdm_swynn0qr0000gn/T/ipykernel_70840/827858591.py:8: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  df = pd.concat(dfs).map(utils.safe_decode)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TYPE</th>\n",
       "      <th>GENE</th>\n",
       "      <th>gnomad_MAX_MAF</th>\n",
       "      <th>gnomad_MAX_MAF_SOURCE</th>\n",
       "      <th>gnomad_AC_afr_pass</th>\n",
       "      <th>gnomad_AC_amr_pass</th>\n",
       "      <th>gnomad_AC_eas_pass</th>\n",
       "      <th>gnomad_AC_nfe_pass</th>\n",
       "      <th>gnomad_AN_afr_pass</th>\n",
       "      <th>gnomad_AN_amr_pass</th>\n",
       "      <th>...</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>Surface Area</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "      <th>ca...</th>\n",
       "      <th>in...</th>\n",
       "      <th>calc...</th>\n",
       "      <th>second part</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>172.951510</td>\n",
       "      <td>0.752780</td>\n",
       "      <td>32.73</td>\n",
       "      <td>-1.4</td>\n",
       "      <td>1.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8602.0</td>\n",
       "      <td>810.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>139.065932</td>\n",
       "      <td>0.883870</td>\n",
       "      <td>38.06</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>0.000105</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1492.0</td>\n",
       "      <td>274.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>180.703642</td>\n",
       "      <td>0.670700</td>\n",
       "      <td>37.23</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>0.031875</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>319.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>10008.0</td>\n",
       "      <td>1046.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>76.425233</td>\n",
       "      <td>0.601022</td>\n",
       "      <td>34.61</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.401913</td>\n",
       "      <td>0.578281</td>\n",
       "      <td>31.28</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8702.0</td>\n",
       "      <td>848.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>0.000123</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>16248.0</td>\n",
       "      <td>34588.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16256.0</td>\n",
       "      <td>34588.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>128.397680</td>\n",
       "      <td>0.816259</td>\n",
       "      <td>44.11</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>24966.0</td>\n",
       "      <td>35438.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>67.643291</td>\n",
       "      <td>0.709449</td>\n",
       "      <td>37.21</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128786 rows Ã— 236 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             TYPE   GENE  gnomad_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                               \n",
       "20:21686358:T>C               SNV   PAX1             NaN   \n",
       "20:21686361:C>T               SNV   PAX1        0.000067   \n",
       "20:21686379:G>A               SNV   PAX1        0.000105   \n",
       "20:21686382:C>T               SNV   PAX1        0.031875   \n",
       "20:21686390:G>T               SNV   PAX1             NaN   \n",
       "...                           ...    ...             ...   \n",
       "10:8115915:C>T                SNV  GATA3             NaN   \n",
       "10:8115916:C>G                SNV  GATA3        0.000065   \n",
       "10:8115916:C>T                SNV  GATA3        0.000123   \n",
       "10:8115927:C>T                SNV  GATA3        0.000009   \n",
       "10:8115945:G>A                SNV  GATA3        0.000015   \n",
       "\n",
       "                             gnomad_MAX_MAF_SOURCE  gnomad_AC_afr_pass  \\\n",
       "Genomic Description (GRCh37)                                             \n",
       "20:21686358:T>C                                NaN                 NaN   \n",
       "20:21686361:C>T                 gnomad_AF_nfe_pass                 0.0   \n",
       "20:21686379:G>A                 gnomad_AF_nfe_pass                 0.0   \n",
       "20:21686382:C>T                 gnomad_AF_afr_pass               319.0   \n",
       "20:21686390:G>T                                NaN                 NaN   \n",
       "...                                            ...                 ...   \n",
       "10:8115915:C>T                                 NaN                 NaN   \n",
       "10:8115916:C>G                  gnomad_AF_nfe_pass                 0.0   \n",
       "10:8115916:C>T                  gnomad_AF_afr_pass                 2.0   \n",
       "10:8115927:C>T                  gnomad_AF_nfe_pass                 0.0   \n",
       "10:8115945:G>A                  gnomad_AF_nfe_pass                 0.0   \n",
       "\n",
       "                              gnomad_AC_amr_pass  gnomad_AC_eas_pass  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "20:21686358:T>C                              NaN                 NaN   \n",
       "20:21686361:C>T                              0.0                 0.0   \n",
       "20:21686379:G>A                              0.0                 0.0   \n",
       "20:21686382:C>T                              1.0                 0.0   \n",
       "20:21686390:G>T                              NaN                 NaN   \n",
       "...                                          ...                 ...   \n",
       "10:8115915:C>T                               NaN                 NaN   \n",
       "10:8115916:C>G                               0.0                 0.0   \n",
       "10:8115916:C>T                               1.0                 0.0   \n",
       "10:8115927:C>T                               0.0                 0.0   \n",
       "10:8115945:G>A                               0.0                 0.0   \n",
       "\n",
       "                              gnomad_AC_nfe_pass  gnomad_AN_afr_pass  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "20:21686358:T>C                              NaN                 NaN   \n",
       "20:21686361:C>T                              1.0              8602.0   \n",
       "20:21686379:G>A                              1.0              1492.0   \n",
       "20:21686382:C>T                              4.0             10008.0   \n",
       "20:21686390:G>T                              NaN                 NaN   \n",
       "...                                          ...                 ...   \n",
       "10:8115915:C>T                               NaN                 NaN   \n",
       "10:8115916:C>G                               1.0              8702.0   \n",
       "10:8115916:C>T                               2.0             16248.0   \n",
       "10:8115927:C>T                               1.0             16256.0   \n",
       "10:8115945:G>A                               2.0             24966.0   \n",
       "\n",
       "                              gnomad_AN_amr_pass  ...      Max MAF Source  \\\n",
       "Genomic Description (GRCh37)                      ...                       \n",
       "20:21686358:T>C                              NaN  ...                 NaN   \n",
       "20:21686361:C>T                            810.0  ...  gnomad_AF_nfe_pass   \n",
       "20:21686379:G>A                            274.0  ...  gnomad_AF_nfe_pass   \n",
       "20:21686382:C>T                           1046.0  ...  gnomad_AF_afr_pass   \n",
       "20:21686390:G>T                              NaN  ...                 NaN   \n",
       "...                                          ...  ...                 ...   \n",
       "10:8115915:C>T                               NaN  ...                 NaN   \n",
       "10:8115916:C>G                             848.0  ...  gnomad_AF_nfe_pass   \n",
       "10:8115916:C>T                           34588.0  ...  gnomad_AF_afr_pass   \n",
       "10:8115927:C>T                           34588.0  ...  gnomad_AF_nfe_pass   \n",
       "10:8115945:G>A                           35438.0  ...  gnomad_AF_nfe_pass   \n",
       "\n",
       "                              Surface Area  Normalized SA  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                 172.951510       0.752780             32.73   \n",
       "20:21686361:C>T                 139.065932       0.883870             38.06   \n",
       "20:21686379:G>A                 180.703642       0.670700             37.23   \n",
       "20:21686382:C>T                  76.425233       0.601022             34.61   \n",
       "20:21686390:G>T                  99.401913       0.578281             31.28   \n",
       "...                                    ...            ...               ...   \n",
       "10:8115915:C>T                  136.946574       0.870607             47.29   \n",
       "10:8115916:C>G                  136.946574       0.870607             47.29   \n",
       "10:8115916:C>T                  136.946574       0.870607             47.29   \n",
       "10:8115927:C>T                  128.397680       0.816259             44.11   \n",
       "10:8115945:G>A                   67.643291       0.709449             37.21   \n",
       "\n",
       "                              ddG  |ddG|  ca...  in...  calc...  second part  \n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C              -1.4    1.4    NaN    NaN      NaN          NaN  \n",
       "20:21686361:C>T              -0.0    0.0    NaN    NaN      NaN          NaN  \n",
       "20:21686379:G>A              -0.3    0.3    NaN    NaN      NaN          NaN  \n",
       "20:21686382:C>T              -0.3    0.3    NaN    NaN      NaN          NaN  \n",
       "20:21686390:G>T              -0.6    0.6    NaN    NaN      NaN          NaN  \n",
       "...                           ...    ...    ...    ...      ...          ...  \n",
       "10:8115915:C>T               -0.3    0.3    NaN    NaN      NaN          NaN  \n",
       "10:8115916:C>G               -0.3    0.3    NaN    NaN      NaN          NaN  \n",
       "10:8115916:C>T               -0.2    0.2    NaN    NaN      NaN          NaN  \n",
       "10:8115927:C>T               -0.1    0.1    NaN    NaN      NaN          NaN  \n",
       "10:8115945:G>A                0.2    0.2    NaN    NaN      NaN          NaN  \n",
       "\n",
       "[128786 rows x 236 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = []\n",
    "id_cols = ['CHROM', 'POS', 'REF', 'ALT']\n",
    "dup_cols = [f\"{base_col}_{alt}\" for base_col in id_cols for alt in ['x', 'y']]\n",
    "for f in os.scandir('chromosomes_w_ddg'):\n",
    "    df = pd.read_csv(f.path, index_col=0, encoding='utf-8', low_memory=False)\n",
    "    df = df.drop(columns=dup_cols)\n",
    "    dfs.append(df)\n",
    "df = pd.concat(dfs).map(utils.safe_decode)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Columns to Drop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "cols_to_drop = []\n",
    "\n",
    "# Colummns containing various identifiers from other sources\n",
    "id_cols = df.columns[df.columns.str.contains('ID') & ~df.columns.str.contains('CONFIDENCE')].tolist()\n",
    "\n",
    "# gnomad columns - mostly just allele frequency filters\n",
    "gnomad_filter_cols = df.columns[df.columns.str.contains('gnomad') & df.columns.str.contains('pass')].tolist()\n",
    "gnomad_filter_cols.append('gnomad_filters')\n",
    "\n",
    "# Columns containing minor allele frequency (MAF)\n",
    "# max_maf_cols = df.columns[df.columns.str.contains('MAF')].tolist()\n",
    "\n",
    "# Errant columns, unknown origin\n",
    "errant_cols = [' second part', ' in...', ' ca...', ' calc...']\n",
    "\n",
    "# DVD/Kafeen Columns\n",
    "dvd_kafeen_cols = ['CLASSIFY_TRAIL', 'CLASSIFY_TRAIL_KEYS', 'FINAL_COMMENTS', 'FINAL_PATHOGENICITY_REASON', 'CURATED_DECISION_DATE', 'PREF_TRANSCRIPT']\n",
    "\n",
    "# Columns containing variant location information\n",
    "genomic_protein_loc_cols = ['VEP_HGVS_P', 'VEP_HGVS_C', 'CLINVAR_HGVS_P', 'CLINVAR_HGVS_C', 'Protein Position']\n",
    "gene_description_cols = ['HGMD_OMIM_REF'] + ['VEP_FEATURE'] # Unsure what VEP_FEATURE is, but it looks like an identifier\n",
    "\n",
    "# Comment columns\n",
    "comment_cols = ['FINAL_COMMENTS', 'CURATED_COMMENTS']\n",
    "\n",
    "# Misc columns\n",
    "misc = ['VEP_OTHER_FEATURES']\n",
    "\n",
    "cols_to_drop += id_cols + gnomad_filter_cols + errant_cols + dvd_kafeen_cols + genomic_protein_loc_cols + gene_description_cols + misc + comment_cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "keep_cols = ['NUM_PATH_PREDS',\n",
    " 'TOTAL_NUM_PREDS',\n",
    " 'FINAL_PRED',\n",
    " 'GENE',\n",
    " 'OVERALL_MAX_MAF',\n",
    " 'OVERALL_MAX_MAF_SOURCE',\n",
    " 'FINAL_PATHOGENICITY',\n",
    " 'CURATED_PATHOGENICITY']\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Alternate Approach"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "feature_sorting_df = pd.read_csv('feature_sorting_updated.csv', header=None, index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['NUM_PATH_PREDS',\n",
       " 'TOTAL_NUM_PREDS',\n",
       " 'FINAL_PRED',\n",
       " 'GENE',\n",
       " 'OVERALL_MAX_MAF',\n",
       " 'OVERALL_MAX_MAF_SOURCE',\n",
       " 'FINAL_PATHOGENICITY',\n",
       " 'CURATED_PATHOGENICITY']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_sorting_df.loc['Definitely Keep (used previously)'].dropna().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "kept_features_df = feature_sorting_df.loc[~feature_sorting_df.index.str.contains('Not Keep'), :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "46"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kept_features_list = sorted(kept_features_df.unstack().dropna().unique().tolist())\n",
    "len(kept_features_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "pathogenicity_label_cols = ['CLINVAR_CLNSIG', 'CLINVAR_PATHOGENIC', 'CURATED_PATHOGENICITY']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Selecting Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CADD Phred',\n",
       " 'CLINVAR_CLNSIG',\n",
       " 'CLINVAR_PATHOGENIC',\n",
       " 'CURATED_PATHOGENICITY',\n",
       " 'Confidence Score',\n",
       " 'Consequence',\n",
       " 'DBNSFP_GERP_RS',\n",
       " 'DBNSFP_GERP_RS_PRED',\n",
       " 'DBNSFP_INTERPRO_DOMAIN',\n",
       " 'DBNSFP_LRT_PRED',\n",
       " 'DBNSFP_LRT_SCORE',\n",
       " 'DBNSFP_MUTATIONTASTER_PRED',\n",
       " 'DBNSFP_MUTATIONTASTER_SCORE',\n",
       " 'DBNSFP_PHYLOP30WAY_MAMMALIAN',\n",
       " 'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED',\n",
       " 'DBNSFP_POLYPHEN2_HDIV_PRED',\n",
       " 'DBNSFP_POLYPHEN2_HDIV_SCORE',\n",
       " 'DBNSFP_SIFT_PRED',\n",
       " 'DBNSFP_SIFT_SCORE',\n",
       " 'FINAL_PATHOGENICITY',\n",
       " 'FINAL_PRED',\n",
       " 'GENE',\n",
       " 'Impact',\n",
       " 'Interpro Domain',\n",
       " 'Max MAF (%)',\n",
       " 'Max MAF Source',\n",
       " 'NUM_PATH_PREDS',\n",
       " 'Normalized SA',\n",
       " 'OVERALL_MAX_MAF',\n",
       " 'OVERALL_MAX_MAF_SOURCE',\n",
       " 'Surface Area',\n",
       " 'TOTAL_NUM_PREDS',\n",
       " 'TYPE',\n",
       " 'UNIPROT_DOMAIN',\n",
       " 'VEP_CADD_PHRED',\n",
       " 'VEP_CADD_RAW',\n",
       " 'VEP_CONSEQUENCE',\n",
       " 'VEP_FEATURE',\n",
       " 'VEP_IMPACT',\n",
       " 'VEP_POLYPHEN_PRED',\n",
       " 'VEP_POLYPHEN_SCORE',\n",
       " 'VEP_PROTEIN_POS',\n",
       " 'VEP_SIFT_SCORE',\n",
       " 'Variant Classification',\n",
       " 'ddG',\n",
       " '|ddG|']"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kept_features_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>CLINVAR_CLNSIG</th>\n",
       "      <th>CLINVAR_PATHOGENIC</th>\n",
       "      <th>CURATED_PATHOGENICITY</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>VEP_CONSEQUENCE</th>\n",
       "      <th>VEP_FEATURE</th>\n",
       "      <th>VEP_IMPACT</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>Variant Classification</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>23.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32.73</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.596</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-1.4</td>\n",
       "      <td>1.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.06</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.917</td>\n",
       "      <td>4</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.90</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.23</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2.32</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.787</td>\n",
       "      <td>10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.61</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2.11</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>11</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>22.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.28</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3.51</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.005</td>\n",
       "      <td>14</td>\n",
       "      <td>0.04</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>27.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>27.50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>29.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>14.78</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.11</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3.68</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.097</td>\n",
       "      <td>426</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>27.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.21</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>432</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128786 rows Ã— 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD Phred CLINVAR_CLNSIG  CLINVAR_PATHOGENIC  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                    23.60            NaN                 NaN   \n",
       "20:21686361:C>T                    23.00            NaN                 NaN   \n",
       "20:21686379:G>A                    22.90            NaN                 NaN   \n",
       "20:21686382:C>T                    21.80            NaN                 NaN   \n",
       "20:21686390:G>T                    22.10            NaN                 NaN   \n",
       "...                                  ...            ...                 ...   \n",
       "10:8115915:C>T                     27.20            NaN                 NaN   \n",
       "10:8115916:C>G                     27.50            NaN                 NaN   \n",
       "10:8115916:C>T                     29.20            NaN                 NaN   \n",
       "10:8115927:C>T                     14.78            NaN                 NaN   \n",
       "10:8115945:G>A                     27.60            NaN                 NaN   \n",
       "\n",
       "                             CURATED_PATHOGENICITY  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "20:21686358:T>C                                NaN             32.73   \n",
       "20:21686361:C>T                                NaN             38.06   \n",
       "20:21686379:G>A                                NaN             37.23   \n",
       "20:21686382:C>T                                NaN             34.61   \n",
       "20:21686390:G>T                                NaN             31.28   \n",
       "...                                            ...               ...   \n",
       "10:8115915:C>T                                 NaN             47.29   \n",
       "10:8115916:C>G                                 NaN             47.29   \n",
       "10:8115916:C>T                                 NaN             47.29   \n",
       "10:8115927:C>T                                 NaN             44.11   \n",
       "10:8115945:G>A                                 NaN             37.21   \n",
       "\n",
       "                                   Consequence  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686358:T>C               missense_variant            4.28   \n",
       "20:21686361:C>T               missense_variant            4.28   \n",
       "20:21686379:G>A               missense_variant            2.32   \n",
       "20:21686382:C>T               missense_variant            2.11   \n",
       "20:21686390:G>T               missense_variant            3.51   \n",
       "...                                        ...             ...   \n",
       "10:8115915:C>T                missense_variant            5.59   \n",
       "10:8115916:C>G                missense_variant            5.59   \n",
       "10:8115916:C>T                missense_variant            5.59   \n",
       "10:8115927:C>T                missense_variant            3.68   \n",
       "10:8115945:G>A                missense_variant            5.59   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_INTERPRO_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686358:T>C                                C                    .&.   \n",
       "20:21686361:C>T                                C                    .&.   \n",
       "20:21686379:G>A                                C                    .&.   \n",
       "20:21686382:C>T                                C                    .&.   \n",
       "20:21686390:G>T                                C                    .&.   \n",
       "...                                          ...                    ...   \n",
       "10:8115915:C>T                                 C                  .&.&.   \n",
       "10:8115916:C>G                                 C                  .&.&.   \n",
       "10:8115916:C>T                                 C                  .&.&.   \n",
       "10:8115927:C>T                                 C                  .&.&.   \n",
       "10:8115945:G>A                                 C                  .&.&.   \n",
       "\n",
       "                             DBNSFP_LRT_PRED  ...   VEP_CONSEQUENCE  \\\n",
       "Genomic Description (GRCh37)                  ...                     \n",
       "20:21686358:T>C                          NaN  ...  missense_variant   \n",
       "20:21686361:C>T                          NaN  ...  missense_variant   \n",
       "20:21686379:G>A                          NaN  ...  missense_variant   \n",
       "20:21686382:C>T                          NaN  ...  missense_variant   \n",
       "20:21686390:G>T                          NaN  ...  missense_variant   \n",
       "...                                      ...  ...               ...   \n",
       "10:8115915:C>T                             U  ...  missense_variant   \n",
       "10:8115916:C>G                             U  ...  missense_variant   \n",
       "10:8115916:C>T                             U  ...  missense_variant   \n",
       "10:8115927:C>T                             D  ...  missense_variant   \n",
       "10:8115945:G>A                             D  ...  missense_variant   \n",
       "\n",
       "                                 VEP_FEATURE VEP_IMPACT  VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                                 \n",
       "20:21686358:T>C                  NM_006192.5   MODERATE                  P   \n",
       "20:21686361:C>T                  NM_006192.5   MODERATE                  P   \n",
       "20:21686379:G>A                  NM_006192.5   MODERATE                  P   \n",
       "20:21686382:C>T                  NM_006192.5   MODERATE                  P   \n",
       "20:21686390:G>T                  NM_006192.5   MODERATE                  B   \n",
       "...                                      ...        ...                ...   \n",
       "10:8115915:C>T                NM_001002295.2   MODERATE                  D   \n",
       "10:8115916:C>G                NM_001002295.2   MODERATE                  D   \n",
       "10:8115916:C>T                NM_001002295.2   MODERATE                  D   \n",
       "10:8115927:C>T                NM_001002295.2   MODERATE                  B   \n",
       "10:8115945:G>A                NM_001002295.2   MODERATE                  D   \n",
       "\n",
       "                             VEP_POLYPHEN_SCORE VEP_PROTEIN_POS  \\\n",
       "Genomic Description (GRCh37)                                      \n",
       "20:21686358:T>C                           0.596               3   \n",
       "20:21686361:C>T                           0.917               4   \n",
       "20:21686379:G>A                           0.787              10   \n",
       "20:21686382:C>T                           0.804              11   \n",
       "20:21686390:G>T                           0.005              14   \n",
       "...                                         ...             ...   \n",
       "10:8115915:C>T                            1.000             422   \n",
       "10:8115916:C>G                            1.000             422   \n",
       "10:8115916:C>T                            1.000             422   \n",
       "10:8115927:C>T                            0.097             426   \n",
       "10:8115945:G>A                            0.997             432   \n",
       "\n",
       "                             VEP_SIFT_SCORE Variant Classification  ddG |ddG|  \n",
       "Genomic Description (GRCh37)                                                   \n",
       "20:21686358:T>C                        0.00   Unknown significance -1.4   1.4  \n",
       "20:21686361:C>T                        0.00   Unknown significance -0.0   0.0  \n",
       "20:21686379:G>A                        0.00   Unknown significance -0.3   0.3  \n",
       "20:21686382:C>T                        0.01                 Benign -0.3   0.3  \n",
       "20:21686390:G>T                        0.04   Unknown significance -0.6   0.6  \n",
       "...                                     ...                    ...  ...   ...  \n",
       "10:8115915:C>T                         0.01      Likely pathogenic -0.3   0.3  \n",
       "10:8115916:C>G                         0.00   Unknown significance -0.3   0.3  \n",
       "10:8115916:C>T                         0.00   Unknown significance -0.2   0.2  \n",
       "10:8115927:C>T                         1.00   Unknown significance -0.1   0.1  \n",
       "10:8115945:G>A                         0.00   Unknown significance  0.2   0.2  \n",
       "\n",
       "[128786 rows x 46 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df = df[kept_features_list]\n",
    "working_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Exploring Prediction Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>C</td>\n",
       "      <td>4.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>4.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>C</td>\n",
       "      <td>2.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>2.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>3.51</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_GERP_RS_PRED  DBNSFP_GERP_RS\n",
       "Genomic Description (GRCh37)                                    \n",
       "20:21686358:T>C                                C            4.28\n",
       "20:21686361:C>T                                C            4.28\n",
       "20:21686379:G>A                                C            2.32\n",
       "20:21686382:C>T                                C            2.11\n",
       "20:21686390:G>T                                C            3.51"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# In GERP, RS > 0 => C == conserved\n",
    "# In GERP, RS < 0 => N == not conserved (?)\n",
    "gerp_pos_df = df.loc[df.DBNSFP_GERP_RS > 0, ['DBNSFP_GERP_RS_PRED', 'DBNSFP_GERP_RS']]\n",
    "gerp_pos_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>N</td>\n",
       "      <td>-0.290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686457:G&gt;A</th>\n",
       "      <td>N</td>\n",
       "      <td>-0.296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686459:G&gt;A</th>\n",
       "      <td>N</td>\n",
       "      <td>-8.940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686468:G&gt;A</th>\n",
       "      <td>N</td>\n",
       "      <td>-1.940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686502:G&gt;T</th>\n",
       "      <td>N</td>\n",
       "      <td>-0.914</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_GERP_RS_PRED  DBNSFP_GERP_RS\n",
       "Genomic Description (GRCh37)                                    \n",
       "20:21686402:G>C                                N          -0.290\n",
       "20:21686457:G>A                                N          -0.296\n",
       "20:21686459:G>A                                N          -8.940\n",
       "20:21686468:G>A                                N          -1.940\n",
       "20:21686502:G>T                                N          -0.914"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gerp_neg_df = df.loc[df.DBNSFP_GERP_RS < 0, ['DBNSFP_GERP_RS_PRED', 'DBNSFP_GERP_RS']]\n",
    "gerp_neg_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>VEP_SIFT_PRED</th>\n",
       "      <th>DBNSFP_SIFT_PRED</th>\n",
       "      <th>DBNSFP_SIFT_SCORE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.001&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.0&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.0&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>0.01</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.002&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>0.04</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.006&amp;.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_SIFT_SCORE VEP_SIFT_PRED DBNSFP_SIFT_PRED  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                         0.00          D-LC              D&.   \n",
       "20:21686361:C>T                         0.00          D-LC              D&.   \n",
       "20:21686379:G>A                         0.00          D-LC              D&.   \n",
       "20:21686382:C>T                         0.01          D-LC              D&.   \n",
       "20:21686390:G>T                         0.04          D-LC              D&.   \n",
       "\n",
       "                             DBNSFP_SIFT_SCORE  \n",
       "Genomic Description (GRCh37)                    \n",
       "20:21686358:T>C                        0.001&.  \n",
       "20:21686361:C>T                          0.0&.  \n",
       "20:21686379:G>A                          0.0&.  \n",
       "20:21686382:C>T                        0.002&.  \n",
       "20:21686390:G>T                        0.006&.  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sift_df = df.loc[:, df.columns.str.contains('SIFT')]\n",
    "sift_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>VEP_SIFT_PRED</th>\n",
       "      <th>DBNSFP_SIFT_PRED</th>\n",
       "      <th>DBNSFP_SIFT_SCORE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.001&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.0&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.0&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>0.01</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.002&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>0.04</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.006&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>0.01</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;.&amp;D</td>\n",
       "      <td>0.014&amp;.&amp;0.014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;.&amp;D</td>\n",
       "      <td>0.008&amp;.&amp;0.008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;.&amp;D</td>\n",
       "      <td>0.002&amp;.&amp;0.003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>1.00</td>\n",
       "      <td>T</td>\n",
       "      <td>T&amp;.&amp;T</td>\n",
       "      <td>1.0&amp;.&amp;1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;.&amp;D</td>\n",
       "      <td>0.007&amp;.&amp;0.007</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128737 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_SIFT_SCORE VEP_SIFT_PRED DBNSFP_SIFT_PRED  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                         0.00          D-LC              D&.   \n",
       "20:21686361:C>T                         0.00          D-LC              D&.   \n",
       "20:21686379:G>A                         0.00          D-LC              D&.   \n",
       "20:21686382:C>T                         0.01          D-LC              D&.   \n",
       "20:21686390:G>T                         0.04          D-LC              D&.   \n",
       "...                                      ...           ...              ...   \n",
       "10:8115915:C>T                          0.01             D            D&.&D   \n",
       "10:8115916:C>G                          0.00             D            D&.&D   \n",
       "10:8115916:C>T                          0.00             D            D&.&D   \n",
       "10:8115927:C>T                          1.00             T            T&.&T   \n",
       "10:8115945:G>A                          0.00             D            D&.&D   \n",
       "\n",
       "                             DBNSFP_SIFT_SCORE  \n",
       "Genomic Description (GRCh37)                    \n",
       "20:21686358:T>C                        0.001&.  \n",
       "20:21686361:C>T                          0.0&.  \n",
       "20:21686379:G>A                          0.0&.  \n",
       "20:21686382:C>T                        0.002&.  \n",
       "20:21686390:G>T                        0.006&.  \n",
       "...                                        ...  \n",
       "10:8115915:C>T                   0.014&.&0.014  \n",
       "10:8115916:C>G                   0.008&.&0.008  \n",
       "10:8115916:C>T                   0.002&.&0.003  \n",
       "10:8115927:C>T                       1.0&.&1.0  \n",
       "10:8115945:G>A                   0.007&.&0.007  \n",
       "\n",
       "[128737 rows x 4 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[~df.VEP_SIFT_PRED.isna(), df.columns.str.contains('SIFT')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>DBNSFP_POLYPHEN2_HDIV_PRED</th>\n",
       "      <th>DBNSFP_POLYPHEN2_HDIV_SCORE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>0.596</td>\n",
       "      <td>P</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>0.596&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>0.917</td>\n",
       "      <td>P</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>0.917&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>0.787</td>\n",
       "      <td>P</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>0.787&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>0.804</td>\n",
       "      <td>P</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>0.804&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>0.005</td>\n",
       "      <td>B</td>\n",
       "      <td>B&amp;.</td>\n",
       "      <td>0.005&amp;.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_POLYPHEN_SCORE VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                         \n",
       "20:21686358:T>C                            0.596                 P   \n",
       "20:21686361:C>T                            0.917                 P   \n",
       "20:21686379:G>A                            0.787                 P   \n",
       "20:21686382:C>T                            0.804                 P   \n",
       "20:21686390:G>T                            0.005                 B   \n",
       "\n",
       "                             DBNSFP_POLYPHEN2_HDIV_PRED  \\\n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686358:T>C                                     P&.   \n",
       "20:21686361:C>T                                     P&.   \n",
       "20:21686379:G>A                                     P&.   \n",
       "20:21686382:C>T                                     P&.   \n",
       "20:21686390:G>T                                     B&.   \n",
       "\n",
       "                             DBNSFP_POLYPHEN2_HDIV_SCORE  \n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686358:T>C                                  0.596&.  \n",
       "20:21686361:C>T                                  0.917&.  \n",
       "20:21686379:G>A                                  0.787&.  \n",
       "20:21686382:C>T                                  0.804&.  \n",
       "20:21686390:G>T                                  0.005&.  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##INFO=<ID=DBNSFP_POLYPHEN2_HDIV_PRED,Number=1,Type=String,Description=\"\\\"(from dbNSFP) Polyphen2 prediction based on HumDiv, D (probably damaging, HDIV score in [0.957,1] or rankscore in [0.55859,0.91137]), P (possibly damaging, HDIV score in [0.454,0.956] or rankscore in [0.37043,0.55681]) and B (benign, HDIV score in [0,0.452] or rankscore in [0.03061,0.36974]). Score cutoff for binary classification is 0.5 for HDIV score or 0.38028 for rankscore, i.e. the prediction is neutral if the HDIV score is smaller than 0.5 (rankscore is smaller than 0.38028), and deleterious if the HDIV score is larger than 0.5 (rankscore is larger than 0.38028). Multiple entries are separated by ;, corresponding to Uniprot_acc.\\\"\">\n",
    "##INFO=<ID=DBNSFP_POLYPHEN2_HDIV_SCORE,Number=1,Type=String,Description=\"\\\"(from dbNSFP) Polyphen2 score based on HumDiv, i.e. hdiv_prob. The score ranges from 0 to 1. Multiple entries separated by ;, corresponding to Uniprot_acc.\\\"\">\n",
    "polyphen_df = df.loc[:, df.columns.str.contains('POLYPHEN')]\n",
    "polyphen_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "VEP_POLYPHEN_PRED\n",
       "D    55263\n",
       "B    51063\n",
       "P    22231\n",
       "U      210\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "polyphen_df.VEP_POLYPHEN_PRED.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>N</td>\n",
       "      <td>0.506194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>N</td>\n",
       "      <td>0.758794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>N</td>\n",
       "      <td>0.999272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>N</td>\n",
       "      <td>0.998176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>N</td>\n",
       "      <td>0.999492</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686358:T>C                                       N   \n",
       "20:21686361:C>T                                       N   \n",
       "20:21686379:G>A                                       N   \n",
       "20:21686382:C>T                                       N   \n",
       "20:21686390:G>T                                       N   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_SCORE  \n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686358:T>C                                 0.506194  \n",
       "20:21686361:C>T                                 0.758794  \n",
       "20:21686379:G>A                                 0.999272  \n",
       "20:21686382:C>T                                 0.998176  \n",
       "20:21686390:G>T                                 0.999492  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##INFO=<ID=DBNSFP_MUTATIONTASTER_PRED,Number=1,Type=String,Description=\"\\\"(from dbNSFP) MutationTaster prediction, A (disease_causing_automatic), D (disease_causing), N (polymorphism) or P (polymorphism_automatic). The score cutoff between D and N is 0.5 for MTnew and 0.31733 for the rankscore.\\\"\">\n",
    "##INFO=<ID=DBNSFP_MUTATIONTASTER_SCORE,Number=1,Type=String,Description=\"\\\"(from dbNSFP) MutationTaster p-value (MTori), ranges from 0 to 1. Multiple scores are separated by ;. Information on corresponding transcript(s) can be found by querying http://www.mutationtaster.org/ChrPos.html\\\"\">\n",
    "mutationtaster_df = df.loc[:, df.columns.str.contains('MUTATIONTASTER')]\n",
    "mutationtaster_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "D&D 21946\n",
      "D 17739\n",
      "D&D&D 15324\n",
      "N&N 12588\n",
      "N 10410\n",
      "D&D&D&D 9660\n",
      "N&N&N 6795\n",
      "D&D&D&D&D 5238\n",
      "D&D&D&D&D&D 3929\n",
      "N&N&N&N 2700\n",
      "N&N&N&N&N 2023\n",
      "D&D&D&D&D&D&D&D 1904\n",
      "D&D&D&D&D&D&D 1637\n",
      "D&N 1555\n",
      "D&D&D&D&D&D&D&D&D 1550\n",
      "N&N&N&N&N&N 1368\n",
      "D&D&D&D&D&D&D&D&D&D&D 1123\n",
      "N&N&N&N&N&N&N&N 1033\n",
      "D&N&N 961\n",
      "N&N&N&N&N&N&N 838\n",
      "D&D&N 822\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&D&D&D&D 573\n",
      "N&N&N&N&N&N&N&N&N 541\n",
      "D&N&N&N 337\n",
      "D&D&D&N 334\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&D&D&D 329\n",
      "A&A 318\n",
      "D&D&D&D&D&D&D&D&D&D 275\n",
      "D&D&D&D&N 229\n",
      "D&N&N&N&N 164\n",
      "P&P 155\n",
      "D&D&N&N 148\n",
      "D&D&D&D&N&N&N&N 138\n",
      "D&D&D&D&D&D&D&D&N 129\n",
      "D&D&D&N&N 128\n",
      "D&D&D&D&N&N 124\n",
      "D&D&D&D&D&D&N&N&N 123\n",
      "D&N&N&N&N&N&N 119\n",
      "D&D&N&N&N 119\n",
      "D&D&D&D&D&D&N 96\n",
      "D&D&D&D&D&D&D&D&D&D&D&D 94\n",
      "P 89\n",
      "A 88\n",
      "N&N&N&N&N&N&N&N&N&N&N&N 84\n",
      "D&D&D&D&D&N 84\n",
      "D&D&D&D&D&D&D&N&N 83\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&D&D&D&N 76\n",
      "A&A&A&A 70\n",
      "D&D&D&D&N&N&N&N&N 69\n",
      "D&D&D&D&D&D&D&D&D&D&N&N 69\n",
      "P&P&P 68\n",
      "D&D&N&N&N&N 62\n",
      "D&D&D&N&N&N 58\n",
      "D&N&N&N&N&N&N&N&N 54\n",
      "D&D&D&N&N&D 47\n",
      "D&D&N&N&N&N&N 46\n",
      "A&A&A 44\n",
      "D&D&D&D&D&D&N&N 44\n",
      "A&A&A&A&A&A&A&A&A&A&A 42\n",
      "D&D&D&D&D&D&D&D&N&N&N 41\n",
      "D&D&D&N&N&N&N&N&N&N 41\n",
      "D&D&N&N&N&N&N&N&N&N 40\n",
      "D&N&N&N&N&N&N&N&N&N 36\n",
      "D&D&D&D&D&D&D&D&D&D&N 36\n",
      "D&N&N&N&N&N 36\n",
      "A&A&A&A&A&A 36\n",
      "D&D&D&D&N&N&N&N&N&N&N&N&N&N&N 34\n",
      "N&N&N&N&N&D 34\n",
      "D&D&D&D&D&D&D&N 33\n",
      "N&N&N&N&N&N&N&N&N&N 32\n",
      "D&N&N&N&D&D 32\n",
      "D&N&N&N&N&N&N&N&N&N&N&N&N&N&N&N 29\n",
      "D&N&N&N&N&N&N&N 26\n",
      "A&A&A&A&A 26\n",
      "P&P&P&P 25\n",
      "D&D&D&N&N&N&N&N&N 24\n",
      "P&P&P&P&P 23\n",
      "D&D&D&D&D&N&N&N&N 21\n",
      "D&D&N&N&N&N&N&N 21\n",
      "D&D&D&D&D&N&N&N 21\n",
      "N&N&N&N&N&N&N&N&N&N&N 20\n",
      "N&N&N&D&D 20\n",
      "D&D&N&N&N&N&N&N&N 19\n",
      "A&A&A&A&A&A&A 19\n",
      "N&N&N&N&N&N&N&N&N&N&N&N&N&N&N&N 17\n",
      "D&D&D&D&D&N&N 16\n",
      "D&D&D&D&D&D&N&N&N&N&N&N 14\n",
      "P&P&P&P&P&P&P 13\n",
      "D&D&N&N&N&N&N&N&N&N&N&N&N&N&N&N 13\n",
      "N&N&N&N&N&N&N&N&N&N&N&N&N&N&N 12\n",
      "D&D&D&N&N&N&N&N 12\n",
      "A&A&A&A&A&A&A&A 12\n",
      "D&D&D&D&N&N&N&N&N&N&N&N&N&N&N&N 12\n",
      "P&P&P&P&P&P&P&P&P 11\n",
      "D&D&D&D&N&N&N 11\n",
      "D&D&D&N&N&N&N&N&N&N&N 10\n",
      "P&P&P&P&P&P&P&P 10\n",
      "D&D&D&D&N&N&N&N&N&N&N&N 10\n",
      "D&D&D&N&N&N&N 10\n",
      "D&D&N&N&N&D 9\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&D&N&N&N 9\n",
      "D&D&N&D 8\n",
      "N&D&D 8\n",
      "D&N&N&N&N&N&N&N&N&N&N 6\n",
      "N&N&N&D 6\n",
      "D&D&D&N&N&N&N&N&N&N&N&N&N&N&N&N 6\n",
      "P&P&P&P&P&P 6\n",
      "P&P&P&P&P&P&P&P&P&P&P&P&P&P&P&P 5\n",
      "D&D&D&D&D&N&N&N&N&N&N&N 5\n",
      "A&A&A&A&A&A&A&A&A&A&A&A 5\n",
      "D&D&D&D&D&D&D&D&N&N&N&N&N&N&D&D 4\n",
      "D&D&D&D&D&N&N&N&N&N 4\n",
      "N&D&N 4\n",
      "A&N&N&N&N&N&N&N&N 4\n",
      "D&D&N&N&N&N&N&N&N&N&N&N 4\n",
      "D&D&D&D&N&N&N&N&N&N&N 4\n",
      "D&D&N&N&N&N&N&N&N&N&N 4\n",
      "D&D&D&D&D&D&D&D&D&D&D&N 3\n",
      "D&D&D&D&D&N&N&N&N&N&N&N&N&N&N 3\n",
      "A&A&A&A&A&A&A&A&A 3\n",
      "D&D&D&D&D&D&D&N&N&N&N 3\n",
      "N&N&N&N&N&N&N&N&D 3\n",
      "D&D&D&D&D&D&N&N&N&N&N 3\n",
      "A&D&D 3\n",
      "D&N&N&D 3\n",
      "D&D&D&D&D&D&D&D&D&N 2\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&N&N&N&N 2\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&D&D&N&N 2\n",
      "D&D&D&D&D&N&N&N&N&N&N&N&N&N&N&N 2\n",
      "A&N&N 2\n",
      "D&D&D&D&D&D&D&N&N&N&N&N&N&N&N 2\n",
      "D&D&D&D&D&D&D&D&D&N&N&N&N&N&N 2\n",
      "D&D&D&D&D&D&D&D&D&D&D&N&N&N&N 2\n",
      "A&A&A&A&A&A&A&A&A&A&A&A&A&A&A&A 2\n",
      "N&N&N&N&N&N&N&N&N&N&N&N&N&N&D&D 2\n",
      "A&D 2\n",
      "D&D&N&N&D&D 2\n",
      "N&N&N&N&D&D 2\n",
      "N&D 2\n",
      "D&D&D&D&D&D&D&D&N&N&N&N&N&N&N 2\n",
      "A&D&D&D&D&D&D&D&D&D&D 1\n",
      "D&D&N&N&N&N&D&D&N&N&N&N 1\n",
      "P&P&P&P&P&P&P&P&P&P&P&P 1\n",
      "D&D&D&D&D&D&D&D&D&N&N&N&N&N&N&N 1\n",
      "A&D&D&D&D 1\n",
      "A&N 1\n",
      "A&D&D&D&D&D&D 1\n",
      "D&D&D&D&D&D&D&D&N&N&N&N 1\n",
      "D&D&D&D&D&D&N&N&N&N&N&N&N&N&N&N 1\n",
      "A&D&D&D&D&D 1\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&D&D&N 1\n",
      "A&A&D&D&D&D&D&D 1\n",
      "A&D&D&D&D&D&D&D&D 1\n",
      "D&D&D&D&D&N&N&N&N&N&N 1\n",
      "A&A&D&D&D&N&N&N 1\n",
      "D&D&D&D&D&D&N&N&N&N&N&N&N&N&N 1\n"
     ]
    }
   ],
   "source": [
    "for i, v in mutationtaster_df.DBNSFP_MUTATIONTASTER_PRED.value_counts().items():\n",
    "    print(i,v)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_LRT_PRED  DBNSFP_LRT_SCORE\n",
       "Genomic Description (GRCh37)                                  \n",
       "20:21686358:T>C                          NaN               NaN\n",
       "20:21686361:C>T                          NaN               NaN\n",
       "20:21686379:G>A                          NaN               NaN\n",
       "20:21686382:C>T                          NaN               NaN\n",
       "20:21686390:G>T                          NaN               NaN"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##INFO=<ID=DBNSFP_LRT_PRED,Number=1,Type=String,Description=\"\\\"(from dbNSFP) LRT prediction, D(eleterious), N(eutral) or U(nknown), which is not solely determined by the score.\\\"\">\n",
    "##INFO=<ID=DBNSFP_LRT_SCORE,Number=1,Type=String,Description=\"\\\"(from dbNSFP) The original LRT two-sided p-value (LRTori), ranges from 0 to 1.\\\"\">\n",
    "lrt_df = df.loc[:, df.columns.str.contains('LRT')]\n",
    "lrt_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>1.111</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>0.118</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>1.147</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>0.106</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>1.147</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              DBNSFP_PHYLOP30WAY_MAMMALIAN  \\\n",
       "Genomic Description (GRCh37)                                 \n",
       "20:21686358:T>C                                      1.111   \n",
       "20:21686361:C>T                                      0.118   \n",
       "20:21686379:G>A                                      1.147   \n",
       "20:21686382:C>T                                      0.106   \n",
       "20:21686390:G>T                                      1.147   \n",
       "\n",
       "                             DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED  \n",
       "Genomic Description (GRCh37)                                    \n",
       "20:21686358:T>C                                              C  \n",
       "20:21686361:C>T                                              N  \n",
       "20:21686379:G>A                                              C  \n",
       "20:21686382:C>T                                              N  \n",
       "20:21686390:G>T                                              C  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##INFO=<ID=DBNSFP_PHYLOP30WAY_MAMMALIAN,Number=1,Type=String,Description=\"\\\"(from dbNSFP) phyloP (phylogenetic p-values) conservation score based on the multiple alignments of 30 mammalian genomes (including human). The larger the score, the more conserved the site. Scores range from -20 to 1.312 in dbNSFP.\\\"\">\n",
    "##INFO=<ID=DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED,Number=.,Type=String,Description=\"NA\">\n",
    "phylop_df = df.loc[:, df.columns.str.contains('PHYLOP')]\n",
    "phylop_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['DBNSFP_PHYLOP30WAY_MAMMALIAN', 'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED'], dtype='object')"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns[df.columns.str.contains('PHYLOP')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "score_cols = [\n",
    "    'VEP_SIFT_SCORE', \n",
    "    'VEP_POLYPHEN_SCORE',\n",
    "    'DBNSFP_SIFT_SCORE',\n",
    "    'DBNSFP_POLYPHEN2_HDIV_SCORE',\n",
    "    'DBNSFP_LRT_SCORE',\n",
    "    'DBNSFP_MUTATIONTASTER_SCORE',\n",
    "    'DBNSFP_GERP_RS',\n",
    "    'DBNSFP_PHYLOP30WAY_MAMMALIAN'\n",
    "]\n",
    "\n",
    "predictor_cols = [\n",
    "    'VEP_SIFT_PRED', \n",
    "    'VEP_POLYPHEN_PRED',\n",
    "    'DBNSFP_SIFT_PRED',\n",
    "    'DBNSFP_POLYPHEN2_HDIV_PRED',\n",
    "    'DBNSFP_LRT_PRED',\n",
    "    'DBNSFP_MUTATIONTASTER_PRED',\n",
    "    'DBNSFP_GERP_RS_PRED',\n",
    "    'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED'\n",
    "]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exploring Other Columns in working_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>Interpro Domain</th>\n",
       "      <th>UNIPROT_DOMAIN</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_INTERPRO_DOMAIN Interpro Domain  \\\n",
       "Genomic Description (GRCh37)                                          \n",
       "20:21686358:T>C                                 .&.             NaN   \n",
       "20:21686361:C>T                                 .&.             NaN   \n",
       "20:21686379:G>A                                 .&.             NaN   \n",
       "20:21686382:C>T                                 .&.             NaN   \n",
       "20:21686390:G>T                                 .&.             NaN   \n",
       "\n",
       "                             UNIPROT_DOMAIN  \n",
       "Genomic Description (GRCh37)                 \n",
       "20:21686358:T>C                         NaN  \n",
       "20:21686361:C>T                         NaN  \n",
       "20:21686379:G>A                         NaN  \n",
       "20:21686382:C>T                         NaN  \n",
       "20:21686390:G>T                         NaN  "
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "domain_df = working_df.loc[:, working_df.columns.str.contains('Domain') | working_df.columns.str.contains('DOMAIN')]\n",
    "domain_cols = domain_df.columns.tolist()\n",
    "domain_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>23.6</td>\n",
       "      <td>23.6</td>\n",
       "      <td>3.009015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>2.751772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.9</td>\n",
       "      <td>22.9</td>\n",
       "      <td>2.662314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.8</td>\n",
       "      <td>21.8</td>\n",
       "      <td>2.257539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>22.1</td>\n",
       "      <td>22.1</td>\n",
       "      <td>2.319001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD Phred  VEP_CADD_PHRED  VEP_CADD_RAW\n",
       "Genomic Description (GRCh37)                                          \n",
       "20:21686358:T>C                     23.6            23.6      3.009015\n",
       "20:21686361:C>T                     23.0            23.0      2.751772\n",
       "20:21686379:G>A                     22.9            22.9      2.662314\n",
       "20:21686382:C>T                     21.8            21.8      2.257539\n",
       "20:21686390:G>T                     22.1            22.1      2.319001"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##INFO=<ID=VEP_CADD_RAW,Number=1,Type=String,Description=\"\\\"VEP annotated CADD raw score.\\\"\">\n",
    "##INFO=<ID=VEP_CADD_PHRED,Number=1,Type=String,Description=\"\\\"VEP annotated CADD phred scaled score.\\\"\">\n",
    "cadd_df = working_df.loc[:, working_df.columns.str.contains('Cadd') | working_df.columns.str.contains('CADD')]\n",
    "cadd_cols = cadd_df.columns.tolist()\n",
    "cadd_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# FINAL_PATHOGENICITY: \"Final curated pathogenicity\" -- includes the DVD manual override\n",
    "# FINAL_PRED: \"Final prediction consensus based on majority vote of prediction scores\" -- should be if NUM_PATH_PREDS / TOTAL_NUM_PREDS >= 0.5 -> Likely pathogenic\n",
    "\n",
    "# Our hypothesis - FINAL_PRED is actually the full Azaiez method result, not just the result of voting from the pathogenicty predictions\n",
    "#                - FINAL_PATHOGENICTY is actually the post curation (overridden) label and is what is displayed on the dvd\n",
    "dvd_labeling_cols = ['NUM_PATH_PREDS', 'TOTAL_NUM_PREDS', 'FINAL_PRED', 'FINAL_PATHOGENICITY']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OVERALL_MAX_MAF</th>\n",
       "      <th>OVERALL_MAX_MAF_SOURCE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>6.65956e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>0.000104932</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>0.0318745</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             OVERALL_MAX_MAF OVERALL_MAX_MAF_SOURCE\n",
       "Genomic Description (GRCh37)                                       \n",
       "20:21686358:T>C                          NaN                    NaN\n",
       "20:21686361:C>T                  6.65956e-05     gnomad_AF_nfe_pass\n",
       "20:21686379:G>A                  0.000104932     gnomad_AF_nfe_pass\n",
       "20:21686382:C>T                    0.0318745     gnomad_AF_afr_pass\n",
       "20:21686390:G>T                          NaN                    NaN"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf_df = working_df.loc[:, ['OVERALL_MAX_MAF', 'OVERALL_MAX_MAF_SOURCE']]\n",
    "maf_cols = maf_df.columns.tolist()\n",
    "maf_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "other_ddg_file_cols = ['Confidence Score', 'Max MAF (%)', 'Max MAF Source', 'Surface Area', 'Normalized SA', 'ddG', '|ddG|']\n",
    "other_dvd_cols = ['VEP_IMPACT', 'VEP_CONSEQUENCE', 'VEP_PROTEIN_POS']\n",
    "classification_cols = ['FINAL_PATHOGENICITY', 'FINAL_PRED', 'NUM_PATH_PREDS', 'TOTAL_NUM_PREDS']\n",
    "misc_details_cols = ['GENE', 'TYPE']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Identifying Duplicate Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>CLINVAR_CLNSIG</th>\n",
       "      <th>CLINVAR_PATHOGENIC</th>\n",
       "      <th>CURATED_PATHOGENICITY</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>VEP_CONSEQUENCE</th>\n",
       "      <th>VEP_FEATURE</th>\n",
       "      <th>VEP_IMPACT</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>Variant Classification</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>23.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32.73</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.596</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-1.4</td>\n",
       "      <td>1.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.06</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.917</td>\n",
       "      <td>4</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.90</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.23</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2.32</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.787</td>\n",
       "      <td>10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.61</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2.11</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>11</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>22.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.28</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3.51</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.005</td>\n",
       "      <td>14</td>\n",
       "      <td>0.04</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>27.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>27.50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>29.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>14.78</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.11</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3.68</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.097</td>\n",
       "      <td>426</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>27.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.21</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>432</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128786 rows Ã— 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD Phred CLINVAR_CLNSIG  CLINVAR_PATHOGENIC  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                    23.60            NaN                 NaN   \n",
       "20:21686361:C>T                    23.00            NaN                 NaN   \n",
       "20:21686379:G>A                    22.90            NaN                 NaN   \n",
       "20:21686382:C>T                    21.80            NaN                 NaN   \n",
       "20:21686390:G>T                    22.10            NaN                 NaN   \n",
       "...                                  ...            ...                 ...   \n",
       "10:8115915:C>T                     27.20            NaN                 NaN   \n",
       "10:8115916:C>G                     27.50            NaN                 NaN   \n",
       "10:8115916:C>T                     29.20            NaN                 NaN   \n",
       "10:8115927:C>T                     14.78            NaN                 NaN   \n",
       "10:8115945:G>A                     27.60            NaN                 NaN   \n",
       "\n",
       "                             CURATED_PATHOGENICITY  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "20:21686358:T>C                                NaN             32.73   \n",
       "20:21686361:C>T                                NaN             38.06   \n",
       "20:21686379:G>A                                NaN             37.23   \n",
       "20:21686382:C>T                                NaN             34.61   \n",
       "20:21686390:G>T                                NaN             31.28   \n",
       "...                                            ...               ...   \n",
       "10:8115915:C>T                                 NaN             47.29   \n",
       "10:8115916:C>G                                 NaN             47.29   \n",
       "10:8115916:C>T                                 NaN             47.29   \n",
       "10:8115927:C>T                                 NaN             44.11   \n",
       "10:8115945:G>A                                 NaN             37.21   \n",
       "\n",
       "                                   Consequence  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686358:T>C               missense_variant            4.28   \n",
       "20:21686361:C>T               missense_variant            4.28   \n",
       "20:21686379:G>A               missense_variant            2.32   \n",
       "20:21686382:C>T               missense_variant            2.11   \n",
       "20:21686390:G>T               missense_variant            3.51   \n",
       "...                                        ...             ...   \n",
       "10:8115915:C>T                missense_variant            5.59   \n",
       "10:8115916:C>G                missense_variant            5.59   \n",
       "10:8115916:C>T                missense_variant            5.59   \n",
       "10:8115927:C>T                missense_variant            3.68   \n",
       "10:8115945:G>A                missense_variant            5.59   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_INTERPRO_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686358:T>C                                C                    .&.   \n",
       "20:21686361:C>T                                C                    .&.   \n",
       "20:21686379:G>A                                C                    .&.   \n",
       "20:21686382:C>T                                C                    .&.   \n",
       "20:21686390:G>T                                C                    .&.   \n",
       "...                                          ...                    ...   \n",
       "10:8115915:C>T                                 C                  .&.&.   \n",
       "10:8115916:C>G                                 C                  .&.&.   \n",
       "10:8115916:C>T                                 C                  .&.&.   \n",
       "10:8115927:C>T                                 C                  .&.&.   \n",
       "10:8115945:G>A                                 C                  .&.&.   \n",
       "\n",
       "                             DBNSFP_LRT_PRED  ...   VEP_CONSEQUENCE  \\\n",
       "Genomic Description (GRCh37)                  ...                     \n",
       "20:21686358:T>C                          NaN  ...  missense_variant   \n",
       "20:21686361:C>T                          NaN  ...  missense_variant   \n",
       "20:21686379:G>A                          NaN  ...  missense_variant   \n",
       "20:21686382:C>T                          NaN  ...  missense_variant   \n",
       "20:21686390:G>T                          NaN  ...  missense_variant   \n",
       "...                                      ...  ...               ...   \n",
       "10:8115915:C>T                             U  ...  missense_variant   \n",
       "10:8115916:C>G                             U  ...  missense_variant   \n",
       "10:8115916:C>T                             U  ...  missense_variant   \n",
       "10:8115927:C>T                             D  ...  missense_variant   \n",
       "10:8115945:G>A                             D  ...  missense_variant   \n",
       "\n",
       "                                 VEP_FEATURE VEP_IMPACT  VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                                 \n",
       "20:21686358:T>C                  NM_006192.5   MODERATE                  P   \n",
       "20:21686361:C>T                  NM_006192.5   MODERATE                  P   \n",
       "20:21686379:G>A                  NM_006192.5   MODERATE                  P   \n",
       "20:21686382:C>T                  NM_006192.5   MODERATE                  P   \n",
       "20:21686390:G>T                  NM_006192.5   MODERATE                  B   \n",
       "...                                      ...        ...                ...   \n",
       "10:8115915:C>T                NM_001002295.2   MODERATE                  D   \n",
       "10:8115916:C>G                NM_001002295.2   MODERATE                  D   \n",
       "10:8115916:C>T                NM_001002295.2   MODERATE                  D   \n",
       "10:8115927:C>T                NM_001002295.2   MODERATE                  B   \n",
       "10:8115945:G>A                NM_001002295.2   MODERATE                  D   \n",
       "\n",
       "                             VEP_POLYPHEN_SCORE VEP_PROTEIN_POS  \\\n",
       "Genomic Description (GRCh37)                                      \n",
       "20:21686358:T>C                           0.596               3   \n",
       "20:21686361:C>T                           0.917               4   \n",
       "20:21686379:G>A                           0.787              10   \n",
       "20:21686382:C>T                           0.804              11   \n",
       "20:21686390:G>T                           0.005              14   \n",
       "...                                         ...             ...   \n",
       "10:8115915:C>T                            1.000             422   \n",
       "10:8115916:C>G                            1.000             422   \n",
       "10:8115916:C>T                            1.000             422   \n",
       "10:8115927:C>T                            0.097             426   \n",
       "10:8115945:G>A                            0.997             432   \n",
       "\n",
       "                             VEP_SIFT_SCORE Variant Classification  ddG |ddG|  \n",
       "Genomic Description (GRCh37)                                                   \n",
       "20:21686358:T>C                        0.00   Unknown significance -1.4   1.4  \n",
       "20:21686361:C>T                        0.00   Unknown significance -0.0   0.0  \n",
       "20:21686379:G>A                        0.00   Unknown significance -0.3   0.3  \n",
       "20:21686382:C>T                        0.01                 Benign -0.3   0.3  \n",
       "20:21686390:G>T                        0.04   Unknown significance -0.6   0.6  \n",
       "...                                     ...                    ...  ...   ...  \n",
       "10:8115915:C>T                         0.01      Likely pathogenic -0.3   0.3  \n",
       "10:8115916:C>G                         0.00   Unknown significance -0.3   0.3  \n",
       "10:8115916:C>T                         0.00   Unknown significance -0.2   0.2  \n",
       "10:8115927:C>T                         1.00   Unknown significance -0.1   0.1  \n",
       "10:8115945:G>A                         0.00   Unknown significance  0.2   0.2  \n",
       "\n",
       "[128786 rows x 46 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df = df[kept_features_list]\n",
    "working_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Consquence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_CONSEQUENCE</th>\n",
       "      <th>Consequence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [VEP_CONSEQUENCE, Consequence]\n",
       "Index: []"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[working_df.VEP_CONSEQUENCE != working_df.Consequence, ['VEP_CONSEQUENCE', 'Consequence']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.drop(columns=['VEP_CONSEQUENCE'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### CADD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CADD Phred', 'VEP_CADD_PHRED', 'VEP_CADD_RAW'], dtype='object')"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.columns[working_df.columns.str.contains('CADD') | working_df.columns.str.contains('Cadd')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These values are all missing even if it says they're not matching. Also are all MNV -> dropping these then re-examining"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TYPE</th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:61468569:CCT&gt;GCC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2:228111435:TG&gt;CA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2:73613119:GG&gt;CA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2:73829349:GCTC&gt;ACAT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18:44102214:GC&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18:44219603:AG&gt;CC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4:6302568:TC&gt;AG</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4:6303899:CGC&gt;GAG</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4:6304085:TCA&gt;CCG</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3:69990450:TT&gt;CC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5:89986675:GC&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5:90119324:GT&gt;AA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12:48370339:TC&gt;GA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12:49441832:GC&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12:49445061:TC&gt;CG</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12:49445372:TGTG&gt;CATA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:145583732:AG&gt;CT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:24811038:CTC&gt;TCT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:38285911:GA&gt;TC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:61655487:AA&gt;GC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:61728967:GG&gt;TC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:61736569:GG&gt;AT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:61769079:GC&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17:46606943:GA&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17:72915617:CT&gt;GA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11:17593707:TCT&gt;CCC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11:17662466:CC&gt;AT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11:76910678:CG&gt;TA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11:76916512:GC&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6:133844297:TAC&gt;AAA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6:133846381:GC&gt;AA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1:160011388:GT&gt;AA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1:215848515:AG&gt;CC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1:216595236:AA&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1:247587982:CTG&gt;TTT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X:107865120:AA&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X:107930811:GC&gt;CT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7:5569281:TC&gt;AA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:55568971:CT&gt;GG</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:73466709:TG&gt;CA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             TYPE  CADD Phred  VEP_CADD_PHRED\n",
       "Genomic Description (GRCh37)                                 \n",
       "20:61468569:CCT>GCC           MNV         NaN             NaN\n",
       "2:228111435:TG>CA             MNV         NaN             NaN\n",
       "2:73613119:GG>CA              MNV         NaN             NaN\n",
       "2:73829349:GCTC>ACAT          MNV         NaN             NaN\n",
       "18:44102214:GC>TT             MNV         NaN             NaN\n",
       "18:44219603:AG>CC             MNV         NaN             NaN\n",
       "4:6302568:TC>AG               MNV         NaN             NaN\n",
       "4:6303899:CGC>GAG             MNV         NaN             NaN\n",
       "4:6304085:TCA>CCG             MNV         NaN             NaN\n",
       "3:69990450:TT>CC              MNV         NaN             NaN\n",
       "5:89986675:GC>TT              MNV         NaN             NaN\n",
       "5:90119324:GT>AA              MNV         NaN             NaN\n",
       "12:48370339:TC>GA             MNV         NaN             NaN\n",
       "12:49441832:GC>TT             MNV         NaN             NaN\n",
       "12:49445061:TC>CG             MNV         NaN             NaN\n",
       "12:49445372:TGTG>CATA         MNV         NaN             NaN\n",
       "8:145583732:AG>CT             MNV         NaN             NaN\n",
       "8:24811038:CTC>TCT            MNV         NaN             NaN\n",
       "8:38285911:GA>TC              MNV         NaN             NaN\n",
       "8:61655487:AA>GC              MNV         NaN             NaN\n",
       "8:61728967:GG>TC              MNV         NaN             NaN\n",
       "8:61736569:GG>AT              MNV         NaN             NaN\n",
       "8:61769079:GC>TT              MNV         NaN             NaN\n",
       "17:46606943:GA>TT             MNV         NaN             NaN\n",
       "17:72915617:CT>GA             MNV         NaN             NaN\n",
       "11:17593707:TCT>CCC           MNV         NaN             NaN\n",
       "11:17662466:CC>AT             MNV         NaN             NaN\n",
       "11:76910678:CG>TA             MNV         NaN             NaN\n",
       "11:76916512:GC>TT             MNV         NaN             NaN\n",
       "6:133844297:TAC>AAA           MNV         NaN             NaN\n",
       "6:133846381:GC>AA             MNV         NaN             NaN\n",
       "1:160011388:GT>AA             MNV         NaN             NaN\n",
       "1:215848515:AG>CC             MNV         NaN             NaN\n",
       "1:216595236:AA>TT             MNV         NaN             NaN\n",
       "1:247587982:CTG>TTT           MNV         NaN             NaN\n",
       "X:107865120:AA>TT             MNV         NaN             NaN\n",
       "X:107930811:GC>CT             MNV         NaN             NaN\n",
       "7:5569281:TC>AA               MNV         NaN             NaN\n",
       "10:55568971:CT>GG             MNV         NaN             NaN\n",
       "10:73466709:TG>CA             MNV         NaN             NaN"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[working_df.VEP_CADD_PHRED != working_df['CADD Phred'], ['TYPE', 'CADD Phred', 'VEP_CADD_PHRED']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df[working_df.VEP_CADD_PHRED == working_df['CADD Phred']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TYPE</th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [TYPE, CADD Phred, VEP_CADD_PHRED]\n",
       "Index: []"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[working_df.VEP_CADD_PHRED != working_df['CADD Phred'], ['TYPE', 'CADD Phred', 'VEP_CADD_PHRED']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.drop(columns='VEP_CADD_PHRED')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Impact"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>CLINVAR_CLNSIG</th>\n",
       "      <th>CLINVAR_PATHOGENIC</th>\n",
       "      <th>CURATED_PATHOGENICITY</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_FEATURE</th>\n",
       "      <th>VEP_IMPACT</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>Variant Classification</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows Ã— 44 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [CADD Phred, CLINVAR_CLNSIG, CLINVAR_PATHOGENIC, CURATED_PATHOGENICITY, Confidence Score, Consequence, DBNSFP_GERP_RS, DBNSFP_GERP_RS_PRED, DBNSFP_INTERPRO_DOMAIN, DBNSFP_LRT_PRED, DBNSFP_LRT_SCORE, DBNSFP_MUTATIONTASTER_PRED, DBNSFP_MUTATIONTASTER_SCORE, DBNSFP_PHYLOP30WAY_MAMMALIAN, DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED, DBNSFP_POLYPHEN2_HDIV_PRED, DBNSFP_POLYPHEN2_HDIV_SCORE, DBNSFP_SIFT_PRED, DBNSFP_SIFT_SCORE, FINAL_PATHOGENICITY, FINAL_PRED, GENE, Impact, Interpro Domain, Max MAF (%), Max MAF Source, NUM_PATH_PREDS, Normalized SA, OVERALL_MAX_MAF, OVERALL_MAX_MAF_SOURCE, Surface Area, TOTAL_NUM_PREDS, TYPE, UNIPROT_DOMAIN, VEP_CADD_RAW, VEP_FEATURE, VEP_IMPACT, VEP_POLYPHEN_PRED, VEP_POLYPHEN_SCORE, VEP_PROTEIN_POS, VEP_SIFT_SCORE, Variant Classification, ddG, |ddG|]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 44 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df[working_df['VEP_IMPACT'] != working_df['Impact']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.drop(columns='VEP_IMPACT')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Max MAF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>OVERALL_MAX_MAF</th>\n",
       "      <th>OVERALL_MAX_MAF_SOURCE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>0.006660</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.65956e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>0.010493</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.000104932</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>3.187450</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>0.0318745</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>0.006484</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.48424e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>0.012309</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>0.000123092</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>0.000880</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>8.80065e-06</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>0.001549</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>1.54947e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128746 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Max MAF (%)      Max MAF Source OVERALL_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                                                    \n",
       "20:21686358:T>C                  0.000000                 NaN             NaN   \n",
       "20:21686361:C>T                  0.006660  gnomad_AF_nfe_pass     6.65956e-05   \n",
       "20:21686379:G>A                  0.010493  gnomad_AF_nfe_pass     0.000104932   \n",
       "20:21686382:C>T                  3.187450  gnomad_AF_afr_pass       0.0318745   \n",
       "20:21686390:G>T                  0.000000                 NaN             NaN   \n",
       "...                                   ...                 ...             ...   \n",
       "10:8115915:C>T                   0.000000                 NaN              []   \n",
       "10:8115916:C>G                   0.006484  gnomad_AF_nfe_pass     6.48424e-05   \n",
       "10:8115916:C>T                   0.012309  gnomad_AF_afr_pass     0.000123092   \n",
       "10:8115927:C>T                   0.000880  gnomad_AF_nfe_pass     8.80065e-06   \n",
       "10:8115945:G>A                   0.001549  gnomad_AF_nfe_pass     1.54947e-05   \n",
       "\n",
       "                             OVERALL_MAX_MAF_SOURCE  \n",
       "Genomic Description (GRCh37)                         \n",
       "20:21686358:T>C                                 NaN  \n",
       "20:21686361:C>T                  gnomad_AF_nfe_pass  \n",
       "20:21686379:G>A                  gnomad_AF_nfe_pass  \n",
       "20:21686382:C>T                  gnomad_AF_afr_pass  \n",
       "20:21686390:G>T                                 NaN  \n",
       "...                                             ...  \n",
       "10:8115915:C>T                                   []  \n",
       "10:8115916:C>G                   gnomad_AF_nfe_pass  \n",
       "10:8115916:C>T                   gnomad_AF_afr_pass  \n",
       "10:8115927:C>T                   gnomad_AF_nfe_pass  \n",
       "10:8115945:G>A                   gnomad_AF_nfe_pass  \n",
       "\n",
       "[128746 rows x 4 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df[['Max MAF (%)', 'Max MAF Source', 'OVERALL_MAX_MAF', 'OVERALL_MAX_MAF_SOURCE']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Everywhere that Max MAF Source does not match OVERALL_MAX_MAF_SOURCE also does not have any MAF data\n",
    "\n",
    "Mismatches in Max MAF % are just rounding errors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>OVERALL_MAX_MAF</th>\n",
       "      <th>OVERALL_MAX_MAF_SOURCE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686398:G&gt;T</th>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686443:C&gt;A</th>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686443:C&gt;G</th>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Max MAF (%) Max MAF Source OVERALL_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                                               \n",
       "20:21686358:T>C                       0.0            NaN             NaN   \n",
       "20:21686390:G>T                       0.0            NaN             NaN   \n",
       "20:21686398:G>T                       0.0            NaN             NaN   \n",
       "20:21686443:C>A                       0.0            NaN             NaN   \n",
       "20:21686443:C>G                       0.0            NaN             NaN   \n",
       "\n",
       "                             OVERALL_MAX_MAF_SOURCE  \n",
       "Genomic Description (GRCh37)                         \n",
       "20:21686358:T>C                                 NaN  \n",
       "20:21686390:G>T                                 NaN  \n",
       "20:21686398:G>T                                 NaN  \n",
       "20:21686443:C>A                                 NaN  \n",
       "20:21686443:C>G                                 NaN  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp_df = working_df.loc[working_df['Max MAF Source'] != working_df['OVERALL_MAX_MAF_SOURCE'], ['Max MAF (%)', 'Max MAF Source', 'OVERALL_MAX_MAF', 'OVERALL_MAX_MAF_SOURCE']]\n",
    "temp_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>OVERALL_MAX_MAF</th>\n",
       "      <th>OVERALL_MAX_MAF_SOURCE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>0.006660</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.65956e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>0.010493</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.000104932</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>3.187450</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>0.0318745</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Max MAF (%)      Max MAF Source OVERALL_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                                                    \n",
       "20:21686358:T>C                  0.000000                 NaN             NaN   \n",
       "20:21686361:C>T                  0.006660  gnomad_AF_nfe_pass     6.65956e-05   \n",
       "20:21686379:G>A                  0.010493  gnomad_AF_nfe_pass     0.000104932   \n",
       "20:21686382:C>T                  3.187450  gnomad_AF_afr_pass       0.0318745   \n",
       "20:21686390:G>T                  0.000000                 NaN             NaN   \n",
       "\n",
       "                             OVERALL_MAX_MAF_SOURCE  \n",
       "Genomic Description (GRCh37)                         \n",
       "20:21686358:T>C                                 NaN  \n",
       "20:21686361:C>T                  gnomad_AF_nfe_pass  \n",
       "20:21686379:G>A                  gnomad_AF_nfe_pass  \n",
       "20:21686382:C>T                  gnomad_AF_afr_pass  \n",
       "20:21686390:G>T                                 NaN  "
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp_df = working_df.loc[working_df['Max MAF (%)'] != working_df['OVERALL_MAX_MAF'], ['Max MAF (%)', 'Max MAF Source', 'OVERALL_MAX_MAF', 'OVERALL_MAX_MAF_SOURCE']]\n",
    "temp_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.drop(columns=['OVERALL_MAX_MAF', 'OVERALL_MAX_MAF_SOURCE'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Variant Classification</th>\n",
       "      <th>FINAL_PATHOGENICITY</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Variant Classification, FINAL_PATHOGENICITY]\n",
       "Index: []"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[working_df['Variant Classification'] != working_df['FINAL_PATHOGENICITY'], ['Variant Classification', 'FINAL_PATHOGENICITY']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.drop(columns='FINAL_PATHOGENICITY')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CADD Phred', 'CLINVAR_CLNSIG', 'CLINVAR_PATHOGENIC',\n",
       "       'CURATED_PATHOGENICITY', 'Confidence Score', 'Consequence',\n",
       "       'DBNSFP_GERP_RS', 'DBNSFP_GERP_RS_PRED', 'DBNSFP_INTERPRO_DOMAIN',\n",
       "       'DBNSFP_LRT_PRED', 'DBNSFP_LRT_SCORE', 'DBNSFP_MUTATIONTASTER_PRED',\n",
       "       'DBNSFP_MUTATIONTASTER_SCORE', 'DBNSFP_PHYLOP30WAY_MAMMALIAN',\n",
       "       'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED', 'DBNSFP_POLYPHEN2_HDIV_PRED',\n",
       "       'DBNSFP_POLYPHEN2_HDIV_SCORE', 'DBNSFP_SIFT_PRED', 'DBNSFP_SIFT_SCORE',\n",
       "       'FINAL_PRED', 'GENE', 'Impact', 'Interpro Domain', 'Max MAF (%)',\n",
       "       'Max MAF Source', 'NUM_PATH_PREDS', 'Normalized SA', 'Surface Area',\n",
       "       'TOTAL_NUM_PREDS', 'TYPE', 'UNIPROT_DOMAIN', 'VEP_CADD_RAW',\n",
       "       'VEP_FEATURE', 'VEP_POLYPHEN_PRED', 'VEP_POLYPHEN_SCORE',\n",
       "       'VEP_PROTEIN_POS', 'VEP_SIFT_SCORE', 'Variant Classification', 'ddG',\n",
       "       '|ddG|'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filtering Instances"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df[working_df.TYPE == 'SNV'].drop(columns='TYPE') # Drop multi-nucleotide variants (MNVS), remove TYPE column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df[working_df.Consequence == 'missense_variant'] # Drop missense variants with secondary effects, i.e., splite region variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df[working_df['Max MAF (%)'] >= 1E-10] # Drop variants with MAF > 10^-10 => effectively 0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(95354, 39)"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(95354, 39)"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df[working_df['Max MAF (%)'] != 0].shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Removing Homogenous Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Impact\n",
      "MODERATE    95354\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(working_df.Impact.value_counts())\n",
    "working_df = working_df.drop(columns='Impact')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Consequence\n",
      "missense_variant    95354\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(working_df.Consequence.value_counts())\n",
    "working_df = working_df.drop(columns=['Consequence'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>CLINVAR_CLNSIG</th>\n",
       "      <th>CLINVAR_PATHOGENIC</th>\n",
       "      <th>CURATED_PATHOGENICITY</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "      <th>...</th>\n",
       "      <th>UNIPROT_DOMAIN</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_FEATURE</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>Variant Classification</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.06</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.751772</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>P</td>\n",
       "      <td>0.917</td>\n",
       "      <td>4</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.90</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.23</td>\n",
       "      <td>2.32</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.662314</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>P</td>\n",
       "      <td>0.787</td>\n",
       "      <td>10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.61</td>\n",
       "      <td>2.11</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.257539</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>11</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>7.08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>36.34</td>\n",
       "      <td>-0.29</td>\n",
       "      <td>N</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.289852</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>18</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686445:C&gt;T</th>\n",
       "      <td>20.50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>35.99</td>\n",
       "      <td>3.25</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.089529</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>P</td>\n",
       "      <td>0.947</td>\n",
       "      <td>32</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD Phred CLINVAR_CLNSIG  CLINVAR_PATHOGENIC  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686361:C>T                    23.00            NaN                 NaN   \n",
       "20:21686379:G>A                    22.90            NaN                 NaN   \n",
       "20:21686382:C>T                    21.80            NaN                 NaN   \n",
       "20:21686402:G>C                     7.08            NaN                 NaN   \n",
       "20:21686445:C>T                    20.50            NaN                 NaN   \n",
       "\n",
       "                             CURATED_PATHOGENICITY  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "20:21686361:C>T                                NaN             38.06   \n",
       "20:21686379:G>A                                NaN             37.23   \n",
       "20:21686382:C>T                                NaN             34.61   \n",
       "20:21686402:G>C                                NaN             36.34   \n",
       "20:21686445:C>T                                NaN             35.99   \n",
       "\n",
       "                              DBNSFP_GERP_RS DBNSFP_GERP_RS_PRED  \\\n",
       "Genomic Description (GRCh37)                                       \n",
       "20:21686361:C>T                         4.28                   C   \n",
       "20:21686379:G>A                         2.32                   C   \n",
       "20:21686382:C>T                         2.11                   C   \n",
       "20:21686402:G>C                        -0.29                   N   \n",
       "20:21686445:C>T                         3.25                   C   \n",
       "\n",
       "                             DBNSFP_INTERPRO_DOMAIN DBNSFP_LRT_PRED  \\\n",
       "Genomic Description (GRCh37)                                          \n",
       "20:21686361:C>T                                 .&.             NaN   \n",
       "20:21686379:G>A                                 .&.             NaN   \n",
       "20:21686382:C>T                                 .&.             NaN   \n",
       "20:21686402:G>C                                 .&.             NaN   \n",
       "20:21686445:C>T                                 .&.             NaN   \n",
       "\n",
       "                              DBNSFP_LRT_SCORE  ... UNIPROT_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                    ...                  \n",
       "20:21686361:C>T                            NaN  ...            NaN   \n",
       "20:21686379:G>A                            NaN  ...            NaN   \n",
       "20:21686382:C>T                            NaN  ...            NaN   \n",
       "20:21686402:G>C                            NaN  ...            NaN   \n",
       "20:21686445:C>T                            NaN  ...            NaN   \n",
       "\n",
       "                             VEP_CADD_RAW  VEP_FEATURE VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                               \n",
       "20:21686361:C>T                  2.751772  NM_006192.5                 P   \n",
       "20:21686379:G>A                  2.662314  NM_006192.5                 P   \n",
       "20:21686382:C>T                  2.257539  NM_006192.5                 P   \n",
       "20:21686402:G>C                  0.289852  NM_006192.5                 B   \n",
       "20:21686445:C>T                  2.089529  NM_006192.5                 P   \n",
       "\n",
       "                             VEP_POLYPHEN_SCORE VEP_PROTEIN_POS  \\\n",
       "Genomic Description (GRCh37)                                      \n",
       "20:21686361:C>T                           0.917               4   \n",
       "20:21686379:G>A                           0.787              10   \n",
       "20:21686382:C>T                           0.804              11   \n",
       "20:21686402:G>C                           0.000              18   \n",
       "20:21686445:C>T                           0.947              32   \n",
       "\n",
       "                             VEP_SIFT_SCORE Variant Classification  ddG |ddG|  \n",
       "Genomic Description (GRCh37)                                                   \n",
       "20:21686361:C>T                        0.00   Unknown significance -0.0   0.0  \n",
       "20:21686379:G>A                        0.00   Unknown significance -0.3   0.3  \n",
       "20:21686382:C>T                        0.01                 Benign -0.3   0.3  \n",
       "20:21686402:G>C                        1.00          Likely benign  0.5   0.5  \n",
       "20:21686445:C>T                        0.00   Unknown significance  0.1   0.1  \n",
       "\n",
       "[5 rows x 37 columns]"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Removing columns that are too varied"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "247\n"
     ]
    }
   ],
   "source": [
    "print(len(working_df.VEP_FEATURE.value_counts()))\n",
    "working_df = working_df.drop(columns='VEP_FEATURE')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.rename(columns={'Max MAF (%)': 'MAX_MAF', 'Max MAF Source': 'MAX_MAF_SOURCE'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "CADD Phred                           100.000000\n",
       "CLINVAR_CLNSIG                        12.167292\n",
       "CLINVAR_PATHOGENIC                    12.159951\n",
       "CURATED_PATHOGENICITY                  0.328251\n",
       "Confidence Score                     100.000000\n",
       "DBNSFP_GERP_RS                        99.958051\n",
       "DBNSFP_GERP_RS_PRED                   99.958051\n",
       "DBNSFP_INTERPRO_DOMAIN                89.045032\n",
       "DBNSFP_LRT_PRED                       87.532773\n",
       "DBNSFP_LRT_SCORE                      87.532773\n",
       "DBNSFP_MUTATIONTASTER_PRED            99.409569\n",
       "DBNSFP_MUTATIONTASTER_SCORE           99.409569\n",
       "DBNSFP_PHYLOP30WAY_MAMMALIAN          99.976928\n",
       "DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED     99.976928\n",
       "DBNSFP_POLYPHEN2_HDIV_PRED            99.419007\n",
       "DBNSFP_POLYPHEN2_HDIV_SCORE           99.419007\n",
       "DBNSFP_SIFT_PRED                      98.541225\n",
       "DBNSFP_SIFT_SCORE                     98.541225\n",
       "FINAL_PRED                            99.976928\n",
       "GENE                                 100.000000\n",
       "Interpro Domain                       40.732429\n",
       "MAX_MAF                              100.000000\n",
       "MAX_MAF_SOURCE                       100.000000\n",
       "NUM_PATH_PREDS                        99.976928\n",
       "Normalized SA                        100.000000\n",
       "Surface Area                         100.000000\n",
       "TOTAL_NUM_PREDS                       99.976928\n",
       "UNIPROT_DOMAIN                        38.585691\n",
       "VEP_CADD_RAW                         100.000000\n",
       "VEP_POLYPHEN_PRED                     99.993708\n",
       "VEP_POLYPHEN_SCORE                    99.993708\n",
       "VEP_PROTEIN_POS                      100.000000\n",
       "VEP_SIFT_SCORE                        99.971684\n",
       "Variant Classification               100.000000\n",
       "ddG                                   99.863666\n",
       "|ddG|                                 99.863666\n",
       "dtype: float64"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "percentage_not_nan = working_df.notna().mean() * 100\n",
    "percentage_not_nan"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>Interpro Domain</th>\n",
       "      <th>UNIPROT_DOMAIN</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686445:C&gt;T</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115913:C&gt;T</th>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95354 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_INTERPRO_DOMAIN Interpro Domain  \\\n",
       "Genomic Description (GRCh37)                                          \n",
       "20:21686361:C>T                                 .&.             NaN   \n",
       "20:21686379:G>A                                 .&.             NaN   \n",
       "20:21686382:C>T                                 .&.             NaN   \n",
       "20:21686402:G>C                                 .&.             NaN   \n",
       "20:21686445:C>T                                 .&.             NaN   \n",
       "...                                             ...             ...   \n",
       "10:8115913:C>T                                .&.&.             NaN   \n",
       "10:8115916:C>G                                .&.&.             NaN   \n",
       "10:8115916:C>T                                .&.&.             NaN   \n",
       "10:8115927:C>T                                .&.&.             NaN   \n",
       "10:8115945:G>A                                .&.&.             NaN   \n",
       "\n",
       "                             UNIPROT_DOMAIN  \n",
       "Genomic Description (GRCh37)                 \n",
       "20:21686361:C>T                         NaN  \n",
       "20:21686379:G>A                         NaN  \n",
       "20:21686382:C>T                         NaN  \n",
       "20:21686402:G>C                         NaN  \n",
       "20:21686445:C>T                         NaN  \n",
       "...                                     ...  \n",
       "10:8115913:C>T                          NaN  \n",
       "10:8115916:C>G                          NaN  \n",
       "10:8115916:C>T                          NaN  \n",
       "10:8115927:C>T                          NaN  \n",
       "10:8115945:G>A                          NaN  \n",
       "\n",
       "[95354 rows x 3 columns]"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[:, working_df.columns.str.contains('Domain') | working_df.columns.str.contains('DOMAIN')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Removing columns that are too sparse"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "['CLINVAR_CLNSIG', 'CLINVAR_PATHOGENIC', 'CURATED_PATHOGENICITY'] have values for fewer than a third of the variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CLINVAR_CLNSIG', 'CLINVAR_PATHOGENIC', 'CURATED_PATHOGENICITY']"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "utils.difference(working_df.columns.tolist(), working_df.dropna(thresh=working_df.shape[0] / 3, axis=1).columns.tolist())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.dropna(thresh=working_df.shape[0] / 3, axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Renaming the columns that include spaces"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.rename(columns={'CADD Phred': 'CADD_PHRED', 'Confidence Score': 'CONFIDENCE_SCORE',\n",
    "                           'Surface Area': 'SURFACE_AREA', 'Normalized SA': 'NORMALIZED_SURFACE_AREA',\n",
    "                           'Interpro Domain': 'INTERPRO_DOMAIN', 'Variant Classification': 'VARIANT_CLASSIFICATION'})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "I'm going to ignore unraveling the complicated prediction columns for now"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD_PHRED</th>\n",
       "      <th>CONFIDENCE_SCORE</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN</th>\n",
       "      <th>...</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>UNIPROT_DOMAIN</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>VARIANT_CLASSIFICATION</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.00</td>\n",
       "      <td>38.06</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.758794</td>\n",
       "      <td>0.118</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.751772</td>\n",
       "      <td>P</td>\n",
       "      <td>0.917</td>\n",
       "      <td>4</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.90</td>\n",
       "      <td>37.23</td>\n",
       "      <td>2.32</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.999272</td>\n",
       "      <td>1.147</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.662314</td>\n",
       "      <td>P</td>\n",
       "      <td>0.787</td>\n",
       "      <td>10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.80</td>\n",
       "      <td>34.61</td>\n",
       "      <td>2.11</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.998176</td>\n",
       "      <td>0.106</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.257539</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>11</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>7.08</td>\n",
       "      <td>36.34</td>\n",
       "      <td>-0.29</td>\n",
       "      <td>N</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>0.111</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.289852</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>18</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686445:C&gt;T</th>\n",
       "      <td>20.50</td>\n",
       "      <td>35.99</td>\n",
       "      <td>3.25</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.995526</td>\n",
       "      <td>0.937</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.089529</td>\n",
       "      <td>P</td>\n",
       "      <td>0.947</td>\n",
       "      <td>32</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115913:C&gt;T</th>\n",
       "      <td>28.40</td>\n",
       "      <td>51.75</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.074794</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>421</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>27.50</td>\n",
       "      <td>47.29</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.000643</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.979789</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>29.20</td>\n",
       "      <td>47.29</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.000643</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.163516</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>14.78</td>\n",
       "      <td>44.11</td>\n",
       "      <td>3.68</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.999167&amp;0.999167</td>\n",
       "      <td>1.026</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.223753</td>\n",
       "      <td>B</td>\n",
       "      <td>0.097</td>\n",
       "      <td>426</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>27.60</td>\n",
       "      <td>37.21</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.999993&amp;0.999993</td>\n",
       "      <td>1.176</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.995061</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>432</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95354 rows Ã— 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD_PHRED  CONFIDENCE_SCORE  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                                 \n",
       "20:21686361:C>T                    23.00             38.06            4.28   \n",
       "20:21686379:G>A                    22.90             37.23            2.32   \n",
       "20:21686382:C>T                    21.80             34.61            2.11   \n",
       "20:21686402:G>C                     7.08             36.34           -0.29   \n",
       "20:21686445:C>T                    20.50             35.99            3.25   \n",
       "...                                  ...               ...             ...   \n",
       "10:8115913:C>T                     28.40             51.75            5.59   \n",
       "10:8115916:C>G                     27.50             47.29            5.59   \n",
       "10:8115916:C>T                     29.20             47.29            5.59   \n",
       "10:8115927:C>T                     14.78             44.11            3.68   \n",
       "10:8115945:G>A                     27.60             37.21            5.59   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_INTERPRO_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686361:C>T                                C                    .&.   \n",
       "20:21686379:G>A                                C                    .&.   \n",
       "20:21686382:C>T                                C                    .&.   \n",
       "20:21686402:G>C                                N                    .&.   \n",
       "20:21686445:C>T                                C                    .&.   \n",
       "...                                          ...                    ...   \n",
       "10:8115913:C>T                                 C                  .&.&.   \n",
       "10:8115916:C>G                                 C                  .&.&.   \n",
       "10:8115916:C>T                                 C                  .&.&.   \n",
       "10:8115927:C>T                                 C                  .&.&.   \n",
       "10:8115945:G>A                                 C                  .&.&.   \n",
       "\n",
       "                             DBNSFP_LRT_PRED  DBNSFP_LRT_SCORE  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686361:C>T                          NaN               NaN   \n",
       "20:21686379:G>A                          NaN               NaN   \n",
       "20:21686382:C>T                          NaN               NaN   \n",
       "20:21686402:G>C                          NaN               NaN   \n",
       "20:21686445:C>T                          NaN               NaN   \n",
       "...                                      ...               ...   \n",
       "10:8115913:C>T                             D          0.000000   \n",
       "10:8115916:C>G                             U          0.000643   \n",
       "10:8115916:C>T                             U          0.000643   \n",
       "10:8115927:C>T                             D          0.000030   \n",
       "10:8115945:G>A                             D          0.000008   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686361:C>T                                       N   \n",
       "20:21686379:G>A                                       N   \n",
       "20:21686382:C>T                                       N   \n",
       "20:21686402:G>C                                       N   \n",
       "20:21686445:C>T                                       N   \n",
       "...                                                 ...   \n",
       "10:8115913:C>T                                      D&D   \n",
       "10:8115916:C>G                                      D&D   \n",
       "10:8115916:C>T                                      D&D   \n",
       "10:8115927:C>T                                      D&D   \n",
       "10:8115945:G>A                                      D&D   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_SCORE  \\\n",
       "Genomic Description (GRCh37)                               \n",
       "20:21686361:C>T                                 0.758794   \n",
       "20:21686379:G>A                                 0.999272   \n",
       "20:21686382:C>T                                 0.998176   \n",
       "20:21686402:G>C                                        1   \n",
       "20:21686445:C>T                                 0.995526   \n",
       "...                                                  ...   \n",
       "10:8115913:C>T                                       1&1   \n",
       "10:8115916:C>G                                       1&1   \n",
       "10:8115916:C>T                                       1&1   \n",
       "10:8115927:C>T                         0.999167&0.999167   \n",
       "10:8115945:G>A                         0.999993&0.999993   \n",
       "\n",
       "                              DBNSFP_PHYLOP30WAY_MAMMALIAN  ...  \\\n",
       "Genomic Description (GRCh37)                                ...   \n",
       "20:21686361:C>T                                      0.118  ...   \n",
       "20:21686379:G>A                                      1.147  ...   \n",
       "20:21686382:C>T                                      0.106  ...   \n",
       "20:21686402:G>C                                      0.111  ...   \n",
       "20:21686445:C>T                                      0.937  ...   \n",
       "...                                                    ...  ...   \n",
       "10:8115913:C>T                                       1.026  ...   \n",
       "10:8115916:C>G                                       1.026  ...   \n",
       "10:8115916:C>T                                       1.026  ...   \n",
       "10:8115927:C>T                                       1.026  ...   \n",
       "10:8115945:G>A                                       1.176  ...   \n",
       "\n",
       "                             TOTAL_NUM_PREDS UNIPROT_DOMAIN VEP_CADD_RAW  \\\n",
       "Genomic Description (GRCh37)                                               \n",
       "20:21686361:C>T                          5.0            NaN     2.751772   \n",
       "20:21686379:G>A                          5.0            NaN     2.662314   \n",
       "20:21686382:C>T                          5.0            NaN     2.257539   \n",
       "20:21686402:G>C                          5.0            NaN     0.289852   \n",
       "20:21686445:C>T                          5.0            NaN     2.089529   \n",
       "...                                      ...            ...          ...   \n",
       "10:8115913:C>T                           6.0            NaN     4.074794   \n",
       "10:8115916:C>G                           5.0            NaN     3.979789   \n",
       "10:8115916:C>T                           5.0            NaN     4.163516   \n",
       "10:8115927:C>T                           6.0            NaN     1.223753   \n",
       "10:8115945:G>A                           6.0            NaN     3.995061   \n",
       "\n",
       "                             VEP_POLYPHEN_PRED VEP_POLYPHEN_SCORE  \\\n",
       "Genomic Description (GRCh37)                                        \n",
       "20:21686361:C>T                              P              0.917   \n",
       "20:21686379:G>A                              P              0.787   \n",
       "20:21686382:C>T                              P              0.804   \n",
       "20:21686402:G>C                              B              0.000   \n",
       "20:21686445:C>T                              P              0.947   \n",
       "...                                        ...                ...   \n",
       "10:8115913:C>T                               D              1.000   \n",
       "10:8115916:C>G                               D              1.000   \n",
       "10:8115916:C>T                               D              1.000   \n",
       "10:8115927:C>T                               B              0.097   \n",
       "10:8115945:G>A                               D              0.997   \n",
       "\n",
       "                             VEP_PROTEIN_POS VEP_SIFT_SCORE  \\\n",
       "Genomic Description (GRCh37)                                  \n",
       "20:21686361:C>T                            4           0.00   \n",
       "20:21686379:G>A                           10           0.00   \n",
       "20:21686382:C>T                           11           0.01   \n",
       "20:21686402:G>C                           18           1.00   \n",
       "20:21686445:C>T                           32           0.00   \n",
       "...                                      ...            ...   \n",
       "10:8115913:C>T                           421           0.01   \n",
       "10:8115916:C>G                           422           0.00   \n",
       "10:8115916:C>T                           422           0.00   \n",
       "10:8115927:C>T                           426           1.00   \n",
       "10:8115945:G>A                           432           0.00   \n",
       "\n",
       "                             VARIANT_CLASSIFICATION  ddG |ddG|  \n",
       "Genomic Description (GRCh37)                                    \n",
       "20:21686361:C>T                Unknown significance -0.0   0.0  \n",
       "20:21686379:G>A                Unknown significance -0.3   0.3  \n",
       "20:21686382:C>T                              Benign -0.3   0.3  \n",
       "20:21686402:G>C                       Likely benign  0.5   0.5  \n",
       "20:21686445:C>T                Unknown significance  0.1   0.1  \n",
       "...                                             ...  ...   ...  \n",
       "10:8115913:C>T                    Likely pathogenic -0.1   0.1  \n",
       "10:8115916:C>G                 Unknown significance -0.3   0.3  \n",
       "10:8115916:C>T                 Unknown significance -0.2   0.2  \n",
       "10:8115927:C>T                 Unknown significance -0.1   0.1  \n",
       "10:8115945:G>A                 Unknown significance  0.2   0.2  \n",
       "\n",
       "[95354 rows x 33 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CADD_PHRED', 'CONFIDENCE_SCORE', 'DBNSFP_GERP_RS',\n",
       "       'DBNSFP_GERP_RS_PRED', 'DBNSFP_INTERPRO_DOMAIN', 'DBNSFP_LRT_PRED',\n",
       "       'DBNSFP_LRT_SCORE', 'DBNSFP_MUTATIONTASTER_PRED',\n",
       "       'DBNSFP_MUTATIONTASTER_SCORE', 'DBNSFP_PHYLOP30WAY_MAMMALIAN',\n",
       "       'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED', 'DBNSFP_POLYPHEN2_HDIV_PRED',\n",
       "       'DBNSFP_POLYPHEN2_HDIV_SCORE', 'DBNSFP_SIFT_PRED', 'DBNSFP_SIFT_SCORE',\n",
       "       'FINAL_PRED', 'GENE', 'INTERPRO_DOMAIN', 'MAX_MAF', 'MAX_MAF_SOURCE',\n",
       "       'NUM_PATH_PREDS', 'NORMALIZED_SURFACE_AREA', 'SURFACE_AREA',\n",
       "       'TOTAL_NUM_PREDS', 'UNIPROT_DOMAIN', 'VEP_CADD_RAW',\n",
       "       'VEP_POLYPHEN_PRED', 'VEP_POLYPHEN_SCORE', 'VEP_PROTEIN_POS',\n",
       "       'VEP_SIFT_SCORE', 'VARIANT_CLASSIFICATION', 'ddG', '|ddG|'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True     88539\n",
       "False     6815\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(working_df['VARIANT_CLASSIFICATION'] == working_df['FINAL_PRED']).value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.drop(columns='VEP_PROTEIN_POS')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Preproccesing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Encoding Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD_PHRED</th>\n",
       "      <th>CONFIDENCE_SCORE</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN</th>\n",
       "      <th>...</th>\n",
       "      <th>SURFACE_AREA</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>UNIPROT_DOMAIN</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>VARIANT_CLASSIFICATION</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.00</td>\n",
       "      <td>38.06</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.758794</td>\n",
       "      <td>0.118</td>\n",
       "      <td>...</td>\n",
       "      <td>139.065932</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.751772</td>\n",
       "      <td>P</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.90</td>\n",
       "      <td>37.23</td>\n",
       "      <td>2.32</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.999272</td>\n",
       "      <td>1.147</td>\n",
       "      <td>...</td>\n",
       "      <td>180.703642</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.662314</td>\n",
       "      <td>P</td>\n",
       "      <td>0.787</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.80</td>\n",
       "      <td>34.61</td>\n",
       "      <td>2.11</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.998176</td>\n",
       "      <td>0.106</td>\n",
       "      <td>...</td>\n",
       "      <td>76.425233</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.257539</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>7.08</td>\n",
       "      <td>36.34</td>\n",
       "      <td>-0.29</td>\n",
       "      <td>N</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>0.111</td>\n",
       "      <td>...</td>\n",
       "      <td>68.153896</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.289852</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686445:C&gt;T</th>\n",
       "      <td>20.50</td>\n",
       "      <td>35.99</td>\n",
       "      <td>3.25</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.995526</td>\n",
       "      <td>0.937</td>\n",
       "      <td>...</td>\n",
       "      <td>72.826957</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.089529</td>\n",
       "      <td>P</td>\n",
       "      <td>0.947</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD_PHRED  CONFIDENCE_SCORE  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                                 \n",
       "20:21686361:C>T                    23.00             38.06            4.28   \n",
       "20:21686379:G>A                    22.90             37.23            2.32   \n",
       "20:21686382:C>T                    21.80             34.61            2.11   \n",
       "20:21686402:G>C                     7.08             36.34           -0.29   \n",
       "20:21686445:C>T                    20.50             35.99            3.25   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_INTERPRO_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686361:C>T                                C                    .&.   \n",
       "20:21686379:G>A                                C                    .&.   \n",
       "20:21686382:C>T                                C                    .&.   \n",
       "20:21686402:G>C                                N                    .&.   \n",
       "20:21686445:C>T                                C                    .&.   \n",
       "\n",
       "                             DBNSFP_LRT_PRED  DBNSFP_LRT_SCORE  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686361:C>T                          NaN               NaN   \n",
       "20:21686379:G>A                          NaN               NaN   \n",
       "20:21686382:C>T                          NaN               NaN   \n",
       "20:21686402:G>C                          NaN               NaN   \n",
       "20:21686445:C>T                          NaN               NaN   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686361:C>T                                       N   \n",
       "20:21686379:G>A                                       N   \n",
       "20:21686382:C>T                                       N   \n",
       "20:21686402:G>C                                       N   \n",
       "20:21686445:C>T                                       N   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_SCORE  \\\n",
       "Genomic Description (GRCh37)                               \n",
       "20:21686361:C>T                                 0.758794   \n",
       "20:21686379:G>A                                 0.999272   \n",
       "20:21686382:C>T                                 0.998176   \n",
       "20:21686402:G>C                                        1   \n",
       "20:21686445:C>T                                 0.995526   \n",
       "\n",
       "                              DBNSFP_PHYLOP30WAY_MAMMALIAN  ... SURFACE_AREA  \\\n",
       "Genomic Description (GRCh37)                                ...                \n",
       "20:21686361:C>T                                      0.118  ...   139.065932   \n",
       "20:21686379:G>A                                      1.147  ...   180.703642   \n",
       "20:21686382:C>T                                      0.106  ...    76.425233   \n",
       "20:21686402:G>C                                      0.111  ...    68.153896   \n",
       "20:21686445:C>T                                      0.937  ...    72.826957   \n",
       "\n",
       "                             TOTAL_NUM_PREDS UNIPROT_DOMAIN VEP_CADD_RAW  \\\n",
       "Genomic Description (GRCh37)                                               \n",
       "20:21686361:C>T                          5.0            NaN     2.751772   \n",
       "20:21686379:G>A                          5.0            NaN     2.662314   \n",
       "20:21686382:C>T                          5.0            NaN     2.257539   \n",
       "20:21686402:G>C                          5.0            NaN     0.289852   \n",
       "20:21686445:C>T                          5.0            NaN     2.089529   \n",
       "\n",
       "                             VEP_POLYPHEN_PRED VEP_POLYPHEN_SCORE  \\\n",
       "Genomic Description (GRCh37)                                        \n",
       "20:21686361:C>T                              P              0.917   \n",
       "20:21686379:G>A                              P              0.787   \n",
       "20:21686382:C>T                              P              0.804   \n",
       "20:21686402:G>C                              B              0.000   \n",
       "20:21686445:C>T                              P              0.947   \n",
       "\n",
       "                             VEP_SIFT_SCORE VARIANT_CLASSIFICATION  ddG |ddG|  \n",
       "Genomic Description (GRCh37)                                                   \n",
       "20:21686361:C>T                        0.00   Unknown significance -0.0   0.0  \n",
       "20:21686379:G>A                        0.00   Unknown significance -0.3   0.3  \n",
       "20:21686382:C>T                        0.01                 Benign -0.3   0.3  \n",
       "20:21686402:G>C                        1.00          Likely benign  0.5   0.5  \n",
       "20:21686445:C>T                        0.00   Unknown significance  0.1   0.1  \n",
       "\n",
       "[5 rows x 32 columns]"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = working_df.copy()\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(95354, 32)"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "df2 = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[('CADD_PHRED', 'other'),\n",
       " ('CONFIDENCE_SCORE', 'score'),\n",
       " ('DBNSFP_GERP_RS', 'other'),\n",
       " ('DBNSFP_GERP_RS_PRED', 'predictions'),\n",
       " ('DBNSFP_INTERPRO_DOMAIN', 'other'),\n",
       " ('DBNSFP_LRT_PRED', 'predictions'),\n",
       " ('DBNSFP_LRT_SCORE', 'score'),\n",
       " ('DBNSFP_MUTATIONTASTER_PRED', 'predictions'),\n",
       " ('DBNSFP_MUTATIONTASTER_SCORE', 'score'),\n",
       " ('DBNSFP_PHYLOP30WAY_MAMMALIAN', 'other'),\n",
       " ('DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED', 'predictions'),\n",
       " ('DBNSFP_POLYPHEN2_HDIV_PRED', 'predictions'),\n",
       " ('DBNSFP_POLYPHEN2_HDIV_SCORE', 'score'),\n",
       " ('DBNSFP_SIFT_PRED', 'predictions'),\n",
       " ('DBNSFP_SIFT_SCORE', 'score'),\n",
       " ('FINAL_PRED', 'predictions'),\n",
       " ('GENE', 'other'),\n",
       " ('INTERPRO_DOMAIN', 'other'),\n",
       " ('MAX_MAF', 'other'),\n",
       " ('MAX_MAF_SOURCE', 'other'),\n",
       " ('NUM_PATH_PREDS', 'predictions'),\n",
       " ('NORMALIZED_SURFACE_AREA', 'other'),\n",
       " ('SURFACE_AREA', 'other'),\n",
       " ('TOTAL_NUM_PREDS', 'predictions'),\n",
       " ('UNIPROT_DOMAIN', 'other'),\n",
       " ('VEP_CADD_RAW', 'other'),\n",
       " ('VEP_POLYPHEN_PRED', 'predictions'),\n",
       " ('VEP_POLYPHEN_SCORE', 'score'),\n",
       " ('VEP_SIFT_SCORE', 'score'),\n",
       " ('VARIANT_CLASSIFICATION', 'other'),\n",
       " ('ddG', 'other'),\n",
       " ('|ddG|', 'other')]"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[(col, 'predictions') if 'PRED' in col else (col, 'score') if 'SCORE' in col else (col, 'other') for col in df.columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "cadd = ['CADD_PHRED', 'VEP_CADD_RAW']\n",
    "classification = ['VARIANT_CLASSIFICATION', 'FINAL_PRED', 'NUM_PATH_PREDS', 'TOTAL_NUM_PREDS']\n",
    "gene = ['GENE']\n",
    "maf = ['MAX_MAF', 'MAX_MAF_SOURCE']\n",
    "domain = ['DBNSFP_INTERPRO_DOMAIN', 'UNIPROT_DOMAIN', 'INTERPRO_DOMAIN']\n",
    "score = ['DBNSFP_GERP_RS', 'DBNSFP_LRT_SCORE', 'DBNSFP_MUTATIONTASTER_SCORE', 'DBNSFP_POLYPHEN2_HDIV_SCORE', 'DBNSFP_SIFT_SCORE', 'DBNSFP_PHYLOP30WAY_MAMMALIAN', 'VEP_POLYPHEN_SCORE', 'VEP_SIFT_SCORE']\n",
    "predictions = ['DBNSFP_SIFT_PRED', 'DBNSFP_GERP_RS_PRED', 'DBNSFP_LRT_PRED', 'DBNSFP_MUTATIONTASTER_PRED', 'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED', 'DBNSFP_POLYPHEN2_HDIV_PRED', 'VEP_POLYPHEN_PRED']\n",
    "protein_structure = ['ddG', '|ddG|', 'CONFIDENCE_SCORE', 'SURFACE_AREA', 'NORMALIZED_SURFACE_AREA']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe thead tr:last-of-type th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>CADD_PHRED</th>\n",
       "      <th>CONFIDENCE_SCORE</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN</th>\n",
       "      <th>...</th>\n",
       "      <th>SURFACE_AREA</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>UNIPROT_DOMAIN</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>VARIANT_CLASSIFICATION</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th>other</th>\n",
       "      <th>score</th>\n",
       "      <th>other</th>\n",
       "      <th>predictions</th>\n",
       "      <th>other</th>\n",
       "      <th>predictions</th>\n",
       "      <th>score</th>\n",
       "      <th>predictions</th>\n",
       "      <th>score</th>\n",
       "      <th>other</th>\n",
       "      <th>...</th>\n",
       "      <th>other</th>\n",
       "      <th>predictions</th>\n",
       "      <th>other</th>\n",
       "      <th>other</th>\n",
       "      <th>predictions</th>\n",
       "      <th>score</th>\n",
       "      <th>score</th>\n",
       "      <th>other</th>\n",
       "      <th>other</th>\n",
       "      <th>other</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.00</td>\n",
       "      <td>38.06</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.758794</td>\n",
       "      <td>0.118</td>\n",
       "      <td>...</td>\n",
       "      <td>139.065932</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.751772</td>\n",
       "      <td>P</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.90</td>\n",
       "      <td>37.23</td>\n",
       "      <td>2.32</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.999272</td>\n",
       "      <td>1.147</td>\n",
       "      <td>...</td>\n",
       "      <td>180.703642</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.662314</td>\n",
       "      <td>P</td>\n",
       "      <td>0.787</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.80</td>\n",
       "      <td>34.61</td>\n",
       "      <td>2.11</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.998176</td>\n",
       "      <td>0.106</td>\n",
       "      <td>...</td>\n",
       "      <td>76.425233</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.257539</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>7.08</td>\n",
       "      <td>36.34</td>\n",
       "      <td>-0.29</td>\n",
       "      <td>N</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>0.111</td>\n",
       "      <td>...</td>\n",
       "      <td>68.153896</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.289852</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686445:C&gt;T</th>\n",
       "      <td>20.50</td>\n",
       "      <td>35.99</td>\n",
       "      <td>3.25</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.995526</td>\n",
       "      <td>0.937</td>\n",
       "      <td>...</td>\n",
       "      <td>72.826957</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.089529</td>\n",
       "      <td>P</td>\n",
       "      <td>0.947</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115913:C&gt;T</th>\n",
       "      <td>28.40</td>\n",
       "      <td>51.75</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>...</td>\n",
       "      <td>107.686086</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.074794</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>27.50</td>\n",
       "      <td>47.29</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.000643</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>...</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.979789</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>29.20</td>\n",
       "      <td>47.29</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>0.000643</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>...</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>5.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.163516</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>14.78</td>\n",
       "      <td>44.11</td>\n",
       "      <td>3.68</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.999167&amp;0.999167</td>\n",
       "      <td>1.026</td>\n",
       "      <td>...</td>\n",
       "      <td>128.397680</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.223753</td>\n",
       "      <td>B</td>\n",
       "      <td>0.097</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>27.60</td>\n",
       "      <td>37.21</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.999993&amp;0.999993</td>\n",
       "      <td>1.176</td>\n",
       "      <td>...</td>\n",
       "      <td>67.643291</td>\n",
       "      <td>6.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.995061</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95354 rows Ã— 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             CADD_PHRED CONFIDENCE_SCORE DBNSFP_GERP_RS  \\\n",
       "                                  other            score          other   \n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686361:C>T                   23.00            38.06           4.28   \n",
       "20:21686379:G>A                   22.90            37.23           2.32   \n",
       "20:21686382:C>T                   21.80            34.61           2.11   \n",
       "20:21686402:G>C                    7.08            36.34          -0.29   \n",
       "20:21686445:C>T                   20.50            35.99           3.25   \n",
       "...                                 ...              ...            ...   \n",
       "10:8115913:C>T                    28.40            51.75           5.59   \n",
       "10:8115916:C>G                    27.50            47.29           5.59   \n",
       "10:8115916:C>T                    29.20            47.29           5.59   \n",
       "10:8115927:C>T                    14.78            44.11           3.68   \n",
       "10:8115945:G>A                    27.60            37.21           5.59   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_INTERPRO_DOMAIN  \\\n",
       "                                     predictions                  other   \n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686361:C>T                                C                    .&.   \n",
       "20:21686379:G>A                                C                    .&.   \n",
       "20:21686382:C>T                                C                    .&.   \n",
       "20:21686402:G>C                                N                    .&.   \n",
       "20:21686445:C>T                                C                    .&.   \n",
       "...                                          ...                    ...   \n",
       "10:8115913:C>T                                 C                  .&.&.   \n",
       "10:8115916:C>G                                 C                  .&.&.   \n",
       "10:8115916:C>T                                 C                  .&.&.   \n",
       "10:8115927:C>T                                 C                  .&.&.   \n",
       "10:8115945:G>A                                 C                  .&.&.   \n",
       "\n",
       "                             DBNSFP_LRT_PRED DBNSFP_LRT_SCORE  \\\n",
       "                                 predictions            score   \n",
       "Genomic Description (GRCh37)                                    \n",
       "20:21686361:C>T                          NaN              NaN   \n",
       "20:21686379:G>A                          NaN              NaN   \n",
       "20:21686382:C>T                          NaN              NaN   \n",
       "20:21686402:G>C                          NaN              NaN   \n",
       "20:21686445:C>T                          NaN              NaN   \n",
       "...                                      ...              ...   \n",
       "10:8115913:C>T                             D         0.000000   \n",
       "10:8115916:C>G                             U         0.000643   \n",
       "10:8115916:C>T                             U         0.000643   \n",
       "10:8115927:C>T                             D         0.000030   \n",
       "10:8115945:G>A                             D         0.000008   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "                                            predictions   \n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686361:C>T                                       N   \n",
       "20:21686379:G>A                                       N   \n",
       "20:21686382:C>T                                       N   \n",
       "20:21686402:G>C                                       N   \n",
       "20:21686445:C>T                                       N   \n",
       "...                                                 ...   \n",
       "10:8115913:C>T                                      D&D   \n",
       "10:8115916:C>G                                      D&D   \n",
       "10:8115916:C>T                                      D&D   \n",
       "10:8115927:C>T                                      D&D   \n",
       "10:8115945:G>A                                      D&D   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_SCORE  \\\n",
       "                                                   score   \n",
       "Genomic Description (GRCh37)                               \n",
       "20:21686361:C>T                                 0.758794   \n",
       "20:21686379:G>A                                 0.999272   \n",
       "20:21686382:C>T                                 0.998176   \n",
       "20:21686402:G>C                                        1   \n",
       "20:21686445:C>T                                 0.995526   \n",
       "...                                                  ...   \n",
       "10:8115913:C>T                                       1&1   \n",
       "10:8115916:C>G                                       1&1   \n",
       "10:8115916:C>T                                       1&1   \n",
       "10:8115927:C>T                         0.999167&0.999167   \n",
       "10:8115945:G>A                         0.999993&0.999993   \n",
       "\n",
       "                             DBNSFP_PHYLOP30WAY_MAMMALIAN  ... SURFACE_AREA  \\\n",
       "                                                    other  ...        other   \n",
       "Genomic Description (GRCh37)                               ...                \n",
       "20:21686361:C>T                                     0.118  ...   139.065932   \n",
       "20:21686379:G>A                                     1.147  ...   180.703642   \n",
       "20:21686382:C>T                                     0.106  ...    76.425233   \n",
       "20:21686402:G>C                                     0.111  ...    68.153896   \n",
       "20:21686445:C>T                                     0.937  ...    72.826957   \n",
       "...                                                   ...  ...          ...   \n",
       "10:8115913:C>T                                      1.026  ...   107.686086   \n",
       "10:8115916:C>G                                      1.026  ...   136.946574   \n",
       "10:8115916:C>T                                      1.026  ...   136.946574   \n",
       "10:8115927:C>T                                      1.026  ...   128.397680   \n",
       "10:8115945:G>A                                      1.176  ...    67.643291   \n",
       "\n",
       "                             TOTAL_NUM_PREDS UNIPROT_DOMAIN VEP_CADD_RAW  \\\n",
       "                                 predictions          other        other   \n",
       "Genomic Description (GRCh37)                                               \n",
       "20:21686361:C>T                          5.0            NaN     2.751772   \n",
       "20:21686379:G>A                          5.0            NaN     2.662314   \n",
       "20:21686382:C>T                          5.0            NaN     2.257539   \n",
       "20:21686402:G>C                          5.0            NaN     0.289852   \n",
       "20:21686445:C>T                          5.0            NaN     2.089529   \n",
       "...                                      ...            ...          ...   \n",
       "10:8115913:C>T                           6.0            NaN     4.074794   \n",
       "10:8115916:C>G                           5.0            NaN     3.979789   \n",
       "10:8115916:C>T                           5.0            NaN     4.163516   \n",
       "10:8115927:C>T                           6.0            NaN     1.223753   \n",
       "10:8115945:G>A                           6.0            NaN     3.995061   \n",
       "\n",
       "                             VEP_POLYPHEN_PRED VEP_POLYPHEN_SCORE  \\\n",
       "                                   predictions              score   \n",
       "Genomic Description (GRCh37)                                        \n",
       "20:21686361:C>T                              P              0.917   \n",
       "20:21686379:G>A                              P              0.787   \n",
       "20:21686382:C>T                              P              0.804   \n",
       "20:21686402:G>C                              B              0.000   \n",
       "20:21686445:C>T                              P              0.947   \n",
       "...                                        ...                ...   \n",
       "10:8115913:C>T                               D              1.000   \n",
       "10:8115916:C>G                               D              1.000   \n",
       "10:8115916:C>T                               D              1.000   \n",
       "10:8115927:C>T                               B              0.097   \n",
       "10:8115945:G>A                               D              0.997   \n",
       "\n",
       "                             VEP_SIFT_SCORE VARIANT_CLASSIFICATION   ddG |ddG|  \n",
       "                                      score                  other other other  \n",
       "Genomic Description (GRCh37)                                                    \n",
       "20:21686361:C>T                        0.00   Unknown significance  -0.0   0.0  \n",
       "20:21686379:G>A                        0.00   Unknown significance  -0.3   0.3  \n",
       "20:21686382:C>T                        0.01                 Benign  -0.3   0.3  \n",
       "20:21686402:G>C                        1.00          Likely benign   0.5   0.5  \n",
       "20:21686445:C>T                        0.00   Unknown significance   0.1   0.1  \n",
       "...                                     ...                    ...   ...   ...  \n",
       "10:8115913:C>T                         0.01      Likely pathogenic  -0.1   0.1  \n",
       "10:8115916:C>G                         0.00   Unknown significance  -0.3   0.3  \n",
       "10:8115916:C>T                         0.00   Unknown significance  -0.2   0.2  \n",
       "10:8115927:C>T                         1.00   Unknown significance  -0.1   0.1  \n",
       "10:8115945:G>A                         0.00   Unknown significance   0.2   0.2  \n",
       "\n",
       "[95354 rows x 32 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2 = df.copy()\n",
    "df2.columns = pd.MultiIndex.from_tuples([(col, 'predictions') if 'PRED' in col else (col, 'score') if 'SCORE' in col else (col, 'other') for col in df.columns])\n",
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                              CADD_PHRED  DBNSFP_GERP_RS  \\\n",
      "Genomic Description (GRCh37)                               \n",
      "20:21686361:C>T                    23.00            4.28   \n",
      "20:21686379:G>A                    22.90            2.32   \n",
      "20:21686382:C>T                    21.80            2.11   \n",
      "20:21686402:G>C                     7.08           -0.29   \n",
      "20:21686445:C>T                    20.50            3.25   \n",
      "...                                  ...             ...   \n",
      "10:8115913:C>T                     28.40            5.59   \n",
      "10:8115916:C>G                     27.50            5.59   \n",
      "10:8115916:C>T                     29.20            5.59   \n",
      "10:8115927:C>T                     14.78            3.68   \n",
      "10:8115945:G>A                     27.60            5.59   \n",
      "\n",
      "                             DBNSFP_INTERPRO_DOMAIN  \\\n",
      "Genomic Description (GRCh37)                          \n",
      "20:21686361:C>T                                 .&.   \n",
      "20:21686379:G>A                                 .&.   \n",
      "20:21686382:C>T                                 .&.   \n",
      "20:21686402:G>C                                 .&.   \n",
      "20:21686445:C>T                                 .&.   \n",
      "...                                             ...   \n",
      "10:8115913:C>T                                .&.&.   \n",
      "10:8115916:C>G                                .&.&.   \n",
      "10:8115916:C>T                                .&.&.   \n",
      "10:8115927:C>T                                .&.&.   \n",
      "10:8115945:G>A                                .&.&.   \n",
      "\n",
      "                              DBNSFP_PHYLOP30WAY_MAMMALIAN   GENE  \\\n",
      "Genomic Description (GRCh37)                                        \n",
      "20:21686361:C>T                                      0.118   PAX1   \n",
      "20:21686379:G>A                                      1.147   PAX1   \n",
      "20:21686382:C>T                                      0.106   PAX1   \n",
      "20:21686402:G>C                                      0.111   PAX1   \n",
      "20:21686445:C>T                                      0.937   PAX1   \n",
      "...                                                    ...    ...   \n",
      "10:8115913:C>T                                       1.026  GATA3   \n",
      "10:8115916:C>G                                       1.026  GATA3   \n",
      "10:8115916:C>T                                       1.026  GATA3   \n",
      "10:8115927:C>T                                       1.026  GATA3   \n",
      "10:8115945:G>A                                       1.176  GATA3   \n",
      "\n",
      "                             INTERPRO_DOMAIN   MAX_MAF      MAX_MAF_SOURCE  \\\n",
      "Genomic Description (GRCh37)                                                 \n",
      "20:21686361:C>T                          NaN  0.006660  gnomad_AF_nfe_pass   \n",
      "20:21686379:G>A                          NaN  0.010493  gnomad_AF_nfe_pass   \n",
      "20:21686382:C>T                          NaN  3.187450  gnomad_AF_afr_pass   \n",
      "20:21686402:G>C                          NaN  0.006735  gnomad_AF_nfe_pass   \n",
      "20:21686445:C>T                          NaN  0.259067  gnomad_AF_amr_pass   \n",
      "...                                      ...       ...                 ...   \n",
      "10:8115913:C>T                           NaN  0.003520  gnomad_AF_nfe_pass   \n",
      "10:8115916:C>G                           NaN  0.006484  gnomad_AF_nfe_pass   \n",
      "10:8115916:C>T                           NaN  0.012309  gnomad_AF_afr_pass   \n",
      "10:8115927:C>T                           NaN  0.000880  gnomad_AF_nfe_pass   \n",
      "10:8115945:G>A                           NaN  0.001549  gnomad_AF_nfe_pass   \n",
      "\n",
      "                              NORMALIZED_SURFACE_AREA  SURFACE_AREA  \\\n",
      "Genomic Description (GRCh37)                                          \n",
      "20:21686361:C>T                              0.883870    139.065932   \n",
      "20:21686379:G>A                              0.670700    180.703642   \n",
      "20:21686382:C>T                              0.601022     76.425233   \n",
      "20:21686402:G>C                              0.714805     68.153896   \n",
      "20:21686445:C>T                              0.572725     72.826957   \n",
      "...                                               ...           ...   \n",
      "10:8115913:C>T                               0.684427    107.686086   \n",
      "10:8115916:C>G                               0.870607    136.946574   \n",
      "10:8115916:C>T                               0.870607    136.946574   \n",
      "10:8115927:C>T                               0.816259    128.397680   \n",
      "10:8115945:G>A                               0.709449     67.643291   \n",
      "\n",
      "                             UNIPROT_DOMAIN  VEP_CADD_RAW  \\\n",
      "Genomic Description (GRCh37)                                \n",
      "20:21686361:C>T                         NaN      2.751772   \n",
      "20:21686379:G>A                         NaN      2.662314   \n",
      "20:21686382:C>T                         NaN      2.257539   \n",
      "20:21686402:G>C                         NaN      0.289852   \n",
      "20:21686445:C>T                         NaN      2.089529   \n",
      "...                                     ...           ...   \n",
      "10:8115913:C>T                          NaN      4.074794   \n",
      "10:8115916:C>G                          NaN      3.979789   \n",
      "10:8115916:C>T                          NaN      4.163516   \n",
      "10:8115927:C>T                          NaN      1.223753   \n",
      "10:8115945:G>A                          NaN      3.995061   \n",
      "\n",
      "                             VARIANT_CLASSIFICATION  ddG  |ddG|  \n",
      "Genomic Description (GRCh37)                                     \n",
      "20:21686361:C>T                Unknown significance -0.0    0.0  \n",
      "20:21686379:G>A                Unknown significance -0.3    0.3  \n",
      "20:21686382:C>T                              Benign -0.3    0.3  \n",
      "20:21686402:G>C                       Likely benign  0.5    0.5  \n",
      "20:21686445:C>T                Unknown significance  0.1    0.1  \n",
      "...                                             ...  ...    ...  \n",
      "10:8115913:C>T                    Likely pathogenic -0.1    0.1  \n",
      "10:8115916:C>G                 Unknown significance -0.3    0.3  \n",
      "10:8115916:C>T                 Unknown significance -0.2    0.2  \n",
      "10:8115927:C>T                 Unknown significance -0.1    0.1  \n",
      "10:8115945:G>A                 Unknown significance  0.2    0.2  \n",
      "\n",
      "[95354 rows x 15 columns]\n"
     ]
    }
   ],
   "source": [
    "other_columns = df2.xs('other', axis=1, level=1)\n",
    "print(other_columns)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_copy = df.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df_copy.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD_PHRED</th>\n",
       "      <th>CONFIDENCE_SCORE</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>NORMALIZED_SURFACE_AREA</th>\n",
       "      <th>SURFACE_AREA</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>VARIANT_CLASSIFICATION</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.00</td>\n",
       "      <td>38.06</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.758794</td>\n",
       "      <td>0.118</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.883870</td>\n",
       "      <td>139.065932</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.751772</td>\n",
       "      <td>P</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.90</td>\n",
       "      <td>37.23</td>\n",
       "      <td>2.32</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.999272</td>\n",
       "      <td>1.147</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.670700</td>\n",
       "      <td>180.703642</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.662314</td>\n",
       "      <td>P</td>\n",
       "      <td>0.787</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.80</td>\n",
       "      <td>34.61</td>\n",
       "      <td>2.11</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.998176</td>\n",
       "      <td>0.106</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.601022</td>\n",
       "      <td>76.425233</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.257539</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>7.08</td>\n",
       "      <td>36.34</td>\n",
       "      <td>-0.29</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>0.111</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.714805</td>\n",
       "      <td>68.153896</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.289852</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686445:C&gt;T</th>\n",
       "      <td>20.50</td>\n",
       "      <td>35.99</td>\n",
       "      <td>3.25</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.995526</td>\n",
       "      <td>0.937</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.572725</td>\n",
       "      <td>72.826957</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.089529</td>\n",
       "      <td>P</td>\n",
       "      <td>0.947</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115913:C&gt;T</th>\n",
       "      <td>28.40</td>\n",
       "      <td>51.75</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.684427</td>\n",
       "      <td>107.686086</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.074794</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>27.50</td>\n",
       "      <td>47.29</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>U</td>\n",
       "      <td>0.000643</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.979789</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>29.20</td>\n",
       "      <td>47.29</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>U</td>\n",
       "      <td>0.000643</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.163516</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>14.78</td>\n",
       "      <td>44.11</td>\n",
       "      <td>3.68</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.999167&amp;0.999167</td>\n",
       "      <td>1.026</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.816259</td>\n",
       "      <td>128.397680</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.223753</td>\n",
       "      <td>B</td>\n",
       "      <td>0.097</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>27.60</td>\n",
       "      <td>37.21</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.999993&amp;0.999993</td>\n",
       "      <td>1.176</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.709449</td>\n",
       "      <td>67.643291</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.995061</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95354 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD_PHRED  CONFIDENCE_SCORE  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                                 \n",
       "20:21686361:C>T                    23.00             38.06            4.28   \n",
       "20:21686379:G>A                    22.90             37.23            2.32   \n",
       "20:21686382:C>T                    21.80             34.61            2.11   \n",
       "20:21686402:G>C                     7.08             36.34           -0.29   \n",
       "20:21686445:C>T                    20.50             35.99            3.25   \n",
       "...                                  ...               ...             ...   \n",
       "10:8115913:C>T                     28.40             51.75            5.59   \n",
       "10:8115916:C>G                     27.50             47.29            5.59   \n",
       "10:8115916:C>T                     29.20             47.29            5.59   \n",
       "10:8115927:C>T                     14.78             44.11            3.68   \n",
       "10:8115945:G>A                     27.60             37.21            5.59   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_LRT_PRED  \\\n",
       "Genomic Description (GRCh37)                                       \n",
       "20:21686361:C>T                                C             NaN   \n",
       "20:21686379:G>A                                C             NaN   \n",
       "20:21686382:C>T                                C             NaN   \n",
       "20:21686402:G>C                                N             NaN   \n",
       "20:21686445:C>T                                C             NaN   \n",
       "...                                          ...             ...   \n",
       "10:8115913:C>T                                 C               D   \n",
       "10:8115916:C>G                                 C               U   \n",
       "10:8115916:C>T                                 C               U   \n",
       "10:8115927:C>T                                 C               D   \n",
       "10:8115945:G>A                                 C               D   \n",
       "\n",
       "                              DBNSFP_LRT_SCORE DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                                                \n",
       "20:21686361:C>T                            NaN                          N   \n",
       "20:21686379:G>A                            NaN                          N   \n",
       "20:21686382:C>T                            NaN                          N   \n",
       "20:21686402:G>C                            NaN                          N   \n",
       "20:21686445:C>T                            NaN                          N   \n",
       "...                                        ...                        ...   \n",
       "10:8115913:C>T                        0.000000                        D&D   \n",
       "10:8115916:C>G                        0.000643                        D&D   \n",
       "10:8115916:C>T                        0.000643                        D&D   \n",
       "10:8115927:C>T                        0.000030                        D&D   \n",
       "10:8115945:G>A                        0.000008                        D&D   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_SCORE  \\\n",
       "Genomic Description (GRCh37)                               \n",
       "20:21686361:C>T                                 0.758794   \n",
       "20:21686379:G>A                                 0.999272   \n",
       "20:21686382:C>T                                 0.998176   \n",
       "20:21686402:G>C                                        1   \n",
       "20:21686445:C>T                                 0.995526   \n",
       "...                                                  ...   \n",
       "10:8115913:C>T                                       1&1   \n",
       "10:8115916:C>G                                       1&1   \n",
       "10:8115916:C>T                                       1&1   \n",
       "10:8115927:C>T                         0.999167&0.999167   \n",
       "10:8115945:G>A                         0.999993&0.999993   \n",
       "\n",
       "                              DBNSFP_PHYLOP30WAY_MAMMALIAN  \\\n",
       "Genomic Description (GRCh37)                                 \n",
       "20:21686361:C>T                                      0.118   \n",
       "20:21686379:G>A                                      1.147   \n",
       "20:21686382:C>T                                      0.106   \n",
       "20:21686402:G>C                                      0.111   \n",
       "20:21686445:C>T                                      0.937   \n",
       "...                                                    ...   \n",
       "10:8115913:C>T                                       1.026   \n",
       "10:8115916:C>G                                       1.026   \n",
       "10:8115916:C>T                                       1.026   \n",
       "10:8115927:C>T                                       1.026   \n",
       "10:8115945:G>A                                       1.176   \n",
       "\n",
       "                             DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED  ...  \\\n",
       "Genomic Description (GRCh37)                                    ...   \n",
       "20:21686361:C>T                                              N  ...   \n",
       "20:21686379:G>A                                              C  ...   \n",
       "20:21686382:C>T                                              N  ...   \n",
       "20:21686402:G>C                                              N  ...   \n",
       "20:21686445:C>T                                              N  ...   \n",
       "...                                                        ...  ...   \n",
       "10:8115913:C>T                                               C  ...   \n",
       "10:8115916:C>G                                               C  ...   \n",
       "10:8115916:C>T                                               C  ...   \n",
       "10:8115927:C>T                                               C  ...   \n",
       "10:8115945:G>A                                               C  ...   \n",
       "\n",
       "                             NORMALIZED_SURFACE_AREA SURFACE_AREA  \\\n",
       "Genomic Description (GRCh37)                                        \n",
       "20:21686361:C>T                             0.883870   139.065932   \n",
       "20:21686379:G>A                             0.670700   180.703642   \n",
       "20:21686382:C>T                             0.601022    76.425233   \n",
       "20:21686402:G>C                             0.714805    68.153896   \n",
       "20:21686445:C>T                             0.572725    72.826957   \n",
       "...                                              ...          ...   \n",
       "10:8115913:C>T                              0.684427   107.686086   \n",
       "10:8115916:C>G                              0.870607   136.946574   \n",
       "10:8115916:C>T                              0.870607   136.946574   \n",
       "10:8115927:C>T                              0.816259   128.397680   \n",
       "10:8115945:G>A                              0.709449    67.643291   \n",
       "\n",
       "                             TOTAL_NUM_PREDS VEP_CADD_RAW VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686361:C>T                          5.0     2.751772                 P   \n",
       "20:21686379:G>A                          5.0     2.662314                 P   \n",
       "20:21686382:C>T                          5.0     2.257539                 P   \n",
       "20:21686402:G>C                          5.0     0.289852                 B   \n",
       "20:21686445:C>T                          5.0     2.089529                 P   \n",
       "...                                      ...          ...               ...   \n",
       "10:8115913:C>T                           6.0     4.074794                 D   \n",
       "10:8115916:C>G                           5.0     3.979789                 D   \n",
       "10:8115916:C>T                           5.0     4.163516                 D   \n",
       "10:8115927:C>T                           6.0     1.223753                 B   \n",
       "10:8115945:G>A                           6.0     3.995061                 D   \n",
       "\n",
       "                             VEP_POLYPHEN_SCORE  VEP_SIFT_SCORE  \\\n",
       "Genomic Description (GRCh37)                                      \n",
       "20:21686361:C>T                           0.917            0.00   \n",
       "20:21686379:G>A                           0.787            0.00   \n",
       "20:21686382:C>T                           0.804            0.01   \n",
       "20:21686402:G>C                           0.000            1.00   \n",
       "20:21686445:C>T                           0.947            0.00   \n",
       "...                                         ...             ...   \n",
       "10:8115913:C>T                            1.000            0.01   \n",
       "10:8115916:C>G                            1.000            0.00   \n",
       "10:8115916:C>T                            1.000            0.00   \n",
       "10:8115927:C>T                            0.097            1.00   \n",
       "10:8115945:G>A                            0.997            0.00   \n",
       "\n",
       "                             VARIANT_CLASSIFICATION  ddG  |ddG|  \n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686361:C>T                Unknown significance -0.0    0.0  \n",
       "20:21686379:G>A                Unknown significance -0.3    0.3  \n",
       "20:21686382:C>T                              Benign -0.3    0.3  \n",
       "20:21686402:G>C                       Likely benign  0.5    0.5  \n",
       "20:21686445:C>T                Unknown significance  0.1    0.1  \n",
       "...                                             ...  ...    ...  \n",
       "10:8115913:C>T                    Likely pathogenic -0.1    0.1  \n",
       "10:8115916:C>G                 Unknown significance -0.3    0.3  \n",
       "10:8115916:C>T                 Unknown significance -0.2    0.2  \n",
       "10:8115927:C>T                 Unknown significance -0.1    0.1  \n",
       "10:8115945:G>A                 Unknown significance  0.2    0.2  \n",
       "\n",
       "[95354 rows x 29 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.drop(columns=['DBNSFP_INTERPRO_DOMAIN', 'INTERPRO_DOMAIN', 'UNIPROT_DOMAIN'])\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "HERE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD_PHRED</th>\n",
       "      <th>CONFIDENCE_SCORE</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>VARIANT_CLASSIFICATION</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "      <th>PREDICTION_RATIO</th>\n",
       "      <th>INTERNAL_STATE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.00</td>\n",
       "      <td>38.06</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.758794</td>\n",
       "      <td>0.118</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.751772</td>\n",
       "      <td>P</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.90</td>\n",
       "      <td>37.23</td>\n",
       "      <td>2.32</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.999272</td>\n",
       "      <td>1.147</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.662314</td>\n",
       "      <td>P</td>\n",
       "      <td>0.787</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.800000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.80</td>\n",
       "      <td>34.61</td>\n",
       "      <td>2.11</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.998176</td>\n",
       "      <td>0.106</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.257539</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>7.08</td>\n",
       "      <td>36.34</td>\n",
       "      <td>-0.29</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>0.111</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.289852</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686445:C&gt;T</th>\n",
       "      <td>20.50</td>\n",
       "      <td>35.99</td>\n",
       "      <td>3.25</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.995526</td>\n",
       "      <td>0.937</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.089529</td>\n",
       "      <td>P</td>\n",
       "      <td>0.947</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115913:C&gt;T</th>\n",
       "      <td>28.40</td>\n",
       "      <td>51.75</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.074794</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>27.50</td>\n",
       "      <td>47.29</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>U</td>\n",
       "      <td>0.000643</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.979789</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>29.20</td>\n",
       "      <td>47.29</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>U</td>\n",
       "      <td>0.000643</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.163516</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>14.78</td>\n",
       "      <td>44.11</td>\n",
       "      <td>3.68</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.999167&amp;0.999167</td>\n",
       "      <td>1.026</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.223753</td>\n",
       "      <td>B</td>\n",
       "      <td>0.097</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>27.60</td>\n",
       "      <td>37.21</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.999993&amp;0.999993</td>\n",
       "      <td>1.176</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.995061</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95354 rows Ã— 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD_PHRED  CONFIDENCE_SCORE  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                                 \n",
       "20:21686361:C>T                    23.00             38.06            4.28   \n",
       "20:21686379:G>A                    22.90             37.23            2.32   \n",
       "20:21686382:C>T                    21.80             34.61            2.11   \n",
       "20:21686402:G>C                     7.08             36.34           -0.29   \n",
       "20:21686445:C>T                    20.50             35.99            3.25   \n",
       "...                                  ...               ...             ...   \n",
       "10:8115913:C>T                     28.40             51.75            5.59   \n",
       "10:8115916:C>G                     27.50             47.29            5.59   \n",
       "10:8115916:C>T                     29.20             47.29            5.59   \n",
       "10:8115927:C>T                     14.78             44.11            3.68   \n",
       "10:8115945:G>A                     27.60             37.21            5.59   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_LRT_PRED  \\\n",
       "Genomic Description (GRCh37)                                       \n",
       "20:21686361:C>T                                C             NaN   \n",
       "20:21686379:G>A                                C             NaN   \n",
       "20:21686382:C>T                                C             NaN   \n",
       "20:21686402:G>C                                N             NaN   \n",
       "20:21686445:C>T                                C             NaN   \n",
       "...                                          ...             ...   \n",
       "10:8115913:C>T                                 C               D   \n",
       "10:8115916:C>G                                 C               U   \n",
       "10:8115916:C>T                                 C               U   \n",
       "10:8115927:C>T                                 C               D   \n",
       "10:8115945:G>A                                 C               D   \n",
       "\n",
       "                              DBNSFP_LRT_SCORE DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                                                \n",
       "20:21686361:C>T                            NaN                          N   \n",
       "20:21686379:G>A                            NaN                          N   \n",
       "20:21686382:C>T                            NaN                          N   \n",
       "20:21686402:G>C                            NaN                          N   \n",
       "20:21686445:C>T                            NaN                          N   \n",
       "...                                        ...                        ...   \n",
       "10:8115913:C>T                        0.000000                        D&D   \n",
       "10:8115916:C>G                        0.000643                        D&D   \n",
       "10:8115916:C>T                        0.000643                        D&D   \n",
       "10:8115927:C>T                        0.000030                        D&D   \n",
       "10:8115945:G>A                        0.000008                        D&D   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_SCORE  \\\n",
       "Genomic Description (GRCh37)                               \n",
       "20:21686361:C>T                                 0.758794   \n",
       "20:21686379:G>A                                 0.999272   \n",
       "20:21686382:C>T                                 0.998176   \n",
       "20:21686402:G>C                                        1   \n",
       "20:21686445:C>T                                 0.995526   \n",
       "...                                                  ...   \n",
       "10:8115913:C>T                                       1&1   \n",
       "10:8115916:C>G                                       1&1   \n",
       "10:8115916:C>T                                       1&1   \n",
       "10:8115927:C>T                         0.999167&0.999167   \n",
       "10:8115945:G>A                         0.999993&0.999993   \n",
       "\n",
       "                              DBNSFP_PHYLOP30WAY_MAMMALIAN  \\\n",
       "Genomic Description (GRCh37)                                 \n",
       "20:21686361:C>T                                      0.118   \n",
       "20:21686379:G>A                                      1.147   \n",
       "20:21686382:C>T                                      0.106   \n",
       "20:21686402:G>C                                      0.111   \n",
       "20:21686445:C>T                                      0.937   \n",
       "...                                                    ...   \n",
       "10:8115913:C>T                                       1.026   \n",
       "10:8115916:C>G                                       1.026   \n",
       "10:8115916:C>T                                       1.026   \n",
       "10:8115927:C>T                                       1.026   \n",
       "10:8115945:G>A                                       1.176   \n",
       "\n",
       "                             DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED  ...  \\\n",
       "Genomic Description (GRCh37)                                    ...   \n",
       "20:21686361:C>T                                              N  ...   \n",
       "20:21686379:G>A                                              C  ...   \n",
       "20:21686382:C>T                                              N  ...   \n",
       "20:21686402:G>C                                              N  ...   \n",
       "20:21686445:C>T                                              N  ...   \n",
       "...                                                        ...  ...   \n",
       "10:8115913:C>T                                               C  ...   \n",
       "10:8115916:C>G                                               C  ...   \n",
       "10:8115916:C>T                                               C  ...   \n",
       "10:8115927:C>T                                               C  ...   \n",
       "10:8115945:G>A                                               C  ...   \n",
       "\n",
       "                             TOTAL_NUM_PREDS VEP_CADD_RAW VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686361:C>T                          5.0     2.751772                 P   \n",
       "20:21686379:G>A                          5.0     2.662314                 P   \n",
       "20:21686382:C>T                          5.0     2.257539                 P   \n",
       "20:21686402:G>C                          5.0     0.289852                 B   \n",
       "20:21686445:C>T                          5.0     2.089529                 P   \n",
       "...                                      ...          ...               ...   \n",
       "10:8115913:C>T                           6.0     4.074794                 D   \n",
       "10:8115916:C>G                           5.0     3.979789                 D   \n",
       "10:8115916:C>T                           5.0     4.163516                 D   \n",
       "10:8115927:C>T                           6.0     1.223753                 B   \n",
       "10:8115945:G>A                           6.0     3.995061                 D   \n",
       "\n",
       "                             VEP_POLYPHEN_SCORE VEP_SIFT_SCORE  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686361:C>T                           0.917           0.00   \n",
       "20:21686379:G>A                           0.787           0.00   \n",
       "20:21686382:C>T                           0.804           0.01   \n",
       "20:21686402:G>C                           0.000           1.00   \n",
       "20:21686445:C>T                           0.947           0.00   \n",
       "...                                         ...            ...   \n",
       "10:8115913:C>T                            1.000           0.01   \n",
       "10:8115916:C>G                            1.000           0.00   \n",
       "10:8115916:C>T                            1.000           0.00   \n",
       "10:8115927:C>T                            0.097           1.00   \n",
       "10:8115945:G>A                            0.997           0.00   \n",
       "\n",
       "                             VARIANT_CLASSIFICATION  ddG |ddG|  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686361:C>T                Unknown significance -0.0   0.0   \n",
       "20:21686379:G>A                Unknown significance -0.3   0.3   \n",
       "20:21686382:C>T                              Benign -0.3   0.3   \n",
       "20:21686402:G>C                       Likely benign  0.5   0.5   \n",
       "20:21686445:C>T                Unknown significance  0.1   0.1   \n",
       "...                                             ...  ...   ...   \n",
       "10:8115913:C>T                    Likely pathogenic -0.1   0.1   \n",
       "10:8115916:C>G                 Unknown significance -0.3   0.3   \n",
       "10:8115916:C>T                 Unknown significance -0.2   0.2   \n",
       "10:8115927:C>T                 Unknown significance -0.1   0.1   \n",
       "10:8115945:G>A                 Unknown significance  0.2   0.2   \n",
       "\n",
       "                              PREDICTION_RATIO  INTERNAL_STATE  \n",
       "Genomic Description (GRCh37)                                    \n",
       "20:21686361:C>T                       0.600000               D  \n",
       "20:21686379:G>A                       0.800000               D  \n",
       "20:21686382:C>T                       0.600000               D  \n",
       "20:21686402:G>C                       0.000000               B  \n",
       "20:21686445:C>T                       0.600000               D  \n",
       "...                                        ...             ...  \n",
       "10:8115913:C>T                        1.000000               D  \n",
       "10:8115916:C>G                        1.000000               D  \n",
       "10:8115916:C>T                        1.000000               D  \n",
       "10:8115927:C>T                        0.666667               D  \n",
       "10:8115945:G>A                        1.000000               D  \n",
       "\n",
       "[95354 rows x 31 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['PREDICTION_RATIO'] = df['NUM_PATH_PREDS'] / df['TOTAL_NUM_PREDS']\n",
    "df['INTERNAL_STATE'] = 'VUS'\n",
    "df.loc[df['TOTAL_NUM_PREDS'] >= 5, 'INTERNAL_STATE'] = 'D'\n",
    "df.loc[(df['TOTAL_NUM_PREDS'] >= 5) & (df['PREDICTION_RATIO'] < 0.6), 'INTERNAL_STATE'] = 'B'\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.loc[:, ['VARIANT_CLASSIFICATION', 'FINAL_PRED']] = df[['VARIANT_CLASSIFICATION', 'FINAL_PRED']].replace({'Unknown significance': 'VUS', 'Likely benign': 'LB', 'Likely pathogenic': 'LB', 'Benign': 'B', 'Pathogenic': 'P'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VARIANT_CLASSIFICATION</th>\n",
       "      <th>FINAL_PRED</th>\n",
       "      <th>NUM_PATH_PREDS</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>PREDICTION_RATIO</th>\n",
       "      <th>INTERNAL_STATE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>VUS</td>\n",
       "      <td>VUS</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>VUS</td>\n",
       "      <td>VUS</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.800000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>B</td>\n",
       "      <td>VUS</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>LB</td>\n",
       "      <td>LB</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686445:C&gt;T</th>\n",
       "      <td>VUS</td>\n",
       "      <td>VUS</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115913:C&gt;T</th>\n",
       "      <td>LB</td>\n",
       "      <td>VUS</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>VUS</td>\n",
       "      <td>VUS</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>VUS</td>\n",
       "      <td>VUS</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>VUS</td>\n",
       "      <td>VUS</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>VUS</td>\n",
       "      <td>VUS</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>95354 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             VARIANT_CLASSIFICATION FINAL_PRED  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686361:C>T                                 VUS        VUS   \n",
       "20:21686379:G>A                                 VUS        VUS   \n",
       "20:21686382:C>T                                   B        VUS   \n",
       "20:21686402:G>C                                  LB         LB   \n",
       "20:21686445:C>T                                 VUS        VUS   \n",
       "...                                             ...        ...   \n",
       "10:8115913:C>T                                   LB        VUS   \n",
       "10:8115916:C>G                                  VUS        VUS   \n",
       "10:8115916:C>T                                  VUS        VUS   \n",
       "10:8115927:C>T                                  VUS        VUS   \n",
       "10:8115945:G>A                                  VUS        VUS   \n",
       "\n",
       "                              NUM_PATH_PREDS  TOTAL_NUM_PREDS  \\\n",
       "Genomic Description (GRCh37)                                    \n",
       "20:21686361:C>T                          3.0              5.0   \n",
       "20:21686379:G>A                          4.0              5.0   \n",
       "20:21686382:C>T                          3.0              5.0   \n",
       "20:21686402:G>C                          0.0              5.0   \n",
       "20:21686445:C>T                          3.0              5.0   \n",
       "...                                      ...              ...   \n",
       "10:8115913:C>T                           6.0              6.0   \n",
       "10:8115916:C>G                           5.0              5.0   \n",
       "10:8115916:C>T                           5.0              5.0   \n",
       "10:8115927:C>T                           4.0              6.0   \n",
       "10:8115945:G>A                           6.0              6.0   \n",
       "\n",
       "                              PREDICTION_RATIO INTERNAL_STATE  \n",
       "Genomic Description (GRCh37)                                   \n",
       "20:21686361:C>T                       0.600000              D  \n",
       "20:21686379:G>A                       0.800000              D  \n",
       "20:21686382:C>T                       0.600000              D  \n",
       "20:21686402:G>C                       0.000000              B  \n",
       "20:21686445:C>T                       0.600000              D  \n",
       "...                                        ...            ...  \n",
       "10:8115913:C>T                        1.000000              D  \n",
       "10:8115916:C>G                        1.000000              D  \n",
       "10:8115916:C>T                        1.000000              D  \n",
       "10:8115927:C>T                        0.666667              D  \n",
       "10:8115945:G>A                        1.000000              D  \n",
       "\n",
       "[95354 rows x 6 columns]"
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_classifications_df = df[['VARIANT_CLASSIFICATION', 'FINAL_PRED', 'NUM_PATH_PREDS', 'TOTAL_NUM_PREDS', 'PREDICTION_RATIO', 'INTERNAL_STATE']]\n",
    "final_classifications_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Only 3419 missense variats from the ddG data that are labeled either pathogenic or benign"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.035855863414224885\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VARIANT_CLASSIFICATION</th>\n",
       "      <th>FINAL_PRED</th>\n",
       "      <th>NUM_PATH_PREDS</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>PREDICTION_RATIO</th>\n",
       "      <th>INTERNAL_STATE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>B</td>\n",
       "      <td>VUS</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.600000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686448:T&gt;C</th>\n",
       "      <td>B</td>\n",
       "      <td>VUS</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.800000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686456:C&gt;G</th>\n",
       "      <td>B</td>\n",
       "      <td>LB</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21687286:G&gt;T</th>\n",
       "      <td>P</td>\n",
       "      <td>VUS</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21689216:G&gt;A</th>\n",
       "      <td>P</td>\n",
       "      <td>LB</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:73574874:G&gt;A</th>\n",
       "      <td>P</td>\n",
       "      <td>VUS</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8100506:C&gt;G</th>\n",
       "      <td>B</td>\n",
       "      <td>LB</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.333333</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8100763:G&gt;C</th>\n",
       "      <td>B</td>\n",
       "      <td>VUS</td>\n",
       "      <td>4.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.666667</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8111473:G&gt;C</th>\n",
       "      <td>P</td>\n",
       "      <td>VUS</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115837:G&gt;A</th>\n",
       "      <td>B</td>\n",
       "      <td>VUS</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.833333</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3419 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             VARIANT_CLASSIFICATION FINAL_PRED  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686382:C>T                                   B        VUS   \n",
       "20:21686448:T>C                                   B        VUS   \n",
       "20:21686456:C>G                                   B         LB   \n",
       "20:21687286:G>T                                   P        VUS   \n",
       "20:21689216:G>A                                   P         LB   \n",
       "...                                             ...        ...   \n",
       "10:73574874:G>A                                   P        VUS   \n",
       "10:8100506:C>G                                    B         LB   \n",
       "10:8100763:G>C                                    B        VUS   \n",
       "10:8111473:G>C                                    P        VUS   \n",
       "10:8115837:G>A                                    B        VUS   \n",
       "\n",
       "                              NUM_PATH_PREDS  TOTAL_NUM_PREDS  \\\n",
       "Genomic Description (GRCh37)                                    \n",
       "20:21686382:C>T                          3.0              5.0   \n",
       "20:21686448:T>C                          4.0              5.0   \n",
       "20:21686456:C>G                          1.0              5.0   \n",
       "20:21687286:G>T                          6.0              6.0   \n",
       "20:21689216:G>A                          2.0              6.0   \n",
       "...                                      ...              ...   \n",
       "10:73574874:G>A                          6.0              6.0   \n",
       "10:8100506:C>G                           2.0              6.0   \n",
       "10:8100763:G>C                           4.0              6.0   \n",
       "10:8111473:G>C                           6.0              6.0   \n",
       "10:8115837:G>A                           5.0              6.0   \n",
       "\n",
       "                              PREDICTION_RATIO INTERNAL_STATE  \n",
       "Genomic Description (GRCh37)                                   \n",
       "20:21686382:C>T                       0.600000              D  \n",
       "20:21686448:T>C                       0.800000              D  \n",
       "20:21686456:C>G                       0.200000              B  \n",
       "20:21687286:G>T                       1.000000              D  \n",
       "20:21689216:G>A                       0.333333              B  \n",
       "...                                        ...            ...  \n",
       "10:73574874:G>A                       1.000000              D  \n",
       "10:8100506:C>G                        0.333333              B  \n",
       "10:8100763:G>C                        0.666667              D  \n",
       "10:8111473:G>C                        1.000000              D  \n",
       "10:8115837:G>A                        0.833333              D  \n",
       "\n",
       "[3419 rows x 6 columns]"
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "definitive_classification_df = final_classifications_df[final_classifications_df['VARIANT_CLASSIFICATION'].isin(['P', 'B'])]\n",
    "print(definitive_classification_df.shape[0]/final_classifications_df.shape[0])\n",
    "definitive_classification_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "VARIANT_CLASSIFICATION  INTERNAL_STATE\n",
       "P                       D                 1584\n",
       "B                       B                  937\n",
       "                        D                  618\n",
       "P                       B                  226\n",
       "B                       VUS                 52\n",
       "P                       VUS                  2\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "definitive_classification_df[['VARIANT_CLASSIFICATION', 'INTERNAL_STATE']].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.7373501023691138"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.metrics import accuracy_score\n",
    "\n",
    "accuracy_score(definitive_classification_df.VARIANT_CLASSIFICATION, definitive_classification_df.INTERNAL_STATE.replace('D', 'P'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/3f/85zthwnn17dfrdm_swynn0qr0000gn/T/ipykernel_68115/2360911525.py:1: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  df = pd.concat(dfs).map(utils.safe_decode)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TYPE</th>\n",
       "      <th>GENE</th>\n",
       "      <th>gnomad_MAX_MAF</th>\n",
       "      <th>gnomad_MAX_MAF_SOURCE</th>\n",
       "      <th>gnomad_AC_afr_pass</th>\n",
       "      <th>gnomad_AC_amr_pass</th>\n",
       "      <th>gnomad_AC_eas_pass</th>\n",
       "      <th>gnomad_AC_nfe_pass</th>\n",
       "      <th>gnomad_AN_afr_pass</th>\n",
       "      <th>gnomad_AN_amr_pass</th>\n",
       "      <th>...</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>Surface Area</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "      <th>ca...</th>\n",
       "      <th>in...</th>\n",
       "      <th>calc...</th>\n",
       "      <th>second part</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>172.951510</td>\n",
       "      <td>0.752780</td>\n",
       "      <td>32.73</td>\n",
       "      <td>-1.4</td>\n",
       "      <td>1.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8602.0</td>\n",
       "      <td>810.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>139.065932</td>\n",
       "      <td>0.883870</td>\n",
       "      <td>38.06</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>0.000105</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1492.0</td>\n",
       "      <td>274.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>180.703642</td>\n",
       "      <td>0.670700</td>\n",
       "      <td>37.23</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>0.031875</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>319.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>10008.0</td>\n",
       "      <td>1046.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>76.425233</td>\n",
       "      <td>0.601022</td>\n",
       "      <td>34.61</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.401913</td>\n",
       "      <td>0.578281</td>\n",
       "      <td>31.28</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8702.0</td>\n",
       "      <td>848.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>0.000123</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>16248.0</td>\n",
       "      <td>34588.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16256.0</td>\n",
       "      <td>34588.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>128.397680</td>\n",
       "      <td>0.816259</td>\n",
       "      <td>44.11</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>24966.0</td>\n",
       "      <td>35438.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>67.643291</td>\n",
       "      <td>0.709449</td>\n",
       "      <td>37.21</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128786 rows Ã— 236 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             TYPE   GENE  gnomad_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                               \n",
       "20:21686358:T>C               SNV   PAX1             NaN   \n",
       "20:21686361:C>T               SNV   PAX1        0.000067   \n",
       "20:21686379:G>A               SNV   PAX1        0.000105   \n",
       "20:21686382:C>T               SNV   PAX1        0.031875   \n",
       "20:21686390:G>T               SNV   PAX1             NaN   \n",
       "...                           ...    ...             ...   \n",
       "10:8115915:C>T                SNV  GATA3             NaN   \n",
       "10:8115916:C>G                SNV  GATA3        0.000065   \n",
       "10:8115916:C>T                SNV  GATA3        0.000123   \n",
       "10:8115927:C>T                SNV  GATA3        0.000009   \n",
       "10:8115945:G>A                SNV  GATA3        0.000015   \n",
       "\n",
       "                             gnomad_MAX_MAF_SOURCE  gnomad_AC_afr_pass  \\\n",
       "Genomic Description (GRCh37)                                             \n",
       "20:21686358:T>C                                NaN                 NaN   \n",
       "20:21686361:C>T                 gnomad_AF_nfe_pass                 0.0   \n",
       "20:21686379:G>A                 gnomad_AF_nfe_pass                 0.0   \n",
       "20:21686382:C>T                 gnomad_AF_afr_pass               319.0   \n",
       "20:21686390:G>T                                NaN                 NaN   \n",
       "...                                            ...                 ...   \n",
       "10:8115915:C>T                                 NaN                 NaN   \n",
       "10:8115916:C>G                  gnomad_AF_nfe_pass                 0.0   \n",
       "10:8115916:C>T                  gnomad_AF_afr_pass                 2.0   \n",
       "10:8115927:C>T                  gnomad_AF_nfe_pass                 0.0   \n",
       "10:8115945:G>A                  gnomad_AF_nfe_pass                 0.0   \n",
       "\n",
       "                              gnomad_AC_amr_pass  gnomad_AC_eas_pass  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "20:21686358:T>C                              NaN                 NaN   \n",
       "20:21686361:C>T                              0.0                 0.0   \n",
       "20:21686379:G>A                              0.0                 0.0   \n",
       "20:21686382:C>T                              1.0                 0.0   \n",
       "20:21686390:G>T                              NaN                 NaN   \n",
       "...                                          ...                 ...   \n",
       "10:8115915:C>T                               NaN                 NaN   \n",
       "10:8115916:C>G                               0.0                 0.0   \n",
       "10:8115916:C>T                               1.0                 0.0   \n",
       "10:8115927:C>T                               0.0                 0.0   \n",
       "10:8115945:G>A                               0.0                 0.0   \n",
       "\n",
       "                              gnomad_AC_nfe_pass  gnomad_AN_afr_pass  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "20:21686358:T>C                              NaN                 NaN   \n",
       "20:21686361:C>T                              1.0              8602.0   \n",
       "20:21686379:G>A                              1.0              1492.0   \n",
       "20:21686382:C>T                              4.0             10008.0   \n",
       "20:21686390:G>T                              NaN                 NaN   \n",
       "...                                          ...                 ...   \n",
       "10:8115915:C>T                               NaN                 NaN   \n",
       "10:8115916:C>G                               1.0              8702.0   \n",
       "10:8115916:C>T                               2.0             16248.0   \n",
       "10:8115927:C>T                               1.0             16256.0   \n",
       "10:8115945:G>A                               2.0             24966.0   \n",
       "\n",
       "                              gnomad_AN_amr_pass  ...      Max MAF Source  \\\n",
       "Genomic Description (GRCh37)                      ...                       \n",
       "20:21686358:T>C                              NaN  ...                 NaN   \n",
       "20:21686361:C>T                            810.0  ...  gnomad_AF_nfe_pass   \n",
       "20:21686379:G>A                            274.0  ...  gnomad_AF_nfe_pass   \n",
       "20:21686382:C>T                           1046.0  ...  gnomad_AF_afr_pass   \n",
       "20:21686390:G>T                              NaN  ...                 NaN   \n",
       "...                                          ...  ...                 ...   \n",
       "10:8115915:C>T                               NaN  ...                 NaN   \n",
       "10:8115916:C>G                             848.0  ...  gnomad_AF_nfe_pass   \n",
       "10:8115916:C>T                           34588.0  ...  gnomad_AF_afr_pass   \n",
       "10:8115927:C>T                           34588.0  ...  gnomad_AF_nfe_pass   \n",
       "10:8115945:G>A                           35438.0  ...  gnomad_AF_nfe_pass   \n",
       "\n",
       "                              Surface Area  Normalized SA  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                 172.951510       0.752780             32.73   \n",
       "20:21686361:C>T                 139.065932       0.883870             38.06   \n",
       "20:21686379:G>A                 180.703642       0.670700             37.23   \n",
       "20:21686382:C>T                  76.425233       0.601022             34.61   \n",
       "20:21686390:G>T                  99.401913       0.578281             31.28   \n",
       "...                                    ...            ...               ...   \n",
       "10:8115915:C>T                  136.946574       0.870607             47.29   \n",
       "10:8115916:C>G                  136.946574       0.870607             47.29   \n",
       "10:8115916:C>T                  136.946574       0.870607             47.29   \n",
       "10:8115927:C>T                  128.397680       0.816259             44.11   \n",
       "10:8115945:G>A                   67.643291       0.709449             37.21   \n",
       "\n",
       "                              ddG  |ddG|  ca...  in...  calc...  second part  \n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C              -1.4    1.4    NaN    NaN      NaN          NaN  \n",
       "20:21686361:C>T              -0.0    0.0    NaN    NaN      NaN          NaN  \n",
       "20:21686379:G>A              -0.3    0.3    NaN    NaN      NaN          NaN  \n",
       "20:21686382:C>T              -0.3    0.3    NaN    NaN      NaN          NaN  \n",
       "20:21686390:G>T              -0.6    0.6    NaN    NaN      NaN          NaN  \n",
       "...                           ...    ...    ...    ...      ...          ...  \n",
       "10:8115915:C>T               -0.3    0.3    NaN    NaN      NaN          NaN  \n",
       "10:8115916:C>G               -0.3    0.3    NaN    NaN      NaN          NaN  \n",
       "10:8115916:C>T               -0.2    0.2    NaN    NaN      NaN          NaN  \n",
       "10:8115927:C>T               -0.1    0.1    NaN    NaN      NaN          NaN  \n",
       "10:8115945:G>A                0.2    0.2    NaN    NaN      NaN          NaN  \n",
       "\n",
       "[128786 rows x 236 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.concat(dfs).map(utils.safe_decode)\n",
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "8022\n"
     ]
    }
   ],
   "source": [
    "definitively_benign_or_pathogenic = df.loc[df.FINAL_PATHOGENICITY.isin(['Benign', 'Pathogenic']), :]\n",
    "print(definitively_benign_or_pathogenic.shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7551\n"
     ]
    }
   ],
   "source": [
    "definitively_benign_or_pathogenic_missense = definitively_benign_or_pathogenic[definitively_benign_or_pathogenic.VEP_CONSEQUENCE == 'missense_variant']\n",
    "print(definitively_benign_or_pathogenic_missense.shape[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "definitively_benign_or_pathogenic_missense = definitively_benign_or_pathogenic_missense.loc[:, ['FINAL_PATHOGENICITY', 'FINAL_PRED', 'NUM_PATH_PREDS', 'TOTAL_NUM_PREDS']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "FINAL_PATHOGENICITY  FINAL_PRED          \n",
       "Pathogenic           Unknown significance    5665\n",
       "Benign               Unknown significance     851\n",
       "                     Likely benign            772\n",
       "Pathogenic           Likely benign            249\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "definitively_benign_or_pathogenic_missense.loc[:, ['FINAL_PATHOGENICITY', 'FINAL_PRED']].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "VARIANT_CLASSIFICATION  FINAL_PRED          \n",
       "Pathogenic              Unknown significance    1673\n",
       "Benign                  Unknown significance     839\n",
       "                        Likely benign            768\n",
       "Pathogenic              Likely benign            139\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "definitive_classification_df.loc[:, ['VARIANT_CLASSIFICATION', 'FINAL_PRED']].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD_PHRED</th>\n",
       "      <th>CONFIDENCE_SCORE</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>NORMALIZED_SURFACE_AREA</th>\n",
       "      <th>SURFACE_AREA</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>VARIANT_CLASSIFICATION</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.800</td>\n",
       "      <td>34.61</td>\n",
       "      <td>2.110</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.998176</td>\n",
       "      <td>0.106</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.601022</td>\n",
       "      <td>76.425233</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.257539</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>7.080</td>\n",
       "      <td>36.34</td>\n",
       "      <td>-0.290</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>0.111</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.714805</td>\n",
       "      <td>68.153896</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.289852</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.00</td>\n",
       "      <td>2</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686448:T&gt;C</th>\n",
       "      <td>16.360</td>\n",
       "      <td>32.99</td>\n",
       "      <td>0.193</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>D</td>\n",
       "      <td>0.994473</td>\n",
       "      <td>1.037</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.918470</td>\n",
       "      <td>177.128056</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.515567</td>\n",
       "      <td>B</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686456:C&gt;G</th>\n",
       "      <td>6.810</td>\n",
       "      <td>33.33</td>\n",
       "      <td>0.668</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.295</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.821205</td>\n",
       "      <td>221.253638</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.269085</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686459:G&gt;A</th>\n",
       "      <td>0.005</td>\n",
       "      <td>33.78</td>\n",
       "      <td>-8.940</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>1</td>\n",
       "      <td>-3.370</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.455428</td>\n",
       "      <td>57.911682</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.885486</td>\n",
       "      <td>B</td>\n",
       "      <td>0.052</td>\n",
       "      <td>0.28</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8111473:G&gt;C</th>\n",
       "      <td>28.400</td>\n",
       "      <td>96.95</td>\n",
       "      <td>5.210</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.176</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.023200</td>\n",
       "      <td>3.585018</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.083778</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.01</td>\n",
       "      <td>4</td>\n",
       "      <td>-4.5</td>\n",
       "      <td>4.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115817:A&gt;C</th>\n",
       "      <td>19.950</td>\n",
       "      <td>36.58</td>\n",
       "      <td>3.240</td>\n",
       "      <td>C</td>\n",
       "      <td>N</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.988413&amp;0.988413</td>\n",
       "      <td>-0.339</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>1.046368</td>\n",
       "      <td>167.237965</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.027122</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.15</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115817:A&gt;G</th>\n",
       "      <td>15.350</td>\n",
       "      <td>36.58</td>\n",
       "      <td>3.240</td>\n",
       "      <td>C</td>\n",
       "      <td>N</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.961623&amp;0.961623</td>\n",
       "      <td>-0.339</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>1.046368</td>\n",
       "      <td>167.237965</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.330430</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.00</td>\n",
       "      <td>2</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115837:G&gt;A</th>\n",
       "      <td>22.900</td>\n",
       "      <td>49.46</td>\n",
       "      <td>5.260</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.999998&amp;0.999998</td>\n",
       "      <td>1.172</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.517109</td>\n",
       "      <td>65.754893</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.701225</td>\n",
       "      <td>P</td>\n",
       "      <td>0.588</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115913:C&gt;T</th>\n",
       "      <td>28.400</td>\n",
       "      <td>51.75</td>\n",
       "      <td>5.590</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>1.026</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.684427</td>\n",
       "      <td>107.686086</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.074794</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.01</td>\n",
       "      <td>3</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26790 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD_PHRED  CONFIDENCE_SCORE  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                                 \n",
       "20:21686382:C>T                   21.800             34.61           2.110   \n",
       "20:21686402:G>C                    7.080             36.34          -0.290   \n",
       "20:21686448:T>C                   16.360             32.99           0.193   \n",
       "20:21686456:C>G                    6.810             33.33           0.668   \n",
       "20:21686459:G>A                    0.005             33.78          -8.940   \n",
       "...                                  ...               ...             ...   \n",
       "10:8111473:G>C                    28.400             96.95           5.210   \n",
       "10:8115817:A>C                    19.950             36.58           3.240   \n",
       "10:8115817:A>G                    15.350             36.58           3.240   \n",
       "10:8115837:G>A                    22.900             49.46           5.260   \n",
       "10:8115913:C>T                    28.400             51.75           5.590   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_LRT_PRED  \\\n",
       "Genomic Description (GRCh37)                                       \n",
       "20:21686382:C>T                                C             NaN   \n",
       "20:21686402:G>C                                N             NaN   \n",
       "20:21686448:T>C                                C             NaN   \n",
       "20:21686456:C>G                                C             NaN   \n",
       "20:21686459:G>A                                N             NaN   \n",
       "...                                          ...             ...   \n",
       "10:8111473:G>C                                 C               D   \n",
       "10:8115817:A>C                                 C               N   \n",
       "10:8115817:A>G                                 C               N   \n",
       "10:8115837:G>A                                 C               D   \n",
       "10:8115913:C>T                                 C               D   \n",
       "\n",
       "                              DBNSFP_LRT_SCORE DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                                                \n",
       "20:21686382:C>T                            NaN                          N   \n",
       "20:21686402:G>C                            NaN                          N   \n",
       "20:21686448:T>C                            NaN                          D   \n",
       "20:21686456:C>G                            NaN                          N   \n",
       "20:21686459:G>A                            NaN                          N   \n",
       "...                                        ...                        ...   \n",
       "10:8111473:G>C                        0.000000                        D&D   \n",
       "10:8115817:A>C                        0.000861                        D&D   \n",
       "10:8115817:A>G                        0.000861                        D&D   \n",
       "10:8115837:G>A                        0.000000                        D&D   \n",
       "10:8115913:C>T                        0.000000                        D&D   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_SCORE  \\\n",
       "Genomic Description (GRCh37)                               \n",
       "20:21686382:C>T                                 0.998176   \n",
       "20:21686402:G>C                                        1   \n",
       "20:21686448:T>C                                 0.994473   \n",
       "20:21686456:C>G                                        1   \n",
       "20:21686459:G>A                                        1   \n",
       "...                                                  ...   \n",
       "10:8111473:G>C                                       1&1   \n",
       "10:8115817:A>C                         0.988413&0.988413   \n",
       "10:8115817:A>G                         0.961623&0.961623   \n",
       "10:8115837:G>A                         0.999998&0.999998   \n",
       "10:8115913:C>T                                       1&1   \n",
       "\n",
       "                              DBNSFP_PHYLOP30WAY_MAMMALIAN  \\\n",
       "Genomic Description (GRCh37)                                 \n",
       "20:21686382:C>T                                      0.106   \n",
       "20:21686402:G>C                                      0.111   \n",
       "20:21686448:T>C                                      1.037   \n",
       "20:21686456:C>G                                     -0.295   \n",
       "20:21686459:G>A                                     -3.370   \n",
       "...                                                    ...   \n",
       "10:8111473:G>C                                       1.176   \n",
       "10:8115817:A>C                                      -0.339   \n",
       "10:8115817:A>G                                      -0.339   \n",
       "10:8115837:G>A                                       1.172   \n",
       "10:8115913:C>T                                       1.026   \n",
       "\n",
       "                             DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED  ...  \\\n",
       "Genomic Description (GRCh37)                                    ...   \n",
       "20:21686382:C>T                                              N  ...   \n",
       "20:21686402:G>C                                              N  ...   \n",
       "20:21686448:T>C                                              C  ...   \n",
       "20:21686456:C>G                                              N  ...   \n",
       "20:21686459:G>A                                              N  ...   \n",
       "...                                                        ...  ...   \n",
       "10:8111473:G>C                                               C  ...   \n",
       "10:8115817:A>C                                               N  ...   \n",
       "10:8115817:A>G                                               N  ...   \n",
       "10:8115837:G>A                                               C  ...   \n",
       "10:8115913:C>T                                               C  ...   \n",
       "\n",
       "                             NORMALIZED_SURFACE_AREA SURFACE_AREA  \\\n",
       "Genomic Description (GRCh37)                                        \n",
       "20:21686382:C>T                             0.601022    76.425233   \n",
       "20:21686402:G>C                             0.714805    68.153896   \n",
       "20:21686448:T>C                             0.918470   177.128056   \n",
       "20:21686456:C>G                             0.821205   221.253638   \n",
       "20:21686459:G>A                             0.455428    57.911682   \n",
       "...                                              ...          ...   \n",
       "10:8111473:G>C                              0.023200     3.585018   \n",
       "10:8115817:A>C                              1.046368   167.237965   \n",
       "10:8115817:A>G                              1.046368   167.237965   \n",
       "10:8115837:G>A                              0.517109    65.754893   \n",
       "10:8115913:C>T                              0.684427   107.686086   \n",
       "\n",
       "                             TOTAL_NUM_PREDS VEP_CADD_RAW VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686382:C>T                          5.0     2.257539                 P   \n",
       "20:21686402:G>C                          5.0     0.289852                 B   \n",
       "20:21686448:T>C                          5.0     1.515567                 B   \n",
       "20:21686456:C>G                          5.0     0.269085                 B   \n",
       "20:21686459:G>A                          5.0    -0.885486                 B   \n",
       "...                                      ...          ...               ...   \n",
       "10:8111473:G>C                           6.0     4.083778                 D   \n",
       "10:8115817:A>C                           6.0     2.027122                 B   \n",
       "10:8115817:A>G                           6.0     1.330430                 B   \n",
       "10:8115837:G>A                           6.0     2.701225                 P   \n",
       "10:8115913:C>T                           6.0     4.074794                 D   \n",
       "\n",
       "                             VEP_POLYPHEN_SCORE  VEP_SIFT_SCORE  \\\n",
       "Genomic Description (GRCh37)                                      \n",
       "20:21686382:C>T                           0.804            0.01   \n",
       "20:21686402:G>C                           0.000            1.00   \n",
       "20:21686448:T>C                           0.001            0.00   \n",
       "20:21686456:C>G                           0.000            0.57   \n",
       "20:21686459:G>A                           0.052            0.28   \n",
       "...                                         ...             ...   \n",
       "10:8111473:G>C                            1.000            0.01   \n",
       "10:8115817:A>C                            0.000            0.15   \n",
       "10:8115817:A>G                            0.000            1.00   \n",
       "10:8115837:G>A                            0.588            0.07   \n",
       "10:8115913:C>T                            1.000            0.01   \n",
       "\n",
       "                             VARIANT_CLASSIFICATION  ddG  |ddG|  \n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686382:C>T                                   0 -0.3    0.3  \n",
       "20:21686402:G>C                                   2  0.5    0.5  \n",
       "20:21686448:T>C                                   0 -0.3    0.3  \n",
       "20:21686456:C>G                                   0 -0.1    0.1  \n",
       "20:21686459:G>A                                   2  0.0    0.0  \n",
       "...                                             ...  ...    ...  \n",
       "10:8111473:G>C                                    4 -4.5    4.5  \n",
       "10:8115817:A>C                                    2  0.0    0.0  \n",
       "10:8115817:A>G                                    2  0.0    0.0  \n",
       "10:8115837:G>A                                    0 -0.0    0.0  \n",
       "10:8115913:C>T                                    3 -0.1    0.1  \n",
       "\n",
       "[26790 rows x 29 columns]"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from sklearn.preprocessing import LabelEncoder\n",
    "\n",
    "le = LabelEncoder().fit(variants_w_label_df['VARIANT_CLASSIFICATION'])\n",
    "\n",
    "variants_w_label_df.loc[:, 'VARIANT_CLASSIFICATION'] = le.transform(variants_w_label_df.loc[:, 'VARIANT_CLASSIFICATION'])\n",
    "variants_w_label_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "Can only use .str accessor with string values!",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[79], line 3\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mnumpy\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m \u001b[38;5;21;01mnp\u001b[39;00m\n\u001b[0;32m----> 3\u001b[0m variants_w_label_df\u001b[38;5;241m.\u001b[39mloc[:, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mVARIANT_CLASSIFICATION\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mwhere(\u001b[43mvariants_w_label_df\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mVARIANT_CLASSIFICATION\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mstr\u001b[49m\u001b[38;5;241m.\u001b[39mcontains(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbenign\u001b[39m\u001b[38;5;124m'\u001b[39m, case\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m), \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mbenign\u001b[39m\u001b[38;5;124m'\u001b[39m, variants_w_label_df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mVARIANT_CLASSIFICATION\u001b[39m\u001b[38;5;124m'\u001b[39m])\n\u001b[1;32m      4\u001b[0m variants_w_label_df\u001b[38;5;241m.\u001b[39mloc[:, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mVARIANT_CLASSIFICATION\u001b[39m\u001b[38;5;124m'\u001b[39m] \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mwhere(variants_w_label_df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mVARIANT_CLASSIFICATION\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mstr\u001b[38;5;241m.\u001b[39mcontains(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpathogenic\u001b[39m\u001b[38;5;124m'\u001b[39m, case\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m), \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mpathogenic\u001b[39m\u001b[38;5;124m'\u001b[39m, variants_w_label_df[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mVARIANT_CLASSIFICATION\u001b[39m\u001b[38;5;124m'\u001b[39m])\n\u001b[1;32m      5\u001b[0m variants_w_label_df\u001b[38;5;241m.\u001b[39mVARIANT_CLASSIFICATION\u001b[38;5;241m.\u001b[39mvalue_counts()\n",
      "File \u001b[0;32m~/Documents/GitHub/MORL-DVD-SVM/.venv311/lib/python3.11/site-packages/pandas/core/generic.py:6204\u001b[0m, in \u001b[0;36mNDFrame.__getattr__\u001b[0;34m(self, name)\u001b[0m\n\u001b[1;32m   6197\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m (\n\u001b[1;32m   6198\u001b[0m     name \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_internal_names_set\n\u001b[1;32m   6199\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m name \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_metadata\n\u001b[1;32m   6200\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m name \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_accessors\n\u001b[1;32m   6201\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_info_axis\u001b[38;5;241m.\u001b[39m_can_hold_identifiers_and_holds_name(name)\n\u001b[1;32m   6202\u001b[0m ):\n\u001b[1;32m   6203\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m[name]\n\u001b[0;32m-> 6204\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mobject\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[38;5;21;43m__getattribute__\u001b[39;49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mname\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/Documents/GitHub/MORL-DVD-SVM/.venv311/lib/python3.11/site-packages/pandas/core/accessor.py:224\u001b[0m, in \u001b[0;36mCachedAccessor.__get__\u001b[0;34m(self, obj, cls)\u001b[0m\n\u001b[1;32m    221\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m obj \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    222\u001b[0m     \u001b[38;5;66;03m# we're accessing the attribute of the class, i.e., Dataset.geo\u001b[39;00m\n\u001b[1;32m    223\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_accessor\n\u001b[0;32m--> 224\u001b[0m accessor_obj \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_accessor\u001b[49m\u001b[43m(\u001b[49m\u001b[43mobj\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    225\u001b[0m \u001b[38;5;66;03m# Replace the property with the accessor object. Inspired by:\u001b[39;00m\n\u001b[1;32m    226\u001b[0m \u001b[38;5;66;03m# https://www.pydanny.com/cached-property.html\u001b[39;00m\n\u001b[1;32m    227\u001b[0m \u001b[38;5;66;03m# We need to use object.__setattr__ because we overwrite __setattr__ on\u001b[39;00m\n\u001b[1;32m    228\u001b[0m \u001b[38;5;66;03m# NDFrame\u001b[39;00m\n\u001b[1;32m    229\u001b[0m \u001b[38;5;28mobject\u001b[39m\u001b[38;5;241m.\u001b[39m\u001b[38;5;21m__setattr__\u001b[39m(obj, \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_name, accessor_obj)\n",
      "File \u001b[0;32m~/Documents/GitHub/MORL-DVD-SVM/.venv311/lib/python3.11/site-packages/pandas/core/strings/accessor.py:190\u001b[0m, in \u001b[0;36mStringMethods.__init__\u001b[0;34m(self, data)\u001b[0m\n\u001b[1;32m    187\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__init__\u001b[39m(\u001b[38;5;28mself\u001b[39m, data) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    188\u001b[0m     \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mpandas\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mcore\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01marrays\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01mstring_\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m StringDtype\n\u001b[0;32m--> 190\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_inferred_dtype \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_validate\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdata\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    191\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_is_categorical \u001b[38;5;241m=\u001b[39m \u001b[38;5;28misinstance\u001b[39m(data\u001b[38;5;241m.\u001b[39mdtype, CategoricalDtype)\n\u001b[1;32m    192\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_is_string \u001b[38;5;241m=\u001b[39m \u001b[38;5;28misinstance\u001b[39m(data\u001b[38;5;241m.\u001b[39mdtype, StringDtype)\n",
      "File \u001b[0;32m~/Documents/GitHub/MORL-DVD-SVM/.venv311/lib/python3.11/site-packages/pandas/core/strings/accessor.py:244\u001b[0m, in \u001b[0;36mStringMethods._validate\u001b[0;34m(data)\u001b[0m\n\u001b[1;32m    241\u001b[0m inferred_dtype \u001b[38;5;241m=\u001b[39m lib\u001b[38;5;241m.\u001b[39minfer_dtype(values, skipna\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[1;32m    243\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m inferred_dtype \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m allowed_types:\n\u001b[0;32m--> 244\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mAttributeError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCan only use .str accessor with string values!\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m    245\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m inferred_dtype\n",
      "\u001b[0;31mAttributeError\u001b[0m: Can only use .str accessor with string values!"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "\n",
    "variants_w_label_df.loc[:, 'VARIANT_CLASSIFICATION'] = np.where(variants_w_label_df['VARIANT_CLASSIFICATION'].str.contains('benign', case=False), 'benign', variants_w_label_df['VARIANT_CLASSIFICATION'])\n",
    "variants_w_label_df.loc[:, 'VARIANT_CLASSIFICATION'] = np.where(variants_w_label_df['VARIANT_CLASSIFICATION'].str.contains('pathogenic', case=False), 'pathogenic', variants_w_label_df['VARIANT_CLASSIFICATION'])\n",
    "variants_w_label_df.VARIANT_CLASSIFICATION.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove variants where ddG data is unavailable"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'variants_w_label_df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[90], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m variants_w_label_df \u001b[38;5;241m=\u001b[39m \u001b[43mvariants_w_label_df\u001b[49m[\u001b[38;5;241m~\u001b[39mvariants_w_label_df\u001b[38;5;241m.\u001b[39mddG\u001b[38;5;241m.\u001b[39misna()]\n",
      "\u001b[0;31mNameError\u001b[0m: name 'variants_w_label_df' is not defined"
     ]
    }
   ],
   "source": [
    "variants_w_label_df = variants_w_label_df[~variants_w_label_df.ddG.isna()]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'variants_w_label_df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[91], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mvariants_w_label_df\u001b[49m\u001b[38;5;241m.\u001b[39mVARIANT_CLASSIFICATION\u001b[38;5;241m.\u001b[39mvalue_counts()\n",
      "\u001b[0;31mNameError\u001b[0m: name 'variants_w_label_df' is not defined"
     ]
    }
   ],
   "source": [
    "variants_w_label_df.VARIANT_CLASSIFICATION.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 466,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VARIANT_CLASSIFICATION</th>\n",
       "      <th>FINAL_PRED</th>\n",
       "      <th>NUM_PATH_PREDS</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>DBNSFP_SIFT_PRED</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED</th>\n",
       "      <th>DBNSFP_POLYPHEN2_HDIV_PRED</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>Benign</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>Likely benign</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>T&amp;.</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>B&amp;.</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686448:T&gt;C</th>\n",
       "      <td>Benign</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>T&amp;.</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>D</td>\n",
       "      <td>C</td>\n",
       "      <td>B&amp;.</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686456:C&gt;G</th>\n",
       "      <td>Benign</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>T&amp;.</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>B&amp;.</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686459:G&gt;A</th>\n",
       "      <td>Likely benign</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>T&amp;.</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>N</td>\n",
       "      <td>B&amp;.</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8111473:G&gt;C</th>\n",
       "      <td>Pathogenic</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>D&amp;.&amp;D</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>C</td>\n",
       "      <td>D&amp;D&amp;D</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115817:A&gt;C</th>\n",
       "      <td>Likely benign</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>T&amp;.&amp;T</td>\n",
       "      <td>C</td>\n",
       "      <td>N</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>N</td>\n",
       "      <td>B&amp;B&amp;B</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115817:A&gt;G</th>\n",
       "      <td>Likely benign</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>T&amp;.&amp;T</td>\n",
       "      <td>C</td>\n",
       "      <td>N</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>N</td>\n",
       "      <td>B&amp;B&amp;B</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115837:G&gt;A</th>\n",
       "      <td>Benign</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>T&amp;.&amp;T</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>C</td>\n",
       "      <td>P&amp;P&amp;D</td>\n",
       "      <td>P</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115913:C&gt;T</th>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "      <td>D&amp;.&amp;D</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>C</td>\n",
       "      <td>D&amp;D&amp;D</td>\n",
       "      <td>D</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26781 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             VARIANT_CLASSIFICATION            FINAL_PRED  \\\n",
       "Genomic Description (GRCh37)                                                \n",
       "20:21686382:C>T                              Benign  Unknown significance   \n",
       "20:21686402:G>C                       Likely benign         Likely benign   \n",
       "20:21686448:T>C                              Benign  Unknown significance   \n",
       "20:21686456:C>G                              Benign         Likely benign   \n",
       "20:21686459:G>A                       Likely benign         Likely benign   \n",
       "...                                             ...                   ...   \n",
       "10:8111473:G>C                           Pathogenic  Unknown significance   \n",
       "10:8115817:A>C                        Likely benign         Likely benign   \n",
       "10:8115817:A>G                        Likely benign         Likely benign   \n",
       "10:8115837:G>A                               Benign  Unknown significance   \n",
       "10:8115913:C>T                    Likely pathogenic  Unknown significance   \n",
       "\n",
       "                              NUM_PATH_PREDS  TOTAL_NUM_PREDS  \\\n",
       "Genomic Description (GRCh37)                                    \n",
       "20:21686382:C>T                          3.0              5.0   \n",
       "20:21686402:G>C                          0.0              5.0   \n",
       "20:21686448:T>C                          4.0              5.0   \n",
       "20:21686456:C>G                          1.0              5.0   \n",
       "20:21686459:G>A                          0.0              5.0   \n",
       "...                                      ...              ...   \n",
       "10:8111473:G>C                           6.0              6.0   \n",
       "10:8115817:A>C                           2.0              6.0   \n",
       "10:8115817:A>G                           2.0              6.0   \n",
       "10:8115837:G>A                           5.0              6.0   \n",
       "10:8115913:C>T                           6.0              6.0   \n",
       "\n",
       "                             DBNSFP_SIFT_PRED DBNSFP_GERP_RS_PRED  \\\n",
       "Genomic Description (GRCh37)                                        \n",
       "20:21686382:C>T                           D&.                   C   \n",
       "20:21686402:G>C                           T&.                   N   \n",
       "20:21686448:T>C                           T&.                   C   \n",
       "20:21686456:C>G                           T&.                   C   \n",
       "20:21686459:G>A                           T&.                   N   \n",
       "...                                       ...                 ...   \n",
       "10:8111473:G>C                          D&.&D                   C   \n",
       "10:8115817:A>C                          T&.&T                   C   \n",
       "10:8115817:A>G                          T&.&T                   C   \n",
       "10:8115837:G>A                          T&.&T                   C   \n",
       "10:8115913:C>T                          D&.&D                   C   \n",
       "\n",
       "                             DBNSFP_LRT_PRED DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686382:C>T                          NaN                          N   \n",
       "20:21686402:G>C                          NaN                          N   \n",
       "20:21686448:T>C                          NaN                          D   \n",
       "20:21686456:C>G                          NaN                          N   \n",
       "20:21686459:G>A                          NaN                          N   \n",
       "...                                      ...                        ...   \n",
       "10:8111473:G>C                             D                        D&D   \n",
       "10:8115817:A>C                             N                        D&D   \n",
       "10:8115817:A>G                             N                        D&D   \n",
       "10:8115837:G>A                             D                        D&D   \n",
       "10:8115913:C>T                             D                        D&D   \n",
       "\n",
       "                             DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686382:C>T                                              N   \n",
       "20:21686402:G>C                                              N   \n",
       "20:21686448:T>C                                              C   \n",
       "20:21686456:C>G                                              N   \n",
       "20:21686459:G>A                                              N   \n",
       "...                                                        ...   \n",
       "10:8111473:G>C                                               C   \n",
       "10:8115817:A>C                                               N   \n",
       "10:8115817:A>G                                               N   \n",
       "10:8115837:G>A                                               C   \n",
       "10:8115913:C>T                                               C   \n",
       "\n",
       "                             DBNSFP_POLYPHEN2_HDIV_PRED VEP_POLYPHEN_PRED  \n",
       "Genomic Description (GRCh37)                                               \n",
       "20:21686382:C>T                                     P&.                 P  \n",
       "20:21686402:G>C                                     B&.                 B  \n",
       "20:21686448:T>C                                     B&.                 B  \n",
       "20:21686456:C>G                                     B&.                 B  \n",
       "20:21686459:G>A                                     B&.                 B  \n",
       "...                                                 ...               ...  \n",
       "10:8111473:G>C                                    D&D&D                 D  \n",
       "10:8115817:A>C                                    B&B&B                 B  \n",
       "10:8115817:A>G                                    B&B&B                 B  \n",
       "10:8115837:G>A                                    P&P&D                 P  \n",
       "10:8115913:C>T                                    D&D&D                 D  \n",
       "\n",
       "[26781 rows x 11 columns]"
      ]
     },
     "execution_count": 466,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants_w_label_df[classification + predictions]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Split the columns that have list-like values into lists"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 467,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CADD_PHRED                               0\n",
      "CONFIDENCE_SCORE                         0\n",
      "DBNSFP_GERP_RS                           0\n",
      "DBNSFP_GERP_RS_PRED                      0\n",
      "DBNSFP_LRT_PRED                          0\n",
      "DBNSFP_LRT_SCORE                         0\n",
      "DBNSFP_MUTATIONTASTER_PRED           20359\n",
      "DBNSFP_MUTATIONTASTER_SCORE          20359\n",
      "DBNSFP_PHYLOP30WAY_MAMMALIAN             0\n",
      "DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED        0\n",
      "DBNSFP_POLYPHEN2_HDIV_PRED           21569\n",
      "DBNSFP_POLYPHEN2_HDIV_SCORE          21569\n",
      "DBNSFP_SIFT_PRED                     21569\n",
      "DBNSFP_SIFT_SCORE                    21569\n",
      "FINAL_PRED                               0\n",
      "GENE                                     0\n",
      "MAX_MAF                                  0\n",
      "MAX_MAF_SOURCE                           0\n",
      "NUM_PATH_PREDS                           0\n",
      "NORMALIZED_SURFACE_AREA                  0\n",
      "SURFACE_AREA                             0\n",
      "TOTAL_NUM_PREDS                          0\n",
      "VEP_CADD_RAW                             0\n",
      "VEP_POLYPHEN_PRED                        0\n",
      "VEP_POLYPHEN_SCORE                       0\n",
      "VEP_SIFT_SCORE                           0\n",
      "VARIANT_CLASSIFICATION                   0\n",
      "ddG                                      0\n",
      "|ddG|                                    0\n",
      "dtype: int64\n"
     ]
    }
   ],
   "source": [
    "ampersand_counts = variants_w_label_df.map(lambda x: '&' in str(x)).sum()\n",
    "print(ampersand_counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 383,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "      <th>DBNSFP_POLYPHEN2_HDIV_PRED</th>\n",
       "      <th>DBNSFP_POLYPHEN2_HDIV_SCORE</th>\n",
       "      <th>DBNSFP_SIFT_PRED</th>\n",
       "      <th>DBNSFP_SIFT_SCORE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>[N]</td>\n",
       "      <td>[0.998176]</td>\n",
       "      <td>[P, .]</td>\n",
       "      <td>[0.804, .]</td>\n",
       "      <td>[D, .]</td>\n",
       "      <td>[0.002, .]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>[N]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>[B, .]</td>\n",
       "      <td>[0.0, .]</td>\n",
       "      <td>[T, .]</td>\n",
       "      <td>[0.436, .]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686448:T&gt;C</th>\n",
       "      <td>[D]</td>\n",
       "      <td>[0.994473]</td>\n",
       "      <td>[B, .]</td>\n",
       "      <td>[0.002, .]</td>\n",
       "      <td>[T, .]</td>\n",
       "      <td>[0.065, .]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686456:C&gt;G</th>\n",
       "      <td>[N]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>[B, .]</td>\n",
       "      <td>[0.0, .]</td>\n",
       "      <td>[T, .]</td>\n",
       "      <td>[0.884, .]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686459:G&gt;A</th>\n",
       "      <td>[N]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>[B, .]</td>\n",
       "      <td>[0.052, .]</td>\n",
       "      <td>[T, .]</td>\n",
       "      <td>[0.063, .]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8111473:G&gt;C</th>\n",
       "      <td>[D, D]</td>\n",
       "      <td>[1, 1]</td>\n",
       "      <td>[D, D, D]</td>\n",
       "      <td>[1.0, 1.0, 1.0]</td>\n",
       "      <td>[D, ., D]</td>\n",
       "      <td>[0.0, ., 0.0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115817:A&gt;C</th>\n",
       "      <td>[D, D]</td>\n",
       "      <td>[0.988413, 0.988413]</td>\n",
       "      <td>[B, B, B]</td>\n",
       "      <td>[0.0, 0.0, 0.0]</td>\n",
       "      <td>[T, ., T]</td>\n",
       "      <td>[0.354, ., 0.352]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115817:A&gt;G</th>\n",
       "      <td>[D, D]</td>\n",
       "      <td>[0.961623, 0.961623]</td>\n",
       "      <td>[B, B, B]</td>\n",
       "      <td>[0.0, 0.0, 0.0]</td>\n",
       "      <td>[T, ., T]</td>\n",
       "      <td>[1.0, ., 1.0]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115837:G&gt;A</th>\n",
       "      <td>[D, D]</td>\n",
       "      <td>[0.999998, 0.999998]</td>\n",
       "      <td>[P, P, D]</td>\n",
       "      <td>[0.588, 0.588, 0.999]</td>\n",
       "      <td>[T, ., T]</td>\n",
       "      <td>[0.127, ., 0.128]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115913:C&gt;T</th>\n",
       "      <td>[D, D]</td>\n",
       "      <td>[1, 1]</td>\n",
       "      <td>[D, D, D]</td>\n",
       "      <td>[1.0, 1.0, 1.0]</td>\n",
       "      <td>[D, ., D]</td>\n",
       "      <td>[0.001, ., 0.001]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26781 rows Ã— 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686382:C>T                                     [N]   \n",
       "20:21686402:G>C                                     [N]   \n",
       "20:21686448:T>C                                     [D]   \n",
       "20:21686456:C>G                                     [N]   \n",
       "20:21686459:G>A                                     [N]   \n",
       "...                                                 ...   \n",
       "10:8111473:G>C                                   [D, D]   \n",
       "10:8115817:A>C                                   [D, D]   \n",
       "10:8115817:A>G                                   [D, D]   \n",
       "10:8115837:G>A                                   [D, D]   \n",
       "10:8115913:C>T                                   [D, D]   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_SCORE  \\\n",
       "Genomic Description (GRCh37)                               \n",
       "20:21686382:C>T                               [0.998176]   \n",
       "20:21686402:G>C                                      [1]   \n",
       "20:21686448:T>C                               [0.994473]   \n",
       "20:21686456:C>G                                      [1]   \n",
       "20:21686459:G>A                                      [1]   \n",
       "...                                                  ...   \n",
       "10:8111473:G>C                                    [1, 1]   \n",
       "10:8115817:A>C                      [0.988413, 0.988413]   \n",
       "10:8115817:A>G                      [0.961623, 0.961623]   \n",
       "10:8115837:G>A                      [0.999998, 0.999998]   \n",
       "10:8115913:C>T                                    [1, 1]   \n",
       "\n",
       "                             DBNSFP_POLYPHEN2_HDIV_PRED  \\\n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686382:C>T                                  [P, .]   \n",
       "20:21686402:G>C                                  [B, .]   \n",
       "20:21686448:T>C                                  [B, .]   \n",
       "20:21686456:C>G                                  [B, .]   \n",
       "20:21686459:G>A                                  [B, .]   \n",
       "...                                                 ...   \n",
       "10:8111473:G>C                                [D, D, D]   \n",
       "10:8115817:A>C                                [B, B, B]   \n",
       "10:8115817:A>G                                [B, B, B]   \n",
       "10:8115837:G>A                                [P, P, D]   \n",
       "10:8115913:C>T                                [D, D, D]   \n",
       "\n",
       "                             DBNSFP_POLYPHEN2_HDIV_SCORE DBNSFP_SIFT_PRED  \\\n",
       "Genomic Description (GRCh37)                                                \n",
       "20:21686382:C>T                               [0.804, .]           [D, .]   \n",
       "20:21686402:G>C                                 [0.0, .]           [T, .]   \n",
       "20:21686448:T>C                               [0.002, .]           [T, .]   \n",
       "20:21686456:C>G                                 [0.0, .]           [T, .]   \n",
       "20:21686459:G>A                               [0.052, .]           [T, .]   \n",
       "...                                                  ...              ...   \n",
       "10:8111473:G>C                           [1.0, 1.0, 1.0]        [D, ., D]   \n",
       "10:8115817:A>C                           [0.0, 0.0, 0.0]        [T, ., T]   \n",
       "10:8115817:A>G                           [0.0, 0.0, 0.0]        [T, ., T]   \n",
       "10:8115837:G>A                     [0.588, 0.588, 0.999]        [T, ., T]   \n",
       "10:8115913:C>T                           [1.0, 1.0, 1.0]        [D, ., D]   \n",
       "\n",
       "                              DBNSFP_SIFT_SCORE  \n",
       "Genomic Description (GRCh37)                     \n",
       "20:21686382:C>T                      [0.002, .]  \n",
       "20:21686402:G>C                      [0.436, .]  \n",
       "20:21686448:T>C                      [0.065, .]  \n",
       "20:21686456:C>G                      [0.884, .]  \n",
       "20:21686459:G>A                      [0.063, .]  \n",
       "...                                         ...  \n",
       "10:8111473:G>C                    [0.0, ., 0.0]  \n",
       "10:8115817:A>C                [0.354, ., 0.352]  \n",
       "10:8115817:A>G                    [1.0, ., 1.0]  \n",
       "10:8115837:G>A                [0.127, ., 0.128]  \n",
       "10:8115913:C>T                [0.001, ., 0.001]  \n",
       "\n",
       "[26781 rows x 6 columns]"
      ]
     },
     "execution_count": 383,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cols_w_ampersands = ampersand_counts[ampersand_counts > 0].index.tolist()\n",
    "for col in cols_w_ampersands:\n",
    "    variants_w_label_df.loc[:, col] = variants_w_label_df[col].astype('string').str.split('&')\n",
    "variants_w_label_df[cols_w_ampersands]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 384,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD_PHRED</th>\n",
       "      <th>CONFIDENCE_SCORE</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>NORMALIZED_SURFACE_AREA</th>\n",
       "      <th>SURFACE_AREA</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>VARIANT_CLASSIFICATION</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.800</td>\n",
       "      <td>34.61</td>\n",
       "      <td>2.110</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[N]</td>\n",
       "      <td>[0.998176]</td>\n",
       "      <td>0.106</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.601022</td>\n",
       "      <td>76.425233</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.257539</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>0.01</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>7.080</td>\n",
       "      <td>36.34</td>\n",
       "      <td>-0.290</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[N]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>0.111</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.714805</td>\n",
       "      <td>68.153896</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.289852</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0</td>\n",
       "      <td>0.5</td>\n",
       "      <td>0.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686448:T&gt;C</th>\n",
       "      <td>16.360</td>\n",
       "      <td>32.99</td>\n",
       "      <td>0.193</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[D]</td>\n",
       "      <td>[0.994473]</td>\n",
       "      <td>1.037</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.918470</td>\n",
       "      <td>177.128056</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.515567</td>\n",
       "      <td>B</td>\n",
       "      <td>0.001</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686456:C&gt;G</th>\n",
       "      <td>6.810</td>\n",
       "      <td>33.33</td>\n",
       "      <td>0.668</td>\n",
       "      <td>C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[N]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>-0.295</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.821205</td>\n",
       "      <td>221.253638</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.269085</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.57</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686459:G&gt;A</th>\n",
       "      <td>0.005</td>\n",
       "      <td>33.78</td>\n",
       "      <td>-8.940</td>\n",
       "      <td>N</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[N]</td>\n",
       "      <td>[1]</td>\n",
       "      <td>-3.370</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>0.455428</td>\n",
       "      <td>57.911682</td>\n",
       "      <td>5.0</td>\n",
       "      <td>-0.885486</td>\n",
       "      <td>B</td>\n",
       "      <td>0.052</td>\n",
       "      <td>0.28</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8111473:G&gt;C</th>\n",
       "      <td>28.400</td>\n",
       "      <td>96.95</td>\n",
       "      <td>5.210</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>[D, D]</td>\n",
       "      <td>[1, 1]</td>\n",
       "      <td>1.176</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.023200</td>\n",
       "      <td>3.585018</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.083778</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>-4.5</td>\n",
       "      <td>4.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115817:A&gt;C</th>\n",
       "      <td>19.950</td>\n",
       "      <td>36.58</td>\n",
       "      <td>3.240</td>\n",
       "      <td>C</td>\n",
       "      <td>N</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>[D, D]</td>\n",
       "      <td>[0.988413, 0.988413]</td>\n",
       "      <td>-0.339</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>1.046368</td>\n",
       "      <td>167.237965</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.027122</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>0.15</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115817:A&gt;G</th>\n",
       "      <td>15.350</td>\n",
       "      <td>36.58</td>\n",
       "      <td>3.240</td>\n",
       "      <td>C</td>\n",
       "      <td>N</td>\n",
       "      <td>0.000861</td>\n",
       "      <td>[D, D]</td>\n",
       "      <td>[0.961623, 0.961623]</td>\n",
       "      <td>-0.339</td>\n",
       "      <td>N</td>\n",
       "      <td>...</td>\n",
       "      <td>1.046368</td>\n",
       "      <td>167.237965</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.330430</td>\n",
       "      <td>B</td>\n",
       "      <td>0.000</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115837:G&gt;A</th>\n",
       "      <td>22.900</td>\n",
       "      <td>49.46</td>\n",
       "      <td>5.260</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>[D, D]</td>\n",
       "      <td>[0.999998, 0.999998]</td>\n",
       "      <td>1.172</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.517109</td>\n",
       "      <td>65.754893</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.701225</td>\n",
       "      <td>P</td>\n",
       "      <td>0.588</td>\n",
       "      <td>0.07</td>\n",
       "      <td>0</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115913:C&gt;T</th>\n",
       "      <td>28.400</td>\n",
       "      <td>51.75</td>\n",
       "      <td>5.590</td>\n",
       "      <td>C</td>\n",
       "      <td>D</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>[D, D]</td>\n",
       "      <td>[1, 1]</td>\n",
       "      <td>1.026</td>\n",
       "      <td>C</td>\n",
       "      <td>...</td>\n",
       "      <td>0.684427</td>\n",
       "      <td>107.686086</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.074794</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.01</td>\n",
       "      <td>1</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26781 rows Ã— 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD_PHRED  CONFIDENCE_SCORE  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                                 \n",
       "20:21686382:C>T                   21.800             34.61           2.110   \n",
       "20:21686402:G>C                    7.080             36.34          -0.290   \n",
       "20:21686448:T>C                   16.360             32.99           0.193   \n",
       "20:21686456:C>G                    6.810             33.33           0.668   \n",
       "20:21686459:G>A                    0.005             33.78          -8.940   \n",
       "...                                  ...               ...             ...   \n",
       "10:8111473:G>C                    28.400             96.95           5.210   \n",
       "10:8115817:A>C                    19.950             36.58           3.240   \n",
       "10:8115817:A>G                    15.350             36.58           3.240   \n",
       "10:8115837:G>A                    22.900             49.46           5.260   \n",
       "10:8115913:C>T                    28.400             51.75           5.590   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_LRT_PRED  \\\n",
       "Genomic Description (GRCh37)                                       \n",
       "20:21686382:C>T                                C             NaN   \n",
       "20:21686402:G>C                                N             NaN   \n",
       "20:21686448:T>C                                C             NaN   \n",
       "20:21686456:C>G                                C             NaN   \n",
       "20:21686459:G>A                                N             NaN   \n",
       "...                                          ...             ...   \n",
       "10:8111473:G>C                                 C               D   \n",
       "10:8115817:A>C                                 C               N   \n",
       "10:8115817:A>G                                 C               N   \n",
       "10:8115837:G>A                                 C               D   \n",
       "10:8115913:C>T                                 C               D   \n",
       "\n",
       "                              DBNSFP_LRT_SCORE DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                                                \n",
       "20:21686382:C>T                            NaN                        [N]   \n",
       "20:21686402:G>C                            NaN                        [N]   \n",
       "20:21686448:T>C                            NaN                        [D]   \n",
       "20:21686456:C>G                            NaN                        [N]   \n",
       "20:21686459:G>A                            NaN                        [N]   \n",
       "...                                        ...                        ...   \n",
       "10:8111473:G>C                        0.000000                     [D, D]   \n",
       "10:8115817:A>C                        0.000861                     [D, D]   \n",
       "10:8115817:A>G                        0.000861                     [D, D]   \n",
       "10:8115837:G>A                        0.000000                     [D, D]   \n",
       "10:8115913:C>T                        0.000000                     [D, D]   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_SCORE  \\\n",
       "Genomic Description (GRCh37)                               \n",
       "20:21686382:C>T                               [0.998176]   \n",
       "20:21686402:G>C                                      [1]   \n",
       "20:21686448:T>C                               [0.994473]   \n",
       "20:21686456:C>G                                      [1]   \n",
       "20:21686459:G>A                                      [1]   \n",
       "...                                                  ...   \n",
       "10:8111473:G>C                                    [1, 1]   \n",
       "10:8115817:A>C                      [0.988413, 0.988413]   \n",
       "10:8115817:A>G                      [0.961623, 0.961623]   \n",
       "10:8115837:G>A                      [0.999998, 0.999998]   \n",
       "10:8115913:C>T                                    [1, 1]   \n",
       "\n",
       "                              DBNSFP_PHYLOP30WAY_MAMMALIAN  \\\n",
       "Genomic Description (GRCh37)                                 \n",
       "20:21686382:C>T                                      0.106   \n",
       "20:21686402:G>C                                      0.111   \n",
       "20:21686448:T>C                                      1.037   \n",
       "20:21686456:C>G                                     -0.295   \n",
       "20:21686459:G>A                                     -3.370   \n",
       "...                                                    ...   \n",
       "10:8111473:G>C                                       1.176   \n",
       "10:8115817:A>C                                      -0.339   \n",
       "10:8115817:A>G                                      -0.339   \n",
       "10:8115837:G>A                                       1.172   \n",
       "10:8115913:C>T                                       1.026   \n",
       "\n",
       "                             DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED  ...  \\\n",
       "Genomic Description (GRCh37)                                    ...   \n",
       "20:21686382:C>T                                              N  ...   \n",
       "20:21686402:G>C                                              N  ...   \n",
       "20:21686448:T>C                                              C  ...   \n",
       "20:21686456:C>G                                              N  ...   \n",
       "20:21686459:G>A                                              N  ...   \n",
       "...                                                        ...  ...   \n",
       "10:8111473:G>C                                               C  ...   \n",
       "10:8115817:A>C                                               N  ...   \n",
       "10:8115817:A>G                                               N  ...   \n",
       "10:8115837:G>A                                               C  ...   \n",
       "10:8115913:C>T                                               C  ...   \n",
       "\n",
       "                             NORMALIZED_SURFACE_AREA SURFACE_AREA  \\\n",
       "Genomic Description (GRCh37)                                        \n",
       "20:21686382:C>T                             0.601022    76.425233   \n",
       "20:21686402:G>C                             0.714805    68.153896   \n",
       "20:21686448:T>C                             0.918470   177.128056   \n",
       "20:21686456:C>G                             0.821205   221.253638   \n",
       "20:21686459:G>A                             0.455428    57.911682   \n",
       "...                                              ...          ...   \n",
       "10:8111473:G>C                              0.023200     3.585018   \n",
       "10:8115817:A>C                              1.046368   167.237965   \n",
       "10:8115817:A>G                              1.046368   167.237965   \n",
       "10:8115837:G>A                              0.517109    65.754893   \n",
       "10:8115913:C>T                              0.684427   107.686086   \n",
       "\n",
       "                             TOTAL_NUM_PREDS VEP_CADD_RAW VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686382:C>T                          5.0     2.257539                 P   \n",
       "20:21686402:G>C                          5.0     0.289852                 B   \n",
       "20:21686448:T>C                          5.0     1.515567                 B   \n",
       "20:21686456:C>G                          5.0     0.269085                 B   \n",
       "20:21686459:G>A                          5.0    -0.885486                 B   \n",
       "...                                      ...          ...               ...   \n",
       "10:8111473:G>C                           6.0     4.083778                 D   \n",
       "10:8115817:A>C                           6.0     2.027122                 B   \n",
       "10:8115817:A>G                           6.0     1.330430                 B   \n",
       "10:8115837:G>A                           6.0     2.701225                 P   \n",
       "10:8115913:C>T                           6.0     4.074794                 D   \n",
       "\n",
       "                             VEP_POLYPHEN_SCORE  VEP_SIFT_SCORE  \\\n",
       "Genomic Description (GRCh37)                                      \n",
       "20:21686382:C>T                           0.804            0.01   \n",
       "20:21686402:G>C                           0.000            1.00   \n",
       "20:21686448:T>C                           0.001            0.00   \n",
       "20:21686456:C>G                           0.000            0.57   \n",
       "20:21686459:G>A                           0.052            0.28   \n",
       "...                                         ...             ...   \n",
       "10:8111473:G>C                            1.000            0.01   \n",
       "10:8115817:A>C                            0.000            0.15   \n",
       "10:8115817:A>G                            0.000            1.00   \n",
       "10:8115837:G>A                            0.588            0.07   \n",
       "10:8115913:C>T                            1.000            0.01   \n",
       "\n",
       "                             VARIANT_CLASSIFICATION  ddG  |ddG|  \n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686382:C>T                                   0 -0.3    0.3  \n",
       "20:21686402:G>C                                   0  0.5    0.5  \n",
       "20:21686448:T>C                                   0 -0.3    0.3  \n",
       "20:21686456:C>G                                   0 -0.1    0.1  \n",
       "20:21686459:G>A                                   0  0.0    0.0  \n",
       "...                                             ...  ...    ...  \n",
       "10:8111473:G>C                                    1 -4.5    4.5  \n",
       "10:8115817:A>C                                    0  0.0    0.0  \n",
       "10:8115817:A>G                                    0  0.0    0.0  \n",
       "10:8115837:G>A                                    0 -0.0    0.0  \n",
       "10:8115913:C>T                                    1 -0.1    0.1  \n",
       "\n",
       "[26781 rows x 29 columns]"
      ]
     },
     "execution_count": 384,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants_w_label_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 386,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VARIANT_CLASSIFICATION</th>\n",
       "      <th>FINAL_PRED</th>\n",
       "      <th>NUM_PATH_PREDS</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>0</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>0</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686448:T&gt;C</th>\n",
       "      <td>0</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>4.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686456:C&gt;G</th>\n",
       "      <td>0</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686459:G&gt;A</th>\n",
       "      <td>0</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8111473:G&gt;C</th>\n",
       "      <td>1</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115817:A&gt;C</th>\n",
       "      <td>0</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115817:A&gt;G</th>\n",
       "      <td>0</td>\n",
       "      <td>Likely benign</td>\n",
       "      <td>2.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115837:G&gt;A</th>\n",
       "      <td>0</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>5.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115913:C&gt;T</th>\n",
       "      <td>1</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>6.0</td>\n",
       "      <td>6.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>26781 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             VARIANT_CLASSIFICATION            FINAL_PRED  \\\n",
       "Genomic Description (GRCh37)                                                \n",
       "20:21686382:C>T                                   0  Unknown significance   \n",
       "20:21686402:G>C                                   0         Likely benign   \n",
       "20:21686448:T>C                                   0  Unknown significance   \n",
       "20:21686456:C>G                                   0         Likely benign   \n",
       "20:21686459:G>A                                   0         Likely benign   \n",
       "...                                             ...                   ...   \n",
       "10:8111473:G>C                                    1  Unknown significance   \n",
       "10:8115817:A>C                                    0         Likely benign   \n",
       "10:8115817:A>G                                    0         Likely benign   \n",
       "10:8115837:G>A                                    0  Unknown significance   \n",
       "10:8115913:C>T                                    1  Unknown significance   \n",
       "\n",
       "                              NUM_PATH_PREDS  TOTAL_NUM_PREDS  \n",
       "Genomic Description (GRCh37)                                   \n",
       "20:21686382:C>T                          3.0              5.0  \n",
       "20:21686402:G>C                          0.0              5.0  \n",
       "20:21686448:T>C                          4.0              5.0  \n",
       "20:21686456:C>G                          1.0              5.0  \n",
       "20:21686459:G>A                          0.0              5.0  \n",
       "...                                      ...              ...  \n",
       "10:8111473:G>C                           6.0              6.0  \n",
       "10:8115817:A>C                           2.0              6.0  \n",
       "10:8115817:A>G                           2.0              6.0  \n",
       "10:8115837:G>A                           5.0              6.0  \n",
       "10:8115913:C>T                           6.0              6.0  \n",
       "\n",
       "[26781 rows x 4 columns]"
      ]
     },
     "execution_count": 386,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_classifications_df = variants_w_label_df[['VARIANT_CLASSIFICATION', 'FINAL_PRED', 'NUM_PATH_PREDS', 'TOTAL_NUM_PREDS']]\n",
    "final_classifications_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 385,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "'<' not supported between instances of 'float' and 'str'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[385], line 7\u001b[0m\n\u001b[1;32m      4\u001b[0m azaiez_classification \u001b[38;5;241m=\u001b[39m variants_w_label_df\u001b[38;5;241m.\u001b[39mFINAL_PRED\n\u001b[1;32m      5\u001b[0m majority_vote_classification \u001b[38;5;241m=\u001b[39m variants_w_label_df\u001b[38;5;241m.\u001b[39mNUM_PATH_PREDS \u001b[38;5;241m/\u001b[39m variants_w_label_df\u001b[38;5;241m.\u001b[39mTOTAL_NUM_PREDS\n\u001b[0;32m----> 7\u001b[0m \u001b[43maccuracy_score\u001b[49m\u001b[43m(\u001b[49m\u001b[43mcurated_classification\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mazaiez_classification\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/Documents/GitHub/MORL-DVD-SVM/.venv311/lib/python3.11/site-packages/sklearn/utils/_param_validation.py:213\u001b[0m, in \u001b[0;36mvalidate_params.<locals>.decorator.<locals>.wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    207\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m    208\u001b[0m     \u001b[38;5;28;01mwith\u001b[39;00m config_context(\n\u001b[1;32m    209\u001b[0m         skip_parameter_validation\u001b[38;5;241m=\u001b[39m(\n\u001b[1;32m    210\u001b[0m             prefer_skip_nested_validation \u001b[38;5;129;01mor\u001b[39;00m global_skip_validation\n\u001b[1;32m    211\u001b[0m         )\n\u001b[1;32m    212\u001b[0m     ):\n\u001b[0;32m--> 213\u001b[0m         \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    214\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m InvalidParameterError \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[1;32m    215\u001b[0m     \u001b[38;5;66;03m# When the function is just a wrapper around an estimator, we allow\u001b[39;00m\n\u001b[1;32m    216\u001b[0m     \u001b[38;5;66;03m# the function to delegate validation to the estimator, but we replace\u001b[39;00m\n\u001b[1;32m    217\u001b[0m     \u001b[38;5;66;03m# the name of the estimator by the name of the function in the error\u001b[39;00m\n\u001b[1;32m    218\u001b[0m     \u001b[38;5;66;03m# message to avoid confusion.\u001b[39;00m\n\u001b[1;32m    219\u001b[0m     msg \u001b[38;5;241m=\u001b[39m re\u001b[38;5;241m.\u001b[39msub(\n\u001b[1;32m    220\u001b[0m         \u001b[38;5;124mr\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mparameter of \u001b[39m\u001b[38;5;124m\\\u001b[39m\u001b[38;5;124mw+ must be\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m    221\u001b[0m         \u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mparameter of \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfunc\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__qualname__\u001b[39m\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m must be\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[1;32m    222\u001b[0m         \u001b[38;5;28mstr\u001b[39m(e),\n\u001b[1;32m    223\u001b[0m     )\n",
      "File \u001b[0;32m~/Documents/GitHub/MORL-DVD-SVM/.venv311/lib/python3.11/site-packages/sklearn/metrics/_classification.py:213\u001b[0m, in \u001b[0;36maccuracy_score\u001b[0;34m(y_true, y_pred, normalize, sample_weight)\u001b[0m\n\u001b[1;32m    147\u001b[0m \u001b[38;5;250m\u001b[39m\u001b[38;5;124;03m\"\"\"Accuracy classification score.\u001b[39;00m\n\u001b[1;32m    148\u001b[0m \n\u001b[1;32m    149\u001b[0m \u001b[38;5;124;03mIn multilabel classification, this function computes subset accuracy:\u001b[39;00m\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m    209\u001b[0m \u001b[38;5;124;03m0.5\u001b[39;00m\n\u001b[1;32m    210\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[1;32m    212\u001b[0m \u001b[38;5;66;03m# Compute accuracy for each possible representation\u001b[39;00m\n\u001b[0;32m--> 213\u001b[0m y_type, y_true, y_pred \u001b[38;5;241m=\u001b[39m \u001b[43m_check_targets\u001b[49m\u001b[43m(\u001b[49m\u001b[43my_true\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43my_pred\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    214\u001b[0m check_consistent_length(y_true, y_pred, sample_weight)\n\u001b[1;32m    215\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m y_type\u001b[38;5;241m.\u001b[39mstartswith(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmultilabel\u001b[39m\u001b[38;5;124m\"\u001b[39m):\n",
      "File \u001b[0;32m~/Documents/GitHub/MORL-DVD-SVM/.venv311/lib/python3.11/site-packages/sklearn/metrics/_classification.py:87\u001b[0m, in \u001b[0;36m_check_targets\u001b[0;34m(y_true, y_pred)\u001b[0m\n\u001b[1;32m     85\u001b[0m check_consistent_length(y_true, y_pred)\n\u001b[1;32m     86\u001b[0m type_true \u001b[38;5;241m=\u001b[39m type_of_target(y_true, input_name\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124my_true\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m---> 87\u001b[0m type_pred \u001b[38;5;241m=\u001b[39m \u001b[43mtype_of_target\u001b[49m\u001b[43m(\u001b[49m\u001b[43my_pred\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43minput_name\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43my_pred\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m     89\u001b[0m y_type \u001b[38;5;241m=\u001b[39m {type_true, type_pred}\n\u001b[1;32m     90\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m y_type \u001b[38;5;241m==\u001b[39m {\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mbinary\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmulticlass\u001b[39m\u001b[38;5;124m\"\u001b[39m}:\n",
      "File \u001b[0;32m~/Documents/GitHub/MORL-DVD-SVM/.venv311/lib/python3.11/site-packages/sklearn/utils/multiclass.py:395\u001b[0m, in \u001b[0;36mtype_of_target\u001b[0;34m(y, input_name)\u001b[0m\n\u001b[1;32m    393\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m issparse(first_row):\n\u001b[1;32m    394\u001b[0m     first_row \u001b[38;5;241m=\u001b[39m first_row\u001b[38;5;241m.\u001b[39mdata\n\u001b[0;32m--> 395\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[43mxp\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43munique_values\u001b[49m\u001b[43m(\u001b[49m\u001b[43my\u001b[49m\u001b[43m)\u001b[49m\u001b[38;5;241m.\u001b[39mshape[\u001b[38;5;241m0\u001b[39m] \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m2\u001b[39m \u001b[38;5;129;01mor\u001b[39;00m (y\u001b[38;5;241m.\u001b[39mndim \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m2\u001b[39m \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(first_row) \u001b[38;5;241m>\u001b[39m \u001b[38;5;241m1\u001b[39m):\n\u001b[1;32m    396\u001b[0m     \u001b[38;5;66;03m# [1, 2, 3] or [[1., 2., 3]] or [[1, 2]]\u001b[39;00m\n\u001b[1;32m    397\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmulticlass\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m+\u001b[39m suffix\n\u001b[1;32m    398\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "File \u001b[0;32m~/Documents/GitHub/MORL-DVD-SVM/.venv311/lib/python3.11/site-packages/sklearn/utils/_array_api.py:307\u001b[0m, in \u001b[0;36m_NumPyAPIWrapper.unique_values\u001b[0;34m(self, x)\u001b[0m\n\u001b[1;32m    306\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21munique_values\u001b[39m(\u001b[38;5;28mself\u001b[39m, x):\n\u001b[0;32m--> 307\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mnumpy\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43munique\u001b[49m\u001b[43m(\u001b[49m\u001b[43mx\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m~/Documents/GitHub/MORL-DVD-SVM/.venv311/lib/python3.11/site-packages/numpy/lib/arraysetops.py:274\u001b[0m, in \u001b[0;36munique\u001b[0;34m(ar, return_index, return_inverse, return_counts, axis, equal_nan)\u001b[0m\n\u001b[1;32m    272\u001b[0m ar \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39masanyarray(ar)\n\u001b[1;32m    273\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m axis \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m--> 274\u001b[0m     ret \u001b[38;5;241m=\u001b[39m \u001b[43m_unique1d\u001b[49m\u001b[43m(\u001b[49m\u001b[43mar\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mreturn_index\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mreturn_inverse\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mreturn_counts\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\n\u001b[1;32m    275\u001b[0m \u001b[43m                    \u001b[49m\u001b[43mequal_nan\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mequal_nan\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    276\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m _unpack_tuple(ret)\n\u001b[1;32m    278\u001b[0m \u001b[38;5;66;03m# axis was specified and not None\u001b[39;00m\n",
      "File \u001b[0;32m~/Documents/GitHub/MORL-DVD-SVM/.venv311/lib/python3.11/site-packages/numpy/lib/arraysetops.py:336\u001b[0m, in \u001b[0;36m_unique1d\u001b[0;34m(ar, return_index, return_inverse, return_counts, equal_nan)\u001b[0m\n\u001b[1;32m    334\u001b[0m     aux \u001b[38;5;241m=\u001b[39m ar[perm]\n\u001b[1;32m    335\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m--> 336\u001b[0m     ar\u001b[38;5;241m.\u001b[39msort()\n\u001b[1;32m    337\u001b[0m     aux \u001b[38;5;241m=\u001b[39m ar\n\u001b[1;32m    338\u001b[0m mask \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39mempty(aux\u001b[38;5;241m.\u001b[39mshape, dtype\u001b[38;5;241m=\u001b[39mnp\u001b[38;5;241m.\u001b[39mbool_)\n",
      "\u001b[0;31mTypeError\u001b[0m: '<' not supported between instances of 'float' and 'str'"
     ]
    }
   ],
   "source": [
    "from sklearn.metrics import accuracy_score\n",
    "\n",
    "curated_classification = variants_w_label_df.VARIANT_CLASSIFICATION\n",
    "azaiez_classification = variants_w_label_df.FINAL_PRED\n",
    "majority_vote_classification = variants_w_label_df.NUM_PATH_PREDS / variants_w_label_df.TOTAL_NUM_PREDS\n",
    "\n",
    "accuracy_score(curated_classification, azaiez_classification)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Model Building"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn import feature_selection\n",
    "from sklearn import feature_extraction\n",
    "from sklearn.svm import LinearSVC\n",
    "from sklearn.pipeline import Pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'raw_data' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[56], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m y \u001b[39m=\u001b[39m raw_data[\u001b[39m'\u001b[39m\u001b[39mClass\u001b[39m\u001b[39m'\u001b[39m]\n\u001b[1;32m      2\u001b[0m X \u001b[39m=\u001b[39m raw_data\u001b[39m.\u001b[39mdrop(\u001b[39m'\u001b[39m\u001b[39mClass\u001b[39m\u001b[39m'\u001b[39m,axis\u001b[39m=\u001b[39m\u001b[39m1\u001b[39m)\n\u001b[1;32m      4\u001b[0m imputer \u001b[39m=\u001b[39m SimpleImputer(strategy\u001b[39m=\u001b[39m\u001b[39m'\u001b[39m\u001b[39mconstant\u001b[39m\u001b[39m'\u001b[39m,fill_value\u001b[39m=\u001b[39m\u001b[39m0\u001b[39m, missing_values\u001b[39m=\u001b[39mnp\u001b[39m.\u001b[39mnan)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'raw_data' is not defined"
     ]
    }
   ],
   "source": [
    "y = raw_data['Class']\n",
    "X = raw_data.drop('Class',axis=1)\n",
    "\n",
    "imputer = SimpleImputer(strategy='constant',fill_value=0, missing_values=np.nan)\n",
    "imputer = imputer.fit(X[['CADD Phred','MAF','Surface Area','Normalized SA','Confidence Score','ddG']])\n",
    "X[['CADD Phred','MAF','Surface Area','Normalized SA','Confidence Score','ddG']]=imputer.transform(X[['CADD Phred','MAF','Surface Area','Normalized SA','Confidence Score','ddG']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.model_selection import train_test_split\n",
    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2)\n",
    "from sklearn.svm import SVC\n",
    "svclassifier = SVC(kernel='linear',probability=True)\n",
    "svclassifier.fit(X_train, y_train)\n",
    "\n",
    "y_pred_prob = svclassifier.predict_proba(X_test)\n",
    "y_pred_prob\n",
    "y_pred=svclassifier.predict(X_test)\n",
    "display(pd.DataFrame({'svclassifier.predict': y_pred,\n",
    "                     'svclassifier.predict_prob_a[:,0] (Pathogenic)': y_pred_prob[:,0],\n",
    "                     'svclassifier.predict_prob_a[:,1] (Non-Pathogenic)': y_pred_prob[:,1]}).head(100))\n",
    "y_predictions_prob_a = y_pred_prob[:,0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "coltons_cols = ['Genomic Description (GRCh37)', 'HGVS Nucleotide Change', 'HGVS Protein Change', 'Phenotype', 'Consequence', 'Exon', 'Intron', 'Protein Position', 'Impact', 'Interpro Domain', 'Max MAF Source', '|ddG|']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>Surface Area</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>3.009015</td>\n",
       "      <td>23.60</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>172.951510</td>\n",
       "      <td>0.752780</td>\n",
       "      <td>32.73</td>\n",
       "      <td>-1.4</td>\n",
       "      <td>1.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>2.751772</td>\n",
       "      <td>23.00</td>\n",
       "      <td>0.006660</td>\n",
       "      <td>139.065932</td>\n",
       "      <td>0.883870</td>\n",
       "      <td>38.06</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>2.662314</td>\n",
       "      <td>22.90</td>\n",
       "      <td>0.010493</td>\n",
       "      <td>180.703642</td>\n",
       "      <td>0.670700</td>\n",
       "      <td>37.23</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>2.257539</td>\n",
       "      <td>21.80</td>\n",
       "      <td>3.187450</td>\n",
       "      <td>76.425233</td>\n",
       "      <td>0.601022</td>\n",
       "      <td>34.61</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>2.319001</td>\n",
       "      <td>22.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>99.401913</td>\n",
       "      <td>0.578281</td>\n",
       "      <td>31.28</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>3.938849</td>\n",
       "      <td>27.20</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>3.979789</td>\n",
       "      <td>27.50</td>\n",
       "      <td>0.006484</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>4.163516</td>\n",
       "      <td>29.20</td>\n",
       "      <td>0.012309</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>1.223753</td>\n",
       "      <td>14.78</td>\n",
       "      <td>0.000880</td>\n",
       "      <td>128.397680</td>\n",
       "      <td>0.816259</td>\n",
       "      <td>44.11</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>3.995061</td>\n",
       "      <td>27.60</td>\n",
       "      <td>0.001549</td>\n",
       "      <td>67.643291</td>\n",
       "      <td>0.709449</td>\n",
       "      <td>37.21</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_CADD_RAW  VEP_CADD_PHRED  Max MAF (%)  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686358:T>C                   3.009015           23.60     0.000000   \n",
       "20:21686361:C>T                   2.751772           23.00     0.006660   \n",
       "20:21686379:G>A                   2.662314           22.90     0.010493   \n",
       "20:21686382:C>T                   2.257539           21.80     3.187450   \n",
       "20:21686390:G>T                   2.319001           22.10     0.000000   \n",
       "...                                    ...             ...          ...   \n",
       "10:8115915:C>T                    3.938849           27.20     0.000000   \n",
       "10:8115916:C>G                    3.979789           27.50     0.006484   \n",
       "10:8115916:C>T                    4.163516           29.20     0.012309   \n",
       "10:8115927:C>T                    1.223753           14.78     0.000880   \n",
       "10:8115945:G>A                    3.995061           27.60     0.001549   \n",
       "\n",
       "                              Surface Area  Normalized SA  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                 172.951510       0.752780             32.73   \n",
       "20:21686361:C>T                 139.065932       0.883870             38.06   \n",
       "20:21686379:G>A                 180.703642       0.670700             37.23   \n",
       "20:21686382:C>T                  76.425233       0.601022             34.61   \n",
       "20:21686390:G>T                  99.401913       0.578281             31.28   \n",
       "...                                    ...            ...               ...   \n",
       "10:8115915:C>T                  136.946574       0.870607             47.29   \n",
       "10:8115916:C>G                  136.946574       0.870607             47.29   \n",
       "10:8115916:C>T                  136.946574       0.870607             47.29   \n",
       "10:8115927:C>T                  128.397680       0.816259             44.11   \n",
       "10:8115945:G>A                   67.643291       0.709449             37.21   \n",
       "\n",
       "                              ddG  |ddG|  \n",
       "Genomic Description (GRCh37)              \n",
       "20:21686358:T>C              -1.4    1.4  \n",
       "20:21686361:C>T              -0.0    0.0  \n",
       "20:21686379:G>A              -0.3    0.3  \n",
       "20:21686382:C>T              -0.3    0.3  \n",
       "20:21686390:G>T              -0.6    0.6  \n",
       "...                           ...    ...  \n",
       "10:8115915:C>T               -0.3    0.3  \n",
       "10:8115916:C>G               -0.3    0.3  \n",
       "10:8115916:C>T               -0.2    0.2  \n",
       "10:8115927:C>T               -0.1    0.1  \n",
       "10:8115945:G>A                0.2    0.2  \n",
       "\n",
       "[128712 rows x 8 columns]"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[:, ['VEP_CADD_RAW', 'VEP_CADD_PHRED', 'Max MAF (%)','Surface Area','Normalized SA','Confidence Score', 'ddG', '|ddG|']]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
