{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from importlib import reload\n",
    "import src.utils as utils\n",
    "import glob"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "chroms_names = ['MT', 'X'] + [str(x) for x in range(1,23)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Preprocessing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Constructing df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_5_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_5_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_16_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_16_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_MT_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_MT_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_19_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_19_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_11_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_11_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_2_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_2_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_10_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_10_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_3_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_3_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_18_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_18_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_4_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_4_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_17_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_17_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_15_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_15_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_6_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_6_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_1_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_1_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_12_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_12_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_9_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_9_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_8_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_8_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_13_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_13_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_X_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_X_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_14_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_14_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_7_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_7_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_20_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_20_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_21_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_21_records.tar\n",
      "Decompression complete. Decompressed content saved to 'split_vcf_chromosomes_csvs_compressed/chromosome_22_records.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file split_vcf_chromosomes_csvs_compressed/chromosome_22_records.tar\n"
     ]
    }
   ],
   "source": [
    "chromosome_dir = 'split_vcf_chromosomes_csvs'\n",
    "if not os.path.isdir(chromosome_dir):\n",
    "    os.makedirs(chromosome_dir)\n",
    "\n",
    "compressed_dir = 'split_vcf_chromosomes_csvs_compressed'\n",
    "for x in os.scandir(compressed_dir):\n",
    "    utils.extract_tarball(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_18_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_18_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_19_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_19_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_X_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_X_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_22_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_22_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_13_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_13_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_14_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_14_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_7_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_7_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_6_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_6_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_1_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_1_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_15_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_15_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_12_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_12_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_21_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_21_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_3_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_3_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_4_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_4_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_10_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_10_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_17_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_17_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_20_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_20_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_MT_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_MT_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_16_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_16_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_11_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_11_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_5_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_5_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_2_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_2_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_8_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_8_w_ddg.tar\n",
      "Decompression complete. Decompressed content saved to 'compressed_chromosome_w_ddg/chromosome_9_w_ddg.tar'\n",
      "Unarchiving complete\n",
      "Removed archive file compressed_chromosome_w_ddg/chromosome_9_w_ddg.tar\n"
     ]
    }
   ],
   "source": [
    "compressed_dir = 'compressed_chromosome_w_ddg'\n",
    "if not os.path.isdir(compressed_dir):\n",
    "    os.mkdir(compressed_dir)\n",
    "\n",
    "for x in os.scandir(compressed_dir):\n",
    "    utils.extract_tarball(x)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/wg/8fxgb56x5d97d8mdr976klq40000gn/T/ipykernel_18153/244386787.py:8: FutureWarning: The behavior of DataFrame concatenation with empty or all-NA entries is deprecated. In a future version, this will no longer exclude empty or all-NA columns when determining the result dtypes. To retain the old behavior, exclude the relevant entries before the concat operation.\n",
      "  df = pd.concat(dfs).map(utils.safe_decode)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TYPE</th>\n",
       "      <th>GENE</th>\n",
       "      <th>gnomad_MAX_MAF</th>\n",
       "      <th>gnomad_MAX_MAF_SOURCE</th>\n",
       "      <th>gnomad_AC_afr_pass</th>\n",
       "      <th>gnomad_AC_amr_pass</th>\n",
       "      <th>gnomad_AC_eas_pass</th>\n",
       "      <th>gnomad_AC_nfe_pass</th>\n",
       "      <th>gnomad_AN_afr_pass</th>\n",
       "      <th>gnomad_AN_amr_pass</th>\n",
       "      <th>...</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>Surface Area</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "      <th>ca...</th>\n",
       "      <th>in...</th>\n",
       "      <th>calc...</th>\n",
       "      <th>second part</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>172.951510</td>\n",
       "      <td>0.752780</td>\n",
       "      <td>32.73</td>\n",
       "      <td>-1.4</td>\n",
       "      <td>1.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8602.0</td>\n",
       "      <td>810.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>139.065932</td>\n",
       "      <td>0.883870</td>\n",
       "      <td>38.06</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>0.000105</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1492.0</td>\n",
       "      <td>274.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>180.703642</td>\n",
       "      <td>0.670700</td>\n",
       "      <td>37.23</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>0.031875</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>319.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>10008.0</td>\n",
       "      <td>1046.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>76.425233</td>\n",
       "      <td>0.601022</td>\n",
       "      <td>34.61</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>PAX1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>99.401913</td>\n",
       "      <td>0.578281</td>\n",
       "      <td>31.28</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>0.000065</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8702.0</td>\n",
       "      <td>848.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>0.000123</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>16248.0</td>\n",
       "      <td>34588.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>16256.0</td>\n",
       "      <td>34588.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>128.397680</td>\n",
       "      <td>0.816259</td>\n",
       "      <td>44.11</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>SNV</td>\n",
       "      <td>GATA3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>24966.0</td>\n",
       "      <td>35438.0</td>\n",
       "      <td>...</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>67.643291</td>\n",
       "      <td>0.709449</td>\n",
       "      <td>37.21</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128786 rows × 236 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             TYPE   GENE  gnomad_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                               \n",
       "20:21686358:T>C               SNV   PAX1             NaN   \n",
       "20:21686361:C>T               SNV   PAX1        0.000067   \n",
       "20:21686379:G>A               SNV   PAX1        0.000105   \n",
       "20:21686382:C>T               SNV   PAX1        0.031875   \n",
       "20:21686390:G>T               SNV   PAX1             NaN   \n",
       "...                           ...    ...             ...   \n",
       "10:8115915:C>T                SNV  GATA3             NaN   \n",
       "10:8115916:C>G                SNV  GATA3        0.000065   \n",
       "10:8115916:C>T                SNV  GATA3        0.000123   \n",
       "10:8115927:C>T                SNV  GATA3        0.000009   \n",
       "10:8115945:G>A                SNV  GATA3        0.000015   \n",
       "\n",
       "                             gnomad_MAX_MAF_SOURCE  gnomad_AC_afr_pass  \\\n",
       "Genomic Description (GRCh37)                                             \n",
       "20:21686358:T>C                                NaN                 NaN   \n",
       "20:21686361:C>T                 gnomad_AF_nfe_pass                 0.0   \n",
       "20:21686379:G>A                 gnomad_AF_nfe_pass                 0.0   \n",
       "20:21686382:C>T                 gnomad_AF_afr_pass               319.0   \n",
       "20:21686390:G>T                                NaN                 NaN   \n",
       "...                                            ...                 ...   \n",
       "10:8115915:C>T                                 NaN                 NaN   \n",
       "10:8115916:C>G                  gnomad_AF_nfe_pass                 0.0   \n",
       "10:8115916:C>T                  gnomad_AF_afr_pass                 2.0   \n",
       "10:8115927:C>T                  gnomad_AF_nfe_pass                 0.0   \n",
       "10:8115945:G>A                  gnomad_AF_nfe_pass                 0.0   \n",
       "\n",
       "                              gnomad_AC_amr_pass  gnomad_AC_eas_pass  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "20:21686358:T>C                              NaN                 NaN   \n",
       "20:21686361:C>T                              0.0                 0.0   \n",
       "20:21686379:G>A                              0.0                 0.0   \n",
       "20:21686382:C>T                              1.0                 0.0   \n",
       "20:21686390:G>T                              NaN                 NaN   \n",
       "...                                          ...                 ...   \n",
       "10:8115915:C>T                               NaN                 NaN   \n",
       "10:8115916:C>G                               0.0                 0.0   \n",
       "10:8115916:C>T                               1.0                 0.0   \n",
       "10:8115927:C>T                               0.0                 0.0   \n",
       "10:8115945:G>A                               0.0                 0.0   \n",
       "\n",
       "                              gnomad_AC_nfe_pass  gnomad_AN_afr_pass  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "20:21686358:T>C                              NaN                 NaN   \n",
       "20:21686361:C>T                              1.0              8602.0   \n",
       "20:21686379:G>A                              1.0              1492.0   \n",
       "20:21686382:C>T                              4.0             10008.0   \n",
       "20:21686390:G>T                              NaN                 NaN   \n",
       "...                                          ...                 ...   \n",
       "10:8115915:C>T                               NaN                 NaN   \n",
       "10:8115916:C>G                               1.0              8702.0   \n",
       "10:8115916:C>T                               2.0             16248.0   \n",
       "10:8115927:C>T                               1.0             16256.0   \n",
       "10:8115945:G>A                               2.0             24966.0   \n",
       "\n",
       "                              gnomad_AN_amr_pass  ...      Max MAF Source  \\\n",
       "Genomic Description (GRCh37)                      ...                       \n",
       "20:21686358:T>C                              NaN  ...                 NaN   \n",
       "20:21686361:C>T                            810.0  ...  gnomad_AF_nfe_pass   \n",
       "20:21686379:G>A                            274.0  ...  gnomad_AF_nfe_pass   \n",
       "20:21686382:C>T                           1046.0  ...  gnomad_AF_afr_pass   \n",
       "20:21686390:G>T                              NaN  ...                 NaN   \n",
       "...                                          ...  ...                 ...   \n",
       "10:8115915:C>T                               NaN  ...                 NaN   \n",
       "10:8115916:C>G                             848.0  ...  gnomad_AF_nfe_pass   \n",
       "10:8115916:C>T                           34588.0  ...  gnomad_AF_afr_pass   \n",
       "10:8115927:C>T                           34588.0  ...  gnomad_AF_nfe_pass   \n",
       "10:8115945:G>A                           35438.0  ...  gnomad_AF_nfe_pass   \n",
       "\n",
       "                              Surface Area  Normalized SA  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                 172.951510       0.752780             32.73   \n",
       "20:21686361:C>T                 139.065932       0.883870             38.06   \n",
       "20:21686379:G>A                 180.703642       0.670700             37.23   \n",
       "20:21686382:C>T                  76.425233       0.601022             34.61   \n",
       "20:21686390:G>T                  99.401913       0.578281             31.28   \n",
       "...                                    ...            ...               ...   \n",
       "10:8115915:C>T                  136.946574       0.870607             47.29   \n",
       "10:8115916:C>G                  136.946574       0.870607             47.29   \n",
       "10:8115916:C>T                  136.946574       0.870607             47.29   \n",
       "10:8115927:C>T                  128.397680       0.816259             44.11   \n",
       "10:8115945:G>A                   67.643291       0.709449             37.21   \n",
       "\n",
       "                              ddG  |ddG|  ca...  in...  calc...  second part  \n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C              -1.4    1.4    NaN    NaN      NaN          NaN  \n",
       "20:21686361:C>T              -0.0    0.0    NaN    NaN      NaN          NaN  \n",
       "20:21686379:G>A              -0.3    0.3    NaN    NaN      NaN          NaN  \n",
       "20:21686382:C>T              -0.3    0.3    NaN    NaN      NaN          NaN  \n",
       "20:21686390:G>T              -0.6    0.6    NaN    NaN      NaN          NaN  \n",
       "...                           ...    ...    ...    ...      ...          ...  \n",
       "10:8115915:C>T               -0.3    0.3    NaN    NaN      NaN          NaN  \n",
       "10:8115916:C>G               -0.3    0.3    NaN    NaN      NaN          NaN  \n",
       "10:8115916:C>T               -0.2    0.2    NaN    NaN      NaN          NaN  \n",
       "10:8115927:C>T               -0.1    0.1    NaN    NaN      NaN          NaN  \n",
       "10:8115945:G>A                0.2    0.2    NaN    NaN      NaN          NaN  \n",
       "\n",
       "[128786 rows x 236 columns]"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs = []\n",
    "id_cols = ['CHROM', 'POS', 'REF', 'ALT']\n",
    "dup_cols = [f\"{base_col}_{alt}\" for base_col in id_cols for alt in ['x', 'y']]\n",
    "for f in os.scandir('chromosomes_w_ddg'):\n",
    "    df = pd.read_csv(f.path, index_col=0, encoding='utf-8', low_memory=False)\n",
    "    df.drop(columns=dup_cols, inplace=True)\n",
    "    dfs.append(df)\n",
    "df = pd.concat(dfs).map(utils.safe_decode)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Columns to Drop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "cols_to_drop = []\n",
    "\n",
    "# Colummns containing various identifiers from other sources\n",
    "id_cols = df.columns[df.columns.str.contains('ID') & ~df.columns.str.contains('CONFIDENCE')].tolist()\n",
    "\n",
    "# gnomad columns - mostly just allele frequency filters\n",
    "gnomad_filter_cols = df.columns[df.columns.str.contains('gnomad') & df.columns.str.contains('pass')].tolist()\n",
    "gnomad_filter_cols.append('gnomad_filters')\n",
    "\n",
    "# Columns containing minor allele frequency (MAF)\n",
    "# max_maf_cols = df.columns[df.columns.str.contains('MAF')].tolist()\n",
    "\n",
    "# Errant columns, unknown origin\n",
    "errant_cols = [' second part', ' in...', ' ca...', ' calc...']\n",
    "\n",
    "# DVD/Kafeen Columns\n",
    "dvd_kafeen_cols = ['CLASSIFY_TRAIL', 'CLASSIFY_TRAIL_KEYS', 'FINAL_COMMENTS', 'FINAL_PATHOGENICITY_REASON', 'CURATED_DECISION_DATE', 'PREF_TRANSCRIPT']\n",
    "\n",
    "# Columns containing variant location information\n",
    "genomic_protein_loc_cols = ['VEP_HGVS_P', 'VEP_HGVS_C', 'CLINVAR_HGVS_P', 'CLINVAR_HGVS_C', 'Protein Position']\n",
    "gene_description_cols = ['HGMD_OMIM_REF'] + ['VEP_FEATURE'] # Unsure what VEP_FEATURE is, but it looks like an identifier\n",
    "\n",
    "# Comment columns\n",
    "comment_cols = ['FINAL_COMMENTS', 'CURATED_COMMENTS']\n",
    "\n",
    "# Misc columns\n",
    "misc = ['VEP_OTHER_FEATURES']\n",
    "\n",
    "cols_to_drop += id_cols + gnomad_filter_cols + errant_cols + dvd_kafeen_cols + genomic_protein_loc_cols + gene_description_cols + misc + comment_cols"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "keep_cols = ['NUM_PATH_PREDS',\n",
    " 'TOTAL_NUM_PREDS',\n",
    " 'FINAL_PRED',\n",
    " 'GENE',\n",
    " 'OVERALL_MAX_MAF',\n",
    " 'OVERALL_MAX_MAF_SOURCE',\n",
    " 'FINAL_PATHOGENICITY',\n",
    " 'CURATED_PATHOGENICITY']\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Alternate Approach"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "feature_sorting_df = pd.read_csv('feature_sorting_updated.csv', header=None, index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['NUM_PATH_PREDS',\n",
       " 'TOTAL_NUM_PREDS',\n",
       " 'FINAL_PRED',\n",
       " 'GENE',\n",
       " 'OVERALL_MAX_MAF',\n",
       " 'OVERALL_MAX_MAF_SOURCE',\n",
       " 'FINAL_PATHOGENICITY',\n",
       " 'CURATED_PATHOGENICITY']"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature_sorting_df.loc['Definitely Keep (used previously)'].dropna().tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "kept_features_df = feature_sorting_df.loc[~feature_sorting_df.index.str.contains('Not Keep'), :]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "46"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kept_features_list = sorted(kept_features_df.unstack().dropna().unique().tolist())\n",
    "len(kept_features_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "pathogenicity_label_cols = ['CLINVAR_CLNSIG', 'CLINVAR_PATHOGENIC', 'CURATED_PATHOGENICITY']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "tags": []
   },
   "source": [
    "## Selecting Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CADD Phred',\n",
       " 'CLINVAR_CLNSIG',\n",
       " 'CLINVAR_PATHOGENIC',\n",
       " 'CURATED_PATHOGENICITY',\n",
       " 'Confidence Score',\n",
       " 'Consequence',\n",
       " 'DBNSFP_GERP_RS',\n",
       " 'DBNSFP_GERP_RS_PRED',\n",
       " 'DBNSFP_INTERPRO_DOMAIN',\n",
       " 'DBNSFP_LRT_PRED',\n",
       " 'DBNSFP_LRT_SCORE',\n",
       " 'DBNSFP_MUTATIONTASTER_PRED',\n",
       " 'DBNSFP_MUTATIONTASTER_SCORE',\n",
       " 'DBNSFP_PHYLOP30WAY_MAMMALIAN',\n",
       " 'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED',\n",
       " 'DBNSFP_POLYPHEN2_HDIV_PRED',\n",
       " 'DBNSFP_POLYPHEN2_HDIV_SCORE',\n",
       " 'DBNSFP_SIFT_PRED',\n",
       " 'DBNSFP_SIFT_SCORE',\n",
       " 'FINAL_PATHOGENICITY',\n",
       " 'FINAL_PRED',\n",
       " 'GENE',\n",
       " 'Impact',\n",
       " 'Interpro Domain',\n",
       " 'Max MAF (%)',\n",
       " 'Max MAF Source',\n",
       " 'NUM_PATH_PREDS',\n",
       " 'Normalized SA',\n",
       " 'OVERALL_MAX_MAF',\n",
       " 'OVERALL_MAX_MAF_SOURCE',\n",
       " 'Surface Area',\n",
       " 'TOTAL_NUM_PREDS',\n",
       " 'TYPE',\n",
       " 'UNIPROT_DOMAIN',\n",
       " 'VEP_CADD_PHRED',\n",
       " 'VEP_CADD_RAW',\n",
       " 'VEP_CONSEQUENCE',\n",
       " 'VEP_FEATURE',\n",
       " 'VEP_IMPACT',\n",
       " 'VEP_POLYPHEN_PRED',\n",
       " 'VEP_POLYPHEN_SCORE',\n",
       " 'VEP_PROTEIN_POS',\n",
       " 'VEP_SIFT_SCORE',\n",
       " 'Variant Classification',\n",
       " 'ddG',\n",
       " '|ddG|']"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "kept_features_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>CLINVAR_CLNSIG</th>\n",
       "      <th>CLINVAR_PATHOGENIC</th>\n",
       "      <th>CURATED_PATHOGENICITY</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>VEP_CONSEQUENCE</th>\n",
       "      <th>VEP_FEATURE</th>\n",
       "      <th>VEP_IMPACT</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>Variant Classification</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>23.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32.73</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.596</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-1.4</td>\n",
       "      <td>1.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.06</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.917</td>\n",
       "      <td>4</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.90</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.23</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2.32</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.787</td>\n",
       "      <td>10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.61</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2.11</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>11</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>22.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.28</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3.51</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.005</td>\n",
       "      <td>14</td>\n",
       "      <td>0.04</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>27.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>27.50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>29.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>14.78</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.11</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3.68</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.097</td>\n",
       "      <td>426</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>27.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.21</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>432</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128786 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD Phred CLINVAR_CLNSIG  CLINVAR_PATHOGENIC  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                    23.60            NaN                 NaN   \n",
       "20:21686361:C>T                    23.00            NaN                 NaN   \n",
       "20:21686379:G>A                    22.90            NaN                 NaN   \n",
       "20:21686382:C>T                    21.80            NaN                 NaN   \n",
       "20:21686390:G>T                    22.10            NaN                 NaN   \n",
       "...                                  ...            ...                 ...   \n",
       "10:8115915:C>T                     27.20            NaN                 NaN   \n",
       "10:8115916:C>G                     27.50            NaN                 NaN   \n",
       "10:8115916:C>T                     29.20            NaN                 NaN   \n",
       "10:8115927:C>T                     14.78            NaN                 NaN   \n",
       "10:8115945:G>A                     27.60            NaN                 NaN   \n",
       "\n",
       "                             CURATED_PATHOGENICITY  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "20:21686358:T>C                                NaN             32.73   \n",
       "20:21686361:C>T                                NaN             38.06   \n",
       "20:21686379:G>A                                NaN             37.23   \n",
       "20:21686382:C>T                                NaN             34.61   \n",
       "20:21686390:G>T                                NaN             31.28   \n",
       "...                                            ...               ...   \n",
       "10:8115915:C>T                                 NaN             47.29   \n",
       "10:8115916:C>G                                 NaN             47.29   \n",
       "10:8115916:C>T                                 NaN             47.29   \n",
       "10:8115927:C>T                                 NaN             44.11   \n",
       "10:8115945:G>A                                 NaN             37.21   \n",
       "\n",
       "                                   Consequence  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686358:T>C               missense_variant            4.28   \n",
       "20:21686361:C>T               missense_variant            4.28   \n",
       "20:21686379:G>A               missense_variant            2.32   \n",
       "20:21686382:C>T               missense_variant            2.11   \n",
       "20:21686390:G>T               missense_variant            3.51   \n",
       "...                                        ...             ...   \n",
       "10:8115915:C>T                missense_variant            5.59   \n",
       "10:8115916:C>G                missense_variant            5.59   \n",
       "10:8115916:C>T                missense_variant            5.59   \n",
       "10:8115927:C>T                missense_variant            3.68   \n",
       "10:8115945:G>A                missense_variant            5.59   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_INTERPRO_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686358:T>C                                C                    .&.   \n",
       "20:21686361:C>T                                C                    .&.   \n",
       "20:21686379:G>A                                C                    .&.   \n",
       "20:21686382:C>T                                C                    .&.   \n",
       "20:21686390:G>T                                C                    .&.   \n",
       "...                                          ...                    ...   \n",
       "10:8115915:C>T                                 C                  .&.&.   \n",
       "10:8115916:C>G                                 C                  .&.&.   \n",
       "10:8115916:C>T                                 C                  .&.&.   \n",
       "10:8115927:C>T                                 C                  .&.&.   \n",
       "10:8115945:G>A                                 C                  .&.&.   \n",
       "\n",
       "                             DBNSFP_LRT_PRED  ...   VEP_CONSEQUENCE  \\\n",
       "Genomic Description (GRCh37)                  ...                     \n",
       "20:21686358:T>C                          NaN  ...  missense_variant   \n",
       "20:21686361:C>T                          NaN  ...  missense_variant   \n",
       "20:21686379:G>A                          NaN  ...  missense_variant   \n",
       "20:21686382:C>T                          NaN  ...  missense_variant   \n",
       "20:21686390:G>T                          NaN  ...  missense_variant   \n",
       "...                                      ...  ...               ...   \n",
       "10:8115915:C>T                             U  ...  missense_variant   \n",
       "10:8115916:C>G                             U  ...  missense_variant   \n",
       "10:8115916:C>T                             U  ...  missense_variant   \n",
       "10:8115927:C>T                             D  ...  missense_variant   \n",
       "10:8115945:G>A                             D  ...  missense_variant   \n",
       "\n",
       "                                 VEP_FEATURE VEP_IMPACT  VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                                 \n",
       "20:21686358:T>C                  NM_006192.5   MODERATE                  P   \n",
       "20:21686361:C>T                  NM_006192.5   MODERATE                  P   \n",
       "20:21686379:G>A                  NM_006192.5   MODERATE                  P   \n",
       "20:21686382:C>T                  NM_006192.5   MODERATE                  P   \n",
       "20:21686390:G>T                  NM_006192.5   MODERATE                  B   \n",
       "...                                      ...        ...                ...   \n",
       "10:8115915:C>T                NM_001002295.2   MODERATE                  D   \n",
       "10:8115916:C>G                NM_001002295.2   MODERATE                  D   \n",
       "10:8115916:C>T                NM_001002295.2   MODERATE                  D   \n",
       "10:8115927:C>T                NM_001002295.2   MODERATE                  B   \n",
       "10:8115945:G>A                NM_001002295.2   MODERATE                  D   \n",
       "\n",
       "                             VEP_POLYPHEN_SCORE VEP_PROTEIN_POS  \\\n",
       "Genomic Description (GRCh37)                                      \n",
       "20:21686358:T>C                           0.596               3   \n",
       "20:21686361:C>T                           0.917               4   \n",
       "20:21686379:G>A                           0.787              10   \n",
       "20:21686382:C>T                           0.804              11   \n",
       "20:21686390:G>T                           0.005              14   \n",
       "...                                         ...             ...   \n",
       "10:8115915:C>T                            1.000             422   \n",
       "10:8115916:C>G                            1.000             422   \n",
       "10:8115916:C>T                            1.000             422   \n",
       "10:8115927:C>T                            0.097             426   \n",
       "10:8115945:G>A                            0.997             432   \n",
       "\n",
       "                             VEP_SIFT_SCORE Variant Classification  ddG |ddG|  \n",
       "Genomic Description (GRCh37)                                                   \n",
       "20:21686358:T>C                        0.00   Unknown significance -1.4   1.4  \n",
       "20:21686361:C>T                        0.00   Unknown significance -0.0   0.0  \n",
       "20:21686379:G>A                        0.00   Unknown significance -0.3   0.3  \n",
       "20:21686382:C>T                        0.01                 Benign -0.3   0.3  \n",
       "20:21686390:G>T                        0.04   Unknown significance -0.6   0.6  \n",
       "...                                     ...                    ...  ...   ...  \n",
       "10:8115915:C>T                         0.01      Likely pathogenic -0.3   0.3  \n",
       "10:8115916:C>G                         0.00   Unknown significance -0.3   0.3  \n",
       "10:8115916:C>T                         0.00   Unknown significance -0.2   0.2  \n",
       "10:8115927:C>T                         1.00   Unknown significance -0.1   0.1  \n",
       "10:8115945:G>A                         0.00   Unknown significance  0.2   0.2  \n",
       "\n",
       "[128786 rows x 46 columns]"
      ]
     },
     "execution_count": 123,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df = df[kept_features_list]\n",
    "working_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Exploring Prediction Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>C</td>\n",
       "      <td>4.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>4.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>C</td>\n",
       "      <td>2.32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>2.11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>C</td>\n",
       "      <td>3.51</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_GERP_RS_PRED  DBNSFP_GERP_RS\n",
       "Genomic Description (GRCh37)                                    \n",
       "20:21686358:T>C                                C            4.28\n",
       "20:21686361:C>T                                C            4.28\n",
       "20:21686379:G>A                                C            2.32\n",
       "20:21686382:C>T                                C            2.11\n",
       "20:21686390:G>T                                C            3.51"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# In GERP, RS > 0 => C == conserved\n",
    "# In GERP, RS < 0 => N == not conserved (?)\n",
    "gerp_pos_df = df.loc[df.DBNSFP_GERP_RS > 0, ['DBNSFP_GERP_RS_PRED', 'DBNSFP_GERP_RS']]\n",
    "gerp_pos_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686402:G&gt;C</th>\n",
       "      <td>N</td>\n",
       "      <td>-0.290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686457:G&gt;A</th>\n",
       "      <td>N</td>\n",
       "      <td>-0.296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686459:G&gt;A</th>\n",
       "      <td>N</td>\n",
       "      <td>-8.940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686468:G&gt;A</th>\n",
       "      <td>N</td>\n",
       "      <td>-1.940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686502:G&gt;T</th>\n",
       "      <td>N</td>\n",
       "      <td>-0.914</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_GERP_RS_PRED  DBNSFP_GERP_RS\n",
       "Genomic Description (GRCh37)                                    \n",
       "20:21686402:G>C                                N          -0.290\n",
       "20:21686457:G>A                                N          -0.296\n",
       "20:21686459:G>A                                N          -8.940\n",
       "20:21686468:G>A                                N          -1.940\n",
       "20:21686502:G>T                                N          -0.914"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gerp_neg_df = df.loc[df.DBNSFP_GERP_RS < 0, ['DBNSFP_GERP_RS_PRED', 'DBNSFP_GERP_RS']]\n",
    "gerp_neg_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>VEP_SIFT_PRED</th>\n",
       "      <th>DBNSFP_SIFT_PRED</th>\n",
       "      <th>DBNSFP_SIFT_SCORE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.001&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.0&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.0&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>0.01</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.002&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>0.04</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.006&amp;.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_SIFT_SCORE VEP_SIFT_PRED DBNSFP_SIFT_PRED  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                         0.00          D-LC              D&.   \n",
       "20:21686361:C>T                         0.00          D-LC              D&.   \n",
       "20:21686379:G>A                         0.00          D-LC              D&.   \n",
       "20:21686382:C>T                         0.01          D-LC              D&.   \n",
       "20:21686390:G>T                         0.04          D-LC              D&.   \n",
       "\n",
       "                             DBNSFP_SIFT_SCORE  \n",
       "Genomic Description (GRCh37)                    \n",
       "20:21686358:T>C                        0.001&.  \n",
       "20:21686361:C>T                          0.0&.  \n",
       "20:21686379:G>A                          0.0&.  \n",
       "20:21686382:C>T                        0.002&.  \n",
       "20:21686390:G>T                        0.006&.  "
      ]
     },
     "execution_count": 88,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sift_df = df.loc[:, df.columns.str.contains('SIFT')]\n",
    "sift_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>VEP_SIFT_PRED</th>\n",
       "      <th>DBNSFP_SIFT_PRED</th>\n",
       "      <th>DBNSFP_SIFT_SCORE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.001&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.0&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.0&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>0.01</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.002&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>0.04</td>\n",
       "      <td>D-LC</td>\n",
       "      <td>D&amp;.</td>\n",
       "      <td>0.006&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>0.01</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;.&amp;D</td>\n",
       "      <td>0.014&amp;.&amp;0.014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;.&amp;D</td>\n",
       "      <td>0.008&amp;.&amp;0.008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;.&amp;D</td>\n",
       "      <td>0.002&amp;.&amp;0.003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>1.00</td>\n",
       "      <td>T</td>\n",
       "      <td>T&amp;.&amp;T</td>\n",
       "      <td>1.0&amp;.&amp;1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>0.00</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;.&amp;D</td>\n",
       "      <td>0.007&amp;.&amp;0.007</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128737 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_SIFT_SCORE VEP_SIFT_PRED DBNSFP_SIFT_PRED  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                         0.00          D-LC              D&.   \n",
       "20:21686361:C>T                         0.00          D-LC              D&.   \n",
       "20:21686379:G>A                         0.00          D-LC              D&.   \n",
       "20:21686382:C>T                         0.01          D-LC              D&.   \n",
       "20:21686390:G>T                         0.04          D-LC              D&.   \n",
       "...                                      ...           ...              ...   \n",
       "10:8115915:C>T                          0.01             D            D&.&D   \n",
       "10:8115916:C>G                          0.00             D            D&.&D   \n",
       "10:8115916:C>T                          0.00             D            D&.&D   \n",
       "10:8115927:C>T                          1.00             T            T&.&T   \n",
       "10:8115945:G>A                          0.00             D            D&.&D   \n",
       "\n",
       "                             DBNSFP_SIFT_SCORE  \n",
       "Genomic Description (GRCh37)                    \n",
       "20:21686358:T>C                        0.001&.  \n",
       "20:21686361:C>T                          0.0&.  \n",
       "20:21686379:G>A                          0.0&.  \n",
       "20:21686382:C>T                        0.002&.  \n",
       "20:21686390:G>T                        0.006&.  \n",
       "...                                        ...  \n",
       "10:8115915:C>T                   0.014&.&0.014  \n",
       "10:8115916:C>G                   0.008&.&0.008  \n",
       "10:8115916:C>T                   0.002&.&0.003  \n",
       "10:8115927:C>T                       1.0&.&1.0  \n",
       "10:8115945:G>A                   0.007&.&0.007  \n",
       "\n",
       "[128737 rows x 4 columns]"
      ]
     },
     "execution_count": 89,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.loc[~df.VEP_SIFT_PRED.isna(), df.columns.str.contains('SIFT')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>DBNSFP_POLYPHEN2_HDIV_PRED</th>\n",
       "      <th>DBNSFP_POLYPHEN2_HDIV_SCORE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>0.596</td>\n",
       "      <td>P</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>0.596&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>0.917</td>\n",
       "      <td>P</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>0.917&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>0.787</td>\n",
       "      <td>P</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>0.787&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>0.804</td>\n",
       "      <td>P</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>0.804&amp;.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>0.005</td>\n",
       "      <td>B</td>\n",
       "      <td>B&amp;.</td>\n",
       "      <td>0.005&amp;.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_POLYPHEN_SCORE VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                         \n",
       "20:21686358:T>C                            0.596                 P   \n",
       "20:21686361:C>T                            0.917                 P   \n",
       "20:21686379:G>A                            0.787                 P   \n",
       "20:21686382:C>T                            0.804                 P   \n",
       "20:21686390:G>T                            0.005                 B   \n",
       "\n",
       "                             DBNSFP_POLYPHEN2_HDIV_PRED  \\\n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686358:T>C                                     P&.   \n",
       "20:21686361:C>T                                     P&.   \n",
       "20:21686379:G>A                                     P&.   \n",
       "20:21686382:C>T                                     P&.   \n",
       "20:21686390:G>T                                     B&.   \n",
       "\n",
       "                             DBNSFP_POLYPHEN2_HDIV_SCORE  \n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686358:T>C                                  0.596&.  \n",
       "20:21686361:C>T                                  0.917&.  \n",
       "20:21686379:G>A                                  0.787&.  \n",
       "20:21686382:C>T                                  0.804&.  \n",
       "20:21686390:G>T                                  0.005&.  "
      ]
     },
     "execution_count": 90,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##INFO=<ID=DBNSFP_POLYPHEN2_HDIV_PRED,Number=1,Type=String,Description=\"\\\"(from dbNSFP) Polyphen2 prediction based on HumDiv, D (probably damaging, HDIV score in [0.957,1] or rankscore in [0.55859,0.91137]), P (possibly damaging, HDIV score in [0.454,0.956] or rankscore in [0.37043,0.55681]) and B (benign, HDIV score in [0,0.452] or rankscore in [0.03061,0.36974]). Score cutoff for binary classification is 0.5 for HDIV score or 0.38028 for rankscore, i.e. the prediction is neutral if the HDIV score is smaller than 0.5 (rankscore is smaller than 0.38028), and deleterious if the HDIV score is larger than 0.5 (rankscore is larger than 0.38028). Multiple entries are separated by ;, corresponding to Uniprot_acc.\\\"\">\n",
    "##INFO=<ID=DBNSFP_POLYPHEN2_HDIV_SCORE,Number=1,Type=String,Description=\"\\\"(from dbNSFP) Polyphen2 score based on HumDiv, i.e. hdiv_prob. The score ranges from 0 to 1. Multiple entries separated by ;, corresponding to Uniprot_acc.\\\"\">\n",
    "polyphen_df = df.loc[:, df.columns.str.contains('POLYPHEN')]\n",
    "polyphen_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "VEP_POLYPHEN_PRED\n",
       "D    55263\n",
       "B    51063\n",
       "P    22231\n",
       "U      210\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "polyphen_df.VEP_POLYPHEN_PRED.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>N</td>\n",
       "      <td>0.506194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>N</td>\n",
       "      <td>0.758794</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>N</td>\n",
       "      <td>0.999272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>N</td>\n",
       "      <td>0.998176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>N</td>\n",
       "      <td>0.999492</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686358:T>C                                       N   \n",
       "20:21686361:C>T                                       N   \n",
       "20:21686379:G>A                                       N   \n",
       "20:21686382:C>T                                       N   \n",
       "20:21686390:G>T                                       N   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_SCORE  \n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686358:T>C                                 0.506194  \n",
       "20:21686361:C>T                                 0.758794  \n",
       "20:21686379:G>A                                 0.999272  \n",
       "20:21686382:C>T                                 0.998176  \n",
       "20:21686390:G>T                                 0.999492  "
      ]
     },
     "execution_count": 92,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##INFO=<ID=DBNSFP_MUTATIONTASTER_PRED,Number=1,Type=String,Description=\"\\\"(from dbNSFP) MutationTaster prediction, A (disease_causing_automatic), D (disease_causing), N (polymorphism) or P (polymorphism_automatic). The score cutoff between D and N is 0.5 for MTnew and 0.31733 for the rankscore.\\\"\">\n",
    "##INFO=<ID=DBNSFP_MUTATIONTASTER_SCORE,Number=1,Type=String,Description=\"\\\"(from dbNSFP) MutationTaster p-value (MTori), ranges from 0 to 1. Multiple scores are separated by ;. Information on corresponding transcript(s) can be found by querying http://www.mutationtaster.org/ChrPos.html\\\"\">\n",
    "mutationtaster_df = df.loc[:, df.columns.str.contains('MUTATIONTASTER')]\n",
    "mutationtaster_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "D&D 21946\n",
      "D 17739\n",
      "D&D&D 15324\n",
      "N&N 12588\n",
      "N 10410\n",
      "D&D&D&D 9660\n",
      "N&N&N 6795\n",
      "D&D&D&D&D 5238\n",
      "D&D&D&D&D&D 3929\n",
      "N&N&N&N 2700\n",
      "N&N&N&N&N 2023\n",
      "D&D&D&D&D&D&D&D 1904\n",
      "D&D&D&D&D&D&D 1637\n",
      "D&N 1555\n",
      "D&D&D&D&D&D&D&D&D 1550\n",
      "N&N&N&N&N&N 1368\n",
      "D&D&D&D&D&D&D&D&D&D&D 1123\n",
      "N&N&N&N&N&N&N&N 1033\n",
      "D&N&N 961\n",
      "N&N&N&N&N&N&N 838\n",
      "D&D&N 822\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&D&D&D&D 573\n",
      "N&N&N&N&N&N&N&N&N 541\n",
      "D&N&N&N 337\n",
      "D&D&D&N 334\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&D&D&D 329\n",
      "A&A 318\n",
      "D&D&D&D&D&D&D&D&D&D 275\n",
      "D&D&D&D&N 229\n",
      "D&N&N&N&N 164\n",
      "P&P 155\n",
      "D&D&N&N 148\n",
      "D&D&D&D&N&N&N&N 138\n",
      "D&D&D&D&D&D&D&D&N 129\n",
      "D&D&D&N&N 128\n",
      "D&D&D&D&N&N 124\n",
      "D&D&D&D&D&D&N&N&N 123\n",
      "D&N&N&N&N&N&N 119\n",
      "D&D&N&N&N 119\n",
      "D&D&D&D&D&D&N 96\n",
      "D&D&D&D&D&D&D&D&D&D&D&D 94\n",
      "P 89\n",
      "A 88\n",
      "N&N&N&N&N&N&N&N&N&N&N&N 84\n",
      "D&D&D&D&D&N 84\n",
      "D&D&D&D&D&D&D&N&N 83\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&D&D&D&N 76\n",
      "A&A&A&A 70\n",
      "D&D&D&D&N&N&N&N&N 69\n",
      "D&D&D&D&D&D&D&D&D&D&N&N 69\n",
      "P&P&P 68\n",
      "D&D&N&N&N&N 62\n",
      "D&D&D&N&N&N 58\n",
      "D&N&N&N&N&N&N&N&N 54\n",
      "D&D&D&N&N&D 47\n",
      "D&D&N&N&N&N&N 46\n",
      "A&A&A 44\n",
      "D&D&D&D&D&D&N&N 44\n",
      "A&A&A&A&A&A&A&A&A&A&A 42\n",
      "D&D&D&D&D&D&D&D&N&N&N 41\n",
      "D&D&D&N&N&N&N&N&N&N 41\n",
      "D&D&N&N&N&N&N&N&N&N 40\n",
      "D&N&N&N&N&N&N&N&N&N 36\n",
      "D&D&D&D&D&D&D&D&D&D&N 36\n",
      "D&N&N&N&N&N 36\n",
      "A&A&A&A&A&A 36\n",
      "D&D&D&D&N&N&N&N&N&N&N&N&N&N&N 34\n",
      "N&N&N&N&N&D 34\n",
      "D&D&D&D&D&D&D&N 33\n",
      "N&N&N&N&N&N&N&N&N&N 32\n",
      "D&N&N&N&D&D 32\n",
      "D&N&N&N&N&N&N&N&N&N&N&N&N&N&N&N 29\n",
      "D&N&N&N&N&N&N&N 26\n",
      "A&A&A&A&A 26\n",
      "P&P&P&P 25\n",
      "D&D&D&N&N&N&N&N&N 24\n",
      "P&P&P&P&P 23\n",
      "D&D&D&D&D&N&N&N&N 21\n",
      "D&D&N&N&N&N&N&N 21\n",
      "D&D&D&D&D&N&N&N 21\n",
      "N&N&N&N&N&N&N&N&N&N&N 20\n",
      "N&N&N&D&D 20\n",
      "D&D&N&N&N&N&N&N&N 19\n",
      "A&A&A&A&A&A&A 19\n",
      "N&N&N&N&N&N&N&N&N&N&N&N&N&N&N&N 17\n",
      "D&D&D&D&D&N&N 16\n",
      "D&D&D&D&D&D&N&N&N&N&N&N 14\n",
      "P&P&P&P&P&P&P 13\n",
      "D&D&N&N&N&N&N&N&N&N&N&N&N&N&N&N 13\n",
      "N&N&N&N&N&N&N&N&N&N&N&N&N&N&N 12\n",
      "D&D&D&N&N&N&N&N 12\n",
      "A&A&A&A&A&A&A&A 12\n",
      "D&D&D&D&N&N&N&N&N&N&N&N&N&N&N&N 12\n",
      "P&P&P&P&P&P&P&P&P 11\n",
      "D&D&D&D&N&N&N 11\n",
      "D&D&D&N&N&N&N&N&N&N&N 10\n",
      "P&P&P&P&P&P&P&P 10\n",
      "D&D&D&D&N&N&N&N&N&N&N&N 10\n",
      "D&D&D&N&N&N&N 10\n",
      "D&D&N&N&N&D 9\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&D&N&N&N 9\n",
      "D&D&N&D 8\n",
      "N&D&D 8\n",
      "D&N&N&N&N&N&N&N&N&N&N 6\n",
      "N&N&N&D 6\n",
      "D&D&D&N&N&N&N&N&N&N&N&N&N&N&N&N 6\n",
      "P&P&P&P&P&P 6\n",
      "P&P&P&P&P&P&P&P&P&P&P&P&P&P&P&P 5\n",
      "D&D&D&D&D&N&N&N&N&N&N&N 5\n",
      "A&A&A&A&A&A&A&A&A&A&A&A 5\n",
      "D&D&D&D&D&D&D&D&N&N&N&N&N&N&D&D 4\n",
      "D&D&D&D&D&N&N&N&N&N 4\n",
      "N&D&N 4\n",
      "A&N&N&N&N&N&N&N&N 4\n",
      "D&D&N&N&N&N&N&N&N&N&N&N 4\n",
      "D&D&D&D&N&N&N&N&N&N&N 4\n",
      "D&D&N&N&N&N&N&N&N&N&N 4\n",
      "D&D&D&D&D&D&D&D&D&D&D&N 3\n",
      "D&D&D&D&D&N&N&N&N&N&N&N&N&N&N 3\n",
      "A&A&A&A&A&A&A&A&A 3\n",
      "D&D&D&D&D&D&D&N&N&N&N 3\n",
      "N&N&N&N&N&N&N&N&D 3\n",
      "D&D&D&D&D&D&N&N&N&N&N 3\n",
      "A&D&D 3\n",
      "D&N&N&D 3\n",
      "D&D&D&D&D&D&D&D&D&N 2\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&N&N&N&N 2\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&D&D&N&N 2\n",
      "D&D&D&D&D&N&N&N&N&N&N&N&N&N&N&N 2\n",
      "A&N&N 2\n",
      "D&D&D&D&D&D&D&N&N&N&N&N&N&N&N 2\n",
      "D&D&D&D&D&D&D&D&D&N&N&N&N&N&N 2\n",
      "D&D&D&D&D&D&D&D&D&D&D&N&N&N&N 2\n",
      "A&A&A&A&A&A&A&A&A&A&A&A&A&A&A&A 2\n",
      "N&N&N&N&N&N&N&N&N&N&N&N&N&N&D&D 2\n",
      "A&D 2\n",
      "D&D&N&N&D&D 2\n",
      "N&N&N&N&D&D 2\n",
      "N&D 2\n",
      "D&D&D&D&D&D&D&D&N&N&N&N&N&N&N 2\n",
      "A&D&D&D&D&D&D&D&D&D&D 1\n",
      "D&D&N&N&N&N&D&D&N&N&N&N 1\n",
      "P&P&P&P&P&P&P&P&P&P&P&P 1\n",
      "D&D&D&D&D&D&D&D&D&N&N&N&N&N&N&N 1\n",
      "A&D&D&D&D 1\n",
      "A&N 1\n",
      "A&D&D&D&D&D&D 1\n",
      "D&D&D&D&D&D&D&D&N&N&N&N 1\n",
      "D&D&D&D&D&D&N&N&N&N&N&N&N&N&N&N 1\n",
      "A&D&D&D&D&D 1\n",
      "D&D&D&D&D&D&D&D&D&D&D&D&D&D&N 1\n",
      "A&A&D&D&D&D&D&D 1\n",
      "A&D&D&D&D&D&D&D&D 1\n",
      "D&D&D&D&D&N&N&N&N&N&N 1\n",
      "A&A&D&D&D&N&N&N 1\n",
      "D&D&D&D&D&D&N&N&N&N&N&N&N&N&N 1\n"
     ]
    }
   ],
   "source": [
    "for i, v in mutationtaster_df.DBNSFP_MUTATIONTASTER_PRED.value_counts().items():\n",
    "    print(i,v)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_LRT_PRED  DBNSFP_LRT_SCORE\n",
       "Genomic Description (GRCh37)                                  \n",
       "20:21686358:T>C                          NaN               NaN\n",
       "20:21686361:C>T                          NaN               NaN\n",
       "20:21686379:G>A                          NaN               NaN\n",
       "20:21686382:C>T                          NaN               NaN\n",
       "20:21686390:G>T                          NaN               NaN"
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##INFO=<ID=DBNSFP_LRT_PRED,Number=1,Type=String,Description=\"\\\"(from dbNSFP) LRT prediction, D(eleterious), N(eutral) or U(nknown), which is not solely determined by the score.\\\"\">\n",
    "##INFO=<ID=DBNSFP_LRT_SCORE,Number=1,Type=String,Description=\"\\\"(from dbNSFP) The original LRT two-sided p-value (LRTori), ranges from 0 to 1.\\\"\">\n",
    "lrt_df = df.loc[:, df.columns.str.contains('LRT')]\n",
    "lrt_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>1.111</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>0.118</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>1.147</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>0.106</td>\n",
       "      <td>N</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>1.147</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              DBNSFP_PHYLOP30WAY_MAMMALIAN  \\\n",
       "Genomic Description (GRCh37)                                 \n",
       "20:21686358:T>C                                      1.111   \n",
       "20:21686361:C>T                                      0.118   \n",
       "20:21686379:G>A                                      1.147   \n",
       "20:21686382:C>T                                      0.106   \n",
       "20:21686390:G>T                                      1.147   \n",
       "\n",
       "                             DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED  \n",
       "Genomic Description (GRCh37)                                    \n",
       "20:21686358:T>C                                              C  \n",
       "20:21686361:C>T                                              N  \n",
       "20:21686379:G>A                                              C  \n",
       "20:21686382:C>T                                              N  \n",
       "20:21686390:G>T                                              C  "
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##INFO=<ID=DBNSFP_PHYLOP30WAY_MAMMALIAN,Number=1,Type=String,Description=\"\\\"(from dbNSFP) phyloP (phylogenetic p-values) conservation score based on the multiple alignments of 30 mammalian genomes (including human). The larger the score, the more conserved the site. Scores range from -20 to 1.312 in dbNSFP.\\\"\">\n",
    "##INFO=<ID=DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED,Number=.,Type=String,Description=\"NA\">\n",
    "phylop_df = df.loc[:, df.columns.str.contains('PHYLOP')]\n",
    "phylop_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['DBNSFP_PHYLOP30WAY_MAMMALIAN', 'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED'], dtype='object')"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.columns[df.columns.str.contains('PHYLOP')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "score_cols = [\n",
    "    'VEP_SIFT_SCORE', \n",
    "    'VEP_POLYPHEN_SCORE',\n",
    "    'DBNSFP_SIFT_SCORE',\n",
    "    'DBNSFP_POLYPHEN2_HDIV_SCORE',\n",
    "    'DBNSFP_LRT_SCORE',\n",
    "    'DBNSFP_MUTATIONTASTER_SCORE',\n",
    "    'DBNSFP_GERP_RS',\n",
    "    'DBNSFP_PHYLOP30WAY_MAMMALIAN'\n",
    "]\n",
    "\n",
    "predictor_cols = [\n",
    "    'VEP_SIFT_PRED', \n",
    "    'VEP_POLYPHEN_PRED',\n",
    "    'DBNSFP_SIFT_PRED',\n",
    "    'DBNSFP_POLYPHEN2_HDIV_PRED',\n",
    "    'DBNSFP_LRT_PRED',\n",
    "    'DBNSFP_MUTATIONTASTER_PRED',\n",
    "    'DBNSFP_GERP_RS_PRED',\n",
    "    'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED'\n",
    "]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Exploring Other Columns in working_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>Interpro Domain</th>\n",
       "      <th>UNIPROT_DOMAIN</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             DBNSFP_INTERPRO_DOMAIN Interpro Domain  \\\n",
       "Genomic Description (GRCh37)                                          \n",
       "20:21686358:T>C                                 .&.             NaN   \n",
       "20:21686361:C>T                                 .&.             NaN   \n",
       "20:21686379:G>A                                 .&.             NaN   \n",
       "20:21686382:C>T                                 .&.             NaN   \n",
       "20:21686390:G>T                                 .&.             NaN   \n",
       "\n",
       "                             UNIPROT_DOMAIN  \n",
       "Genomic Description (GRCh37)                 \n",
       "20:21686358:T>C                         NaN  \n",
       "20:21686361:C>T                         NaN  \n",
       "20:21686379:G>A                         NaN  \n",
       "20:21686382:C>T                         NaN  \n",
       "20:21686390:G>T                         NaN  "
      ]
     },
     "execution_count": 98,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "domain_df = working_df.loc[:, working_df.columns.str.contains('Domain') | working_df.columns.str.contains('DOMAIN')]\n",
    "domain_cols = domain_df.columns.tolist()\n",
    "domain_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>23.6</td>\n",
       "      <td>23.6</td>\n",
       "      <td>3.009015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.0</td>\n",
       "      <td>23.0</td>\n",
       "      <td>2.751772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.9</td>\n",
       "      <td>22.9</td>\n",
       "      <td>2.662314</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.8</td>\n",
       "      <td>21.8</td>\n",
       "      <td>2.257539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>22.1</td>\n",
       "      <td>22.1</td>\n",
       "      <td>2.319001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD Phred  VEP_CADD_PHRED  VEP_CADD_RAW\n",
       "Genomic Description (GRCh37)                                          \n",
       "20:21686358:T>C                     23.6            23.6      3.009015\n",
       "20:21686361:C>T                     23.0            23.0      2.751772\n",
       "20:21686379:G>A                     22.9            22.9      2.662314\n",
       "20:21686382:C>T                     21.8            21.8      2.257539\n",
       "20:21686390:G>T                     22.1            22.1      2.319001"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##INFO=<ID=VEP_CADD_RAW,Number=1,Type=String,Description=\"\\\"VEP annotated CADD raw score.\\\"\">\n",
    "##INFO=<ID=VEP_CADD_PHRED,Number=1,Type=String,Description=\"\\\"VEP annotated CADD phred scaled score.\\\"\">\n",
    "cadd_df = working_df.loc[:, working_df.columns.str.contains('Cadd') | working_df.columns.str.contains('CADD')]\n",
    "cadd_cols = cadd_df.columns.tolist()\n",
    "cadd_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>OVERALL_MAX_MAF</th>\n",
       "      <th>OVERALL_MAX_MAF_SOURCE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>6.65956e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>0.000104932</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>0.0318745</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             OVERALL_MAX_MAF OVERALL_MAX_MAF_SOURCE\n",
       "Genomic Description (GRCh37)                                       \n",
       "20:21686358:T>C                          NaN                    NaN\n",
       "20:21686361:C>T                  6.65956e-05     gnomad_AF_nfe_pass\n",
       "20:21686379:G>A                  0.000104932     gnomad_AF_nfe_pass\n",
       "20:21686382:C>T                    0.0318745     gnomad_AF_afr_pass\n",
       "20:21686390:G>T                          NaN                    NaN"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf_df = working_df.loc[:, ['OVERALL_MAX_MAF', 'OVERALL_MAX_MAF_SOURCE']]\n",
    "maf_cols = maf_df.columns.tolist()\n",
    "maf_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "other_ddg_file_cols = ['Confidence Score', 'Max MAF (%)', 'Max MAF Source', 'Surface Area', 'Normalized SA', 'ddG', '|ddG|']\n",
    "other_dvd_cols = ['VEP_IMPACT', 'VEP_CONSEQUENCE', 'VEP_PROTEIN_POS']\n",
    "classification_cols = ['FINAL_PATHOGENICITY', 'FINAL_PRED', 'NUM_PATH_PREDS', 'TOTAL_NUM_PREDS']\n",
    "misc_details_cols = ['GENE', 'TYPE']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Identifying Duplicate Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>CLINVAR_CLNSIG</th>\n",
       "      <th>CLINVAR_PATHOGENIC</th>\n",
       "      <th>CURATED_PATHOGENICITY</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>VEP_CONSEQUENCE</th>\n",
       "      <th>VEP_FEATURE</th>\n",
       "      <th>VEP_IMPACT</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>Variant Classification</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>23.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32.73</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.596</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-1.4</td>\n",
       "      <td>1.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.06</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.917</td>\n",
       "      <td>4</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.90</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.23</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2.32</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.787</td>\n",
       "      <td>10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.61</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2.11</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>11</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>22.10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.28</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3.51</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.005</td>\n",
       "      <td>14</td>\n",
       "      <td>0.04</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>27.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Likely pathogenic</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>27.50</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>29.20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>1.000</td>\n",
       "      <td>422</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>14.78</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>44.11</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3.68</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>0.097</td>\n",
       "      <td>426</td>\n",
       "      <td>1.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>27.60</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.21</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>5.59</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>...</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>NM_001002295.2</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>0.997</td>\n",
       "      <td>432</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128786 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD Phred CLINVAR_CLNSIG  CLINVAR_PATHOGENIC  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                    23.60            NaN                 NaN   \n",
       "20:21686361:C>T                    23.00            NaN                 NaN   \n",
       "20:21686379:G>A                    22.90            NaN                 NaN   \n",
       "20:21686382:C>T                    21.80            NaN                 NaN   \n",
       "20:21686390:G>T                    22.10            NaN                 NaN   \n",
       "...                                  ...            ...                 ...   \n",
       "10:8115915:C>T                     27.20            NaN                 NaN   \n",
       "10:8115916:C>G                     27.50            NaN                 NaN   \n",
       "10:8115916:C>T                     29.20            NaN                 NaN   \n",
       "10:8115927:C>T                     14.78            NaN                 NaN   \n",
       "10:8115945:G>A                     27.60            NaN                 NaN   \n",
       "\n",
       "                             CURATED_PATHOGENICITY  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "20:21686358:T>C                                NaN             32.73   \n",
       "20:21686361:C>T                                NaN             38.06   \n",
       "20:21686379:G>A                                NaN             37.23   \n",
       "20:21686382:C>T                                NaN             34.61   \n",
       "20:21686390:G>T                                NaN             31.28   \n",
       "...                                            ...               ...   \n",
       "10:8115915:C>T                                 NaN             47.29   \n",
       "10:8115916:C>G                                 NaN             47.29   \n",
       "10:8115916:C>T                                 NaN             47.29   \n",
       "10:8115927:C>T                                 NaN             44.11   \n",
       "10:8115945:G>A                                 NaN             37.21   \n",
       "\n",
       "                                   Consequence  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686358:T>C               missense_variant            4.28   \n",
       "20:21686361:C>T               missense_variant            4.28   \n",
       "20:21686379:G>A               missense_variant            2.32   \n",
       "20:21686382:C>T               missense_variant            2.11   \n",
       "20:21686390:G>T               missense_variant            3.51   \n",
       "...                                        ...             ...   \n",
       "10:8115915:C>T                missense_variant            5.59   \n",
       "10:8115916:C>G                missense_variant            5.59   \n",
       "10:8115916:C>T                missense_variant            5.59   \n",
       "10:8115927:C>T                missense_variant            3.68   \n",
       "10:8115945:G>A                missense_variant            5.59   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_INTERPRO_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686358:T>C                                C                    .&.   \n",
       "20:21686361:C>T                                C                    .&.   \n",
       "20:21686379:G>A                                C                    .&.   \n",
       "20:21686382:C>T                                C                    .&.   \n",
       "20:21686390:G>T                                C                    .&.   \n",
       "...                                          ...                    ...   \n",
       "10:8115915:C>T                                 C                  .&.&.   \n",
       "10:8115916:C>G                                 C                  .&.&.   \n",
       "10:8115916:C>T                                 C                  .&.&.   \n",
       "10:8115927:C>T                                 C                  .&.&.   \n",
       "10:8115945:G>A                                 C                  .&.&.   \n",
       "\n",
       "                             DBNSFP_LRT_PRED  ...   VEP_CONSEQUENCE  \\\n",
       "Genomic Description (GRCh37)                  ...                     \n",
       "20:21686358:T>C                          NaN  ...  missense_variant   \n",
       "20:21686361:C>T                          NaN  ...  missense_variant   \n",
       "20:21686379:G>A                          NaN  ...  missense_variant   \n",
       "20:21686382:C>T                          NaN  ...  missense_variant   \n",
       "20:21686390:G>T                          NaN  ...  missense_variant   \n",
       "...                                      ...  ...               ...   \n",
       "10:8115915:C>T                             U  ...  missense_variant   \n",
       "10:8115916:C>G                             U  ...  missense_variant   \n",
       "10:8115916:C>T                             U  ...  missense_variant   \n",
       "10:8115927:C>T                             D  ...  missense_variant   \n",
       "10:8115945:G>A                             D  ...  missense_variant   \n",
       "\n",
       "                                 VEP_FEATURE VEP_IMPACT  VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                                 \n",
       "20:21686358:T>C                  NM_006192.5   MODERATE                  P   \n",
       "20:21686361:C>T                  NM_006192.5   MODERATE                  P   \n",
       "20:21686379:G>A                  NM_006192.5   MODERATE                  P   \n",
       "20:21686382:C>T                  NM_006192.5   MODERATE                  P   \n",
       "20:21686390:G>T                  NM_006192.5   MODERATE                  B   \n",
       "...                                      ...        ...                ...   \n",
       "10:8115915:C>T                NM_001002295.2   MODERATE                  D   \n",
       "10:8115916:C>G                NM_001002295.2   MODERATE                  D   \n",
       "10:8115916:C>T                NM_001002295.2   MODERATE                  D   \n",
       "10:8115927:C>T                NM_001002295.2   MODERATE                  B   \n",
       "10:8115945:G>A                NM_001002295.2   MODERATE                  D   \n",
       "\n",
       "                             VEP_POLYPHEN_SCORE VEP_PROTEIN_POS  \\\n",
       "Genomic Description (GRCh37)                                      \n",
       "20:21686358:T>C                           0.596               3   \n",
       "20:21686361:C>T                           0.917               4   \n",
       "20:21686379:G>A                           0.787              10   \n",
       "20:21686382:C>T                           0.804              11   \n",
       "20:21686390:G>T                           0.005              14   \n",
       "...                                         ...             ...   \n",
       "10:8115915:C>T                            1.000             422   \n",
       "10:8115916:C>G                            1.000             422   \n",
       "10:8115916:C>T                            1.000             422   \n",
       "10:8115927:C>T                            0.097             426   \n",
       "10:8115945:G>A                            0.997             432   \n",
       "\n",
       "                             VEP_SIFT_SCORE Variant Classification  ddG |ddG|  \n",
       "Genomic Description (GRCh37)                                                   \n",
       "20:21686358:T>C                        0.00   Unknown significance -1.4   1.4  \n",
       "20:21686361:C>T                        0.00   Unknown significance -0.0   0.0  \n",
       "20:21686379:G>A                        0.00   Unknown significance -0.3   0.3  \n",
       "20:21686382:C>T                        0.01                 Benign -0.3   0.3  \n",
       "20:21686390:G>T                        0.04   Unknown significance -0.6   0.6  \n",
       "...                                     ...                    ...  ...   ...  \n",
       "10:8115915:C>T                         0.01      Likely pathogenic -0.3   0.3  \n",
       "10:8115916:C>G                         0.00   Unknown significance -0.3   0.3  \n",
       "10:8115916:C>T                         0.00   Unknown significance -0.2   0.2  \n",
       "10:8115927:C>T                         1.00   Unknown significance -0.1   0.1  \n",
       "10:8115945:G>A                         0.00   Unknown significance  0.2   0.2  \n",
       "\n",
       "[128786 rows x 46 columns]"
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df = df[kept_features_list]\n",
    "working_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Consquence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_CONSEQUENCE</th>\n",
       "      <th>Consequence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [VEP_CONSEQUENCE, Consequence]\n",
       "Index: []"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[working_df.VEP_CONSEQUENCE != working_df.Consequence, ['VEP_CONSEQUENCE', 'Consequence']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.drop(columns=['VEP_CONSEQUENCE'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### CADD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CADD Phred', 'VEP_CADD_PHRED', 'VEP_CADD_RAW'], dtype='object')"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.columns[working_df.columns.str.contains('CADD') | working_df.columns.str.contains('Cadd')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These values are all missing even if it says they're not matching. Also are all MNV -> dropping these then re-examining"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TYPE</th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:61468569:CCT&gt;GCC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2:228111435:TG&gt;CA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2:73613119:GG&gt;CA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2:73829349:GCTC&gt;ACAT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18:44102214:GC&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18:44219603:AG&gt;CC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4:6302568:TC&gt;AG</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4:6303899:CGC&gt;GAG</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4:6304085:TCA&gt;CCG</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3:69990450:TT&gt;CC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5:89986675:GC&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5:90119324:GT&gt;AA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12:48370339:TC&gt;GA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12:49441832:GC&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12:49445061:TC&gt;CG</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12:49445372:TGTG&gt;CATA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:145583732:AG&gt;CT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:24811038:CTC&gt;TCT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:38285911:GA&gt;TC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:61655487:AA&gt;GC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:61728967:GG&gt;TC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:61736569:GG&gt;AT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8:61769079:GC&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17:46606943:GA&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17:72915617:CT&gt;GA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11:17593707:TCT&gt;CCC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11:17662466:CC&gt;AT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11:76910678:CG&gt;TA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11:76916512:GC&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6:133844297:TAC&gt;AAA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6:133846381:GC&gt;AA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1:160011388:GT&gt;AA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1:215848515:AG&gt;CC</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1:216595236:AA&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1:247587982:CTG&gt;TTT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X:107865120:AA&gt;TT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>X:107930811:GC&gt;CT</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7:5569281:TC&gt;AA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:55568971:CT&gt;GG</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:73466709:TG&gt;CA</th>\n",
       "      <td>MNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             TYPE  CADD Phred  VEP_CADD_PHRED\n",
       "Genomic Description (GRCh37)                                 \n",
       "20:61468569:CCT>GCC           MNV         NaN             NaN\n",
       "2:228111435:TG>CA             MNV         NaN             NaN\n",
       "2:73613119:GG>CA              MNV         NaN             NaN\n",
       "2:73829349:GCTC>ACAT          MNV         NaN             NaN\n",
       "18:44102214:GC>TT             MNV         NaN             NaN\n",
       "18:44219603:AG>CC             MNV         NaN             NaN\n",
       "4:6302568:TC>AG               MNV         NaN             NaN\n",
       "4:6303899:CGC>GAG             MNV         NaN             NaN\n",
       "4:6304085:TCA>CCG             MNV         NaN             NaN\n",
       "3:69990450:TT>CC              MNV         NaN             NaN\n",
       "5:89986675:GC>TT              MNV         NaN             NaN\n",
       "5:90119324:GT>AA              MNV         NaN             NaN\n",
       "12:48370339:TC>GA             MNV         NaN             NaN\n",
       "12:49441832:GC>TT             MNV         NaN             NaN\n",
       "12:49445061:TC>CG             MNV         NaN             NaN\n",
       "12:49445372:TGTG>CATA         MNV         NaN             NaN\n",
       "8:145583732:AG>CT             MNV         NaN             NaN\n",
       "8:24811038:CTC>TCT            MNV         NaN             NaN\n",
       "8:38285911:GA>TC              MNV         NaN             NaN\n",
       "8:61655487:AA>GC              MNV         NaN             NaN\n",
       "8:61728967:GG>TC              MNV         NaN             NaN\n",
       "8:61736569:GG>AT              MNV         NaN             NaN\n",
       "8:61769079:GC>TT              MNV         NaN             NaN\n",
       "17:46606943:GA>TT             MNV         NaN             NaN\n",
       "17:72915617:CT>GA             MNV         NaN             NaN\n",
       "11:17593707:TCT>CCC           MNV         NaN             NaN\n",
       "11:17662466:CC>AT             MNV         NaN             NaN\n",
       "11:76910678:CG>TA             MNV         NaN             NaN\n",
       "11:76916512:GC>TT             MNV         NaN             NaN\n",
       "6:133844297:TAC>AAA           MNV         NaN             NaN\n",
       "6:133846381:GC>AA             MNV         NaN             NaN\n",
       "1:160011388:GT>AA             MNV         NaN             NaN\n",
       "1:215848515:AG>CC             MNV         NaN             NaN\n",
       "1:216595236:AA>TT             MNV         NaN             NaN\n",
       "1:247587982:CTG>TTT           MNV         NaN             NaN\n",
       "X:107865120:AA>TT             MNV         NaN             NaN\n",
       "X:107930811:GC>CT             MNV         NaN             NaN\n",
       "7:5569281:TC>AA               MNV         NaN             NaN\n",
       "10:55568971:CT>GG             MNV         NaN             NaN\n",
       "10:73466709:TG>CA             MNV         NaN             NaN"
      ]
     },
     "execution_count": 170,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[working_df.VEP_CADD_PHRED != working_df['CADD Phred'], ['TYPE', 'CADD Phred', 'VEP_CADD_PHRED']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df[working_df.VEP_CADD_PHRED == working_df['CADD Phred']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TYPE</th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [TYPE, CADD Phred, VEP_CADD_PHRED]\n",
       "Index: []"
      ]
     },
     "execution_count": 172,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[working_df.VEP_CADD_PHRED != working_df['CADD Phred'], ['TYPE', 'CADD Phred', 'VEP_CADD_PHRED']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.drop(columns='VEP_CADD_PHRED')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Impact"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>CLINVAR_CLNSIG</th>\n",
       "      <th>CLINVAR_PATHOGENIC</th>\n",
       "      <th>CURATED_PATHOGENICITY</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_FEATURE</th>\n",
       "      <th>VEP_IMPACT</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>Variant Classification</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>0 rows × 44 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [CADD Phred, CLINVAR_CLNSIG, CLINVAR_PATHOGENIC, CURATED_PATHOGENICITY, Confidence Score, Consequence, DBNSFP_GERP_RS, DBNSFP_GERP_RS_PRED, DBNSFP_INTERPRO_DOMAIN, DBNSFP_LRT_PRED, DBNSFP_LRT_SCORE, DBNSFP_MUTATIONTASTER_PRED, DBNSFP_MUTATIONTASTER_SCORE, DBNSFP_PHYLOP30WAY_MAMMALIAN, DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED, DBNSFP_POLYPHEN2_HDIV_PRED, DBNSFP_POLYPHEN2_HDIV_SCORE, DBNSFP_SIFT_PRED, DBNSFP_SIFT_SCORE, FINAL_PATHOGENICITY, FINAL_PRED, GENE, Impact, Interpro Domain, Max MAF (%), Max MAF Source, NUM_PATH_PREDS, Normalized SA, OVERALL_MAX_MAF, OVERALL_MAX_MAF_SOURCE, Surface Area, TOTAL_NUM_PREDS, TYPE, UNIPROT_DOMAIN, VEP_CADD_RAW, VEP_FEATURE, VEP_IMPACT, VEP_POLYPHEN_PRED, VEP_POLYPHEN_SCORE, VEP_PROTEIN_POS, VEP_SIFT_SCORE, Variant Classification, ddG, |ddG|]\n",
       "Index: []\n",
       "\n",
       "[0 rows x 44 columns]"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df[working_df['VEP_IMPACT'] != working_df['Impact']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.drop(columns='VEP_IMPACT')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Max MAF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>OVERALL_MAX_MAF</th>\n",
       "      <th>OVERALL_MAX_MAF_SOURCE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>0.006660</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.65956e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>0.010493</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.000104932</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>3.187450</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>0.0318745</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>0.006484</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.48424e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>0.012309</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>0.000123092</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>0.000880</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>8.80065e-06</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>0.001549</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>1.54947e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Max MAF (%)      Max MAF Source OVERALL_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                                                    \n",
       "20:21686358:T>C                  0.000000                 NaN             NaN   \n",
       "20:21686361:C>T                  0.006660  gnomad_AF_nfe_pass     6.65956e-05   \n",
       "20:21686379:G>A                  0.010493  gnomad_AF_nfe_pass     0.000104932   \n",
       "20:21686382:C>T                  3.187450  gnomad_AF_afr_pass       0.0318745   \n",
       "20:21686390:G>T                  0.000000                 NaN             NaN   \n",
       "...                                   ...                 ...             ...   \n",
       "10:8115915:C>T                   0.000000                 NaN              []   \n",
       "10:8115916:C>G                   0.006484  gnomad_AF_nfe_pass     6.48424e-05   \n",
       "10:8115916:C>T                   0.012309  gnomad_AF_afr_pass     0.000123092   \n",
       "10:8115927:C>T                   0.000880  gnomad_AF_nfe_pass     8.80065e-06   \n",
       "10:8115945:G>A                   0.001549  gnomad_AF_nfe_pass     1.54947e-05   \n",
       "\n",
       "                             OVERALL_MAX_MAF_SOURCE  \n",
       "Genomic Description (GRCh37)                         \n",
       "20:21686358:T>C                                 NaN  \n",
       "20:21686361:C>T                  gnomad_AF_nfe_pass  \n",
       "20:21686379:G>A                  gnomad_AF_nfe_pass  \n",
       "20:21686382:C>T                  gnomad_AF_afr_pass  \n",
       "20:21686390:G>T                                 NaN  \n",
       "...                                             ...  \n",
       "10:8115915:C>T                                   []  \n",
       "10:8115916:C>G                   gnomad_AF_nfe_pass  \n",
       "10:8115916:C>T                   gnomad_AF_afr_pass  \n",
       "10:8115927:C>T                   gnomad_AF_nfe_pass  \n",
       "10:8115945:G>A                   gnomad_AF_nfe_pass  \n",
       "\n",
       "[128712 rows x 4 columns]"
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df[['Max MAF (%)', 'Max MAF Source', 'OVERALL_MAX_MAF', 'OVERALL_MAX_MAF_SOURCE']]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Everywhere that Max MAF Source does not match OVERALL_MAX_MAF_SOURCE also does not have any MAF data\n",
    "\n",
    "Mismatches in Max MAF % are just rounding errors"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>OVERALL_MAX_MAF</th>\n",
       "      <th>OVERALL_MAX_MAF_SOURCE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686398:G&gt;T</th>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686443:C&gt;A</th>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686443:C&gt;G</th>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Max MAF (%) Max MAF Source OVERALL_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                                               \n",
       "20:21686358:T>C                       0.0            NaN             NaN   \n",
       "20:21686390:G>T                       0.0            NaN             NaN   \n",
       "20:21686398:G>T                       0.0            NaN             NaN   \n",
       "20:21686443:C>A                       0.0            NaN             NaN   \n",
       "20:21686443:C>G                       0.0            NaN             NaN   \n",
       "\n",
       "                             OVERALL_MAX_MAF_SOURCE  \n",
       "Genomic Description (GRCh37)                         \n",
       "20:21686358:T>C                                 NaN  \n",
       "20:21686390:G>T                                 NaN  \n",
       "20:21686398:G>T                                 NaN  \n",
       "20:21686443:C>A                                 NaN  \n",
       "20:21686443:C>G                                 NaN  "
      ]
     },
     "execution_count": 196,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp_df = working_df.loc[working_df['Max MAF Source'] != working_df['OVERALL_MAX_MAF_SOURCE'], ['Max MAF (%)', 'Max MAF Source', 'OVERALL_MAX_MAF', 'OVERALL_MAX_MAF_SOURCE']]\n",
    "temp_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>OVERALL_MAX_MAF</th>\n",
       "      <th>OVERALL_MAX_MAF_SOURCE</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>0.006660</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.65956e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>0.010493</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.000104932</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>3.187450</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>0.0318745</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              Max MAF (%)      Max MAF Source OVERALL_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                                                    \n",
       "20:21686358:T>C                  0.000000                 NaN             NaN   \n",
       "20:21686361:C>T                  0.006660  gnomad_AF_nfe_pass     6.65956e-05   \n",
       "20:21686379:G>A                  0.010493  gnomad_AF_nfe_pass     0.000104932   \n",
       "20:21686382:C>T                  3.187450  gnomad_AF_afr_pass       0.0318745   \n",
       "20:21686390:G>T                  0.000000                 NaN             NaN   \n",
       "\n",
       "                             OVERALL_MAX_MAF_SOURCE  \n",
       "Genomic Description (GRCh37)                         \n",
       "20:21686358:T>C                                 NaN  \n",
       "20:21686361:C>T                  gnomad_AF_nfe_pass  \n",
       "20:21686379:G>A                  gnomad_AF_nfe_pass  \n",
       "20:21686382:C>T                  gnomad_AF_afr_pass  \n",
       "20:21686390:G>T                                 NaN  "
      ]
     },
     "execution_count": 201,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp_df = working_df.loc[working_df['Max MAF (%)'] != working_df['OVERALL_MAX_MAF'], ['Max MAF (%)', 'Max MAF Source', 'OVERALL_MAX_MAF', 'OVERALL_MAX_MAF_SOURCE']]\n",
    "temp_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df.drop(columns=['OVERALL_MAX_MAF', 'OVERALL_MAX_MAF_SOURCE'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CADD Phred', 'CLINVAR_CLNSIG', 'CLINVAR_PATHOGENIC',\n",
       "       'CURATED_PATHOGENICITY', 'Confidence Score', 'Consequence',\n",
       "       'DBNSFP_GERP_RS', 'DBNSFP_GERP_RS_PRED', 'DBNSFP_INTERPRO_DOMAIN',\n",
       "       'DBNSFP_LRT_PRED', 'DBNSFP_LRT_SCORE', 'DBNSFP_MUTATIONTASTER_PRED',\n",
       "       'DBNSFP_MUTATIONTASTER_SCORE', 'DBNSFP_PHYLOP30WAY_MAMMALIAN',\n",
       "       'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED', 'DBNSFP_POLYPHEN2_HDIV_PRED',\n",
       "       'DBNSFP_POLYPHEN2_HDIV_SCORE', 'DBNSFP_SIFT_PRED', 'DBNSFP_SIFT_SCORE',\n",
       "       'FINAL_PATHOGENICITY', 'FINAL_PRED', 'GENE', 'Interpro Domain',\n",
       "       'Max MAF (%)', 'Max MAF Source', 'NUM_PATH_PREDS', 'Normalized SA',\n",
       "       'Surface Area', 'TOTAL_NUM_PREDS', 'UNIPROT_DOMAIN', 'VEP_CADD_RAW',\n",
       "       'VEP_POLYPHEN_PRED', 'VEP_POLYPHEN_SCORE', 'VEP_PROTEIN_POS',\n",
       "       'VEP_SIFT_SCORE', 'Variant Classification', 'ddG', '|ddG|'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Filtering Instances"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df[working_df.TYPE == 'SNV'].drop(columns='TYPE') # Drop multi-nucleotide variants (MNVS), remove TYPE column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "working_df = working_df[working_df.Consequence == 'missense_variant'] # Drop missense variants with secondary effects, i.e., splite region variant"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(128712, 38)"
      ]
     },
     "execution_count": 204,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(98433, 38)"
      ]
     },
     "execution_count": 206,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df[working_df['Max MAF (%)'] != 0].shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Removing Homogenous Columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Impact\n",
      "MODERATE    128712\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "print(working_df.Impact.value_counts())\n",
    "working_df = working_df.drop(columns='Impact')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD Phred</th>\n",
       "      <th>CLINVAR_CLNSIG</th>\n",
       "      <th>CLINVAR_PATHOGENIC</th>\n",
       "      <th>CURATED_PATHOGENICITY</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>...</th>\n",
       "      <th>UNIPROT_DOMAIN</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_FEATURE</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>Variant Classification</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>23.6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>32.73</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.009015</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>P</td>\n",
       "      <td>0.596</td>\n",
       "      <td>3</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-1.4</td>\n",
       "      <td>1.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>23.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>38.06</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>4.28</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.751772</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>P</td>\n",
       "      <td>0.917</td>\n",
       "      <td>4</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>22.9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37.23</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2.32</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.662314</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>P</td>\n",
       "      <td>0.787</td>\n",
       "      <td>10</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>21.8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>34.61</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2.11</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.257539</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>P</td>\n",
       "      <td>0.804</td>\n",
       "      <td>11</td>\n",
       "      <td>0.01</td>\n",
       "      <td>Benign</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>22.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>31.28</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3.51</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.319001</td>\n",
       "      <td>NM_006192.5</td>\n",
       "      <td>B</td>\n",
       "      <td>0.005</td>\n",
       "      <td>14</td>\n",
       "      <td>0.04</td>\n",
       "      <td>Unknown significance</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CADD Phred CLINVAR_CLNSIG  CLINVAR_PATHOGENIC  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                     23.6            NaN                 NaN   \n",
       "20:21686361:C>T                     23.0            NaN                 NaN   \n",
       "20:21686379:G>A                     22.9            NaN                 NaN   \n",
       "20:21686382:C>T                     21.8            NaN                 NaN   \n",
       "20:21686390:G>T                     22.1            NaN                 NaN   \n",
       "\n",
       "                             CURATED_PATHOGENICITY  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                           \n",
       "20:21686358:T>C                                NaN             32.73   \n",
       "20:21686361:C>T                                NaN             38.06   \n",
       "20:21686379:G>A                                NaN             37.23   \n",
       "20:21686382:C>T                                NaN             34.61   \n",
       "20:21686390:G>T                                NaN             31.28   \n",
       "\n",
       "                                   Consequence  DBNSFP_GERP_RS  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686358:T>C               missense_variant            4.28   \n",
       "20:21686361:C>T               missense_variant            4.28   \n",
       "20:21686379:G>A               missense_variant            2.32   \n",
       "20:21686382:C>T               missense_variant            2.11   \n",
       "20:21686390:G>T               missense_variant            3.51   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_INTERPRO_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686358:T>C                                C                    .&.   \n",
       "20:21686361:C>T                                C                    .&.   \n",
       "20:21686379:G>A                                C                    .&.   \n",
       "20:21686382:C>T                                C                    .&.   \n",
       "20:21686390:G>T                                C                    .&.   \n",
       "\n",
       "                             DBNSFP_LRT_PRED  ...  UNIPROT_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                  ...                   \n",
       "20:21686358:T>C                          NaN  ...             NaN   \n",
       "20:21686361:C>T                          NaN  ...             NaN   \n",
       "20:21686379:G>A                          NaN  ...             NaN   \n",
       "20:21686382:C>T                          NaN  ...             NaN   \n",
       "20:21686390:G>T                          NaN  ...             NaN   \n",
       "\n",
       "                             VEP_CADD_RAW  VEP_FEATURE  VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                                \n",
       "20:21686358:T>C                  3.009015  NM_006192.5                  P   \n",
       "20:21686361:C>T                  2.751772  NM_006192.5                  P   \n",
       "20:21686379:G>A                  2.662314  NM_006192.5                  P   \n",
       "20:21686382:C>T                  2.257539  NM_006192.5                  P   \n",
       "20:21686390:G>T                  2.319001  NM_006192.5                  B   \n",
       "\n",
       "                             VEP_POLYPHEN_SCORE VEP_PROTEIN_POS  \\\n",
       "Genomic Description (GRCh37)                                      \n",
       "20:21686358:T>C                           0.596               3   \n",
       "20:21686361:C>T                           0.917               4   \n",
       "20:21686379:G>A                           0.787              10   \n",
       "20:21686382:C>T                           0.804              11   \n",
       "20:21686390:G>T                           0.005              14   \n",
       "\n",
       "                             VEP_SIFT_SCORE Variant Classification  ddG |ddG|  \n",
       "Genomic Description (GRCh37)                                                   \n",
       "20:21686358:T>C                        0.00   Unknown significance -1.4   1.4  \n",
       "20:21686361:C>T                        0.00   Unknown significance -0.0   0.0  \n",
       "20:21686379:G>A                        0.00   Unknown significance -0.3   0.3  \n",
       "20:21686382:C>T                        0.01                 Benign -0.3   0.3  \n",
       "20:21686390:G>T                        0.04   Unknown significance -0.6   0.6  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 183,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Removing columns that are too varied"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "247\n"
     ]
    }
   ],
   "source": [
    "print(len(working_df.VEP_FEATURE.value_counts()))\n",
    "working_df = working_df.drop(columns='VEP_FEATURE')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CADD Phred', 'CLINVAR_CLNSIG', 'CLINVAR_PATHOGENIC',\n",
       "       'CURATED_PATHOGENICITY', 'Confidence Score', 'Consequence',\n",
       "       'DBNSFP_GERP_RS', 'DBNSFP_GERP_RS_PRED', 'DBNSFP_INTERPRO_DOMAIN',\n",
       "       'DBNSFP_LRT_PRED', 'DBNSFP_LRT_SCORE', 'DBNSFP_MUTATIONTASTER_PRED',\n",
       "       'DBNSFP_MUTATIONTASTER_SCORE', 'DBNSFP_PHYLOP30WAY_MAMMALIAN',\n",
       "       'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED', 'DBNSFP_POLYPHEN2_HDIV_PRED',\n",
       "       'DBNSFP_POLYPHEN2_HDIV_SCORE', 'DBNSFP_SIFT_PRED', 'DBNSFP_SIFT_SCORE',\n",
       "       'FINAL_PATHOGENICITY', 'FINAL_PRED', 'GENE', 'Interpro Domain',\n",
       "       'Max MAF (%)', 'Max MAF Source', 'NUM_PATH_PREDS', 'Normalized SA',\n",
       "       'OVERALL_MAX_MAF', 'OVERALL_MAX_MAF_SOURCE', 'Surface Area',\n",
       "       'TOTAL_NUM_PREDS', 'UNIPROT_DOMAIN', 'VEP_CADD_RAW', 'VEP_FEATURE',\n",
       "       'VEP_POLYPHEN_PRED', 'VEP_POLYPHEN_SCORE', 'VEP_PROTEIN_POS',\n",
       "       'VEP_SIFT_SCORE', 'Variant Classification', 'ddG', '|ddG|'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 184,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.columns"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sort by numerical vs categorical"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Categorical"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Reviewing other columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['CLINVAR_PATHOGENIC', 'Confidence Score', 'DBNSFP_GERP_RS', 'DBNSFP_LRT_SCORE', 'DBNSFP_PHYLOP30WAY_MAMMALIAN', 'Max MAF (%)', 'NUM_PATH_PREDS', 'Normalized SA', 'Surface Area', 'TOTAL_NUM_PREDS', 'VEP_CADD_PHRED', 'VEP_CADD_RAW', 'VEP_POLYPHEN_SCORE', 'VEP_SIFT_SCORE', 'ddG', '|ddG|']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CLINVAR_PATHOGENIC</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>DBNSFP_GERP_RS</th>\n",
       "      <th>DBNSFP_LRT_SCORE</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN</th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>NUM_PATH_PREDS</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>Surface Area</th>\n",
       "      <th>TOTAL_NUM_PREDS</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_POLYPHEN_SCORE</th>\n",
       "      <th>VEP_SIFT_SCORE</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>NaN</td>\n",
       "      <td>32.73</td>\n",
       "      <td>4.28</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.111</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.752780</td>\n",
       "      <td>172.951510</td>\n",
       "      <td>5.0</td>\n",
       "      <td>23.60</td>\n",
       "      <td>3.009015</td>\n",
       "      <td>0.596</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-1.4</td>\n",
       "      <td>1.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>38.06</td>\n",
       "      <td>4.28</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.118</td>\n",
       "      <td>0.006660</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.883870</td>\n",
       "      <td>139.065932</td>\n",
       "      <td>5.0</td>\n",
       "      <td>23.00</td>\n",
       "      <td>2.751772</td>\n",
       "      <td>0.917</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>NaN</td>\n",
       "      <td>37.23</td>\n",
       "      <td>2.32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.147</td>\n",
       "      <td>0.010493</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.670700</td>\n",
       "      <td>180.703642</td>\n",
       "      <td>5.0</td>\n",
       "      <td>22.90</td>\n",
       "      <td>2.662314</td>\n",
       "      <td>0.787</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>34.61</td>\n",
       "      <td>2.11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.106</td>\n",
       "      <td>3.187450</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.601022</td>\n",
       "      <td>76.425233</td>\n",
       "      <td>5.0</td>\n",
       "      <td>21.80</td>\n",
       "      <td>2.257539</td>\n",
       "      <td>0.804</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>31.28</td>\n",
       "      <td>3.51</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.147</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.578281</td>\n",
       "      <td>99.401913</td>\n",
       "      <td>5.0</td>\n",
       "      <td>22.10</td>\n",
       "      <td>2.319001</td>\n",
       "      <td>0.005</td>\n",
       "      <td>0.04</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>5.59</td>\n",
       "      <td>0.000643</td>\n",
       "      <td>1.026</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>5.0</td>\n",
       "      <td>27.20</td>\n",
       "      <td>3.938849</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.01</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>5.59</td>\n",
       "      <td>0.000643</td>\n",
       "      <td>1.026</td>\n",
       "      <td>0.006484</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>5.0</td>\n",
       "      <td>27.50</td>\n",
       "      <td>3.979789</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>47.29</td>\n",
       "      <td>5.59</td>\n",
       "      <td>0.000643</td>\n",
       "      <td>1.026</td>\n",
       "      <td>0.012309</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>5.0</td>\n",
       "      <td>29.20</td>\n",
       "      <td>4.163516</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.00</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>44.11</td>\n",
       "      <td>3.68</td>\n",
       "      <td>0.000030</td>\n",
       "      <td>1.026</td>\n",
       "      <td>0.000880</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.816259</td>\n",
       "      <td>128.397680</td>\n",
       "      <td>6.0</td>\n",
       "      <td>14.78</td>\n",
       "      <td>1.223753</td>\n",
       "      <td>0.097</td>\n",
       "      <td>1.00</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>NaN</td>\n",
       "      <td>37.21</td>\n",
       "      <td>5.59</td>\n",
       "      <td>0.000008</td>\n",
       "      <td>1.176</td>\n",
       "      <td>0.001549</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.709449</td>\n",
       "      <td>67.643291</td>\n",
       "      <td>6.0</td>\n",
       "      <td>27.60</td>\n",
       "      <td>3.995061</td>\n",
       "      <td>0.997</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 16 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              CLINVAR_PATHOGENIC  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                         \n",
       "20:21686358:T>C                              NaN             32.73   \n",
       "20:21686361:C>T                              NaN             38.06   \n",
       "20:21686379:G>A                              NaN             37.23   \n",
       "20:21686382:C>T                              NaN             34.61   \n",
       "20:21686390:G>T                              NaN             31.28   \n",
       "...                                          ...               ...   \n",
       "10:8115915:C>T                               NaN             47.29   \n",
       "10:8115916:C>G                               NaN             47.29   \n",
       "10:8115916:C>T                               NaN             47.29   \n",
       "10:8115927:C>T                               NaN             44.11   \n",
       "10:8115945:G>A                               NaN             37.21   \n",
       "\n",
       "                              DBNSFP_GERP_RS  DBNSFP_LRT_SCORE  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686358:T>C                         4.28               NaN   \n",
       "20:21686361:C>T                         4.28               NaN   \n",
       "20:21686379:G>A                         2.32               NaN   \n",
       "20:21686382:C>T                         2.11               NaN   \n",
       "20:21686390:G>T                         3.51               NaN   \n",
       "...                                      ...               ...   \n",
       "10:8115915:C>T                          5.59          0.000643   \n",
       "10:8115916:C>G                          5.59          0.000643   \n",
       "10:8115916:C>T                          5.59          0.000643   \n",
       "10:8115927:C>T                          3.68          0.000030   \n",
       "10:8115945:G>A                          5.59          0.000008   \n",
       "\n",
       "                              DBNSFP_PHYLOP30WAY_MAMMALIAN  Max MAF (%)  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686358:T>C                                      1.111     0.000000   \n",
       "20:21686361:C>T                                      0.118     0.006660   \n",
       "20:21686379:G>A                                      1.147     0.010493   \n",
       "20:21686382:C>T                                      0.106     3.187450   \n",
       "20:21686390:G>T                                      1.147     0.000000   \n",
       "...                                                    ...          ...   \n",
       "10:8115915:C>T                                       1.026     0.000000   \n",
       "10:8115916:C>G                                       1.026     0.006484   \n",
       "10:8115916:C>T                                       1.026     0.012309   \n",
       "10:8115927:C>T                                       1.026     0.000880   \n",
       "10:8115945:G>A                                       1.176     0.001549   \n",
       "\n",
       "                              NUM_PATH_PREDS  Normalized SA  Surface Area  \\\n",
       "Genomic Description (GRCh37)                                                \n",
       "20:21686358:T>C                          4.0       0.752780    172.951510   \n",
       "20:21686361:C>T                          3.0       0.883870    139.065932   \n",
       "20:21686379:G>A                          4.0       0.670700    180.703642   \n",
       "20:21686382:C>T                          3.0       0.601022     76.425233   \n",
       "20:21686390:G>T                          3.0       0.578281     99.401913   \n",
       "...                                      ...            ...           ...   \n",
       "10:8115915:C>T                           5.0       0.870607    136.946574   \n",
       "10:8115916:C>G                           5.0       0.870607    136.946574   \n",
       "10:8115916:C>T                           5.0       0.870607    136.946574   \n",
       "10:8115927:C>T                           4.0       0.816259    128.397680   \n",
       "10:8115945:G>A                           6.0       0.709449     67.643291   \n",
       "\n",
       "                              TOTAL_NUM_PREDS  VEP_CADD_PHRED  VEP_CADD_RAW  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                           5.0           23.60      3.009015   \n",
       "20:21686361:C>T                           5.0           23.00      2.751772   \n",
       "20:21686379:G>A                           5.0           22.90      2.662314   \n",
       "20:21686382:C>T                           5.0           21.80      2.257539   \n",
       "20:21686390:G>T                           5.0           22.10      2.319001   \n",
       "...                                       ...             ...           ...   \n",
       "10:8115915:C>T                            5.0           27.20      3.938849   \n",
       "10:8115916:C>G                            5.0           27.50      3.979789   \n",
       "10:8115916:C>T                            5.0           29.20      4.163516   \n",
       "10:8115927:C>T                            6.0           14.78      1.223753   \n",
       "10:8115945:G>A                            6.0           27.60      3.995061   \n",
       "\n",
       "                              VEP_POLYPHEN_SCORE  VEP_SIFT_SCORE  ddG  |ddG|  \n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                            0.596            0.00 -1.4    1.4  \n",
       "20:21686361:C>T                            0.917            0.00 -0.0    0.0  \n",
       "20:21686379:G>A                            0.787            0.00 -0.3    0.3  \n",
       "20:21686382:C>T                            0.804            0.01 -0.3    0.3  \n",
       "20:21686390:G>T                            0.005            0.04 -0.6    0.6  \n",
       "...                                          ...             ...  ...    ...  \n",
       "10:8115915:C>T                             1.000            0.01 -0.3    0.3  \n",
       "10:8115916:C>G                             1.000            0.00 -0.3    0.3  \n",
       "10:8115916:C>T                             1.000            0.00 -0.2    0.2  \n",
       "10:8115927:C>T                             0.097            1.00 -0.1    0.1  \n",
       "10:8115945:G>A                             0.997            0.00  0.2    0.2  \n",
       "\n",
       "[128712 rows x 16 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "numerical_columns = working_df.select_dtypes(include=[np.number]).columns.tolist()\n",
    "print(numerical_columns)\n",
    "working_df[numerical_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['CLINVAR_CLNSIG', 'CURATED_PATHOGENICITY', 'DBNSFP_GERP_RS_PRED', 'DBNSFP_INTERPRO_DOMAIN', 'DBNSFP_LRT_PRED', 'DBNSFP_MUTATIONTASTER_PRED', 'DBNSFP_MUTATIONTASTER_SCORE', 'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED', 'DBNSFP_POLYPHEN2_HDIV_PRED', 'DBNSFP_POLYPHEN2_HDIV_SCORE', 'DBNSFP_SIFT_PRED', 'DBNSFP_SIFT_SCORE', 'FINAL_PATHOGENICITY', 'FINAL_PRED', 'GENE', 'Interpro Domain', 'Max MAF Source', 'OVERALL_MAX_MAF', 'OVERALL_MAX_MAF_SOURCE', 'TYPE', 'UNIPROT_DOMAIN', 'VEP_CONSEQUENCE', 'VEP_IMPACT', 'VEP_POLYPHEN_PRED', 'VEP_PROTEIN_POS']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CLINVAR_CLNSIG</th>\n",
       "      <th>CURATED_PATHOGENICITY</th>\n",
       "      <th>DBNSFP_GERP_RS_PRED</th>\n",
       "      <th>DBNSFP_INTERPRO_DOMAIN</th>\n",
       "      <th>DBNSFP_LRT_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_PRED</th>\n",
       "      <th>DBNSFP_MUTATIONTASTER_SCORE</th>\n",
       "      <th>DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED</th>\n",
       "      <th>DBNSFP_POLYPHEN2_HDIV_PRED</th>\n",
       "      <th>DBNSFP_POLYPHEN2_HDIV_SCORE</th>\n",
       "      <th>...</th>\n",
       "      <th>Interpro Domain</th>\n",
       "      <th>Max MAF Source</th>\n",
       "      <th>OVERALL_MAX_MAF</th>\n",
       "      <th>OVERALL_MAX_MAF_SOURCE</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>UNIPROT_DOMAIN</th>\n",
       "      <th>VEP_CONSEQUENCE</th>\n",
       "      <th>VEP_IMPACT</th>\n",
       "      <th>VEP_POLYPHEN_PRED</th>\n",
       "      <th>VEP_PROTEIN_POS</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.506194</td>\n",
       "      <td>C</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>0.596&amp;.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.758794</td>\n",
       "      <td>N</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>0.917&amp;.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.65956e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.999272</td>\n",
       "      <td>C</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>0.787&amp;.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>0.000104932</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.998176</td>\n",
       "      <td>N</td>\n",
       "      <td>P&amp;.</td>\n",
       "      <td>0.804&amp;.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>0.0318745</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>P</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>N</td>\n",
       "      <td>0.999492</td>\n",
       "      <td>C</td>\n",
       "      <td>B&amp;.</td>\n",
       "      <td>0.005&amp;.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>C</td>\n",
       "      <td>D&amp;D&amp;D</td>\n",
       "      <td>1.0&amp;1.0&amp;1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "      <td>[]</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>C</td>\n",
       "      <td>D&amp;D&amp;D</td>\n",
       "      <td>1.0&amp;1.0&amp;1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>6.48424e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>U</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>1&amp;1</td>\n",
       "      <td>C</td>\n",
       "      <td>D&amp;D&amp;D</td>\n",
       "      <td>1.0&amp;1.0&amp;1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>0.000123092</td>\n",
       "      <td>gnomad_AF_afr_pass</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.999167&amp;0.999167</td>\n",
       "      <td>C</td>\n",
       "      <td>B&amp;B&amp;B</td>\n",
       "      <td>0.097&amp;0.097&amp;0.001</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>8.80065e-06</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>B</td>\n",
       "      <td>426</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C</td>\n",
       "      <td>.&amp;.&amp;.</td>\n",
       "      <td>D</td>\n",
       "      <td>D&amp;D</td>\n",
       "      <td>0.999993&amp;0.999993</td>\n",
       "      <td>C</td>\n",
       "      <td>D&amp;D&amp;P</td>\n",
       "      <td>0.997&amp;0.997&amp;0.745</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>1.54947e-05</td>\n",
       "      <td>gnomad_AF_nfe_pass</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>D</td>\n",
       "      <td>432</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                             CLINVAR_CLNSIG CURATED_PATHOGENICITY  \\\n",
       "Genomic Description (GRCh37)                                        \n",
       "20:21686358:T>C                         NaN                   NaN   \n",
       "20:21686361:C>T                         NaN                   NaN   \n",
       "20:21686379:G>A                         NaN                   NaN   \n",
       "20:21686382:C>T                         NaN                   NaN   \n",
       "20:21686390:G>T                         NaN                   NaN   \n",
       "...                                     ...                   ...   \n",
       "10:8115915:C>T                          NaN                   NaN   \n",
       "10:8115916:C>G                          NaN                   NaN   \n",
       "10:8115916:C>T                          NaN                   NaN   \n",
       "10:8115927:C>T                          NaN                   NaN   \n",
       "10:8115945:G>A                          NaN                   NaN   \n",
       "\n",
       "                             DBNSFP_GERP_RS_PRED DBNSFP_INTERPRO_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686358:T>C                                C                    .&.   \n",
       "20:21686361:C>T                                C                    .&.   \n",
       "20:21686379:G>A                                C                    .&.   \n",
       "20:21686382:C>T                                C                    .&.   \n",
       "20:21686390:G>T                                C                    .&.   \n",
       "...                                          ...                    ...   \n",
       "10:8115915:C>T                                 C                  .&.&.   \n",
       "10:8115916:C>G                                 C                  .&.&.   \n",
       "10:8115916:C>T                                 C                  .&.&.   \n",
       "10:8115927:C>T                                 C                  .&.&.   \n",
       "10:8115945:G>A                                 C                  .&.&.   \n",
       "\n",
       "                             DBNSFP_LRT_PRED DBNSFP_MUTATIONTASTER_PRED  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686358:T>C                          NaN                          N   \n",
       "20:21686361:C>T                          NaN                          N   \n",
       "20:21686379:G>A                          NaN                          N   \n",
       "20:21686382:C>T                          NaN                          N   \n",
       "20:21686390:G>T                          NaN                          N   \n",
       "...                                      ...                        ...   \n",
       "10:8115915:C>T                             U                        D&D   \n",
       "10:8115916:C>G                             U                        D&D   \n",
       "10:8115916:C>T                             U                        D&D   \n",
       "10:8115927:C>T                             D                        D&D   \n",
       "10:8115945:G>A                             D                        D&D   \n",
       "\n",
       "                             DBNSFP_MUTATIONTASTER_SCORE  \\\n",
       "Genomic Description (GRCh37)                               \n",
       "20:21686358:T>C                                 0.506194   \n",
       "20:21686361:C>T                                 0.758794   \n",
       "20:21686379:G>A                                 0.999272   \n",
       "20:21686382:C>T                                 0.998176   \n",
       "20:21686390:G>T                                 0.999492   \n",
       "...                                                  ...   \n",
       "10:8115915:C>T                                       1&1   \n",
       "10:8115916:C>G                                       1&1   \n",
       "10:8115916:C>T                                       1&1   \n",
       "10:8115927:C>T                         0.999167&0.999167   \n",
       "10:8115945:G>A                         0.999993&0.999993   \n",
       "\n",
       "                             DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED  \\\n",
       "Genomic Description (GRCh37)                                     \n",
       "20:21686358:T>C                                              C   \n",
       "20:21686361:C>T                                              N   \n",
       "20:21686379:G>A                                              C   \n",
       "20:21686382:C>T                                              N   \n",
       "20:21686390:G>T                                              C   \n",
       "...                                                        ...   \n",
       "10:8115915:C>T                                               C   \n",
       "10:8115916:C>G                                               C   \n",
       "10:8115916:C>T                                               C   \n",
       "10:8115927:C>T                                               C   \n",
       "10:8115945:G>A                                               C   \n",
       "\n",
       "                             DBNSFP_POLYPHEN2_HDIV_PRED  \\\n",
       "Genomic Description (GRCh37)                              \n",
       "20:21686358:T>C                                     P&.   \n",
       "20:21686361:C>T                                     P&.   \n",
       "20:21686379:G>A                                     P&.   \n",
       "20:21686382:C>T                                     P&.   \n",
       "20:21686390:G>T                                     B&.   \n",
       "...                                                 ...   \n",
       "10:8115915:C>T                                    D&D&D   \n",
       "10:8115916:C>G                                    D&D&D   \n",
       "10:8115916:C>T                                    D&D&D   \n",
       "10:8115927:C>T                                    B&B&B   \n",
       "10:8115945:G>A                                    D&D&P   \n",
       "\n",
       "                             DBNSFP_POLYPHEN2_HDIV_SCORE  ... Interpro Domain  \\\n",
       "Genomic Description (GRCh37)                              ...                   \n",
       "20:21686358:T>C                                  0.596&.  ...             NaN   \n",
       "20:21686361:C>T                                  0.917&.  ...             NaN   \n",
       "20:21686379:G>A                                  0.787&.  ...             NaN   \n",
       "20:21686382:C>T                                  0.804&.  ...             NaN   \n",
       "20:21686390:G>T                                  0.005&.  ...             NaN   \n",
       "...                                                  ...  ...             ...   \n",
       "10:8115915:C>T                               1.0&1.0&1.0  ...             NaN   \n",
       "10:8115916:C>G                               1.0&1.0&1.0  ...             NaN   \n",
       "10:8115916:C>T                               1.0&1.0&1.0  ...             NaN   \n",
       "10:8115927:C>T                         0.097&0.097&0.001  ...             NaN   \n",
       "10:8115945:G>A                         0.997&0.997&0.745  ...             NaN   \n",
       "\n",
       "                                  Max MAF Source OVERALL_MAX_MAF  \\\n",
       "Genomic Description (GRCh37)                                       \n",
       "20:21686358:T>C                              NaN             NaN   \n",
       "20:21686361:C>T               gnomad_AF_nfe_pass     6.65956e-05   \n",
       "20:21686379:G>A               gnomad_AF_nfe_pass     0.000104932   \n",
       "20:21686382:C>T               gnomad_AF_afr_pass       0.0318745   \n",
       "20:21686390:G>T                              NaN             NaN   \n",
       "...                                          ...             ...   \n",
       "10:8115915:C>T                               NaN              []   \n",
       "10:8115916:C>G                gnomad_AF_nfe_pass     6.48424e-05   \n",
       "10:8115916:C>T                gnomad_AF_afr_pass     0.000123092   \n",
       "10:8115927:C>T                gnomad_AF_nfe_pass     8.80065e-06   \n",
       "10:8115945:G>A                gnomad_AF_nfe_pass     1.54947e-05   \n",
       "\n",
       "                             OVERALL_MAX_MAF_SOURCE TYPE UNIPROT_DOMAIN  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686358:T>C                                 NaN  SNV            NaN   \n",
       "20:21686361:C>T                  gnomad_AF_nfe_pass  SNV            NaN   \n",
       "20:21686379:G>A                  gnomad_AF_nfe_pass  SNV            NaN   \n",
       "20:21686382:C>T                  gnomad_AF_afr_pass  SNV            NaN   \n",
       "20:21686390:G>T                                 NaN  SNV            NaN   \n",
       "...                                             ...  ...            ...   \n",
       "10:8115915:C>T                                   []  SNV            NaN   \n",
       "10:8115916:C>G                   gnomad_AF_nfe_pass  SNV            NaN   \n",
       "10:8115916:C>T                   gnomad_AF_afr_pass  SNV            NaN   \n",
       "10:8115927:C>T                   gnomad_AF_nfe_pass  SNV            NaN   \n",
       "10:8115945:G>A                   gnomad_AF_nfe_pass  SNV            NaN   \n",
       "\n",
       "                               VEP_CONSEQUENCE VEP_IMPACT VEP_POLYPHEN_PRED  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C               missense_variant   MODERATE                 P   \n",
       "20:21686361:C>T               missense_variant   MODERATE                 P   \n",
       "20:21686379:G>A               missense_variant   MODERATE                 P   \n",
       "20:21686382:C>T               missense_variant   MODERATE                 P   \n",
       "20:21686390:G>T               missense_variant   MODERATE                 B   \n",
       "...                                        ...        ...               ...   \n",
       "10:8115915:C>T                missense_variant   MODERATE                 D   \n",
       "10:8115916:C>G                missense_variant   MODERATE                 D   \n",
       "10:8115916:C>T                missense_variant   MODERATE                 D   \n",
       "10:8115927:C>T                missense_variant   MODERATE                 B   \n",
       "10:8115945:G>A                missense_variant   MODERATE                 D   \n",
       "\n",
       "                             VEP_PROTEIN_POS  \n",
       "Genomic Description (GRCh37)                  \n",
       "20:21686358:T>C                            3  \n",
       "20:21686361:C>T                            4  \n",
       "20:21686379:G>A                           10  \n",
       "20:21686382:C>T                           11  \n",
       "20:21686390:G>T                           14  \n",
       "...                                      ...  \n",
       "10:8115915:C>T                           422  \n",
       "10:8115916:C>G                           422  \n",
       "10:8115916:C>T                           422  \n",
       "10:8115927:C>T                           426  \n",
       "10:8115945:G>A                           432  \n",
       "\n",
       "[128712 rows x 25 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "categorical_columns = working_df.select_dtypes(include='object').columns.tolist()\n",
    "print(categorical_columns)\n",
    "working_df[categorical_columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn import feature_selection\n",
    "from sklearn import feature_extraction\n",
    "from sklearn.svm import LinearSVC\n",
    "from sklearn.pipeline import Pipeline\n",
    "from sklearn.preprocessing import LabelEncoder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "# FINAL_PATHOGENICITY: \"Final curated pathogenicity\" -- includes the DVD manual override\n",
    "# FINAL_PRED: \"Final prediction consensus based on majority vote of prediction scores\" -- should be if NUM_PATH_PREDS / TOTAL_NUM_PREDS >= 0.5 -> Likely pathogenic\n",
    "\n",
    "# Our hypothesis - FINAL_PRED is actually the full Azaiez method result, not just the result of voting from the pathogenicty predictions\n",
    "#                - FINAL_PATHOGENICTY is actually the post curation (overridden) label and is what is displayed on the dvd\n",
    "dvd_labeling_cols = ['NUM_PATH_PREDS', 'TOTAL_NUM_PREDS', 'FINAL_PRED', 'FINAL_PATHOGENICITY']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Up-stream Predictions\n",
    "# From Colton's Jan 6th, 2023 presentation (slide 12): PhyloP, SIFT, LRT, MutationTaster, PolyPhen HDIV, GERP\n",
    "prediction_cols = ['DBNSFP_GERP_RS_PRED', 'DBNSFP_LRT_PRED', 'DBNSFP_MUTATIONTASTER_PRED', 'DBNSFP_PHYLOP30WAY_MAMMALIAN_PRED', 'DBNSFP_POLYPHEN2_HDIV_PRED', 'DBNSFP_SIFT_PRED', 'VEP_POLYPHEN_PRED', 'NUM_PATH_PREDS', 'TOTAL_NUM_PREDS']\n",
    "\n",
    "# Up-stream Scores\n",
    "score_cols = ['DBNSFP_GERP_RS', 'DBNSFP_LRT_SCORE', 'DBNSFP_MUTATIONTASTER_SCORE', 'DBNSFP_PHYLOP30WAY_MAMMALIAN', 'DBNSFP_POLYPHEN2_HDIV_SCORE', 'DBNSFP_SIFT_SCORE', 'VEP_CADD_PHRED', 'VEP_CADD_RAW', 'VEP_POLYPHEN_SCORE', 'VEP_SIFT_SCORE']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'working_df' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[1], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mworking_df\u001b[49m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'working_df' is not defined"
     ]
    }
   ],
   "source": [
    "working_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'raw_data' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[56], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m y \u001b[39m=\u001b[39m raw_data[\u001b[39m'\u001b[39m\u001b[39mClass\u001b[39m\u001b[39m'\u001b[39m]\n\u001b[1;32m      2\u001b[0m X \u001b[39m=\u001b[39m raw_data\u001b[39m.\u001b[39mdrop(\u001b[39m'\u001b[39m\u001b[39mClass\u001b[39m\u001b[39m'\u001b[39m,axis\u001b[39m=\u001b[39m\u001b[39m1\u001b[39m)\n\u001b[1;32m      4\u001b[0m imputer \u001b[39m=\u001b[39m SimpleImputer(strategy\u001b[39m=\u001b[39m\u001b[39m'\u001b[39m\u001b[39mconstant\u001b[39m\u001b[39m'\u001b[39m,fill_value\u001b[39m=\u001b[39m\u001b[39m0\u001b[39m, missing_values\u001b[39m=\u001b[39mnp\u001b[39m.\u001b[39mnan)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'raw_data' is not defined"
     ]
    }
   ],
   "source": [
    "y = raw_data['Class']\n",
    "X = raw_data.drop('Class',axis=1)\n",
    "\n",
    "imputer = SimpleImputer(strategy='constant',fill_value=0, missing_values=np.nan)\n",
    "imputer = imputer.fit(X[['CADD Phred','MAF','Surface Area','Normalized SA','Confidence Score','ddG']])\n",
    "X[['CADD Phred','MAF','Surface Area','Normalized SA','Confidence Score','ddG']]=imputer.transform(X[['CADD Phred','MAF','Surface Area','Normalized SA','Confidence Score','ddG']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from sklearn.model_selection import train_test_split\n",
    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2)\n",
    "from sklearn.svm import SVC\n",
    "svclassifier = SVC(kernel='linear',probability=True)\n",
    "svclassifier.fit(X_train, y_train)\n",
    "\n",
    "y_pred_prob = svclassifier.predict_proba(X_test)\n",
    "y_pred_prob\n",
    "y_pred=svclassifier.predict(X_test)\n",
    "display(pd.DataFrame({'svclassifier.predict': y_pred,\n",
    "                     'svclassifier.predict_prob_a[:,0] (Pathogenic)': y_pred_prob[:,0],\n",
    "                     'svclassifier.predict_prob_a[:,1] (Non-Pathogenic)': y_pred_prob[:,1]}).head(100))\n",
    "y_predictions_prob_a = y_pred_prob[:,0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "coltons_cols = ['Genomic Description (GRCh37)', 'HGVS Nucleotide Change', 'HGVS Protein Change', 'Phenotype', 'Consequence', 'Exon', 'Intron', 'Protein Position', 'Impact', 'Interpro Domain', 'Max MAF Source', '|ddG|']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>VEP_CADD_RAW</th>\n",
       "      <th>VEP_CADD_PHRED</th>\n",
       "      <th>Max MAF (%)</th>\n",
       "      <th>Surface Area</th>\n",
       "      <th>Normalized SA</th>\n",
       "      <th>Confidence Score</th>\n",
       "      <th>ddG</th>\n",
       "      <th>|ddG|</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic Description (GRCh37)</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20:21686358:T&gt;C</th>\n",
       "      <td>3.009015</td>\n",
       "      <td>23.60</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>172.951510</td>\n",
       "      <td>0.752780</td>\n",
       "      <td>32.73</td>\n",
       "      <td>-1.4</td>\n",
       "      <td>1.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686361:C&gt;T</th>\n",
       "      <td>2.751772</td>\n",
       "      <td>23.00</td>\n",
       "      <td>0.006660</td>\n",
       "      <td>139.065932</td>\n",
       "      <td>0.883870</td>\n",
       "      <td>38.06</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686379:G&gt;A</th>\n",
       "      <td>2.662314</td>\n",
       "      <td>22.90</td>\n",
       "      <td>0.010493</td>\n",
       "      <td>180.703642</td>\n",
       "      <td>0.670700</td>\n",
       "      <td>37.23</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686382:C&gt;T</th>\n",
       "      <td>2.257539</td>\n",
       "      <td>21.80</td>\n",
       "      <td>3.187450</td>\n",
       "      <td>76.425233</td>\n",
       "      <td>0.601022</td>\n",
       "      <td>34.61</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20:21686390:G&gt;T</th>\n",
       "      <td>2.319001</td>\n",
       "      <td>22.10</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>99.401913</td>\n",
       "      <td>0.578281</td>\n",
       "      <td>31.28</td>\n",
       "      <td>-0.6</td>\n",
       "      <td>0.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115915:C&gt;T</th>\n",
       "      <td>3.938849</td>\n",
       "      <td>27.20</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;G</th>\n",
       "      <td>3.979789</td>\n",
       "      <td>27.50</td>\n",
       "      <td>0.006484</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.3</td>\n",
       "      <td>0.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115916:C&gt;T</th>\n",
       "      <td>4.163516</td>\n",
       "      <td>29.20</td>\n",
       "      <td>0.012309</td>\n",
       "      <td>136.946574</td>\n",
       "      <td>0.870607</td>\n",
       "      <td>47.29</td>\n",
       "      <td>-0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115927:C&gt;T</th>\n",
       "      <td>1.223753</td>\n",
       "      <td>14.78</td>\n",
       "      <td>0.000880</td>\n",
       "      <td>128.397680</td>\n",
       "      <td>0.816259</td>\n",
       "      <td>44.11</td>\n",
       "      <td>-0.1</td>\n",
       "      <td>0.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10:8115945:G&gt;A</th>\n",
       "      <td>3.995061</td>\n",
       "      <td>27.60</td>\n",
       "      <td>0.001549</td>\n",
       "      <td>67.643291</td>\n",
       "      <td>0.709449</td>\n",
       "      <td>37.21</td>\n",
       "      <td>0.2</td>\n",
       "      <td>0.2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>128712 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                              VEP_CADD_RAW  VEP_CADD_PHRED  Max MAF (%)  \\\n",
       "Genomic Description (GRCh37)                                              \n",
       "20:21686358:T>C                   3.009015           23.60     0.000000   \n",
       "20:21686361:C>T                   2.751772           23.00     0.006660   \n",
       "20:21686379:G>A                   2.662314           22.90     0.010493   \n",
       "20:21686382:C>T                   2.257539           21.80     3.187450   \n",
       "20:21686390:G>T                   2.319001           22.10     0.000000   \n",
       "...                                    ...             ...          ...   \n",
       "10:8115915:C>T                    3.938849           27.20     0.000000   \n",
       "10:8115916:C>G                    3.979789           27.50     0.006484   \n",
       "10:8115916:C>T                    4.163516           29.20     0.012309   \n",
       "10:8115927:C>T                    1.223753           14.78     0.000880   \n",
       "10:8115945:G>A                    3.995061           27.60     0.001549   \n",
       "\n",
       "                              Surface Area  Normalized SA  Confidence Score  \\\n",
       "Genomic Description (GRCh37)                                                  \n",
       "20:21686358:T>C                 172.951510       0.752780             32.73   \n",
       "20:21686361:C>T                 139.065932       0.883870             38.06   \n",
       "20:21686379:G>A                 180.703642       0.670700             37.23   \n",
       "20:21686382:C>T                  76.425233       0.601022             34.61   \n",
       "20:21686390:G>T                  99.401913       0.578281             31.28   \n",
       "...                                    ...            ...               ...   \n",
       "10:8115915:C>T                  136.946574       0.870607             47.29   \n",
       "10:8115916:C>G                  136.946574       0.870607             47.29   \n",
       "10:8115916:C>T                  136.946574       0.870607             47.29   \n",
       "10:8115927:C>T                  128.397680       0.816259             44.11   \n",
       "10:8115945:G>A                   67.643291       0.709449             37.21   \n",
       "\n",
       "                              ddG  |ddG|  \n",
       "Genomic Description (GRCh37)              \n",
       "20:21686358:T>C              -1.4    1.4  \n",
       "20:21686361:C>T              -0.0    0.0  \n",
       "20:21686379:G>A              -0.3    0.3  \n",
       "20:21686382:C>T              -0.3    0.3  \n",
       "20:21686390:G>T              -0.6    0.6  \n",
       "...                           ...    ...  \n",
       "10:8115915:C>T               -0.3    0.3  \n",
       "10:8115916:C>G               -0.3    0.3  \n",
       "10:8115916:C>T               -0.2    0.2  \n",
       "10:8115927:C>T               -0.1    0.1  \n",
       "10:8115945:G>A                0.2    0.2  \n",
       "\n",
       "[128712 rows x 8 columns]"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "working_df.loc[:, ['VEP_CADD_RAW', 'VEP_CADD_PHRED', 'Max MAF (%)','Surface Area','Normalized SA','Confidence Score', 'ddG', '|ddG|']]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
